{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "01a4c137",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting litstudy\n",
      "  Downloading litstudy-1.0.3.tar.gz (38 kB)\n",
      "  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: networkx in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from litstudy) (2.8.4)\n",
      "Requirement already satisfied: numpy in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from litstudy) (1.21.5)\n",
      "Collecting wordcloud\n",
      "  Downloading wordcloud-1.8.2.2-cp39-cp39-macosx_10_9_x86_64.whl (160 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m160.5/160.5 kB\u001b[0m \u001b[31m1,000.0 kB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: unidecode in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from litstudy) (1.2.0)\n",
      "Collecting pyvis\n",
      "  Downloading pyvis-0.3.1.tar.gz (748 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m748.9/748.9 kB\u001b[0m \u001b[31m1.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting sklearn\n",
      "  Downloading sklearn-0.0.post1.tar.gz (3.6 kB)\n",
      "  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: requests in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from litstudy) (2.28.1)\n",
      "Collecting pybliometrics\n",
      "  Downloading pybliometrics-3.4.0.tar.gz (920 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m920.4/920.4 kB\u001b[0m \u001b[31m1.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: pandas in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from litstudy) (1.4.4)\n",
      "Requirement already satisfied: gensim in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from litstudy) (4.1.2)\n",
      "Requirement already satisfied: matplotlib in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from litstudy) (3.5.2)\n",
      "Collecting feedparser\n",
      "  Downloading feedparser-6.0.10-py3-none-any.whl (81 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m81.1/81.1 kB\u001b[0m \u001b[31m904.7 kB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: seaborn in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from litstudy) (0.11.2)\n",
      "Collecting bibtexparser\n",
      "  Downloading bibtexparser-1.4.0.tar.gz (51 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m51.9/51.9 kB\u001b[0m \u001b[31m787.8 kB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: pyparsing>=2.0.3 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from bibtexparser->litstudy) (3.0.9)\n",
      "Collecting sgmllib3k\n",
      "  Downloading sgmllib3k-1.0.0.tar.gz (5.8 kB)\n",
      "  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: scipy>=0.18.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from gensim->litstudy) (1.9.1)\n",
      "Requirement already satisfied: smart-open>=1.8.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from gensim->litstudy) (5.2.1)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->litstudy) (1.4.2)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->litstudy) (2.8.2)\n",
      "Requirement already satisfied: packaging>=20.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->litstudy) (21.3)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->litstudy) (4.25.0)\n",
      "Requirement already satisfied: pillow>=6.2.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->litstudy) (9.2.0)\n",
      "Requirement already satisfied: cycler>=0.10 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->litstudy) (0.11.0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pandas->litstudy) (2022.1)\n",
      "Collecting pbr\n",
      "  Using cached pbr-5.11.1-py2.py3-none-any.whl (112 kB)\n",
      "Collecting simplejson\n",
      "  Downloading simplejson-3.18.1-cp39-cp39-macosx_10_9_x86_64.whl (75 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m75.9/75.9 kB\u001b[0m \u001b[31m823.5 kB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: tqdm in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pybliometrics->litstudy) (4.64.1)\n",
      "Requirement already satisfied: jinja2>=2.9.6 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pyvis->litstudy) (2.11.3)\n",
      "Requirement already satisfied: ipython>=5.3.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pyvis->litstudy) (7.31.1)\n",
      "Collecting jsonpickle>=1.4.1\n",
      "  Downloading jsonpickle-3.0.1-py2.py3-none-any.whl (40 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m40.5/40.5 kB\u001b[0m \u001b[31m636.8 kB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: idna<4,>=2.5 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests->litstudy) (3.3)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests->litstudy) (2022.9.24)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests->litstudy) (2.0.4)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests->litstudy) (1.26.11)\n",
      "Requirement already satisfied: pexpect>4.3 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (4.8.0)\n",
      "Requirement already satisfied: backcall in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (0.2.0)\n",
      "Requirement already satisfied: appnope in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (0.1.2)\n",
      "Requirement already satisfied: decorator in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (5.1.1)\n",
      "Requirement already satisfied: setuptools>=18.5 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (63.4.1)\n",
      "Requirement already satisfied: jedi>=0.16 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (0.18.1)\n",
      "Requirement already satisfied: traitlets>=4.2 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (5.1.1)\n",
      "Requirement already satisfied: pickleshare in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (0.7.5)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (3.0.20)\n",
      "Requirement already satisfied: pygments in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (2.11.2)\n",
      "Requirement already satisfied: matplotlib-inline in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from ipython>=5.3.0->pyvis->litstudy) (0.1.6)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from jinja2>=2.9.6->pyvis->litstudy) (2.0.1)\n",
      "Requirement already satisfied: six>=1.5 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from python-dateutil>=2.7->matplotlib->litstudy) (1.16.0)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from jedi>=0.16->ipython>=5.3.0->pyvis->litstudy) (0.8.3)\n",
      "Requirement already satisfied: ptyprocess>=0.5 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pexpect>4.3->ipython>=5.3.0->pyvis->litstudy) (0.7.0)\n",
      "Requirement already satisfied: wcwidth in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=5.3.0->pyvis->litstudy) (0.2.5)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building wheels for collected packages: litstudy, bibtexparser, pybliometrics, pyvis, sklearn, sgmllib3k\n",
      "  Building wheel for litstudy (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for litstudy: filename=litstudy-1.0.3-py3-none-any.whl size=43342 sha256=29b2ad6387e60c1f36ce745fbb7b78a4cae56911200df44d804fd05880c63925\n",
      "  Stored in directory: /private/var/root/Library/Caches/pip/wheels/cd/59/0b/7a9da0f14bde03abd8e812e0246af5ad495a16d64386182f06\n",
      "  Building wheel for bibtexparser (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for bibtexparser: filename=bibtexparser-1.4.0-py3-none-any.whl size=42430 sha256=4b5e15dc331a9d5ac4b8171c0a3a98c9a9af9a2281069de283661aee4099c18c\n",
      "  Stored in directory: /private/var/root/Library/Caches/pip/wheels/83/e1/e3/2311be27728119eefd014e0a6039eee58470560d8ab31fd1fa\n",
      "  Building wheel for pybliometrics (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for pybliometrics: filename=pybliometrics-3.4.0-py2.py3-none-any.whl size=73911 sha256=3ccc657e3264bc2eda48dc9fc7c8327e8ead9993fc1f7b7fde7341421f19b6e2\n",
      "  Stored in directory: /private/var/root/Library/Caches/pip/wheels/6a/32/d2/79e913d29c27162765fa8264987d89c59cbe94318b8539d32f\n",
      "  Building wheel for pyvis (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for pyvis: filename=pyvis-0.3.1-py3-none-any.whl size=755835 sha256=fba49738f5a7387ca9531e6a9564efb42b70502432149c9435aa87baf355e8d7\n",
      "  Stored in directory: /private/var/root/Library/Caches/pip/wheels/55/7b/4b/dc896cf96ad07e0dc92d30e1984ab583c0d0feec208756e2b2\n",
      "  Building wheel for sklearn (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for sklearn: filename=sklearn-0.0.post1-py3-none-any.whl size=2936 sha256=ad4b2f3413c8c6e49fba189068441b0b4d8081d0c2975a8d1655f5ed514fcbdd\n",
      "  Stored in directory: /private/var/root/Library/Caches/pip/wheels/f8/e0/3d/9d0c2020c44a519b9f02ab4fa6d2a4a996c98d79ab2f569fa1\n",
      "  Building wheel for sgmllib3k (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for sgmllib3k: filename=sgmllib3k-1.0.0-py3-none-any.whl size=6048 sha256=6b709856e10f1fcee50bba5e54b1d7b5fe289fe52e56f02be7cb1a0daa554b9d\n",
      "  Stored in directory: /private/var/root/Library/Caches/pip/wheels/65/7a/a7/78c287f64e401255dff4c13fdbc672fed5efbfd21c530114e1\n",
      "Successfully built litstudy bibtexparser pybliometrics pyvis sklearn sgmllib3k\n",
      "Installing collected packages: sklearn, sgmllib3k, simplejson, pbr, jsonpickle, feedparser, bibtexparser, pybliometrics, wordcloud, pyvis, litstudy\n",
      "Successfully installed bibtexparser-1.4.0 feedparser-6.0.10 jsonpickle-3.0.1 litstudy-1.0.3 pbr-5.11.1 pybliometrics-3.4.0 pyvis-0.3.1 sgmllib3k-1.0.0 simplejson-3.18.1 sklearn-0.0.post1 wordcloud-1.8.2.2\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!sudo -H pip install litstudy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2e332204",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: scikit-learn in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (1.0.2)\n",
      "Requirement already satisfied: joblib>=0.11 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from scikit-learn) (1.1.0)\n",
      "Requirement already satisfied: scipy>=1.1.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from scikit-learn) (1.9.1)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from scikit-learn) (2.2.0)\n",
      "Requirement already satisfied: numpy>=1.14.6 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from scikit-learn) (1.21.5)\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!sudo -H pip install scikit-learn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "853fbed8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mWARNING: The directory '/Users/jennifer/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\u001b[0m\u001b[33m\n",
      "\u001b[0mRequirement already satisfied: networkx in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (2.8.4)\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!pip install networkx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "e62d5830",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mWARNING: The directory '/Users/jennifer/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\u001b[0m\u001b[33m\n",
      "\u001b[0mCollecting d3graph\n",
      "  Downloading d3graph-2.3.10-py3-none-any.whl (100 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m100.1/100.1 kB\u001b[0m \u001b[31m5.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: markupsafe==2.0.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from d3graph) (2.0.1)\n",
      "Requirement already satisfied: networkx>2 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from d3graph) (2.8.4)\n",
      "Collecting colourmap\n",
      "  Downloading colourmap-1.1.10-py3-none-any.whl (7.4 kB)\n",
      "Requirement already satisfied: packaging in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from d3graph) (21.3)\n",
      "Collecting python-louvain\n",
      "  Downloading python-louvain-0.16.tar.gz (204 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m204.6/204.6 kB\u001b[0m \u001b[31m21.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: numpy in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from d3graph) (1.21.5)\n",
      "Requirement already satisfied: pandas in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from d3graph) (1.4.4)\n",
      "Requirement already satisfied: jinja2 in /Users/jennifer/.local/lib/python3.9/site-packages (from d3graph) (3.1.2)\n",
      "Collecting ismember\n",
      "  Downloading ismember-1.0.2-py3-none-any.whl (7.5 kB)\n",
      "Requirement already satisfied: matplotlib in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from colourmap->d3graph) (3.5.2)\n",
      "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from packaging->d3graph) (3.0.9)\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pandas->d3graph) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pandas->d3graph) (2022.1)\n",
      "Requirement already satisfied: six>=1.5 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from python-dateutil>=2.8.1->pandas->d3graph) (1.16.0)\n",
      "Requirement already satisfied: pillow>=6.2.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->colourmap->d3graph) (9.2.0)\n",
      "Requirement already satisfied: cycler>=0.10 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->colourmap->d3graph) (0.11.0)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->colourmap->d3graph) (4.25.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from matplotlib->colourmap->d3graph) (1.4.2)\n",
      "Building wheels for collected packages: python-louvain\n",
      "  Building wheel for python-louvain (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for python-louvain: filename=python_louvain-0.16-py3-none-any.whl size=9388 sha256=ac9c90083c7735ec827f4c5f34d028fe3ab2a30bbfaa710ba1b09631110fe975\n",
      "  Stored in directory: /private/tmp/pip-ephem-wheel-cache-seykvxv4/wheels/4d/7c/b6/79b198e4ec43f915fbdf967953d48b89a18893b12aa9df0ae2\n",
      "Successfully built python-louvain\n",
      "Installing collected packages: python-louvain, ismember, colourmap, d3graph\n",
      "Successfully installed colourmap-1.1.10 d3graph-2.3.10 ismember-1.0.2 python-louvain-0.16\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!pip install d3graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "948ca1b2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mWARNING: The directory '/Users/jennifer/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\u001b[0m\u001b[33m\n",
      "\u001b[0mRequirement already satisfied: pybliometrics in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (3.4.0)\n",
      "Requirement already satisfied: simplejson in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pybliometrics) (3.18.1)\n",
      "Requirement already satisfied: pbr in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pybliometrics) (5.11.1)\n",
      "Requirement already satisfied: requests in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pybliometrics) (2.28.1)\n",
      "Requirement already satisfied: tqdm in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from pybliometrics) (4.64.1)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests->pybliometrics) (3.3)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests->pybliometrics) (2022.9.24)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests->pybliometrics) (1.26.11)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests->pybliometrics) (2.0.4)\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
      "\u001b[0mCould not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))) - skipping\n",
      "\u001b[33mWARNING: There was an error checking the latest version of pip.\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!pip install pybliometrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "90ff21b9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mWARNING: The directory '/Users/jennifer/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\u001b[0m\u001b[33m\n",
      "\u001b[0m\u001b[33mWARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /simple/scholarly/\u001b[0m\u001b[33m\n",
      "\u001b[0m\u001b[33mWARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /simple/scholarly/\u001b[0m\u001b[33m\n",
      "\u001b[0m\u001b[33mWARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /simple/scholarly/\u001b[0m\u001b[33m\n",
      "\u001b[0m\u001b[33mWARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /simple/scholarly/\u001b[0m\u001b[33m\n",
      "\u001b[0m\u001b[33mWARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))': /simple/scholarly/\u001b[0m\u001b[33m\n",
      "\u001b[0mCould not fetch URL https://pypi.org/simple/scholarly/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/scholarly/ (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))) - skipping\n",
      "\u001b[31mERROR: Could not find a version that satisfies the requirement scholarly (from versions: none)\u001b[0m\u001b[31m\n",
      "\u001b[0m\u001b[31mERROR: No matching distribution found for scholarly\u001b[0m\u001b[31m\n",
      "\u001b[0mCould not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)'))) - skipping\n",
      "\u001b[33mWARNING: There was an error checking the latest version of pip.\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!pip install scholarly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "02b6215d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting scholarly\n",
      "  Downloading scholarly-1.7.11-py3-none-any.whl (39 kB)\n",
      "Collecting free-proxy\n",
      "  Downloading free_proxy-1.1.0.tar.gz (5.0 kB)\n",
      "  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting sphinx-rtd-theme\n",
      "  Downloading sphinx_rtd_theme-1.2.0-py2.py3-none-any.whl (2.8 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.8/2.8 MB\u001b[0m \u001b[31m5.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0mm\n",
      "\u001b[?25hRequirement already satisfied: requests[socks] in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from scholarly) (2.28.1)\n",
      "Requirement already satisfied: arrow in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from scholarly) (1.2.2)\n",
      "Collecting python-dotenv\n",
      "  Downloading python_dotenv-0.21.1-py3-none-any.whl (19 kB)\n",
      "Requirement already satisfied: beautifulsoup4 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from scholarly) (4.11.1)\n",
      "Requirement already satisfied: bibtexparser in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from scholarly) (1.4.0)\n",
      "Collecting deprecated\n",
      "  Downloading Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)\n",
      "Requirement already satisfied: typing-extensions in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from scholarly) (4.3.0)\n",
      "Collecting selenium\n",
      "  Downloading selenium-4.8.0-py3-none-any.whl (6.3 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m6.3/6.3 MB\u001b[0m \u001b[31m6.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0mm\n",
      "\u001b[?25hCollecting httpx\n",
      "  Downloading httpx-0.23.3-py3-none-any.whl (71 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m71.5/71.5 kB\u001b[0m \u001b[31m2.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting fake-useragent\n",
      "  Downloading fake_useragent-1.1.1-py3-none-any.whl (50 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m50.4/50.4 kB\u001b[0m \u001b[31m1.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: python-dateutil>=2.7.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from arrow->scholarly) (2.8.2)\n",
      "Requirement already satisfied: soupsieve>1.2 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from beautifulsoup4->scholarly) (2.3.1)\n",
      "Requirement already satisfied: pyparsing>=2.0.3 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from bibtexparser->scholarly) (3.0.9)\n",
      "Requirement already satisfied: wrapt<2,>=1.10 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from deprecated->scholarly) (1.14.1)\n",
      "Collecting importlib-resources>=5.0\n",
      "  Downloading importlib_resources-5.10.2-py3-none-any.whl (34 kB)\n",
      "Requirement already satisfied: lxml in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from free-proxy->scholarly) (4.9.1)\n",
      "Requirement already satisfied: certifi in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from httpx->scholarly) (2022.9.24)\n",
      "Requirement already satisfied: sniffio in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from httpx->scholarly) (1.2.0)\n",
      "Collecting httpcore<0.17.0,>=0.15.0\n",
      "  Downloading httpcore-0.16.3-py3-none-any.whl (69 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m69.6/69.6 kB\u001b[0m \u001b[31m2.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting rfc3986[idna2008]<2,>=1.3\n",
      "  Downloading rfc3986-1.5.0-py2.py3-none-any.whl (31 kB)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests[socks]->scholarly) (2.0.4)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests[socks]->scholarly) (1.26.11)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests[socks]->scholarly) (3.3)\n",
      "Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from requests[socks]->scholarly) (1.7.1)\n",
      "Collecting trio-websocket~=0.9\n",
      "  Downloading trio_websocket-0.9.2-py3-none-any.whl (16 kB)\n",
      "Collecting trio~=0.17\n",
      "  Downloading trio-0.22.0-py3-none-any.whl (384 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m384.9/384.9 kB\u001b[0m \u001b[31m6.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: sphinx<7,>=1.6 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx-rtd-theme->scholarly) (5.0.2)\n",
      "Requirement already satisfied: docutils<0.19 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx-rtd-theme->scholarly) (0.18.1)\n",
      "Collecting sphinxcontrib-jquery!=3.0.0,>=2.0.0\n",
      "  Downloading sphinxcontrib_jquery-2.0.0-py3-none-any.whl (3.2 kB)\n",
      "Requirement already satisfied: anyio<5.0,>=3.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from httpcore<0.17.0,>=0.15.0->httpx->scholarly) (3.5.0)\n",
      "Collecting h11<0.15,>=0.13\n",
      "  Downloading h11-0.14.0-py3-none-any.whl (58 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m58.3/58.3 kB\u001b[0m \u001b[31m1.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: zipp>=3.1.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from importlib-resources>=5.0->fake-useragent->scholarly) (3.8.0)\n",
      "Requirement already satisfied: six>=1.5 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from python-dateutil>=2.7.0->arrow->scholarly) (1.16.0)\n",
      "Requirement already satisfied: importlib-metadata>=4.4 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (4.11.3)\n",
      "Requirement already satisfied: sphinxcontrib-htmlhelp>=2.0.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (2.0.0)\n",
      "Requirement already satisfied: sphinxcontrib-serializinghtml>=1.1.5 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (1.1.5)\n",
      "Requirement already satisfied: Jinja2>=2.3 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (2.11.3)\n",
      "Requirement already satisfied: sphinxcontrib-devhelp in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (1.0.2)\n",
      "Requirement already satisfied: sphinxcontrib-jsmath in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (1.0.1)\n",
      "Requirement already satisfied: sphinxcontrib-applehelp in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (1.0.2)\n",
      "Requirement already satisfied: snowballstemmer>=1.1 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (2.2.0)\n",
      "Requirement already satisfied: imagesize in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (1.4.1)\n",
      "Requirement already satisfied: alabaster<0.8,>=0.7 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (0.7.12)\n",
      "Requirement already satisfied: babel>=1.3 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (2.9.1)\n",
      "Requirement already satisfied: Pygments>=2.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (2.11.2)\n",
      "Requirement already satisfied: packaging in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (21.3)\n",
      "Requirement already satisfied: sphinxcontrib-qthelp in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (1.0.3)\n",
      "Requirement already satisfied: setuptools in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from sphinxcontrib-jquery!=3.0.0,>=2.0.0->sphinx-rtd-theme->scholarly) (63.4.1)\n",
      "Requirement already satisfied: attrs>=19.2.0 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from trio~=0.17->selenium->scholarly) (21.4.0)\n",
      "Collecting outcome\n",
      "  Downloading outcome-1.2.0-py2.py3-none-any.whl (9.7 kB)\n",
      "Collecting exceptiongroup>=1.0.0rc9\n",
      "  Downloading exceptiongroup-1.1.0-py3-none-any.whl (14 kB)\n",
      "Requirement already satisfied: async-generator>=1.9 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from trio~=0.17->selenium->scholarly) (1.10)\n",
      "Requirement already satisfied: sortedcontainers in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from trio~=0.17->selenium->scholarly) (2.4.0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting wsproto>=0.14\n",
      "  Downloading wsproto-1.2.0-py3-none-any.whl (24 kB)\n",
      "Requirement already satisfied: pytz>=2015.7 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from babel>=1.3->sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (2022.1)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in /Users/jennifer/opt/anaconda3/lib/python3.9/site-packages (from Jinja2>=2.3->sphinx<7,>=1.6->sphinx-rtd-theme->scholarly) (2.0.1)\n",
      "Building wheels for collected packages: free-proxy\n",
      "  Building wheel for free-proxy (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for free-proxy: filename=free_proxy-1.1.0-py3-none-any.whl size=5551 sha256=ec743f910ee12da5943ebd8b748b6c87eda33b97e663a721fad0f8107cd4dfac\n",
      "  Stored in directory: /private/var/root/Library/Caches/pip/wheels/3a/f5/c6/16538d0b36be48f9636442b14356fec6537b6d3516422947b3\n",
      "Successfully built free-proxy\n",
      "Installing collected packages: rfc3986, sphinxcontrib-jquery, python-dotenv, outcome, importlib-resources, h11, exceptiongroup, deprecated, wsproto, trio, httpcore, free-proxy, fake-useragent, trio-websocket, sphinx-rtd-theme, httpx, selenium, scholarly\n",
      "Successfully installed deprecated-1.2.13 exceptiongroup-1.1.0 fake-useragent-1.1.1 free-proxy-1.1.0 h11-0.14.0 httpcore-0.16.3 httpx-0.23.3 importlib-resources-5.10.2 outcome-1.2.0 python-dotenv-0.21.1 rfc3986-1.5.0 scholarly-1.7.11 selenium-4.8.0 sphinx-rtd-theme-1.2.0 sphinxcontrib-jquery-2.0.0 trio-0.22.0 trio-websocket-0.9.2 wsproto-1.2.0\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!sudo -H pip install scholarly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c4671708",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sbs\n",
    "import unittest\n",
    "import requests\n",
    "import json\n",
    "import litstudy\n",
    "import itertools\n",
    "from pyvis.network import Network\n",
    "import networkx as nx\n",
    "from d3graph import d3graph\n",
    "import gensim\n",
    "from gensim import corpora\n",
    "import matplotlib\n",
    "import warnings\n",
    "import string\n",
    "import re\n",
    "\n",
    "warnings.filterwarnings('ignore')  # To ignore all warnings that arise here to enhance clarity\n",
    " \n",
    "from gensim.models.coherencemodel import CoherenceModel\n",
    "from gensim.models.ldamodel import LdaModel\n",
    "from scholarly import scholarly"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85fe664f",
   "metadata": {},
   "source": [
    "## Springer dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b445133",
   "metadata": {},
   "outputs": [],
   "source": [
    "docs = litstudy.sources.scopus.search_scopus(\"network\")\n",
    "litstudy.plot.plot_author_histogram(docs)\n",
    "\n",
    "# https://api.elsevier.com/content/search/scopus?facets=country(include=\"United States\")&field=dc:identifier,authkeywords,affiliation-country&query=issn(19384122)&count=200&date=2018&httpAccept=application/json&apiKey=KEY&insttoken=TOKEN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b607e934",
   "metadata": {},
   "outputs": [],
   "source": [
    "litstudy.load_springer_csv()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "96fa501c",
   "metadata": {},
   "source": [
    "Filter Settings:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "9c274f68",
   "metadata": {},
   "outputs": [],
   "source": [
    "keyword = \"network\"\n",
    "date_int = '2022'\n",
    "springer_keyword = \"?q=(\"+ \"%22\" + keyword.replace(\" \", \"%20\") + \"%22\" + \"%20AND%20year:\" + str(date_int) + \")\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "24642ece",
   "metadata": {},
   "outputs": [],
   "source": [
    "springer_api_key = \"d38abc6c90282c27f83cf11a4442683c\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "cac57bb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "base_url_springer = 'http://api.springer.com/metadata/json'\n",
    "#springer_keyword = \"network\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "da0f9e07",
   "metadata": {},
   "outputs": [],
   "source": [
    "country_api = \"http://api.springernature.com/openaccess/jats?q=country:%22New%20Netherlands%22&api_key=d38abc6c90282c27f83cf11a4442683c\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78640079",
   "metadata": {},
   "outputs": [],
   "source": [
    "url_params_springer = {\"network analysis\"}\n",
    "url_params_springer[\"api_key\"] = springer_api_key\n",
    "url_params_springer[\"p\"] = 200"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "698371f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'api_key': 'd38abc6c90282c27f83cf11a4442683c', 'p': 200}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "url_params_springer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "bb614086",
   "metadata": {},
   "outputs": [],
   "source": [
    "try: \n",
    "    ###Pick out articles' SCOPUS_ID \n",
    "    for_springer = open('springer_abstract.txt').read()\n",
    "    #transit text dictionary to dictionary\n",
    "    d_springer = json.loads(for_springer)\n",
    "\n",
    "except:\n",
    "    #request 200 articles from Scopus, search articles which contain \"rain garden\" and are published in 2016\n",
    "    #see document https://dev.springer.com/adding-constraints\n",
    "    d_springer = requests.get(base_url_springer + springer_keyword\n",
    "                                ,params=url_params_springer).json()\n",
    "    #for Gitbash\n",
    "    print (\"--------Result of Springer Metadata API-----------\")\n",
    "    #print pretty(d_springer)[:2000]\n",
    "    # cache the data from Scopus and collected\n",
    "    fr_springer = open(\"springer_abstract.txt\",\"w\")\n",
    "    fr_springer.write(json.dumps(d_springer))\n",
    "    fr_springer.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "908c49d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "abstract_list_springer = [i[\"abstract\"] for i in d_springer[\"records\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "5b75aea9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['The use of static graphs for modelling and analysis of biological and biomedical data plays a key role in biomedical research. However, many real-world scenarios present dynamic behaviours resulting in both node and edges modification as well as feature evolution. Consequently, ad-hoc models for capturing these evolutions along the time have been introduced, also referred to as dynamic, temporal, time-varying graphs. Here, we focus on temporal graphs, i.e., graphs whose evolution is represented by a sequence of time-ordered snapshots. Each snapshot represents a graph active in a particular timestamp. We survey temporal graph models and related algorithms, presenting fundamentals aspects and the recent advances. We formally define temporal graphs, focusing on the problem setting and we present their main applications in biology and medicine. We also present temporal graph embedding and the application to recent problems such as epidemic modelling. Finally, we further state some promising research directions in the area. Main results of this study include a systematic review of fundamental temporal network problems and their algorithmic solutions considered in the literature, in particular those having application in computational biology and medicine. We also include the main software developed in this context.',\n",
       " '',\n",
       " 'Network functions virtualization (NFV) can realize flexible and diverse network services by replacing the conventional network equipment with the combination of virtual network functions (VNFs) and commodity servers. A certain network service can be composed of a sequence of VNFs, i.e., service (function) chain. The service chaining (SC) problem aims to establish an appropriate service path from the origin node to the destination node, which holds both the resource constraints and service chain requirements of executing the required VNFs in the designated order. SC belongs to the complexity class NP -hard. In the previous work, inspired by the similarity between the SC problem and the shortest path tour problem (SPTP), we showed the capacitated SPTP (CSPTP) based ILP for the SC problem, where CSPTP is a generalized version of the SPTP with both the node and link capacity constraints. In this paper, we propose Lagrangian heuristics to solve the CSPTP-based ILP for the SC in a speedy and efficient manner. We further present that the proposed heuristics can also solve both the service chaining and function placement by slightly extending the network model called an augmented network. Through numerical results, we show that the proposed heuristics for the SC is competitive with the optimal resource allocation while executing much faster than the combination of the CSPTP-based ILP and the existing solver, i.e., CPLEX. Furthermore, we also show that the proposed heuristics for both the service chaining and function placement can still balance the solution optimality and computational complexity, thanks to the parallel computation architectures.',\n",
       " 'In this paper, we address the problem of implementing distributed slice allocation algorithms for tactical networks deployed in regions lacking a communication infrastructure. We propose a multi-agent game implementation in which Application Slice Agents ( ASAs ) compete for shared computational and bandwidth resources in the network. We prove that the game can be implemented as a two stage potential game, with the first stage ensuring the selection of computational servers with load balancing, and the second stage improving connectivity and network throughput by promoting inter-slice cooperation. Our proposed analysis framework also incorporates a theoretical model to capture the impact of mitigation methods for timing channel leaks for the shared resources for different slices, and proposes a theoretical framework that allows for a security-delay tradeoff analysis.',\n",
       " 'Image denoising is an important ill-posed problem of image processing. The main goal in image denoising is to suppress noise while protecting textures of the image. A plethora of methods and different approaches have targeted image denoising problem. In this manuscript, we propose a novel method which brings dense and residual blocks together with dilated convolutions in image denoising applications. The proposed method introduces new strategies for proper combination of dense residual blocks and dilated convolution layers. The resulting approach is called as Densely connected Dilated Residual Network (DDR-Net). The proposed DDR-Net extracts multi-scale information by employing dilated convolutions. Use of dilated convolutions leads to improved receptive field performance while keeping the complexity of the network in check. Residual and dense connections on the other hand prevent loss of information along the DDR-Net pipeline. Simulations are performed in both color and grayscale image denoising settings. Quantitative and qualitative results indicate that the proposed network architecture leads to improved results when compared to multiple high performance convolutional denoising networks from the recent literature.',\n",
       " 'Permissionless blockchains such as Ethereum offer decentralization and security. However, their performance is low compared to permissioned and traditional centralized systems, which may hinder their adoption. Transaction latency is an important performance metric that impacts the use of blockchain systems. That is, there is a need for a systematic evaluation to analyze the factors that may contribute to transaction latency. In this article, we propose a queueing model and conduct extensive simulation experiments to evaluate transaction latency within the Ethereum blockchain, considering all the contributing factors. Our simulation results suggest that transaction fees, block limit, block interval time, transaction arrival rate, and the behavior of the network nodes can all significantly contribute to the latency figures. We find the behavior of the network nodes can significantly impact the waiting time for transactions, especially since nodes are not enforced to utilize their blocks properly in the current implementation of permissionless blockchains. Finally, we suggest improvements to transaction latency by (1) increasing the block limit, (2) reducing the block interval time, and (3) encouraging nodes to fill their blocks with transactions as well as discussing the relevant security issues.',\n",
       " 'This article is dedicated to Eduardo D. Sontag on the occasion of his 70th birthday. We build upon fundamental stability concepts developed by Sontag, such as input-to-state stability and its related properties, to study a relevant application in industrial internet of things, namely estimation for wireless networked control systems. Particularly, we study emulation-based state estimation for nonlinear plants that communicate with a remote observer over a shared wireless network subject to packet losses. To reduce bandwidth usage, a stochastic communication protocol is employed to determine which node should be given access to the network. Each node has a different successful transmission probability. We describe the overall closed-loop system as a stochastic hybrid model, which allows us to capture the behaviour both between and at transmission instants, whilst covering network features such as random transmission instants, packet losses and stochastic scheduling. We then provide sufficient conditions on the transmission rate that guarantee an input-to-state stability property (in expectation) for the corresponding estimation error system. We illustrate our results in the design of circle criterion observers.',\n",
       " 'Existing state-of-the-art 3D point clouds understanding methods merely perform well in a fully supervised manner. To the best of our knowledge, there exists no unified framework which simultaneously solves the downstream high-level understanding tasks including both segmentation and detection, especially when labels are extremely limited. This work presents a general and simple framework to tackle point clouds understanding when labels are limited. The first contribution is that we have done extensive methodology comparisons of traditional and learnt 3D descriptors for the task of weakly supervised 3D scene understanding, and validated that our adapted traditional PFH-based 3D descriptors show excellent generalization ability across different domains. The second contribution is that we proposed a learning-based region merging strategy based on the affinity provided by both the traditional/learnt 3D descriptors and learnt semantics. The merging process takes both low-level geometric and high-level semantic feature correlations into consideration. Experimental results demonstrate that our framework has the best performance among the three most important weakly supervised point clouds understanding tasks including semantic segmentation, instance segmentation, and object detection even when very limited number of points are labeled. Our method, termed Region Merging 3D (RM3D), has superior performance on ScanNet data-efficient learning online benchmarks and other four large-scale 3D understanding benchmarks under various experimental settings, outperforming current arts by a margin for various 3D understanding tasks without complicated learning strategies such as active learning.',\n",
       " 'In the practice of deep learning-based rotating machinery fault diagnosis method, how to improve the accuracy of cross-working condition diagnosis under noise background is an urgent problem to be solved, which limits the application of deep learning method in engineering practice. Although domain adaptation methods have been widely researched to solve the problem, they often face the problem of domain shift. In this paper, two domain adaptation methods: domain adversarial learning and distribution matching methods, are used to train the deep network, to make the model realize the fault diagnosis across load conditions and signal-to-noise ratio. Moreover, in order to suppress the domain shift phenomenon which often occurred in domain adaptation tasks. A new network architecture based on the wide convolution kernel, multi-scale attention module, and self-adaptive soft threshold function is proposed so that the network is more suitable for feature extraction in cross-working condition fault diagnosis, and can avoid the influence of noise in vibration signals. Compared with other methods, the proposed method has good performance in faults diagnosis across load conditions and under noise. Feature visualization proved that the proposed network can effectively extract the conjoint fault features across different load conditions from the signal; hence, the fault diagnosis across load conditions and under the noise background can be realized and the domain shift can be suppressed effectively.',\n",
       " 'The automatic recognition of the medical examination report table (MERT) is receiving increasing attention in recent years as it is an essential step for intelligent healthcare and medical treatment. However, there are still some challenges in the table prediction when it is applied practically. In this paper, a recognition network (CoT_SRN) for medical examination reports is proposed to improve the recognition accuracy of MERT structure and reconstruct the image into a spreadsheet. The network is based on contextual transformer sequence and consists of CoT encoder and SRN decoder. In the encoder, the CNN backbone is constructed to extract the MERT image structure features based on the Contextual Transformer (CoT) proposed in this paper. In the decoder, an attention head with gated recurrent unit (GRU) was used for feature sequence recognition to obtain the cell location and table structure represented by a structured language. In addition, MERT structure labels are defined as character-level HTML formats, which are added in the training of the table structure recognition. The proposed method can achieve competitive tree-edit-distance-based similarity (TEDS) scores on the English datasets, such as PubTabNet and SciTSR, and Chinese datasets, such as the Chinese medical document dataset (CMDD). It demonstrates that the Cot_SRN is helpful to preserve the good performance across multi-language MERT structure recognition. Additionally, the performance of the proposed method is verified on the practical examples with folds and small angle deflection. The experimental results show that the proposed method is promising in practical application.',\n",
       " 'Identifying a set of vital nodes to achieve influence maximization is a topic of general interest in network science. Many algorithms have been proposed to solve the influence maximization problem in complex networks. Most of them just use topology information of networks to measure the node influence. However, the node attribute is also an important factor for measuring node influence in attributed networks. To tackle this problem, we first propose an extension model of linear threshold (LT) propagation model to simulate the information propagation in attributed networks. Then, we propose a novel community-based method to identify a set of vital nodes for influence maximization in attributed networks. The proposed method considers both topology influence and attribute influence of nodes, which is more suitable for identifying vital nodes in attributed networks. A series of experiments are carried out on five real world networks and a large scale synthetic network. Compared with CELF, IMM, CoFIM, HGD, NCVoteRank and K-Shell methods, experimental results based on different propagation models show that the proposed method improves the influence spread by $$-2.28\\\\% \\\\, \\\\textrm{to} \\\\, 4.76\\\\%$$ - 2.28 % to 4.76 % , $$-2.50\\\\% \\\\, \\\\textrm{to} \\\\, 16.97\\\\%$$ - 2.50 % to 16.97 % , $$0.18\\\\% \\\\, \\\\textrm{to} \\\\, 16.07\\\\%$$ 0.18 % to 16.07 % , $$0.22\\\\% \\\\, \\\\textrm{to} \\\\, 41.82\\\\%$$ 0.22 % to 41.82 % , $$0.23\\\\% \\\\, \\\\textrm{to} \\\\, 11.24\\\\%$$ 0.23 % to 11.24 % and $$10.78\\\\% \\\\, \\\\textrm{to} \\\\, 75.22\\\\%$$ 10.78 % to 75.22 % .',\n",
       " 'Recent advancements in deep learning have enabled 3D human body reconstruction from a monocular image, which has broad applications in multiple domains. In this paper, we propose SHARP ( SH ape A ware R econstruction of P eople in loose clothing), a novel end-to-end trainable network that accurately recovers the 3D geometry and appearance of humans in loose clothing from a monocular image. SHARP uses a sparse and efficient fusion strategy to combine parametric body prior with a non-parametric 2D representation of clothed humans. The parametric body prior enforces geometrical consistency on the body shape and pose, while the non-parametric representation models loose clothing and handles self-occlusions as well. We also leverage the sparseness of the non-parametric representation for faster training of our network while using losses on 2D maps. Another key contribution is 3DHumans , our new life-like dataset of 3D human body scans with rich geometrical and textural details. We evaluate SHARP on 3DHumans and other publicly available datasets, and show superior qualitative and quantitative performance than existing state-of-the-art methods.',\n",
       " 'An intelligent malware prediction system is needed for monitoring the network traffic of traditional and future network attacks. The Internet of Things (IoT) is fully connected and remotely monitored in many places. These devices are more vulnerable and easier to attack remotely. Securing and monitoring the activities of these devices must be done at the edge level. Machine Learning (ML) models are broadly used for supervising defence purposes, and these models are computationally expensive. Light Weight Gradient Ensemble Machine (LWGEM) with voting-based feature selection techniques and tuning the hyper-parameter are used to enhance the model accuracy and reduce the computational expenses. Performance of the model was evaluated with benchmark dataset UNSW_NB15 which shows the prominent result over the existing model and state of art models. The proposed model performs better in predicting unknown abnormalities in the IoT network.',\n",
       " 'Through the integration of distributed generation into the distribution network, several benefits can be achieved, including minimization of system losses, enhancement of voltage profiles, and improvement of network reliability and security. One of the best ways to maximize the benefits of distributed generation units is to optimize the location and size of these units in the network. The main objective of this paper is the application and comparison of five recently developed optimization algorithms for optimizing the allocation of multiple distributed generators in the distribution network. The algorithms include the hybrid grey wolf optimizer and cuckoo search, bald eagle search, marine predator algorithm, artificial ecosystem optimizer, and slime mould algorithm. Three objectives are set in the problem of determining the optimal allocation of distributed generators, including the reduction of energy losses, enhancement of voltage profile, and improvement of voltage stability. The proposed optimization algorithms are evaluated and compared using IEEE 33-bus and IEEE 69-bus radial distribution networks. Numerical results show that the proposed algorithms are very competitive in their ability to allocate distributed generators optimally. However, artificial ecosystem optimizer and bald eagle search outperform other algorithms.',\n",
       " 'Arbitrary-style-per-model (ASPM) style transfer algorithms transfer arbitrary styles based on a single model. Statistics-based learning algorithms of ASPM, represented by adaptive instance normalization (AdaIN), conduct instance normalization and then perform an affine transformation on target features. These algorithms are computationally efficient and easy to embed in convolutional neural networks. Consequently, they are widely used in image synthesis tasks to control the style of the resulting images. However, the style of stylized images may be a combination of content and stylized images, which suggests that these methods do not transform styles accurately. In this work, we rethink the function of AdaIN in controlling style. We show that the role of AdaIN is to (1) give each input content image a specific optimization target, (2) dynamically set cross-channel correlations, and (3) act as a feature selector after combining it with an activation function. Accordingly, we propose adaptive style modulation (AdaSM), which fully leverages the three roles mentioned above and thereby enables more precise control of global style. Experimental results show that AdaSM provides superior style controllability, alleviates the style blending problem, and outperforms state-of-the-art methods in artistic style transfer tasks.',\n",
       " 'This study deals with the intelligent modeling of reactive yellow 84 dye removal from aqueous solutions by zinc oxide (ZnO) nanoparticles based on inorganic light expanded clay aggregates. Accordingly, a set of experimental data were utilized to develop robust models by adjusting the initial feed concentration, initial pH, de-colorization time, catalyst concentration, and lamp power as input factors. The Leverage method, a common method for detecting outliers, proved all measured data were reliable. After definition of input factors, four intelligent approaches, i.e., multilayer perceptron, Gaussian process regression (GPR), radial basis function, and adaptive neuro fuzzy inference system with subtractive clustering were used to establish exact models for predicting the dye removal. Among them, the GPR-based model produced the best results with AARE, RRMSE, and R 2 values of 4.87%, 6.21%, and 97.31%, respectively, for the test dataset. The remaining models also provided satisfactory outputs with AAREs between 6.59% and 9.81%. In addition, the influence of each operating parameter on the dye removal was properly described for the novel models. Finally, the most effective parameters on removal performance were determined by using a sensitivity analysis based on the GPR model.',\n",
       " 'Abstract The problem of classifying turbulent environments from partial observation is key for some theoretical and applied fields, from engineering to earth observation and astrophysics, e.g., to precondition searching of optimal control policies in different turbulent backgrounds, to predict the probability of rare events and/or to infer physical parameters labeling different turbulent setups. To achieve such goal one can use different tools depending on the system’s knowledge and on the quality and quantity of the accessible data. In this context, we assume to work in a model-free setup completely blind to all dynamical laws, but with a large quantity of (good quality) data for training. As a prototype of complex flows with different attractors, and different multi-scale statistical properties we selected 10 turbulent ‘ensembles’ by changing the rotation frequency of the frame of reference of the 3d domain and we suppose to have access to a set of partial observations limited to the instantaneous kinetic energy distribution in a 2d plane, as it is often the case in geophysics and astrophysics. We compare results obtained by a machine learning (ML) approach consisting of a state-of-the-art deep convolutional neural network (DCNN) against Bayesian inference which exploits the information on velocity and entropy moments. First, we discuss the supremacy of the ML approach, presenting also results at changing the number of training data and of the hyper-parameters. Second, we present an ablation study on the input data aimed to perform a ranking on the importance of the flow features used by the DCNN, helping to identify the main physical contents used by the classifier. Finally, we discuss the main limitations of such data-driven methods and potential interesting applications. Graphical Abstract ',\n",
       " 'Recent surge in technology and integration of IoT into Cloud computing has resulted in increasingly heterogeneous workloads with unprecedented compute and storage demands. As Cloud workloads evolve, so also their computational demands, resulting in server loads interspersed by random peaks and troughs and by continuous and periodic demands. A predictive model that accounts for these diverse load patterns can generate more realistic future resource demands which is essential for efficient capacity planning and meeting service level objectives with minimal energy consumption. Long Short-Term Memory (LSTM), the prevalent approach for host load prediction suffers from information decay with long inputs, while hybrid methods using Convolution Neural Network (CNN) and LSTM fail to effectively model different host load patterns. To overcome above limitations, we propose a multistep CPU usage prediction approach named RCP-CL , to model r andom fluctuations and novel c ontinuous and p eriodic patterns from contiguous and non-contiguous CPU load values augmented with daily and weekly time patterns, by integrating 1-Dimensional C NN (1D-CNN) and L STM networks. RCP-CL uses parallel and stacked 1D-CNN layers with kernel size and dilation rates that are guided by the autocorrelation and partial autocorrelation analysis of CPU usage while LSTM derives temporal dependencies from the learnt spatial patterns. Experimental evaluations with Google Trace and Alibaba trace demonstrate impressive learning skills of RCP-CL over the state-of-the-art LSTM and 1D-CNN based host load prediction models, achieving up to 20% improvement in mean squared error for Google trace and up to 22% improvement for Alibaba trace.',\n",
       " 'The path integral is not typically utilized for analyzing entanglement experiments, in part because there is no standard toolbox for converting an arbitrary experiment into a form allowing a simple sum-over-history calculation. After completing the last portion of this toolbox (a technique for implementing multi-particle measurements in an entangled basis), some interesting 4- and 6-particle experiments are analyzed with this alternate technique. While the joint probabilities of measurement outcomes are always equivalent to conventional quantum mechanics, differences in the calculations motivate a number of foundational insights, concerning nonlocality, retrocausality, and the objectivity of entanglement itself.',\n",
       " 'This paper presents an offline trained artificial neural network (ANN)-assisted perturb and observe (P &O) algorithm to track the global maximum power of a solar PV under partial shading conditions. The conventional hill climbing algorithms work very well to extract maximum power under uniform shading conditions, but fails under partial shading conditions. On the other hand, the offline trained ANN can track the power closer to global maximum power point (GMPP) with less oscillations. Hence, two methods are combined to design the MPPT controller for tracking the global maximum power accurately with less oscillations. Initially, ANN predicts the voltage near to the global maximum power point with solar PV voltage and current as the inputs. This predicted voltage from ANN is used as the initial values for the P &O algorithm, and the P &O algorithm is triggered for tracking the actual GMPP. The proposed MPPT controller is validated in the MATLAB/Simulink environment and the performance of the ANN model for the prediction of GMPP is validated with the experimental data sets.',\n",
       " 'After sequencing, it is common to screen ncRNA according to expression differences. But this may lose a lot of valuable information and there is currently no indicator to characterize the regulatory function and participation degree of ncRNA on transcriptome. Based on existing pathway enrichment methods, we developed a new algorithm to calculating the participation degree of ncRNA in transcriptome (PDNT). Here we analyzed multiple data sets, and differentially expressed genes (DEGs) were used for pathway enrichment analysis. The PDNT algorithm was used to calculate the Contribution value (C value) of each ncRNA based on its target genes and the pathways they participates in. The results showed that compared with ncRNAs screened by log2 fold change (FC) and p-value, those screened by C value regulated more DEGs in IPA canonical pathways, and their target DEGs were more concentrated in the core region of the protein–protein interaction (PPI) network. The ranking of disease critical ncRNAs increased integrally after sorting with C value. Collectively, we found that the PDNT algorithm provides a measure from another view compared with the log2FC and p-value and it may provide more clues to effectively evaluate ncRNA.',\n",
       " 'Accurate assessment of Trees outside forest (TOF) is an essential input for the\\xa0management and preservation of these vast resources, due to their importance in carbon sequestration and biodiversity conservation. Many studies have demonstrated the\\xa0use of High-resolution satellite (HRS) imageries for mapping and monitoring of TOF in urban and rural landscapes. The conventional per-pixel classifier is not a suitable option for HRS data classification, owing to inherent high intra-class variability and high interclass similarity. In the present study, a convolution neural network-based approach was employed for TOF mapping using HRS images, predominantly in the\\xa0urban landscape of Bengaluru city, India. We developed a semi-automated procedure for the\\xa0generation of labelled training samples using object-based image analysis (OBIA), minimizing time requirement for input training data preparation for deep learning (DL) model development. These inputs were utilized to develop a DL model for the\\xa0classification of TOF using U-Net, which is a semantic segmentation-based DL architecture. The results indicated that the model performed well for the\\xa0classification of urban TOF with an\\xa0overall classification accuracy of 89.65 percent and an\\xa0F1 score of 93.03 percent, in comparison with OBIA (overall accuracy of 80.73 percent & F1 score of 86.44 percent). We demonstrated utilisation of a\\xa0developed methodology for the\\xa0assessment of TOF in a distinctive urban landscape, which can be extended to agriculture dominated regions in diverse landscapes.',\n",
       " 'Graph convolutional networks (GCNs) have been shown to be effective in performing skeleton-based action recognition, as graph topology has advantages in representing the natural connectivity of the human bodies. Nevertheless, it is challenging to effectively model the human joints spatially and temporally, and we are lacking attentional mechanisms for critical temporal frames and important skeletal points. In this work, we propose a novel GCNs combined with channel-wise joints and temporal attention for skeleton-based action recognition. Our temporal attention module captures the long-term dependence of time and then enhances the temporal semantics of key frames. In addition, we design a channel-wise attention module that fuses multi-channel joint weights with the topological map to capture the attention of nodes at different actions along the channel dimension. We propose to concatenate joint and bone together along the channel dimension as the joint& bone (J& B) modality, J& B modality can extract hybrid action patterns under the coalition of channel-wise joint attention. We prove the powerful spatio-temporal modeling capability of our model on three widely used dataset, NTU-RGB D, NTU RGB+D 120 and Northwestern-UCLA. Compared with leading GCN-based methods, we achieve performance comparable to the-state-of-art.',\n",
       " \"Background The high number of COVID-19 deaths is a serious threat to the world. Demographic and clinical biomarkers are significantly associated with the mortality risk of this disease. This study aimed to implement Generalized Neural Additive Model (GNAM) as an interpretable machine learning method to predict the COVID-19 mortality of patients. Methods This cohort study included 2181 COVID-19 patients admitted from February 2020 to July 2021 in Sina and Besat hospitals in Hamadan, west of Iran. A total of 22 baseline features including patients' demographic information and clinical biomarkers were collected. Four strategies including removing missing values, mean, K-Nearest Neighbor (KNN), and Multivariate Imputation by Chained Equations (MICE) imputation methods were used to deal with missing data. Firstly, the important features for predicting binary outcome (1: death, 0: recovery) were selected using the Random Forest (RF) method. Also, synthetic minority over-sampling technique (SMOTE) method was used for handling imbalanced data. Next, considering the selected features, the predictive performance of GNAM for predicting mortality outcome was compared with logistic regression, RF, generalized additive model (GAMs), gradient boosting decision tree (GBDT), and deep neural networks (DNNs) classification models. Each model trained on fifty different subsets of a train-test dataset to ensure a model performance. The average accuracy, F1-score and area under the curve (AUC) evaluation indices were used for comparison of the predictive performance of the models. Results Out of the 2181 COVID-19 patients, 624 died during hospitalization and 1557 recovered. The missing rate was 3 percent for each patient. The mean age of dead patients (71.17\\u2009±\\u200914.44\\xa0years) was statistically significant higher than recovered patients (58.25\\u2009±\\u200916.52\\xa0years). Based on RF, 10 features with the highest relative importance were selected as the best influential features; including blood urea nitrogen (BUN), lymphocytes (Lym), age, blood sugar (BS), serum glutamic-oxaloacetic transaminase (SGOT), monocytes (Mono), blood creatinine (CR), neutrophils (NUT), alkaline phosphatase (ALP) and hematocrit (HCT). The results of predictive performance comparisons showed GNAM with the mean accuracy, F1-score, and mean AUC in the test dataset of 0.847, 0.691, and 0.774, respectively, had the best performance. The smooth function graphs learned from the GNAM were descending for the Lym and ascending for the other important features. Conclusions Interpretable GNAM can perform well in predicting the mortality of COVID-19 patients. Therefore, the use of such a reliable model can help physicians to prioritize some important demographic and clinical biomarkers by identifying the effective features and the type of predictive trend in disease progression.\",\n",
       " 'Citrus is an agricultural product with significant added value in the world. Depending on market demands and causing possible economic losses, disease detection in citrus fruits at an early stage is important as well as for all agricultural products. Therefore, it is critical to detect both early-stage citrus diseases and physically damaged citrus fruits with technological solutions. In this study, a new convolutional neural network (CNN)-based model, CitrusNet, was proposed for the classification of defective and deformed citrus fruits. In this study, 5149 images of citrus fruits were collected from citrus orchards in Antalya, Turkey. In the experiments conducted with four different CNN models, the CitrusNet and ResNet50 models obtained the best classification results. In the other phase of the study, experiments were carried out to detect Alternaria alternata and Thrips diseases, which are common in Turkey, using five different CNN models. In the first phase of the study, a dataset of 3582 images of citrus diseases was obtained. The experimental results show that the YOLOv5 and Mask R-CNN models adequately detected the citrus diseases compared to the other models. These models achieved the best performance with an average precision (AP) of 0.99.',\n",
       " 'Helopeltis theivora or the tea mosquito bug (TMB) is reportedly one of the most devastating pests of tea plant ( Camellia sinensis ) causing threat to the beverage crop. Long non-coding RNAs (lncRNAs) constitute a group of endogenous RNAs that play gene regulatory roles in eukaryotes. In the present study, 9502 lncRNAs were identified from healthy and TMB-infested C. sinensis tissues using high-throughput strand-specific RNA sequencing, out of which 80 lncRNAs got differentially expressed in response to TMB infestation. Determination of genes that could act as potential targets of lncRNAs revealed that the identified lncRNAs could possibly target as many as 5804 genes. Differential gene expression (DGE) analysis led to the identification of 3665 differentially expressed genes (DEGs), of which, the expression of 1767 genes got upregulated and 1898 genes got downregulated during tea plant’s response to TMB. Gene Ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) enrichment analysis of DEGs and lncRNA-target genes have shown that TMB infestation might have triggered transcriptomic reprogramming leading to altered primary and secondary metabolism in C. sinensis . LncRNAs can act as competing endogenous RNAs (ceRNAs) to bind with common microRNA (miRNA) response elements (MREs) involving a competition between mRNAs and lncRNAs. We report 11 lncRNAs competing with 14 mRNAs to bind with 28 miRNAs forming the ceRNA network. The expression of 6 DEGs and 5 differentially expressed lncRNAs (DELs) has been validated by qRT-PCR.',\n",
       " 'U-Net and its variants show excellent performance in medical image segmentation. However, due to the information loss caused by down-sampling and the blurry boundary of the ultrasound images itself, it is hard to extract and recover boundary information. Additionally, due to the intensity similarity and low contrast between tumor and non-tumor regions in breast ultrasound images, obtaining discriminative features is difficult. So we propose an optimized U-Net breast ultrasound image segmentation model ECU-Net. Firstly, a multi-scale salient boundary detection module is proposed to enhance the model’s perception of tumor boundaries. Secondly, a contrast feature enhancement module is proposed to highlight the otherness between tumor and normal tissues. Then we combine the two modules to optimize skip-connection through boundaries completion and contrast information enhancement. Exhaustive experiments are conducted on two publicly available datasets, and experimental results show our ECU-Net performs better than U-Net and its classic variants.',\n",
       " 'Artificial intelligence (AI)-based approaches can now use electrocardiograms (ECGs) to provide expert-level performance in detecting heart abnormalities and diagnosing disease. Additionally, patient age predicted from ECGs by AI models has shown great potential as a biomarker for cardiovascular age, where recent work has found its deviation from chronological age (“delta age”) to be associated with mortality and co-morbidities. However, despite being crucial for understanding underlying individual risk, the genetic underpinning of delta age is unknown. In this work we performed a genome-wide association study using UK Biobank data (n=34,432) and identified eight loci associated with delta age ( $$p\\\\le 5 \\\\times 10^{-8}$$ p ≤ 5 × 10 - 8 ), including genes linked to cardiovascular disease (CVD) (e.g. SCN5A ) and (heart) muscle development (e.g. TTN ). Our results indicate that the genetic basis of cardiovascular ageing is predominantly determined by genes directly involved with the cardiovascular system rather than those connected to more general mechanisms of ageing. Our insights inform the epidemiology of CVD, with implications for preventative and precision medicine.',\n",
       " 'The use of bibliometrics, based on statistical and mathematical tools, makes it possible to measure the contributions of researchers to science. This is a widely used tool to assess scientific production in several areas of knowledge. Such methodology analyzes publication trends, author networks, structures of co-citation, journals and even the scientific contribution of renowned scholars in science. The precursor of bibliometrics, Eugene Garfield, who proposes the retrieval of information from the indexing of citations, was the object of a scientometric review aimed at assessing his impact on science. Given such relevance, this article presents the academic contribution of Jürgen Habermas based on a preliminary scientometric review of his studies. Jürgen Habermas is regarded not only as an active scholar in the social and political process, but also as a productive, controversial and influential contemporary author. The correct understanding of his works is a great challenge, as the bases of his thinking are so broad that they allow an interface between different approaches. We elaborated a design of his scientific work with the advancement to a connection between his main ideas through the use of bibliometric software. Bibliometrics, of the scientometric type, allows the understanding of how recognized patterns in citations can develop information relevant to the scientific field. The results confirm the multidisciplinary contribution of Habermas’s studies and highlight his main fields of research and works, which serve as a foundation for clearly understanding and applying his concepts.',\n",
       " 'The increasing energy demand of the burgeoning population in India has propelled the development of new thermal power plants in the country. Availability of plenty of power-grade coal and water in the form of Govind Ballabh Pant Sagar Dam led to the establishment of several thermal power plants (TPP) in the Singrauli district of Madhya Pradesh. After the construction and post-construction activities of these TPP in the district, the pristine dense forests have completely been transformed into degraded land. In the present work, change detection analysis is carried out for three-time gradients, i.e., from 2005 to 2015 at the interval of five years using Quantum GIS software to understand the land use dynamics. It is observed that the area has witnessed significant transformation during the three-time gradients, including approximately a\\xa06.9% decline in forest cover and a\\xa023% increase in the\\xa0barren surface. Further, artificial neural network (ANN) is employed in the present work to simulate and predict land use of the year 2025. The projected LULC for 2025 reveals that as the power plants operational activities will continue and expand further, there will be a substantial increase in fallow land and scrub land by the year 2025. The percentage of the\\xa0correctness of the\\xa0predicted LULC map is 98.21 with\\xa0an\\xa0 overall Kappa index of 0.97 and a\\xa0kappa histogram of 0.98. Details of the past, present and predicted LULC of an area aid the urban planners in informed decision making so as to choose, design, and implement future land use plans of the area in a sustainable manner.',\n",
       " 'Traditional Chinese Medicine (TCM) has been widely used in the treatment of various diseases for millennia. In the modernization process of TCM, TCM ingredient databases are playing more and more important roles. However, most of the existing TCM ingredient databases do not provide simplification function for extracting key ingredients in each herb or formula, which hinders the research on the mechanism of actions of the ingredients in TCM databases. The lack of quality control and standardization of the data in most of these existing databases is also a prominent disadvantage. Therefore, we developed a Traditional Chinese Medicine Simplified Integrated Database (TCMSID) with high storage, high quality and standardization. The database includes 499 herbs registered in the Chinese pharmacopeia with 20,015 ingredients, 3270 targets as well as corresponding detailed information. TCMSID is not only a database of herbal ingredients, but also a TCM simplification platform. Key ingredients from TCM herbs are available to be screened out and regarded as representatives to explore the mechanism of TCM herbs by implementing multi-tool target prediction and multilevel network construction. TCMSID provides abundant data sources and analysis platforms for TCM simplification and drug discovery, which is expected to promote modernization and internationalization of TCM and enhance its international status in the future. TCMSID is freely available at https://tcm.scbdd.com .',\n",
       " 'The COVID-19 pandemic has underscored the necessity for strategic approaches to manage global catastrophes. This study argues that disaster risk management (DRM) is an essential approach to mitigate the impact of global calamities, such as a pandemic. However, due to the uncertainty emerging from variables such as time and demand, managing DRM effectively involves high complexity. Blockchain technology (BcT) can be implemented to help address these challenges due to its potential to build trust, transparency, and accountability in complex operations. However, no research has quantitatively examined the applicability of BcT in the sub-dimensions of DRM to address the COVID-19 pandemic. The goal and scope of this study is to explore the role of BcT in DRM operations during the COVID-19 pandemic through an Intuitionistic Fuzzy Multi-Criteria Decision Making (IF-MCDM) framework. More specifically, the Intuitionistic Fuzzy Analytic Network Process (IF-ANP) method was utilized to calculate the weights of key criteria (i.e., BcT benefits), while Intuitionistic Fuzzy VIKOR (IF-VIKOR) was used to prioritize the alternatives (i.e., the sub-dimensions of DRM). The findings of this study are threefold. First, supporting effective coordination turned out to be the most essential benefit of BcT to build resilience in response to the COVID-19 pandemic. Second, disaster management was found to be the most appropriate DRM sub-dimension for possible BcT implementation during the COVID-19 pandemic. Lastly, eleven distinct activities involved in disaster management and governance and financial protection were discovered to be the most applicable for BcT. The findings of this study could assist disaster risk managers to assess whether (or not) BcT is suitable for the sub-dimensions of DRM to build national and organizational resilience in the wake of the COVID-19 pandemic.',\n",
       " 'Crim1 has been implicated in cataracts in mice and is of great importance in the development of the eye in both humans and mice. Therefore, we aimed to clarify how Crim1 mutations affect lens development and the molecular mechanism of cataracts in mice through comprehensive bioinformatics analysis. The microarray chip was downloaded from the GEO database to obtain the gene expression profile data set. Differentially expressed genes (DEGs) were screened using the limma package. GO and KEGG analyses of DEGs were performed using the DAVID database. Then, we established the protein–protein interaction (PPI) network in Cytoscape. Next, we used MCODE to analyze the data. We obtained 750 DEGs in total, including 407 upregulated DEGs and 343 downregulated DEGs. GO analysis showed that the DEGs were mainly related to biological processes, such as apoptosis, cell translation and the immune system. KEGG analysis showed that the enriched functions and pathways were related to the processing and presentation of ribosomes, lysosomes, and antigens. We identified 18 HUB genes, among which four core genes, C1qa, C1qb, C1qc, and Cd74, were closely related to congenital cataracts induced by Crim1 mutation. This study reveals the molecular pathogenesis of congenital cataracts induced by Crim1, and this information is expected to facilitate clinical genetic testing, molecular diagnosis, prognosis, and individualized chemotherapy for congenital cataracts (CC).',\n",
       " \"High-performance cloud computing has revolutionized computing. Resource provisioning has become an indispensable barrier in serving the high-end service facilitating computing paradigm. Energy efficient provisioning has become the major obstacle in the cloud as it downsizes the cost of the cloud system. This paper has proposed a technique to resolve the provisioning problem by scheduling the tasks to virtual machines (VMs) along with special consideration for energy usage. The technique aimed to elevate the potency of the scheduling algorithm in the real-time cloud application by curtailing down the system's energy consumption, makespan, and average start time. The technique has been inspired by the Artificial Neural Network (ANN), utilizing the Particle Swarm Optimization (PSO) algorithm to train the datasets. The technique has been implemented on the Cloudsim tool and compared with state of the art. The result obtained has shown that the technique has turned out to be performing better by improving the energy efficiency by 12%, makespan by an average of 20%, and the average start time by more than 50%.\",\n",
       " 'Nowadays, social media plays an essential role in spreading the news with low cost and high speed in publishing, and easy availability. Given that, anyone can publish any news on social networks, with some of them to be fake. These fake stories should be detected as soon as possible since they might have negative impacts on the society. To address this issue, most researches consider fake news detection as a binary classification problem. However, as some news are half-true, recently, multi-class detection has gained more attention. This paper investigates an early detection of fake news using multi-class classification. This is achieved by extracting the content features, such as sentiment and semantic features, from the news. The proposed model employs five classifiers (Random Forest, Support Vector Machine, Decision Tree, LightGBM, and XGBoost) as primary classifiers. Furthermore, AdaBoost is used for the meta-learning algorithm to develop a stacking generalization model. Stacking generalization is an ensemble learning method that uses all data produced by the first-level algorithms. We trained our model with PolitiFact data for the evaluation, and the model performance was evaluated by Accuracy, Precision, Recall, and F1 score. Excremental evaluation of the real-world datasets showed that our proposed model outperformed all previous works in both binary and multi-class classifications.',\n",
       " 'The degree of severity of a cybersecurity event of potentially malicious activity is crucial to determine an appropriate response. Machine learning techniques are used to obtain models to achieve a correct characterization of events as well as an accurate maliciousness measure. A model is constructed by setting up an effective collection of features that are able to characterize the maliciousness of IP addresses. As a first approach, a set of contextual features has been selected. These features are simple to extract and require no high costs. That set contains 23 features: 18 obtained through time series analysis, and the other 5 extracted directly related with the spatial (geolocation) and temporal (time of occurrence) similarity of the events. The test of the feature set is conducted with statistical analyses to determine which features are the most effective and the impact of hyperparameter selection in the construction of a time series of features. In addition, the effect of IP geo-localization shifts on model performance is studied through concept drift theory. The results conclude that, overall, the feature set provides adequate performance for our task, although more complex features may be required to achieve better performance. The best results are obtained with the geolocation and time occurrence features.',\n",
       " 'Diminished ovarian reserve (DOR) is an etiologically heterogeneous disorder that usually leads to poor reproductive outcomes. Does a specific or common pathogenesis exist for DOR subtypes with different etiologies? Two frequently used mouse models, age-related DOR (AR-DOR) and cyclophosphamide (CTX)-induced DOR (CTX-DOR), were successfully established, and RNA sequencing was performed on ovarian tissue samples. Differentially expressed genes (DEGs) in each subtype and common DEGs (co-DEGs) in the two subtypes were identified. Subsequently, we performed comprehensive bioinformatics analyses, including an evaluation of immune cell infiltration. Finally, the genes of interest were further validated by performing RT-qPCR and immunohistochemistry. In AR-DOR mice, functional enrichment analyses showed that upregulated DEGs were mainly involved in the inflammatory/immune response, while downregulated DEGs were involved in DNA damage repair. In CTX-DOR mice, the inflammatory/immune response and cell apoptosis played significant roles. Meanwhile, 406 co-DEGs were identified from the two models. The biological functions of these co-DEGs were associated with inflammatory/immune responses. The analysis of immune cell infiltration showed reduced infiltration of Treg cells, as well as increased infiltration of M0 macrophages, NK resting, and T cells CD4 follicular in both DOR subtypes. The results of the validation experiments were consistent with the RNA sequencing data. In conclusion, the inflammatory/immune response might be the common pathogenesis for the two DOR subtypes, while DNA repair and cell apoptosis may have different roles in the two subtypes. These results may provide potential insights for mechanistic research and therapeutic targets of DOR.',\n",
       " 'Mg alloys are considered a promising alternative to use as biodegradable implants. However, the main limitations of these alloys are the high susceptibility to hydrogen evolution and corrosion. Consequently, it is necessary to control the corrosion rate through surface treatment techniques. In this regard, a double layer system, based on the first deposition of a MgO coating and the subsequent deposition of an organo-silane sol-gel coating (anodized/SGMI) was proposed to increase the biocompatibility and reduce the corrosion rate of Mg alloy. Sol-gel coatings were prepared using methyltriethoxysilane (MTES), (glycidyloxypropyl) trimethoxysilane (GPTMS), SiO 2 nanoparticles, and 1-methyl imidazole (1-MI). 29 Si and 13 C NMR measurements were performed to investigate the inorganic-organic network formation during the sol-gel synthesis, and to characterize the structure of the cured silane coating. According to the results, the addition of 1-MI mainly promotes the condensation reaction of GPTMS and MTES. After sol-gel deposition, a higher cross-linking coating network was obtained and the degradation of AZ31B Mg alloy in simulated body fluid (SBF) was slowed down. The control of the hydrogen and pH evolution over 450\\u2009h suggest that the anodized/SGMI coating system may be suitable for short-term implants applications; particularly for applications that are expected to degrade after 10-12 days such as surgical skin staples. Graphical Abstract To control the corrosion of AZ31B Mg alloy in simulated body fluid (SBF) using a double layer system based on anodized and sol-gel coating. The corrosion rate of AZ31B Mg alloy in simulated body fluid (SBF) solution was controlled by using anodizing and sol-gel technique. A highly cross-linked silica network was successfully prepared using methyltriethoxysilane (MTES), (glycidyloxypropyl) trimethoxysilane (GPTMS), SiO 2 nanoparticles, and 1-methyl imidazole (1-MI). The crack-free epoxy-alkyl silane coating deposited on the top of the anodized substrates allows the evolution of hydrogen and pH to be controlled. The hydrophilic nature and in vitro results of the coated AZ31B Mg alloy substrates suggest that the proposed coating system is suitable for short-term implant applications.',\n",
       " 'In the current business scenario, real-time analysis of enterprise data through Business Intelligence (BI) is crucial for supporting operational activities and taking any strategic decision. The automated ETL (extraction, transformation, and load) process ensures data ingestion into the data warehouse in near real-time, and insights are generated through the BI process based on real-time data. In this paper, we have concentrated on automated credit risk assessment in the financial domain based on the machine learning approach. The machine learning-based classification techniques can furnish a self-regulating process to categorize data. Establishing an automated credit decision-making system helps the lending institution to manage the risks, increase operational efficiency and comply with regulators. In this paper, an empirical approach is taken for credit risk assessment using logistic regression and neural network classification method in compliance with Basel II standards. Here, Basel II standards are adopted to calculate the expected loss. The required data integration for building machine learning models is done through an automated ETL process. We have concluded this research work by evaluating this new methodology for credit risk assessment.',\n",
       " 'Research has shown that several variables affect language control among bilingual speakers but the effect of affective processing remains unexplored. Chinese–English bilinguals participated in a novel prime-target language switching experiment in which they first judged the affective valence (i.e., positive or negative) of auditorily presented words and then named pictures with neutral emotional valence in either the same (non-switch trial) or different language (switch trial). Brain activity was monitored using functional magnetic resonance imaging (fMRI). The behavioral performance showed that the typical switch cost (i.e., the calculated difference between switch and non-switch trials) emerged after processing positive words but not after negative words. Brain imaging demonstrated that processing negative words immediately before non-switch picturing naming trials (but not for switch trials) increased activation in brain areas associated with domain-general cognitive control. The opposite patterns were found after processing positive words. These findings suggest that an (emotional) negative priming effect is induced by spontaneous exposure to negative words and that these priming effects may be triggered by reactive emotional processing and that they may interact with higher level cognitive functions.',\n",
       " 'Carbon fiber-reinforced polymer composites have permeated every aspect of modern life. Significant progress has been made in engineering communities to recycle composite wastes. While emerging recycling techniques can recover both fibers and resins at mild processing conditions, they are currently limited on the laboratory scale. It remains unclear if they could lead to market viability and environmental sustainability when they are used for industrial applications; if not, what the limiting factors are and how to improve them. In this study, the environmental and economic impacts of existing recycling techniques are studied. The average recycling costs are analyzed, and the environmental impacts are examined. It reveals that while the emerging techniques can recycle composites with extremely low environmental impact, their costs are significant compared to conventional techniques, primarily due to the expensive solvents and catalysts used to enable recycling. However, the study shows that it is possible to combine conventional waste processing techniques with emerging ones to enable recycling scenarios that are both cost-effective and environmentally beneficial. Overall, the study in this paper provides valuable insights for the design of future waste management supply chains and appropriate business models for the sustainable development of society. Graphical abstract ',\n",
       " 'Purpose The human body is a complex structure. Its strength is ensured by the collagen protein which exists under the form of fibers. The quantitative analysis of these fibers in biological tissues can be very interesting to establish a relationship between the microstructure and their functions. This analysis is usually performed using two-photon microscopy and second harmonic generated (SHG) images. Lately, more and more researchers focused on the use of SHG images since it is a non-invasive technique and allows the capture of collagen fibers only. Many image-processing techniques can be used to extract quantitative information from those images such as fiber orientations, dimensions, and density. Therefore, accurate measure extraction depends mainly on the used image processing methods and, thus, it is necessary to know what processing technique to use. Methods The main purpose of this article is to exhibit the most used techniques in collagen fiber quantitative analysis then categorize them according to the information to extract. A comparison of three most used methods in fiber orientation’s estimation is carried out. Result and conclusion Despite the considerable number of papers aiming to quantitatively analyze collagen fibers from SHG images, two main aspects were not deeply covered. First, the use of deep learning algorithms is still limited even for segmentation and denoizing applications. Second, most of the studies processed in this review focused on two-dimensional SHG images and did not take into consideration collagen fibers as a three-dimensional volume.',\n",
       " 'Background and purpose To predict treatment-related cardiovascular disease (CVD) and second cancer 30-year absolute mortality risks (AMR 30 ) for patients with mediastinal Hodgkin lymphoma in a large multicentre radiation oncology network in Ireland. Material and methods This study includes consecutive patients treated for mediastinal lymphoma using chemotherapy and involved site radiotherapy (RT) 2016–2019. Radiation doses to heart, left ventricle, cardiac valves, lungs, oesophagus, carotid arteries and female breasts were calculated. Individual CVD and second cancer AMR 30 were predicted using Irish background population rates and dose–response relationships. Results Forty-four patients with Hodgkin lymphoma were identified, 23 females, median age 28 years. Ninety-eight percent received anthracycline, 80% received 4–6 cycles ABVD. Volumetric modulated arc therapy (VMAT)\\u2009±\\u2009deep inspiration breath hold (DIBH) was delivered, median total prescribed dose 30\\xa0Gy. Average mean heart dose 9.8\\xa0Gy (range 0.2–23.8\\xa0Gy). Excess treatment-related mean AMR 30 from CVD was 2.18% (0.79, 0.90, 0.01, 0.13 and 0.35% for coronary disease, heart failure, valvular disease, stroke and other cardiac diseases), 1.07% due to chemotherapy and a further 1.11% from RT. Excess mean AMR 30 for second cancers following RT were: lung cancer 2.20%, breast cancer in females 0.34%, and oesophageal cancer 0.28%. Conclusion For patients with mediastinal lymphoma excess mortality risks from CVD and second cancers remain clinically significant despite contemporary chemotherapy and photon-RT. Efforts to reduce the toxicity of combined modality treatment, for example, using DIBH, reduced margins and advanced RT, e.g. proton beam therapy, should be continued to further reduce potentially fatal treatment effects.',\n",
       " 'Periodic inspection, commonly performed by a technician, of weld seam quality is important for assessing equipment reliability. To save labor costs and improve efficiency, an autonomous navigation and inspection robot is developed. The development process involves the design of chassis damping, target detection mechanism, control system, and algorithms. For performing weld inspection in complex, outdoor, environments, an algorithm is developed for the robot to avoid any obstacles. This algorithm for planning the inspection route is based on an improved timed-elastic-band (TEB) algorithm. The developed robot is capable of conducting inspection tasks in complex and dangerous environments efficiently and autonomously.',\n",
       " 'Age-related macular degeneration (AMD) is the most widespread cause of blindness and the identification of baseline AMD features or biomarkers is critical for early intervention. Optical coherence tomography (OCT) imaging produces a 3D volume consisting of cross sections of retinal tissue while fundus fluorescence (FAF) imaging produces a 2D mapping of retina. FAF has been a good standard for assessing dry AMD late-stage geographic atrophy (GA) while OCT has been used for assessing early AMD biomarkers beyond as well. However, previous approaches in large extent defined AMD features subjectively based on clinicians’ observation. Deep learning—an objective artificial intelligence approach, may enable to discover ’true’ salient AMD features. We develop a novel reverse engineering approach which bases on the backbone of a fully convolutional neural network to objectively identify and visualize AMD early biomarkers in OCT from baseline exams before significant atrophy occurs. Utilizing manually annotated GA regions on FAF from a follow-up visit as ground truth, we segment GA regions and reconstruct early AMD features in baseline OCT volumes. In this preliminary exploration, compared with ground truth, we achieve baseline GA segmentation accuracy of 0.95 and overlapping ratio of 0.65. The reconstructions consistently highlight that large druse and druse clusters with or without mixed hyper-reflective focus lesion on baseline OCT cause the conversion of GA after 12\\xa0months. However, hyper-reflective focus lesions and subretinal drusenoid deposit lesions alone are not seen such conversion after 12\\xa0months. Further research with larger dataset would be needed to verify these findings.',\n",
       " 'The total number of published articles and the resulting citations are generally acknowledged as suitable criteria of the scientist’s evaluation. However, it is challenging to determine the ranking of scientists as the value of their scientific work (at times) is not directly reflective of the abovementioned aspects. In this regard, multiple other elements needs to be examined in combination for better evaluating the scientific worth of an individual. This work presents a learning-based technique, i.e., an Artificial Intelligence (AI)-based solution towards categorizing scientists utilizing a multifaceted criteria. In this context, a novel ranking metric is proposed which is grounded on authorship, experience, publications count, total citations, i10-index, and h -index. To assess the proposed framework’s performance, a dataset is collected considering the world’s top ten computing departments and ten domestic ones. This results in a data of 1000 computer scientists. The dataset is preprocessed and afterwards three techniques for feature selection are employed, i.e., Mutual Information (MI), Chi-Square ( X 2 ), and Fisher-Test (F-Test) to rank the features in the data. To validate the collected data, the framework has three clustering techniques as well, namely, k -medoids, k -means, and spectral clustering to identify the optimum number of heterogeneous groups. Three cluster validity indices are used to evaluate the clustering outcomes, namely, Calinski-Harabasz Index (CHI), Davies Bouldin Index (DBI), and Silhouette Coefficient (SC). Once the optimum clusters are obtained, five classification procedures are used, including, Artificial Neural Network (ANN), k -Nearest Neighbor ( k -NN), Decision Tree (DT), Gaussian Naive Bayes (GNB), and Linear Regression Classifier (LRC) to predict the category of a previously unknown scientist. Among all classifiers, an average accuracy of 94.44% is shown by the ANN to predict an unknown/new scientist category. The current proposal is also compared with closely related past works. The proposed framework offers the possibility to independently classify scientists based on AI techniques.',\n",
       " 'Two-dimensional metal carbide or nitride materials (MXenes) are widely used in electromagnetic wave absorption because of their unique structure. Herein, a novel composite preparation strategy has been proposed to design dendritic nanofibers based on the electrostatic spinning methods. The multifunctional MXene nanosheets are used as the dendritic matrix, and magnetic nanoparticles are embedded in the nanosheets as magnetic loss units. Multidimensional nanocomposites have interlaced carbon fiber networks, large-scale magnetically coupled networks, and a lot of multi-heterojunction interface structures, which endow the composites with extraordinary conduction loss, magnetic loss, and polarization loss capabilities, respectively. The impedance matching and loss mechanisms of the composites are improved by optimizing the synergistic relationship between the components and building a suitable structure. The optimum reflection loss (RL) of −54.1 dB is achieved at 2.7 mm and a wide effective absorption bandwidth (EAB, RL below −10 dB) of 7.76 GHz is obtained at a small thickness of 2.1 mm for the nanocomposites. The distinctive microstructures of the nanofibrous membranes give rise to their flexibility, waterproof, and electromagnetic wave absorption performance and endow the nanofibrous membranes potential to be utilized as lightweight, efficient electromagnetic wave protective fabric in harsh environment.',\n",
       " 'Image stitching aims to stitch multiple images with overlapping areas into a high-resolution image with natural appearance, no ghosting and no seam, so as to obtain a wide field of vision. And the stitching is required to be as fast as possible. In this article, we discuss the loss of foreground object caused by the seam-cut method for image stitching. Moreover, we propose an improved seam-cut method based on superpixel to solve this problem. The proposed method uses the matching information of feature points and superpixel segmentation to divide the scene into foreground and background. By adjusting the penalty value of foreground superpixels in the superpixel-based seam-cut method, the foreground object can avoid the cutting of optimal seam. The experimental results demonstrate that our method will not cause the ghosting of foreground object or the loss of foreground object in the stitching result in the scene with large parallax. In this method, the foreground object will be preserved completely.',\n",
       " 'Background Non-modifiable risk factors of Alzheimer’s disease (AD) have lifelong effects on cortical integrity that could be mitigated if identified at early stages. However, it remains unknown whether cortical microstructure is affected in older individuals with non-modifiable AD risk factors and whether altered cortical tissue integrity produces abnormalities in brain functional networks in this AD-risk population. Methods Using relative T1w/T2w (rT1w/T2w) ratio maps, we have compared tissue integrity of normal-appearing cortical GM between controls and cognitively normal older adults with either APOE4 ( N = 50), with a first-degree family history (FH) of AD ( N = 52), or with the co-occurrence of both AD risk factors (APOE4+FH) ( N = 35). Additionally, individuals with only one risk factor (APOE4 or FH) were combined into one group ( N = 102) and compared with controls. The same number of controls matched in age, sex, and years of education was employed for each of these comparisons. Group differences in resting state functional connectivity (rs-FC) patterns were also investigated, using as FC seeds those cortical regions showing significant changes in rT1w/T2w ratios. Results Overall, individuals with non-modifiable AD risk factors exhibited significant variations in rT1w/T2w ratios compared to controls, being APOE4 and APOE4+FH at opposite ends of a continuum. The co-occurrence of APOE4 and FH was further accompanied by altered patterns of rs-FC. Conclusions These findings may have practical implications for early detection of cortical abnormalities in older populations with APOE4 and/or FH\\xa0of AD and open new avenues to monitor changes in cortical tissue integrity associated with non-modifiable AD risk factors.',\n",
       " 'In this paper, advanced modulation techniques based on external modulation, such as I-AMI, AMI, and DBRZ, are designed to improve long-haul transmission and lower high power requirements in the FSO communication system. For long-haul 16\\u2009×\\u2009100\\xa0Gbps FSO systems, an alternate mark inversion modulation approach with high-data rate distortion less transmission is possible. The results demonstrate improved performance in terms of performance indicators such as Q-factor (29.62) and BER (3.8e −193 ), as well as improved data transmission capabilities of up to 100\\xa0Gbps over a distance of 9\\xa0km in clear weather. Because here, AMI technique uses a low input power of 5\\xa0dBm instead of a conventional power of 10\\xa0dBm, the achieved findings demonstrate a reasonable reduction in signal degradation, especially nonlinear effects.']"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abstract_list_springer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "3592ad19",
   "metadata": {},
   "outputs": [],
   "source": [
    "class Springer_Article():\n",
    "    def __init__(self, records={}):#input each result\n",
    "        self.records= records\n",
    "        self.title = records['title'].encode('utf-8')\n",
    "        self.journal = records['publicationName'].encode('utf-8')\n",
    "        self.date = records['publicationDate'].encode('utf-8')\n",
    "        self.citesnumber = 'N/A'\n",
    "        self.abstract = records['abstract'].encode('utf-8')[8:]\n",
    "\n",
    "    def author(self):\n",
    "        for data in self.records:\n",
    "            if 'creators' in data:\n",
    "                #name_lst = data['creators']\n",
    "                #combine_name = ', '.join(name_lst)\n",
    "                #return combine_name\n",
    "                return data\n",
    "\n",
    "    def significant(self):\n",
    "        sentences_lst = (self.abstract).split('.')\n",
    "        i = 0\n",
    "        for s in sentences_lst :\n",
    "            if 'significan' in s:\n",
    "                return s\n",
    "                i +=1\n",
    "            elif 'associat' in s:\n",
    "                return s\n",
    "                i +=1\n",
    "        if i == 0:\n",
    "            return 'None'\n",
    "\n",
    "    def conclusion(self):\n",
    "        sentences_lst = (self.abstract).split('.')\n",
    "        i = 0\n",
    "        for s in sentences_lst :\n",
    "            if 'conclu' in s:\n",
    "                return s\n",
    "                i +=1\n",
    "        if i == 0:\n",
    "            return 'None'\n",
    "\n",
    "    def suggest(self):\n",
    "        sentences_lst = (self.abstract).split('.')\n",
    "        i = 0\n",
    "        for s in sentences_lst :\n",
    "            if 'suggest' in s:\n",
    "                return s\n",
    "                i +=1\n",
    "        if i == 0:\n",
    "            return 'None'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d165dc12",
   "metadata": {},
   "outputs": [],
   "source": [
    "article_insts2 = [Springer_Article(records) for records in d_springer['records']]\n",
    "titlelst2 = [i.title for i in article_insts2]\n",
    "\n",
    "#springer and scopus (remove duplicate between them)\n",
    "for i in article_insts2:\n",
    "\tif i.title in titlelst2:\n",
    "\t\tarticle_insts2.remove(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "37862798",
   "metadata": {},
   "outputs": [],
   "source": [
    "#springer article id \n",
    "article_insts2 = [Springer_Article(records) for records in d_springer['records']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "2ed8d3ca",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "authorlst2 = [i.author() for i in article_insts2]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "a54ca646",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[b'Temporal networks in biology and medicine: a survey on models, algorithms, and tools',\n",
       " b'ECR 2022 Book of Abstracts',\n",
       " b'Speedy and Efficient Service Chaining and Function Placement Based on Lagrangian Heuristics for Capacitated Shortest Path Tour Problem',\n",
       " b'A Game Theoretic Framework for Distributed Mission Slice Allocation and Management for Tactical Networks',\n",
       " b'Densely Connected Dilated Residual Network for Image Denoising: DDR-Net',\n",
       " b'Transaction Latency Within Permissionless Blockchains: Analysis, Improvement, and Security Considerations',\n",
       " b'On state estimation for nonlinear systems under random access wireless protocols',\n",
       " b'RM3D: Robust Data-Efficient 3D Scene Parsing via Traditional and Learnt 3D Descriptors-Based Semantic Region Merging',\n",
       " b'A fault diagnosis network based on domain adversarial learning and distribution matching for rotating machine vibration signal with noise and across-load conditions',\n",
       " b'Contextual transformer sequence-based recognition network for medical examination reports',\n",
       " b'Identifying vital nodes for influence maximization in attributed networks',\n",
       " b'SHARP: Shape-Aware Reconstruction of People in Loose Clothing',\n",
       " b'Light Weight Gradient Ensemble Model for detecting network attack at the edge of the IoT network',\n",
       " b'Performance Comparison of Recent Nature-Inspired Optimization Algorithms in Optimal Placement and Sizing of Distributed Energy Resources',\n",
       " b'Adaptive Style Modulation for Artistic Style Transfer',\n",
       " b'Intelligent modeling of photocatalytically reactive yellow 84 azo dye removal from aqueous solutions by ZnO-light expanded clay aggregate nanoparticles',\n",
       " b'Inferring turbulent environments via machine learning',\n",
       " b'An Integrated Deep Learning Prediction Approach for Efficient Modelling of Host Load Patterns in Cloud Computing',\n",
       " b'Entanglement and the Path Integral',\n",
       " b'Artificial Neural Network Assisted P &O-Based MPPT Controller for a Partially Shaded Grid-Connected Solar PV Panel',\n",
       " b'A pathway analysis-based algorithm for calculating the participation degree of ncRNA in transcriptome',\n",
       " b'Assessment of Trees Outside Forest (TOF) in Urban Landscape Using High-Resolution Satellite Images and Deep Learning Techniques',\n",
       " b'Combining channel-wise joint attention and temporal attention in graph convolutional networks for skeleton-based action recognition',\n",
       " b'Interpretable generalized neural additive models for mortality prediction of COVID-19 hospitalized patients in Hamadan, Iran',\n",
       " b'Disease detection and physical disorders classification for citrus fruit images using convolutional neural network',\n",
       " b'Helopeltis theivora\\xc2\\xa0Responsive Transcriptomic Reprogramming Uncovers Long Non-coding RNAs as Possible Regulators of Primary and Secondary Metabolism in Tea Plant',\n",
       " b'ECU-Net: multi-scale salient boundary detection and contrast feature enhancement U-Net for breast ultrasound image segmentation',\n",
       " b'Deep learning-derived cardiovascular age shares a genetic basis with other cardiac phenotypes',\n",
       " b'The impact of J\\xc3\\xbcrgen Habermas\\xe2\\x80\\x99s scientific production: a scientometric review',\n",
       " b'Predicting Spatial and Decadal LULC Changes in the Singrauli District of Madhya Pradesh Through Artificial Neural Network Models Using Geospatial Technology',\n",
       " b'TCMSID: a simplified integrated database for drug discovery from traditional chinese medicine',\n",
       " b'Leveraging blockchain in response to a pandemic through disaster risk management: an IF-MCDM framework',\n",
       " b'Integrative Bioinformatics Analysis of mRNA Expression Profiles of Mice to Explore the Key Genes Involved in Crim1 Mutation-Induced Congenital Cataracts',\n",
       " b'An energy efficient robust resource provisioning based on improved PSO-ANN',\n",
       " b'Early multi-class ensemble-based fake news detection using content features',\n",
       " b'A mathematical analysis about the geo-temporal characterization of the multi-class maliciousness of an IP address',\n",
       " b'Identification of Common and Specific Genes Involved in Mouse Models of Age-Related and Cyclophosphamide-Induced Diminished Ovarian Reserve',\n",
       " b'In-vitro degradation behavior of hybrid epoxy-alkyl sol\\xe2\\x80\\x93gel/anodized composite coating on AZ31B Mg alloy',\n",
       " b'Automated credit assessment framework using ETL process and machine learning',\n",
       " b'How processing emotion affects language control in bilinguals',\n",
       " b'Economic and environmental analysis of composite recycling techniques for the design of future waste management supply chains',\n",
       " b'Quantitative analysis of second harmonic generated images of collagen fibers: a review',\n",
       " b'Predicted cardiac and second cancer risks for patients undergoing VMAT for mediastinal Hodgkin lymphoma',\n",
       " b'Design and analysis of welding inspection robot',\n",
       " b'Reverse engineering for reconstructing baseline features of dry age-related macular degeneration in optical coherence tomography',\n",
       " b'An artificial intelligence-based framework for data-driven categorization of computer scientists: a case study of world\\xe2\\x80\\x99s Top 10 computing departments',\n",
       " b'Electrospun Fe0.64Ni0.36/MXene/CNFs nanofibrous membranes with multicomponent heterostructures as flexible electromagnetic wave absorbers',\n",
       " b'Superpixel-based foreground-preserving image stitching',\n",
       " b'Effects of non-modifiable risk factors of Alzheimer\\xe2\\x80\\x99s disease on intracortical myelin content',\n",
       " b'Design of 16\\xe2\\x80\\x89\\xc3\\x97\\xe2\\x80\\x89100\\xc2\\xa0Gbps free-space optical system using advanced modulation techniques']"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "titlelst2 = [i.title for i in article_insts2]\n",
    "titlelst2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "aa3c9080",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[b'Network Modeling Analysis in Health Informatics and Bioinformatics',\n",
       " b'Insights into Imaging',\n",
       " b'Journal of Network and Systems Management',\n",
       " b'Journal of Network and Systems Management',\n",
       " b'Neural Processing Letters',\n",
       " b'Journal of Network and Systems Management',\n",
       " b'Mathematics of Control, Signals, and Systems',\n",
       " b'International Journal of Computer Vision',\n",
       " b'Journal of the Brazilian Society of Mechanical Sciences and Engineering',\n",
       " b'Applied Intelligence',\n",
       " b'Scientific Reports',\n",
       " b'International Journal of Computer Vision',\n",
       " b'International Journal of Information Technology',\n",
       " b'Process Integration and Optimization for Sustainability',\n",
       " b'Neural Processing Letters',\n",
       " b'International Journal of Environmental Science and Technology',\n",
       " b'The European Physical Journal E',\n",
       " b'Journal of Grid Computing',\n",
       " b'Foundations of Physics',\n",
       " b'Arabian Journal for Science and Engineering',\n",
       " b'Scientific Reports',\n",
       " b'Journal of the Indian Society of Remote Sensing',\n",
       " b'Signal, Image and Video Processing',\n",
       " b'BMC Medical Research Methodology',\n",
       " b'Journal of Food Measurement and Characterization',\n",
       " b'Journal of Plant Growth Regulation',\n",
       " b'Signal, Image and Video Processing',\n",
       " b'Scientific Reports',\n",
       " b'Scientometrics',\n",
       " b'Journal of the Indian Society of Remote Sensing',\n",
       " b'Journal of Cheminformatics',\n",
       " b'Operations Management Research',\n",
       " b'Biochemical Genetics',\n",
       " b'International Journal of Information Technology',\n",
       " b'Social Network Analysis and Mining',\n",
       " b'Wireless Networks',\n",
       " b'Reproductive Sciences',\n",
       " b'Journal of Sol-Gel Science and Technology',\n",
       " b'Innovations in Systems and Software Engineering',\n",
       " b'Brain Structure and Function',\n",
       " b'Clean Technologies and Environmental Policy',\n",
       " b'Research on Biomedical Engineering',\n",
       " b'Clinical and Translational Oncology',\n",
       " b'Scientific Reports',\n",
       " b'Scientific Reports',\n",
       " b'Scientometrics',\n",
       " b'Nano Research',\n",
       " b'Machine Vision and Applications',\n",
       " b\"Alzheimer's Research & Therapy\",\n",
       " b'Journal of Optics']"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "journallst2  = [i.journal for i in article_insts2]\n",
    "journallst2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "38fcb6f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['apiMessage', 'query', 'apiKey', 'nextPage', 'result', 'records', 'facets'])"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_springer.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "fc814da7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s13721-022-00406-x',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s13721-022-00406-x'}],\n",
       "  'title': 'Temporal networks in biology and medicine: a survey on models, algorithms, and tools',\n",
       "  'creators': [{'creator': 'Hosseinzadeh, Mohammad Mehdi'},\n",
       "   {'creator': 'Cannataro, Mario'},\n",
       "   {'ORCID': '0000-0001-5542-2997', 'creator': 'Guzzi, Pietro Hiram'},\n",
       "   {'creator': 'Dondi, Riccardo'}],\n",
       "  'publicationName': 'Network Modeling Analysis in Health Informatics and Bioinformatics',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s13721-022-00406-x',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2192-6670',\n",
       "  'volume': '12',\n",
       "  'number': '1',\n",
       "  'genre': ['ReviewPaper', 'Review Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '22',\n",
       "  'journalId': '13721',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer-Verlag GmbH Austria, part of Springer Nature',\n",
       "  'abstract': 'The use of static graphs for modelling and analysis of biological and biomedical data plays a key role in biomedical research. However, many real-world scenarios present dynamic behaviours resulting in both node and edges modification as well as feature evolution. Consequently, ad-hoc models for capturing these evolutions along the time have been introduced, also referred to as dynamic, temporal, time-varying graphs. Here, we focus on temporal graphs, i.e., graphs whose evolution is represented by a sequence of time-ordered snapshots. Each snapshot represents a graph active in a particular timestamp. We survey temporal graph models and related algorithms, presenting fundamentals aspects and the recent advances. We formally define temporal graphs, focusing on the problem setting and we present their main applications in biology and medicine. We also present temporal graph embedding and the application to recent problems such as epidemic modelling. Finally, we further state some promising research directions in the area. Main results of this study include a systematic review of fundamental temporal network problems and their algorithmic solutions considered in the literature, in particular those having application in computational biology and medicine. We also include the main software developed in this context.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Health Informatics',\n",
       "   'Applications of Graph Theory and Complex Networks',\n",
       "   'Bioinformatics',\n",
       "   'Computational Biology/Bioinformatics']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1186/s13244-022-01337-x',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1186/s13244-022-01337-x'}],\n",
       "  'title': 'ECR 2022 Book of Abstracts',\n",
       "  'publicationName': 'Insights into Imaging',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1186/s13244-022-01337-x',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1869-4101',\n",
       "  'volume': '13',\n",
       "  'number': '4',\n",
       "  'genre': ['Abstract', 'Meeting abstracts'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '454',\n",
       "  'journalId': '13244',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': '',\n",
       "  'subjects': ['Medicine & Public Health',\n",
       "   'Imaging / Radiology',\n",
       "   'Diagnostic Radiology',\n",
       "   'Interventional Radiology',\n",
       "   'Neuroradiology',\n",
       "   'Ultrasound',\n",
       "   'Internal Medicine']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s10922-022-09715-y',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s10922-022-09715-y'}],\n",
       "  'title': 'Speedy and Efficient Service Chaining and Function Placement Based on Lagrangian Heuristics for Capacitated Shortest Path Tour Problem',\n",
       "  'creators': [{'creator': 'Hara, Takanori'}, {'creator': 'Sasabe, Masahiro'}],\n",
       "  'publicationName': 'Journal of Network and Systems Management',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s10922-022-09715-y',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-7705',\n",
       "  'volume': '31',\n",
       "  'number': '1',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '34',\n",
       "  'journalId': '10922',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'Network functions virtualization (NFV) can realize flexible and diverse network services by replacing the conventional network equipment with the combination of virtual network functions (VNFs) and commodity servers. A certain network service can be composed of a sequence of VNFs, i.e., service (function) chain. The service chaining (SC) problem aims to establish an appropriate service path from the origin node to the destination node, which holds both the resource constraints and service chain requirements of executing the required VNFs in the designated order. SC belongs to the complexity class NP -hard. In the previous work, inspired by the similarity between the SC problem and the shortest path tour problem (SPTP), we showed the capacitated SPTP (CSPTP) based ILP for the SC problem, where CSPTP is a generalized version of the SPTP with both the node and link capacity constraints. In this paper, we propose Lagrangian heuristics to solve the CSPTP-based ILP for the SC in a speedy and efficient manner. We further present that the proposed heuristics can also solve both the service chaining and function placement by slightly extending the network model called an augmented network. Through numerical results, we show that the proposed heuristics for the SC is competitive with the optimal resource allocation while executing much faster than the combination of the CSPTP-based ILP and the existing solver, i.e., CPLEX. Furthermore, we also show that the proposed heuristics for both the service chaining and function placement can still balance the solution optimality and computational complexity, thanks to the parallel computation architectures.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Computer Communication Networks',\n",
       "   'Computer Systems Organization and Communication Networks',\n",
       "   'Communications Engineering, Networks',\n",
       "   'Information Systems and Communication Service',\n",
       "   'Operations Research/Decision Theory']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s10922-022-09709-w',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s10922-022-09709-w'}],\n",
       "  'title': 'A Game Theoretic Framework for Distributed Mission Slice Allocation and Management for Tactical Networks',\n",
       "  'creators': [{'creator': 'Yang, Fan'},\n",
       "   {'creator': 'Comaniciu, Cristina'},\n",
       "   {'creator': 'Krishnan, P.'},\n",
       "   {'creator': 'Sabnani, Krishan'},\n",
       "   {'creator': 'Greenwald, Lloyd'}],\n",
       "  'publicationName': 'Journal of Network and Systems Management',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s10922-022-09709-w',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-7705',\n",
       "  'volume': '31',\n",
       "  'number': '1',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '28',\n",
       "  'journalId': '10922',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'In this paper, we address the problem of implementing distributed slice allocation algorithms for tactical networks deployed in regions lacking a communication infrastructure. We propose a multi-agent game implementation in which Application Slice Agents ( ASAs ) compete for shared computational and bandwidth resources in the network. We prove that the game can be implemented as a two stage potential game, with the first stage ensuring the selection of computational servers with load balancing, and the second stage improving connectivity and network throughput by promoting inter-slice cooperation. Our proposed analysis framework also incorporates a theoretical model to capture the impact of mitigation methods for timing channel leaks for the shared resources for different slices, and proposes a theoretical framework that allows for a security-delay tradeoff analysis.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Computer Communication Networks',\n",
       "   'Computer Systems Organization and Communication Networks',\n",
       "   'Communications Engineering, Networks',\n",
       "   'Information Systems and Communication Service',\n",
       "   'Operations Research/Decision Theory']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11063-022-11100-4',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11063-022-11100-4'}],\n",
       "  'title': 'Densely Connected Dilated Residual Network for Image Denoising: DDR-Net',\n",
       "  'creators': [{'ORCID': '0000-0002-8321-2070', 'creator': 'Acar, Vedat'},\n",
       "   {'creator': 'Eksioglu, Ender M.'}],\n",
       "  'publicationName': 'Neural Processing Letters',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11063-022-11100-4',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-773X',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '15',\n",
       "  'journalId': '11063',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'Image denoising is an important ill-posed problem of image processing. The main goal in image denoising is to suppress noise while protecting textures of the image. A plethora of methods and different approaches have targeted image denoising problem. In this manuscript, we propose a novel method which brings dense and residual blocks together with dilated convolutions in image denoising applications. The proposed method introduces new strategies for proper combination of dense residual blocks and dilated convolution layers. The resulting approach is called as Densely connected Dilated Residual Network (DDR-Net). The proposed DDR-Net extracts multi-scale information by employing dilated convolutions. Use of dilated convolutions leads to improved receptive field performance while keeping the complexity of the network in check. Residual and dense connections on the other hand prevent loss of information along the DDR-Net pipeline. Simulations are performed in both color and grayscale image denoising settings. Quantitative and qualitative results indicate that the proposed network architecture leads to improved results when compared to multiple high performance convolutional denoising networks from the recent literature.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Artificial Intelligence',\n",
       "   'Complex Systems',\n",
       "   'Computational Intelligence']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s10922-022-09717-w',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s10922-022-09717-w'}],\n",
       "  'title': 'Transaction Latency Within Permissionless Blockchains: Analysis, Improvement, and Security Considerations',\n",
       "  'creators': [{'creator': 'Alharby, Maher'}],\n",
       "  'publicationName': 'Journal of Network and Systems Management',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s10922-022-09717-w',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-7705',\n",
       "  'volume': '31',\n",
       "  'number': '1',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '25',\n",
       "  'journalId': '10922',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'Permissionless blockchains such as Ethereum offer decentralization and security. However, their performance is low compared to permissioned and traditional centralized systems, which may hinder their adoption. Transaction latency is an important performance metric that impacts the use of blockchain systems. That is, there is a need for a systematic evaluation to analyze the factors that may contribute to transaction latency. In this article, we propose a queueing model and conduct extensive simulation experiments to evaluate transaction latency within the Ethereum blockchain, considering all the contributing factors. Our simulation results suggest that transaction fees, block limit, block interval time, transaction arrival rate, and the behavior of the network nodes can all significantly contribute to the latency figures. We find the behavior of the network nodes can significantly impact the waiting time for transactions, especially since nodes are not enforced to utilize their blocks properly in the current implementation of permissionless blockchains. Finally, we suggest improvements to transaction latency by (1) increasing the block limit, (2) reducing the block interval time, and (3) encouraging nodes to fill their blocks with transactions as well as discussing the relevant security issues.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Computer Communication Networks',\n",
       "   'Computer Systems Organization and Communication Networks',\n",
       "   'Communications Engineering, Networks',\n",
       "   'Information Systems and Communication Service',\n",
       "   'Operations Research/Decision Theory']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s00498-022-00337-y',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s00498-022-00337-y'}],\n",
       "  'title': 'On state estimation for nonlinear systems under random access wireless protocols',\n",
       "  'creators': [{'ORCID': '0000-0001-9717-4354',\n",
       "    'creator': 'Maass, Alejandro I.'},\n",
       "   {'creator': 'Nešić, Dragan'},\n",
       "   {'creator': 'Postoyan, Romain'},\n",
       "   {'creator': 'Tan, Ying'}],\n",
       "  'publicationName': 'Mathematics of Control, Signals, and Systems',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s00498-022-00337-y',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1435-568X',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '27',\n",
       "  'journalId': '498',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer-Verlag London Ltd., part of Springer Nature',\n",
       "  'abstract': 'This article is dedicated to Eduardo D. Sontag on the occasion of his 70th birthday. We build upon fundamental stability concepts developed by Sontag, such as input-to-state stability and its related properties, to study a relevant application in industrial internet of things, namely estimation for wireless networked control systems. Particularly, we study emulation-based state estimation for nonlinear plants that communicate with a remote observer over a shared wireless network subject to packet losses. To reduce bandwidth usage, a stochastic communication protocol is employed to determine which node should be given access to the network. Each node has a different successful transmission probability. We describe the overall closed-loop system as a stochastic hybrid model, which allows us to capture the behaviour both between and at transmission instants, whilst covering network features such as random transmission instants, packet losses and stochastic scheduling. We then provide sufficient conditions on the transmission rate that guarantee an input-to-state stability property (in expectation) for the corresponding estimation error system. We illustrate our results in the design of circle criterion observers.',\n",
       "  'subjects': ['Mathematics',\n",
       "   'Systems Theory, Control',\n",
       "   'Control, Robotics, Mechatronics',\n",
       "   'Communications Engineering, Networks']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11263-022-01740-3',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11263-022-01740-3'}],\n",
       "  'title': 'RM3D: Robust Data-Efficient 3D Scene Parsing via Traditional and Learnt 3D Descriptors-Based Semantic Region Merging',\n",
       "  'creators': [{'ORCID': '0000-0002-8387-3565', 'creator': 'Liu, Kangcheng'}],\n",
       "  'publicationName': 'International Journal of Computer Vision',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11263-022-01740-3',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-1405',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '30',\n",
       "  'journalId': '11263',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'Existing state-of-the-art 3D point clouds understanding methods merely perform well in a fully supervised manner. To the best of our knowledge, there exists no unified framework which simultaneously solves the downstream high-level understanding tasks including both segmentation and detection, especially when labels are extremely limited. This work presents a general and simple framework to tackle point clouds understanding when labels are limited. The first contribution is that we have done extensive methodology comparisons of traditional and learnt 3D descriptors for the task of weakly supervised 3D scene understanding, and validated that our adapted traditional PFH-based 3D descriptors show excellent generalization ability across different domains. The second contribution is that we proposed a learning-based region merging strategy based on the affinity provided by both the traditional/learnt 3D descriptors and learnt semantics. The merging process takes both low-level geometric and high-level semantic feature correlations into consideration. Experimental results demonstrate that our framework has the best performance among the three most important weakly supervised point clouds understanding tasks including semantic segmentation, instance segmentation, and object detection even when very limited number of points are labeled. Our method, termed Region Merging 3D (RM3D), has superior performance on ScanNet data-efficient learning online benchmarks and other four large-scale 3D understanding benchmarks under various experimental settings, outperforming current arts by a margin for various 3D understanding tasks without complicated learning strategies such as active learning.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Computer Imaging, Vision, Pattern Recognition and Graphics',\n",
       "   'Artificial Intelligence',\n",
       "   'Image Processing and Computer Vision',\n",
       "   'Pattern Recognition']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s40430-022-03974-1',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s40430-022-03974-1'}],\n",
       "  'title': 'A fault diagnosis network based on domain adversarial learning and distribution matching for rotating machine vibration signal with noise and across-load conditions',\n",
       "  'creators': [{'ORCID': '0000-0002-3726-3416', 'creator': 'Gao, Sheng'},\n",
       "   {'ORCID': '0000-0003-1074-7776', 'creator': 'Sun, Hongchun'},\n",
       "   {'creator': 'Ma, Sihan'}],\n",
       "  'publicationName': 'Journal of the Brazilian Society of Mechanical Sciences and Engineering',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s40430-022-03974-1',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1806-3691',\n",
       "  'volume': '45',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Technical Paper'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '17',\n",
       "  'journalId': '40430',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to The Brazilian Society of Mechanical Sciences and Engineering',\n",
       "  'abstract': 'In the practice of deep learning-based rotating machinery fault diagnosis method, how to improve the accuracy of cross-working condition diagnosis under noise background is an urgent problem to be solved, which limits the application of deep learning method in engineering practice. Although domain adaptation methods have been widely researched to solve the problem, they often face the problem of domain shift. In this paper, two domain adaptation methods: domain adversarial learning and distribution matching methods, are used to train the deep network, to make the model realize the fault diagnosis across load conditions and signal-to-noise ratio. Moreover, in order to suppress the domain shift phenomenon which often occurred in domain adaptation tasks. A new network architecture based on the wide convolution kernel, multi-scale attention module, and self-adaptive soft threshold function is proposed so that the network is more suitable for feature extraction in cross-working condition fault diagnosis, and can avoid the influence of noise in vibration signals. Compared with other methods, the proposed method has good performance in faults diagnosis across load conditions and under noise. Feature visualization proved that the proposed network can effectively extract the conjoint fault features across different load conditions from the signal; hence, the fault diagnosis across load conditions and under the noise background can be realized and the domain shift can be suppressed effectively.',\n",
       "  'subjects': ['Engineering', 'Mechanical Engineering']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s10489-022-04420-4',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s10489-022-04420-4'}],\n",
       "  'title': 'Contextual transformer sequence-based recognition network for medical examination reports',\n",
       "  'creators': [{'creator': 'Wan, Honglin'},\n",
       "   {'creator': 'Zhong, Zongfeng'},\n",
       "   {'creator': 'Li, Tianping'},\n",
       "   {'creator': 'Zhang, Huaxiang'},\n",
       "   {'ORCID': '0000-0001-6157-2051', 'creator': 'Sun, Jiande'}],\n",
       "  'publicationName': 'Applied Intelligence',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s10489-022-04420-4',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-7497',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '18',\n",
       "  'journalId': '10489',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'The automatic recognition of the medical examination report table (MERT) is receiving increasing attention in recent years as it is an essential step for intelligent healthcare and medical treatment. However, there are still some challenges in the table prediction when it is applied practically. In this paper, a recognition network (CoT_SRN) for medical examination reports is proposed to improve the recognition accuracy of MERT structure and reconstruct the image into a spreadsheet. The network is based on contextual transformer sequence and consists of CoT encoder and SRN decoder. In the encoder, the CNN backbone is constructed to extract the MERT image structure features based on the Contextual Transformer (CoT) proposed in this paper. In the decoder, an attention head with gated recurrent unit (GRU) was used for feature sequence recognition to obtain the cell location and table structure represented by a structured language. In addition, MERT structure labels are defined as character-level HTML formats, which are added in the training of the table structure recognition. The proposed method can achieve competitive tree-edit-distance-based similarity (TEDS) scores on the English datasets, such as PubTabNet and SciTSR, and Chinese datasets, such as the Chinese medical document dataset (CMDD). It demonstrates that the Cot_SRN is helpful to preserve the good performance across multi-language MERT structure recognition. Additionally, the performance of the proposed method is verified on the practical examples with folds and small angle deflection. The experimental results show that the proposed method is promising in practical application.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Artificial Intelligence',\n",
       "   'Mechanical Engineering',\n",
       "   'Manufacturing, Machines, Tools, Processes']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1038/s41598-022-27145-3',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1038/s41598-022-27145-3'}],\n",
       "  'title': 'Identifying vital nodes for influence maximization in attributed networks',\n",
       "  'creators': [{'creator': 'Wang, Ying'},\n",
       "   {'creator': 'Zheng, Yunan'},\n",
       "   {'creator': 'Liu, Yiguang'}],\n",
       "  'publicationName': 'Scientific Reports',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1038/s41598-022-27145-3',\n",
       "  'publisher': 'Nature',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2045-2322',\n",
       "  'volume': '12',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '12',\n",
       "  'journalId': '41598',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Identifying a set of vital nodes to achieve influence maximization is a topic of general interest in network science. Many algorithms have been proposed to solve the influence maximization problem in complex networks. Most of them just use topology information of networks to measure the node influence. However, the node attribute is also an important factor for measuring node influence in attributed networks. To tackle this problem, we first propose an extension model of linear threshold (LT) propagation model to simulate the information propagation in attributed networks. Then, we propose a novel community-based method to identify a set of vital nodes for influence maximization in attributed networks. The proposed method considers both topology influence and attribute influence of nodes, which is more suitable for identifying vital nodes in attributed networks. A series of experiments are carried out on five real world networks and a large scale synthetic network. Compared with CELF, IMM, CoFIM, HGD, NCVoteRank and K-Shell methods, experimental results based on different propagation models show that the proposed method improves the influence spread by $$-2.28\\\\% \\\\, \\\\textrm{to} \\\\, 4.76\\\\%$$ - 2.28 % to 4.76 % , $$-2.50\\\\% \\\\, \\\\textrm{to} \\\\, 16.97\\\\%$$ - 2.50 % to 16.97 % , $$0.18\\\\% \\\\, \\\\textrm{to} \\\\, 16.07\\\\%$$ 0.18 % to 16.07 % , $$0.22\\\\% \\\\, \\\\textrm{to} \\\\, 41.82\\\\%$$ 0.22 % to 41.82 % , $$0.23\\\\% \\\\, \\\\textrm{to} \\\\, 11.24\\\\%$$ 0.23 % to 11.24 % and $$10.78\\\\% \\\\, \\\\textrm{to} \\\\, 75.22\\\\%$$ 10.78 % to 75.22 % .',\n",
       "  'subjects': ['Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, multidisciplinary']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11263-022-01736-z',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11263-022-01736-z'}],\n",
       "  'title': 'SHARP: Shape-Aware Reconstruction of People in Loose Clothing',\n",
       "  'creators': [{'creator': 'Jinka, Sai Sagar'},\n",
       "   {'ORCID': '0000-0001-6600-722X', 'creator': 'Srivastava, Astitva'},\n",
       "   {'creator': 'Pokhariya, Chandradeep'},\n",
       "   {'creator': 'Sharma, Avinash'},\n",
       "   {'creator': 'Narayanan, P. J.'}],\n",
       "  'publicationName': 'International Journal of Computer Vision',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11263-022-01736-z',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-1405',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '20',\n",
       "  'journalId': '11263',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'Recent advancements in deep learning have enabled 3D human body reconstruction from a monocular image, which has broad applications in multiple domains. In this paper, we propose SHARP ( SH ape A ware R econstruction of P eople in loose clothing), a novel end-to-end trainable network that accurately recovers the 3D geometry and appearance of humans in loose clothing from a monocular image. SHARP uses a sparse and efficient fusion strategy to combine parametric body prior with a non-parametric 2D representation of clothed humans. The parametric body prior enforces geometrical consistency on the body shape and pose, while the non-parametric representation models loose clothing and handles self-occlusions as well. We also leverage the sparseness of the non-parametric representation for faster training of our network while using losses on 2D maps. Another key contribution is 3DHumans , our new life-like dataset of 3D human body scans with rich geometrical and textural details. We evaluate SHARP on 3DHumans and other publicly available datasets, and show superior qualitative and quantitative performance than existing state-of-the-art methods.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Computer Imaging, Vision, Pattern Recognition and Graphics',\n",
       "   'Artificial Intelligence',\n",
       "   'Image Processing and Computer Vision',\n",
       "   'Pattern Recognition']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s41870-022-01140-3',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s41870-022-01140-3'}],\n",
       "  'title': 'Light Weight Gradient Ensemble Model for detecting network attack at the edge of the IoT network',\n",
       "  'creators': [{'ORCID': '0000-0002-2734-7747', 'creator': 'Santhadevi, D.'},\n",
       "   {'creator': 'Janet, B.'}],\n",
       "  'publicationName': 'International Journal of Information Technology',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s41870-022-01140-3',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2511-2112',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Research'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '10',\n",
       "  'journalId': '41870',\n",
       "  'copyright': \"©2022 The Author(s), under exclusive licence to Bharati Vidyapeeth's Institute of Computer Applications and Management\",\n",
       "  'abstract': 'An intelligent malware prediction system is needed for monitoring the network traffic of traditional and future network attacks. The Internet of Things (IoT) is fully connected and remotely monitored in many places. These devices are more vulnerable and easier to attack remotely. Securing and monitoring the activities of these devices must be done at the edge level. Machine Learning (ML) models are broadly used for supervising defence purposes, and these models are computationally expensive. Light Weight Gradient Ensemble Machine (LWGEM) with voting-based feature selection techniques and tuning the hyper-parameter are used to enhance the model accuracy and reduce the computational expenses. Performance of the model was evaluated with benchmark dataset UNSW_NB15 which shows the prominent result over the existing model and state of art models. The proposed model performs better in predicting unknown abnormalities in the IoT network.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Computer Science, general',\n",
       "   'Artificial Intelligence',\n",
       "   'Machine Learning',\n",
       "   'Image Processing and Computer Vision',\n",
       "   'Software Engineering',\n",
       "   'Computer Imaging, Vision, Pattern Recognition and Graphics']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s41660-022-00306-7',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s41660-022-00306-7'}],\n",
       "  'title': 'Performance Comparison of Recent Nature-Inspired Optimization Algorithms in Optimal Placement and Sizing of Distributed Energy Resources',\n",
       "  'creators': [{'creator': 'Amroune, Mohammed'},\n",
       "   {'creator': 'Boukaroura, Abdelkader'},\n",
       "   {'creator': 'Bourzami, Arif'},\n",
       "   {'creator': 'Slimani, Linda'},\n",
       "   {'creator': 'Musirin, Ismail'}],\n",
       "  'publicationName': 'Process Integration and Optimization for Sustainability',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s41660-022-00306-7',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2509-4246',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Research Paper'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '14',\n",
       "  'journalId': '41660',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Nature Singapore Pte Ltd.',\n",
       "  'abstract': 'Through the integration of distributed generation into the distribution network, several benefits can be achieved, including minimization of system losses, enhancement of voltage profiles, and improvement of network reliability and security. One of the best ways to maximize the benefits of distributed generation units is to optimize the location and size of these units in the network. The main objective of this paper is the application and comparison of five recently developed optimization algorithms for optimizing the allocation of multiple distributed generators in the distribution network. The algorithms include the hybrid grey wolf optimizer and cuckoo search, bald eagle search, marine predator algorithm, artificial ecosystem optimizer, and slime mould algorithm. Three objectives are set in the problem of determining the optimal allocation of distributed generators, including the reduction of energy losses, enhancement of voltage profile, and improvement of voltage stability. The proposed optimization algorithms are evaluated and compared using IEEE 33-bus and IEEE 69-bus radial distribution networks. Numerical results show that the proposed algorithms are very competitive in their ability to allocate distributed generators optimally. However, artificial ecosystem optimizer and bald eagle search outperform other algorithms.',\n",
       "  'subjects': ['Engineering',\n",
       "   'Industrial and Production Engineering',\n",
       "   'Sustainable Development',\n",
       "   'Industrial Chemistry/Chemical Engineering',\n",
       "   'Energy Policy, Economics and Management',\n",
       "   'Waste Management/Waste Technology']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11063-022-11135-7',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11063-022-11135-7'}],\n",
       "  'title': 'Adaptive Style Modulation for Artistic Style Transfer',\n",
       "  'creators': [{'creator': 'Zhang, Yipeng'},\n",
       "   {'creator': 'Hu, Bingliang'},\n",
       "   {'creator': 'Huang, Yingying'},\n",
       "   {'creator': 'Gao, Chi'},\n",
       "   {'creator': 'Wang, Quan'}],\n",
       "  'publicationName': 'Neural Processing Letters',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11063-022-11135-7',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-773X',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '18',\n",
       "  'journalId': '11063',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'Arbitrary-style-per-model (ASPM) style transfer algorithms transfer arbitrary styles based on a single model. Statistics-based learning algorithms of ASPM, represented by adaptive instance normalization (AdaIN), conduct instance normalization and then perform an affine transformation on target features. These algorithms are computationally efficient and easy to embed in convolutional neural networks. Consequently, they are widely used in image synthesis tasks to control the style of the resulting images. However, the style of stylized images may be a combination of content and stylized images, which suggests that these methods do not transform styles accurately. In this work, we rethink the function of AdaIN in controlling style. We show that the role of AdaIN is to (1) give each input content image a specific optimization target, (2) dynamically set cross-channel correlations, and (3) act as a feature selector after combining it with an activation function. Accordingly, we propose adaptive style modulation (AdaSM), which fully leverages the three roles mentioned above and thereby enables more precise control of global style. Experimental results show that AdaSM provides superior style controllability, alleviates the style blending problem, and outperforms state-of-the-art methods in artistic style transfer tasks.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Artificial Intelligence',\n",
       "   'Complex Systems',\n",
       "   'Computational Intelligence']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s13762-022-04728-1',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s13762-022-04728-1'}],\n",
       "  'title': 'Intelligent modeling of photocatalytically reactive yellow 84 azo dye removal from aqueous solutions by ZnO-light expanded clay aggregate nanoparticles',\n",
       "  'creators': [{'creator': 'Moradi, M.'},\n",
       "   {'creator': 'Moradkhani, M. A.'},\n",
       "   {'creator': 'Hosseini, S. H.'},\n",
       "   {'creator': 'Olazar, M.'}],\n",
       "  'publicationName': 'International Journal of Environmental Science and Technology',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s13762-022-04728-1',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1735-2630',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Paper'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '14',\n",
       "  'journalId': '13762',\n",
       "  'copyright': '©2022 The Author(s) under exclusive licence to Iranian Society of Environmentalists (IRSEN) and Science and Research Branch, Islamic Azad University',\n",
       "  'abstract': 'This study deals with the intelligent modeling of reactive yellow 84 dye removal from aqueous solutions by zinc oxide (ZnO) nanoparticles based on inorganic light expanded clay aggregates. Accordingly, a set of experimental data were utilized to develop robust models by adjusting the initial feed concentration, initial pH, de-colorization time, catalyst concentration, and lamp power as input factors. The Leverage method, a common method for detecting outliers, proved all measured data were reliable. After definition of input factors, four intelligent approaches, i.e., multilayer perceptron, Gaussian process regression (GPR), radial basis function, and adaptive neuro fuzzy inference system with subtractive clustering were used to establish exact models for predicting the dye removal. Among them, the GPR-based model produced the best results with AARE, RRMSE, and R 2 values of 4.87%, 6.21%, and 97.31%, respectively, for the test dataset. The remaining models also provided satisfactory outputs with AAREs between 6.59% and 9.81%. In addition, the influence of each operating parameter on the dye removal was properly described for the novel models. Finally, the most effective parameters on removal performance were determined by using a sensitivity analysis based on the GPR model.',\n",
       "  'subjects': ['Environment',\n",
       "   'Environment, general',\n",
       "   'Environmental Science and Engineering',\n",
       "   'Environmental Chemistry',\n",
       "   'Waste Water Technology / Water Pollution Control / Water Management / Aquatic Pollution',\n",
       "   'Soil Science & Conservation',\n",
       "   'Ecotoxicology']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1140/epje/s10189-022-00258-3',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1140/epje/s10189-022-00258-3'}],\n",
       "  'title': 'Inferring turbulent environments via machine learning',\n",
       "  'creators': [{'ORCID': '0000-0002-7162-5038',\n",
       "    'creator': 'Buzzicotti, Michele'},\n",
       "   {'creator': 'Bonaccorso, Fabio'}],\n",
       "  'publicationName': 'The European Physical Journal E',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1140/epje/s10189-022-00258-3',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1292-895X',\n",
       "  'volume': '45',\n",
       "  'number': '12',\n",
       "  'genre': ['OriginalPaper', 'Regular Article - Flowing Matter'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '17',\n",
       "  'journalId': '10189',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to EDP Sciences, SIF and Springer-Verlag GmbH Germany, part of Springer Nature',\n",
       "  'abstract': 'Abstract The problem of classifying turbulent environments from partial observation is key for some theoretical and applied fields, from engineering to earth observation and astrophysics, e.g., to precondition searching of optimal control policies in different turbulent backgrounds, to predict the probability of rare events and/or to infer physical parameters labeling different turbulent setups. To achieve such goal one can use different tools depending on the system’s knowledge and on the quality and quantity of the accessible data. In this context, we assume to work in a model-free setup completely blind to all dynamical laws, but with a large quantity of (good quality) data for training. As a prototype of complex flows with different attractors, and different multi-scale statistical properties we selected 10 turbulent ‘ensembles’ by changing the rotation frequency of the frame of reference of the 3d domain and we suppose to have access to a set of partial observations limited to the instantaneous kinetic energy distribution in a 2d plane, as it is often the case in geophysics and astrophysics. We compare results obtained by a machine learning (ML) approach consisting of a state-of-the-art deep convolutional neural network (DCNN) against Bayesian inference which exploits the information on velocity and entropy moments. First, we discuss the supremacy of the ML approach, presenting also results at changing the number of training data and of the hyper-parameters. Second, we present an ablation study on the input data aimed to perform a ranking on the importance of the flow features used by the DCNN, helping to identify the main physical contents used by the classifier. Finally, we discuss the main limitations of such data-driven methods and potential interesting applications. Graphical Abstract ',\n",
       "  'subjects': ['Physics',\n",
       "   'Soft and Granular Matter, Complex Fluids and Microfluidics',\n",
       "   'Biological and Medical Physics, Biophysics',\n",
       "   'Surfaces and Interfaces, Thin Films',\n",
       "   'Nanotechnology',\n",
       "   'Polymer Sciences',\n",
       "   'Complex Systems']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s10723-022-09639-6',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s10723-022-09639-6'}],\n",
       "  'title': 'An Integrated Deep Learning Prediction Approach for Efficient Modelling of Host Load Patterns in Cloud Computing',\n",
       "  'creators': [{'creator': 'Patel, Eva'},\n",
       "   {'creator': 'Kushwaha, Dharmender Singh'}],\n",
       "  'publicationName': 'Journal of Grid Computing',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s10723-022-09639-6',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1572-9184',\n",
       "  'volume': '21',\n",
       "  'number': '1',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '24',\n",
       "  'journalId': '10723',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Nature B.V.',\n",
       "  'abstract': 'Recent surge in technology and integration of IoT into Cloud computing has resulted in increasingly heterogeneous workloads with unprecedented compute and storage demands. As Cloud workloads evolve, so also their computational demands, resulting in server loads interspersed by random peaks and troughs and by continuous and periodic demands. A predictive model that accounts for these diverse load patterns can generate more realistic future resource demands which is essential for efficient capacity planning and meeting service level objectives with minimal energy consumption. Long Short-Term Memory (LSTM), the prevalent approach for host load prediction suffers from information decay with long inputs, while hybrid methods using Convolution Neural Network (CNN) and LSTM fail to effectively model different host load patterns. To overcome above limitations, we propose a multistep CPU usage prediction approach named RCP-CL , to model r andom fluctuations and novel c ontinuous and p eriodic patterns from contiguous and non-contiguous CPU load values augmented with daily and weekly time patterns, by integrating 1-Dimensional C NN (1D-CNN) and L STM networks. RCP-CL uses parallel and stacked 1D-CNN layers with kernel size and dilation rates that are guided by the autocorrelation and partial autocorrelation analysis of CPU usage while LSTM derives temporal dependencies from the learnt spatial patterns. Experimental evaluations with Google Trace and Alibaba trace demonstrate impressive learning skills of RCP-CL over the state-of-the-art LSTM and 1D-CNN based host load prediction models, achieving up to 20% improvement in mean squared error for Google trace and up to 22% improvement for Alibaba trace.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Processor Architectures',\n",
       "   'Management of Computing and Information Systems',\n",
       "   'User Interfaces and Human Computer Interaction']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s10701-022-00664-9',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s10701-022-00664-9'}],\n",
       "  'title': 'Entanglement and the Path Integral',\n",
       "  'creators': [{'creator': 'Wharton, Ken'}, {'creator': 'Liu, Raylor'}],\n",
       "  'publicationName': 'Foundations of Physics',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s10701-022-00664-9',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1572-9516',\n",
       "  'volume': '53',\n",
       "  'number': '1',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '23',\n",
       "  'journalId': '10701',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'The path integral is not typically utilized for analyzing entanglement experiments, in part because there is no standard toolbox for converting an arbitrary experiment into a form allowing a simple sum-over-history calculation. After completing the last portion of this toolbox (a technique for implementing multi-particle measurements in an entangled basis), some interesting 4- and 6-particle experiments are analyzed with this alternate technique. While the joint probabilities of measurement outcomes are always equivalent to conventional quantum mechanics, differences in the calculations motivate a number of foundational insights, concerning nonlocality, retrocausality, and the objectivity of entanglement itself.',\n",
       "  'subjects': ['Physics',\n",
       "   'History and Philosophical Foundations of Physics',\n",
       "   'Quantum Physics',\n",
       "   'Classical and Quantum Gravitation, Relativity Theory',\n",
       "   'Statistical Physics and Dynamical Systems',\n",
       "   'Classical Mechanics',\n",
       "   'Philosophy of Science']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s13369-022-07566-y',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s13369-022-07566-y'}],\n",
       "  'title': 'Artificial Neural Network Assisted P &O-Based MPPT Controller for a Partially Shaded Grid-Connected Solar PV Panel',\n",
       "  'creators': [{'ORCID': '0000-0001-9454-513X',\n",
       "    'creator': 'Sakthivel, Senthamizh Selvan'},\n",
       "   {'creator': 'Arunachalam, Venkadesan'}],\n",
       "  'publicationName': 'Arabian Journal for Science and Engineering',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s13369-022-07566-y',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2191-4281',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Research Article-Electrical Engineering '],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '12',\n",
       "  'journalId': '13369',\n",
       "  'copyright': '©2022 King Fahd University of Petroleum & Minerals',\n",
       "  'abstract': 'This paper presents an offline trained artificial neural network (ANN)-assisted perturb and observe (P &O) algorithm to track the global maximum power of a solar PV under partial shading conditions. The conventional hill climbing algorithms work very well to extract maximum power under uniform shading conditions, but fails under partial shading conditions. On the other hand, the offline trained ANN can track the power closer to global maximum power point (GMPP) with less oscillations. Hence, two methods are combined to design the MPPT controller for tracking the global maximum power accurately with less oscillations. Initially, ANN predicts the voltage near to the global maximum power point with solar PV voltage and current as the inputs. This predicted voltage from ANN is used as the initial values for the P &O algorithm, and the P &O algorithm is triggered for tracking the actual GMPP. The proposed MPPT controller is validated in the MATLAB/Simulink environment and the performance of the ANN model for the prediction of GMPP is validated with the experimental data sets.',\n",
       "  'subjects': ['Engineering',\n",
       "   'Engineering, general',\n",
       "   'Science, Humanities and Social Sciences, multidisciplinary']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1038/s41598-022-27178-8',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1038/s41598-022-27178-8'}],\n",
       "  'title': 'A pathway analysis-based algorithm for calculating the participation degree of ncRNA in transcriptome',\n",
       "  'creators': [{'creator': 'Gu, Xinyi'},\n",
       "   {'creator': 'Wang, Shen'},\n",
       "   {'creator': 'Jin, Bo'},\n",
       "   {'creator': 'Qi, Zhidan'},\n",
       "   {'creator': 'Deng, Jin'},\n",
       "   {'creator': 'Huang, Chen'},\n",
       "   {'creator': 'Yin, Xiaofeng'}],\n",
       "  'publicationName': 'Scientific Reports',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1038/s41598-022-27178-8',\n",
       "  'publisher': 'Nature',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2045-2322',\n",
       "  'volume': '12',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '13',\n",
       "  'journalId': '41598',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'After sequencing, it is common to screen ncRNA according to expression differences. But this may lose a lot of valuable information and there is currently no indicator to characterize the regulatory function and participation degree of ncRNA on transcriptome. Based on existing pathway enrichment methods, we developed a new algorithm to calculating the participation degree of ncRNA in transcriptome (PDNT). Here we analyzed multiple data sets, and differentially expressed genes (DEGs) were used for pathway enrichment analysis. The PDNT algorithm was used to calculate the Contribution value (C value) of each ncRNA based on its target genes and the pathways they participates in. The results showed that compared with ncRNAs screened by log2 fold change (FC) and p-value, those screened by C value regulated more DEGs in IPA canonical pathways, and their target DEGs were more concentrated in the core region of the protein–protein interaction (PPI) network. The ranking of disease critical ncRNAs increased integrally after sorting with C value. Collectively, we found that the PDNT algorithm provides a measure from another view compared with the log2FC and p-value and it may provide more clues to effectively evaluate ncRNA.',\n",
       "  'subjects': ['Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, multidisciplinary']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s12524-022-01646-0',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s12524-022-01646-0'}],\n",
       "  'title': 'Assessment of Trees Outside Forest (TOF) in Urban Landscape Using High-Resolution Satellite Images and Deep Learning Techniques',\n",
       "  'creators': [{'ORCID': '0000-0002-8851-0958', 'creator': 'Vinod, P. V.'},\n",
       "   {'ORCID': '0000-0003-1402-089X', 'creator': 'Trivedi, Shivam'},\n",
       "   {'ORCID': '0000-0003-3685-7295', 'creator': 'Hebbar, R.'},\n",
       "   {'ORCID': '0000-0002-3560-4993', 'creator': 'Jha, C. S.'}],\n",
       "  'publicationName': 'Journal of the Indian Society of Remote Sensing',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s12524-022-01646-0',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '0974-3006',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Research Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '16',\n",
       "  'journalId': '12524',\n",
       "  'copyright': '©2022 Indian Society of Remote Sensing',\n",
       "  'abstract': 'Accurate assessment of Trees outside forest (TOF) is an essential input for the\\xa0management and preservation of these vast resources, due to their importance in carbon sequestration and biodiversity conservation. Many studies have demonstrated the\\xa0use of High-resolution satellite (HRS) imageries for mapping and monitoring of TOF in urban and rural landscapes. The conventional per-pixel classifier is not a suitable option for HRS data classification, owing to inherent high intra-class variability and high interclass similarity. In the present study, a convolution neural network-based approach was employed for TOF mapping using HRS images, predominantly in the\\xa0urban landscape of Bengaluru city, India. We developed a semi-automated procedure for the\\xa0generation of labelled training samples using object-based image analysis (OBIA), minimizing time requirement for input training data preparation for deep learning (DL) model development. These inputs were utilized to develop a DL model for the\\xa0classification of TOF using U-Net, which is a semantic segmentation-based DL architecture. The results indicated that the model performed well for the\\xa0classification of urban TOF with an\\xa0overall classification accuracy of 89.65 percent and an\\xa0F1 score of 93.03 percent, in comparison with OBIA (overall accuracy of 80.73 percent & F1 score of 86.44 percent). We demonstrated utilisation of a\\xa0developed methodology for the\\xa0assessment of TOF in a distinctive urban landscape, which can be extended to agriculture dominated regions in diverse landscapes.',\n",
       "  'subjects': ['Earth Sciences',\n",
       "   'Earth Sciences, general',\n",
       "   'Remote Sensing/Photogrammetry']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11760-022-02465-z',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11760-022-02465-z'}],\n",
       "  'title': 'Combining channel-wise joint attention and temporal attention in graph convolutional networks for skeleton-based action recognition',\n",
       "  'creators': [{'creator': 'Sun, Zhonghua'},\n",
       "   {'creator': 'Wang, Tianyi'},\n",
       "   {'creator': 'Dai, Meng'}],\n",
       "  'publicationName': 'Signal, Image and Video Processing',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11760-022-02465-z',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1863-1711',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Paper'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '8',\n",
       "  'journalId': '11760',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer-Verlag London Ltd., part of Springer Nature',\n",
       "  'abstract': 'Graph convolutional networks (GCNs) have been shown to be effective in performing skeleton-based action recognition, as graph topology has advantages in representing the natural connectivity of the human bodies. Nevertheless, it is challenging to effectively model the human joints spatially and temporally, and we are lacking attentional mechanisms for critical temporal frames and important skeletal points. In this work, we propose a novel GCNs combined with channel-wise joints and temporal attention for skeleton-based action recognition. Our temporal attention module captures the long-term dependence of time and then enhances the temporal semantics of key frames. In addition, we design a channel-wise attention module that fuses multi-channel joint weights with the topological map to capture the attention of nodes at different actions along the channel dimension. We propose to concatenate joint and bone together along the channel dimension as the joint& bone (J& B) modality, J& B modality can extract hybrid action patterns under the coalition of channel-wise joint attention. We prove the powerful spatio-temporal modeling capability of our model on three widely used dataset, NTU-RGB D, NTU RGB+D 120 and Northwestern-UCLA. Compared with leading GCN-based methods, we achieve performance comparable to the-state-of-art.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Image Processing and Computer Vision',\n",
       "   'Signal,Image and Speech Processing',\n",
       "   'Computer Imaging, Vision, Pattern Recognition and Graphics',\n",
       "   'Multimedia Information Systems']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1186/s12874-022-01827-y',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1186/s12874-022-01827-y'}],\n",
       "  'title': 'Interpretable generalized neural additive models for mortality prediction of COVID-19 hospitalized patients in Hamadan, Iran',\n",
       "  'creators': [{'ORCID': '0000-0003-1597-7327', 'creator': 'Moslehi, Samad'},\n",
       "   {'ORCID': '0000-0002-9375-3807', 'creator': 'Mahjub, Hossein'},\n",
       "   {'ORCID': '0000-0002-6054-9850', 'creator': 'Farhadian, Maryam'},\n",
       "   {'ORCID': '0000-0002-7483-3502', 'creator': 'Soltanian, Ali Reza'},\n",
       "   {'ORCID': '0000-0001-5802-6367', 'creator': 'Mamani, Mojgan'}],\n",
       "  'publicationName': 'BMC Medical Research Methodology',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1186/s12874-022-01827-y',\n",
       "  'publisher': 'BioMed Central',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1471-2288',\n",
       "  'volume': '22',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Research'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '14',\n",
       "  'journalId': '12874',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': \"Background The high number of COVID-19 deaths is a serious threat to the world. Demographic and clinical biomarkers are significantly associated with the mortality risk of this disease. This study aimed to implement Generalized Neural Additive Model (GNAM) as an interpretable machine learning method to predict the COVID-19 mortality of patients. Methods This cohort study included 2181 COVID-19 patients admitted from February 2020 to July 2021 in Sina and Besat hospitals in Hamadan, west of Iran. A total of 22 baseline features including patients' demographic information and clinical biomarkers were collected. Four strategies including removing missing values, mean, K-Nearest Neighbor (KNN), and Multivariate Imputation by Chained Equations (MICE) imputation methods were used to deal with missing data. Firstly, the important features for predicting binary outcome (1: death, 0: recovery) were selected using the Random Forest (RF) method. Also, synthetic minority over-sampling technique (SMOTE) method was used for handling imbalanced data. Next, considering the selected features, the predictive performance of GNAM for predicting mortality outcome was compared with logistic regression, RF, generalized additive model (GAMs), gradient boosting decision tree (GBDT), and deep neural networks (DNNs) classification models. Each model trained on fifty different subsets of a train-test dataset to ensure a model performance. The average accuracy, F1-score and area under the curve (AUC) evaluation indices were used for comparison of the predictive performance of the models. Results Out of the 2181 COVID-19 patients, 624 died during hospitalization and 1557 recovered. The missing rate was 3 percent for each patient. The mean age of dead patients (71.17\\u2009±\\u200914.44\\xa0years) was statistically significant higher than recovered patients (58.25\\u2009±\\u200916.52\\xa0years). Based on RF, 10 features with the highest relative importance were selected as the best influential features; including blood urea nitrogen (BUN), lymphocytes (Lym), age, blood sugar (BS), serum glutamic-oxaloacetic transaminase (SGOT), monocytes (Mono), blood creatinine (CR), neutrophils (NUT), alkaline phosphatase (ALP) and hematocrit (HCT). The results of predictive performance comparisons showed GNAM with the mean accuracy, F1-score, and mean AUC in the test dataset of 0.847, 0.691, and 0.774, respectively, had the best performance. The smooth function graphs learned from the GNAM were descending for the Lym and ascending for the other important features. Conclusions Interpretable GNAM can perform well in predicting the mortality of COVID-19 patients. Therefore, the use of such a reliable model can help physicians to prioritize some important demographic and clinical biomarkers by identifying the effective features and the type of predictive trend in disease progression.\",\n",
       "  'subjects': ['Medicine & Public Health',\n",
       "   'Theory of Medicine/Bioethics',\n",
       "   'Statistical Theory and Methods',\n",
       "   'Statistics for Life Sciences, Medicine, Health Sciences']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11694-022-01795-3',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11694-022-01795-3'}],\n",
       "  'title': 'Disease detection and physical disorders classification for citrus fruit images using convolutional neural network',\n",
       "  'creators': [{'ORCID': '0000-0003-4397-6196', 'creator': 'Uğuz, Sinan'},\n",
       "   {'creator': 'Şikaroğlu, Gulhan'},\n",
       "   {'creator': 'Yağız, Abdullah'}],\n",
       "  'publicationName': 'Journal of Food Measurement and Characterization',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11694-022-01795-3',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2193-4134',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Paper'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '10',\n",
       "  'journalId': '11694',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'Citrus is an agricultural product with significant added value in the world. Depending on market demands and causing possible economic losses, disease detection in citrus fruits at an early stage is important as well as for all agricultural products. Therefore, it is critical to detect both early-stage citrus diseases and physically damaged citrus fruits with technological solutions. In this study, a new convolutional neural network (CNN)-based model, CitrusNet, was proposed for the classification of defective and deformed citrus fruits. In this study, 5149 images of citrus fruits were collected from citrus orchards in Antalya, Turkey. In the experiments conducted with four different CNN models, the CitrusNet and ResNet50 models obtained the best classification results. In the other phase of the study, experiments were carried out to detect Alternaria alternata and Thrips diseases, which are common in Turkey, using five different CNN models. In the first phase of the study, a dataset of 3582 images of citrus diseases was obtained. The experimental results show that the YOLOv5 and Mask R-CNN models adequately detected the citrus diseases compared to the other models. These models achieved the best performance with an average precision (AP) of 0.99.',\n",
       "  'subjects': ['Chemistry',\n",
       "   'Food Science',\n",
       "   'Chemistry/Food Science, general',\n",
       "   'Engineering, general']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s00344-022-10893-x',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s00344-022-10893-x'}],\n",
       "  'title': 'Helopeltis theivora\\xa0Responsive Transcriptomic Reprogramming Uncovers Long Non-coding RNAs as Possible Regulators of Primary and Secondary Metabolism in Tea Plant',\n",
       "  'creators': [{'creator': 'Bordoloi, Kuntala Sarma'},\n",
       "   {'creator': 'Baruah, Pooja Moni'},\n",
       "   {'creator': 'Tanti, Bhaben'},\n",
       "   {'creator': 'Gill, Sarvajeet Singh'},\n",
       "   {'ORCID': '0000-0002-1144-7743', 'creator': 'Agarwala, Niraj'}],\n",
       "  'publicationName': 'Journal of Plant Growth Regulation',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s00344-022-10893-x',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1435-8107',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '26',\n",
       "  'journalId': '344',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'Helopeltis theivora or the tea mosquito bug (TMB) is reportedly one of the most devastating pests of tea plant ( Camellia sinensis ) causing threat to the beverage crop. Long non-coding RNAs (lncRNAs) constitute a group of endogenous RNAs that play gene regulatory roles in eukaryotes. In the present study, 9502 lncRNAs were identified from healthy and TMB-infested C. sinensis tissues using high-throughput strand-specific RNA sequencing, out of which 80 lncRNAs got differentially expressed in response to TMB infestation. Determination of genes that could act as potential targets of lncRNAs revealed that the identified lncRNAs could possibly target as many as 5804 genes. Differential gene expression (DGE) analysis led to the identification of 3665 differentially expressed genes (DEGs), of which, the expression of 1767 genes got upregulated and 1898 genes got downregulated during tea plant’s response to TMB. Gene Ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) enrichment analysis of DEGs and lncRNA-target genes have shown that TMB infestation might have triggered transcriptomic reprogramming leading to altered primary and secondary metabolism in C. sinensis . LncRNAs can act as competing endogenous RNAs (ceRNAs) to bind with common microRNA (miRNA) response elements (MREs) involving a competition between mRNAs and lncRNAs. We report 11 lncRNAs competing with 14 mRNAs to bind with 28 miRNAs forming the ceRNA network. The expression of 6 DEGs and 5 differentially expressed lncRNAs (DELs) has been validated by qRT-PCR.',\n",
       "  'subjects': ['Life Sciences',\n",
       "   'Plant Sciences',\n",
       "   'Plant Anatomy/Development',\n",
       "   'Plant Physiology',\n",
       "   'Agriculture']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11760-022-02445-3',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11760-022-02445-3'}],\n",
       "  'title': 'ECU-Net: multi-scale salient boundary detection and contrast feature enhancement U-Net for breast ultrasound image segmentation',\n",
       "  'creators': [{'creator': 'Han, Cailing'},\n",
       "   {'creator': 'Huang, Xi'},\n",
       "   {'creator': 'Zhang, Yimin'},\n",
       "   {'creator': 'Wang, Minghui'}],\n",
       "  'publicationName': 'Signal, Image and Video Processing',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11760-022-02445-3',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1863-1711',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Paper'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '9',\n",
       "  'journalId': '11760',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer-Verlag London Ltd., part of Springer Nature',\n",
       "  'abstract': 'U-Net and its variants show excellent performance in medical image segmentation. However, due to the information loss caused by down-sampling and the blurry boundary of the ultrasound images itself, it is hard to extract and recover boundary information. Additionally, due to the intensity similarity and low contrast between tumor and non-tumor regions in breast ultrasound images, obtaining discriminative features is difficult. So we propose an optimized U-Net breast ultrasound image segmentation model ECU-Net. Firstly, a multi-scale salient boundary detection module is proposed to enhance the model’s perception of tumor boundaries. Secondly, a contrast feature enhancement module is proposed to highlight the otherness between tumor and normal tissues. Then we combine the two modules to optimize skip-connection through boundaries completion and contrast information enhancement. Exhaustive experiments are conducted on two publicly available datasets, and experimental results show our ECU-Net performs better than U-Net and its classic variants.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Image Processing and Computer Vision',\n",
       "   'Signal,Image and Speech Processing',\n",
       "   'Computer Imaging, Vision, Pattern Recognition and Graphics',\n",
       "   'Multimedia Information Systems']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1038/s41598-022-27254-z',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1038/s41598-022-27254-z'}],\n",
       "  'title': 'Deep learning-derived cardiovascular age shares a genetic basis with other cardiac phenotypes',\n",
       "  'creators': [{'creator': 'Libiseller-Egger, Julian'},\n",
       "   {'creator': 'Phelan, Jody E.'},\n",
       "   {'creator': 'Attia, Zachi I.'},\n",
       "   {'creator': 'Benavente, Ernest Diez'},\n",
       "   {'creator': 'Campino, Susana'},\n",
       "   {'creator': 'Friedman, Paul A.'},\n",
       "   {'creator': 'Lopez-Jimenez, Francisco'},\n",
       "   {'creator': 'Leon, David A.'},\n",
       "   {'creator': 'Clark, Taane G.'}],\n",
       "  'publicationName': 'Scientific Reports',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1038/s41598-022-27254-z',\n",
       "  'publisher': 'Nature',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2045-2322',\n",
       "  'volume': '12',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '11',\n",
       "  'journalId': '41598',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Artificial intelligence (AI)-based approaches can now use electrocardiograms (ECGs) to provide expert-level performance in detecting heart abnormalities and diagnosing disease. Additionally, patient age predicted from ECGs by AI models has shown great potential as a biomarker for cardiovascular age, where recent work has found its deviation from chronological age (“delta age”) to be associated with mortality and co-morbidities. However, despite being crucial for understanding underlying individual risk, the genetic underpinning of delta age is unknown. In this work we performed a genome-wide association study using UK Biobank data (n=34,432) and identified eight loci associated with delta age ( $$p\\\\le 5 \\\\times 10^{-8}$$ p ≤ 5 × 10 - 8 ), including genes linked to cardiovascular disease (CVD) (e.g. SCN5A ) and (heart) muscle development (e.g. TTN ). Our results indicate that the genetic basis of cardiovascular ageing is predominantly determined by genes directly involved with the cardiovascular system rather than those connected to more general mechanisms of ageing. Our insights inform the epidemiology of CVD, with implications for preventative and precision medicine.',\n",
       "  'subjects': ['Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, multidisciplinary']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11192-022-04625-x',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11192-022-04625-x'}],\n",
       "  'title': 'The impact of Jürgen Habermas’s scientific production: a scientometric review',\n",
       "  'creators': [{'ORCID': '0000-0002-7260-5063', 'creator': 'Carmo, Gisleine'},\n",
       "   {'creator': 'Felizardo, Luiz Flávio'},\n",
       "   {'creator': 'Castro Alcântara, Valderí'},\n",
       "   {'creator': 'Silva, Cristiane Aparecida'},\n",
       "   {'creator': 'Prado, José Willer'}],\n",
       "  'publicationName': 'Scientometrics',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11192-022-04625-x',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1588-2861',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '23',\n",
       "  'journalId': '11192',\n",
       "  'copyright': '©2022 Akadémiai Kiadó, Budapest, Hungary',\n",
       "  'abstract': 'The use of bibliometrics, based on statistical and mathematical tools, makes it possible to measure the contributions of researchers to science. This is a widely used tool to assess scientific production in several areas of knowledge. Such methodology analyzes publication trends, author networks, structures of co-citation, journals and even the scientific contribution of renowned scholars in science. The precursor of bibliometrics, Eugene Garfield, who proposes the retrieval of information from the indexing of citations, was the object of a scientometric review aimed at assessing his impact on science. Given such relevance, this article presents the academic contribution of Jürgen Habermas based on a preliminary scientometric review of his studies. Jürgen Habermas is regarded not only as an active scholar in the social and political process, but also as a productive, controversial and influential contemporary author. The correct understanding of his works is a great challenge, as the bases of his thinking are so broad that they allow an interface between different approaches. We elaborated a design of his scientific work with the advancement to a connection between his main ideas through the use of bibliometric software. Bibliometrics, of the scientometric type, allows the understanding of how recognized patterns in citations can develop information relevant to the scientific field. The results confirm the multidisciplinary contribution of Habermas’s studies and highlight his main fields of research and works, which serve as a foundation for clearly understanding and applying his concepts.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Information Storage and Retrieval',\n",
       "   'Library Science']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s12524-022-01641-5',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s12524-022-01641-5'}],\n",
       "  'title': 'Predicting Spatial and Decadal LULC Changes in the Singrauli District of Madhya Pradesh Through Artificial Neural Network Models Using Geospatial Technology',\n",
       "  'creators': [{'ORCID': '0000-0001-7479-2803', 'creator': 'Kumari, Maya'},\n",
       "   {'creator': 'Sarma, Kiranmay'},\n",
       "   {'creator': 'Sharma, Richa'}],\n",
       "  'publicationName': 'Journal of the Indian Society of Remote Sensing',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s12524-022-01641-5',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '0974-3006',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Research Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '12',\n",
       "  'journalId': '12524',\n",
       "  'copyright': '©2022 Indian Society of Remote Sensing',\n",
       "  'abstract': 'The increasing energy demand of the burgeoning population in India has propelled the development of new thermal power plants in the country. Availability of plenty of power-grade coal and water in the form of Govind Ballabh Pant Sagar Dam led to the establishment of several thermal power plants (TPP) in the Singrauli district of Madhya Pradesh. After the construction and post-construction activities of these TPP in the district, the pristine dense forests have completely been transformed into degraded land. In the present work, change detection analysis is carried out for three-time gradients, i.e., from 2005 to 2015 at the interval of five years using Quantum GIS software to understand the land use dynamics. It is observed that the area has witnessed significant transformation during the three-time gradients, including approximately a\\xa06.9% decline in forest cover and a\\xa023% increase in the\\xa0barren surface. Further, artificial neural network (ANN) is employed in the present work to simulate and predict land use of the year 2025. The projected LULC for 2025 reveals that as the power plants operational activities will continue and expand further, there will be a substantial increase in fallow land and scrub land by the year 2025. The percentage of the\\xa0correctness of the\\xa0predicted LULC map is 98.21 with\\xa0an\\xa0 overall Kappa index of 0.97 and a\\xa0kappa histogram of 0.98. Details of the past, present and predicted LULC of an area aid the urban planners in informed decision making so as to choose, design, and implement future land use plans of the area in a sustainable manner.',\n",
       "  'subjects': ['Earth Sciences',\n",
       "   'Earth Sciences, general',\n",
       "   'Remote Sensing/Photogrammetry']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1186/s13321-022-00670-z',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1186/s13321-022-00670-z'}],\n",
       "  'title': 'TCMSID: a simplified integrated database for drug discovery from traditional chinese medicine',\n",
       "  'creators': [{'creator': 'Zhang, Liu-Xia'},\n",
       "   {'creator': 'Dong, Jie'},\n",
       "   {'creator': 'Wei, Hui'},\n",
       "   {'creator': 'Shi, Shao-Hua'},\n",
       "   {'creator': 'Lu, Ai-Ping'},\n",
       "   {'creator': 'Deng, Gui-Ming'},\n",
       "   {'creator': 'Cao, Dong-Sheng'}],\n",
       "  'publicationName': 'Journal of Cheminformatics',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1186/s13321-022-00670-z',\n",
       "  'publisher': 'BioMed Central',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1758-2946',\n",
       "  'volume': '14',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Database'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '11',\n",
       "  'journalId': '13321',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Traditional Chinese Medicine (TCM) has been widely used in the treatment of various diseases for millennia. In the modernization process of TCM, TCM ingredient databases are playing more and more important roles. However, most of the existing TCM ingredient databases do not provide simplification function for extracting key ingredients in each herb or formula, which hinders the research on the mechanism of actions of the ingredients in TCM databases. The lack of quality control and standardization of the data in most of these existing databases is also a prominent disadvantage. Therefore, we developed a Traditional Chinese Medicine Simplified Integrated Database (TCMSID) with high storage, high quality and standardization. The database includes 499 herbs registered in the Chinese pharmacopeia with 20,015 ingredients, 3270 targets as well as corresponding detailed information. TCMSID is not only a database of herbal ingredients, but also a TCM simplification platform. Key ingredients from TCM herbs are available to be screened out and regarded as representatives to explore the mechanism of TCM herbs by implementing multi-tool target prediction and multilevel network construction. TCMSID provides abundant data sources and analysis platforms for TCM simplification and drug discovery, which is expected to promote modernization and internationalization of TCM and enhance its international status in the future. TCMSID is freely available at https://tcm.scbdd.com .',\n",
       "  'subjects': ['Chemistry',\n",
       "   'Computer Applications in Chemistry',\n",
       "   'Documentation and Information in Chemistry',\n",
       "   'Theoretical and Computational Chemistry',\n",
       "   'Computational Biology/Bioinformatics']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s12063-022-00340-1',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s12063-022-00340-1'}],\n",
       "  'title': 'Leveraging blockchain in response to a pandemic through disaster risk management: an IF-MCDM framework',\n",
       "  'creators': [{'ORCID': '0000-0001-6402-5117', 'creator': 'Peker, Iskender'},\n",
       "   {'creator': 'AR, Ilker Murat'},\n",
       "   {'creator': 'Erol, Ismail'},\n",
       "   {'creator': 'Searcy, Cory'}],\n",
       "  'publicationName': 'Operations Management Research',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s12063-022-00340-1',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1936-9743',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '26',\n",
       "  'journalId': '12063',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'The COVID-19 pandemic has underscored the necessity for strategic approaches to manage global catastrophes. This study argues that disaster risk management (DRM) is an essential approach to mitigate the impact of global calamities, such as a pandemic. However, due to the uncertainty emerging from variables such as time and demand, managing DRM effectively involves high complexity. Blockchain technology (BcT) can be implemented to help address these challenges due to its potential to build trust, transparency, and accountability in complex operations. However, no research has quantitatively examined the applicability of BcT in the sub-dimensions of DRM to address the COVID-19 pandemic. The goal and scope of this study is to explore the role of BcT in DRM operations during the COVID-19 pandemic through an Intuitionistic Fuzzy Multi-Criteria Decision Making (IF-MCDM) framework. More specifically, the Intuitionistic Fuzzy Analytic Network Process (IF-ANP) method was utilized to calculate the weights of key criteria (i.e., BcT benefits), while Intuitionistic Fuzzy VIKOR (IF-VIKOR) was used to prioritize the alternatives (i.e., the sub-dimensions of DRM). The findings of this study are threefold. First, supporting effective coordination turned out to be the most essential benefit of BcT to build resilience in response to the COVID-19 pandemic. Second, disaster management was found to be the most appropriate DRM sub-dimension for possible BcT implementation during the COVID-19 pandemic. Lastly, eleven distinct activities involved in disaster management and governance and financial protection were discovered to be the most applicable for BcT. The findings of this study could assist disaster risk managers to assess whether (or not) BcT is suitable for the sub-dimensions of DRM to build national and organizational resilience in the wake of the COVID-19 pandemic.',\n",
       "  'subjects': ['Business and Management',\n",
       "   'Operations Management',\n",
       "   'Engineering Economics, Organization, Logistics, Marketing',\n",
       "   'Industrial and Production Engineering',\n",
       "   'Operations Research/Decision Theory',\n",
       "   'Management',\n",
       "   'Innovation/Technology Management']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s10528-022-10323-3',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s10528-022-10323-3'}],\n",
       "  'title': 'Integrative Bioinformatics Analysis of mRNA Expression Profiles of Mice to Explore the Key Genes Involved in Crim1 Mutation-Induced Congenital Cataracts',\n",
       "  'creators': [{'creator': 'Zhang, Ziran'},\n",
       "   {'creator': 'Meng, Fanke'},\n",
       "   {'creator': 'Zhou, Jing'},\n",
       "   {'creator': 'Zhang, Zhihan'},\n",
       "   {'creator': 'Liang, Xiaotian'},\n",
       "   {'creator': 'Meng, Meijun'},\n",
       "   {'creator': 'Yi, Guoguo'},\n",
       "   {'creator': 'Fu, Min'}],\n",
       "  'publicationName': 'Biochemical Genetics',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1007/s10528-022-10323-3',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-4927',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '17',\n",
       "  'journalId': '10528',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Crim1 has been implicated in cataracts in mice and is of great importance in the development of the eye in both humans and mice. Therefore, we aimed to clarify how Crim1 mutations affect lens development and the molecular mechanism of cataracts in mice through comprehensive bioinformatics analysis. The microarray chip was downloaded from the GEO database to obtain the gene expression profile data set. Differentially expressed genes (DEGs) were screened using the limma package. GO and KEGG analyses of DEGs were performed using the DAVID database. Then, we established the protein–protein interaction (PPI) network in Cytoscape. Next, we used MCODE to analyze the data. We obtained 750 DEGs in total, including 407 upregulated DEGs and 343 downregulated DEGs. GO analysis showed that the DEGs were mainly related to biological processes, such as apoptosis, cell translation and the immune system. KEGG analysis showed that the enriched functions and pathways were related to the processing and presentation of ribosomes, lysosomes, and antigens. We identified 18 HUB genes, among which four core genes, C1qa, C1qb, C1qc, and Cd74, were closely related to congenital cataracts induced by Crim1 mutation. This study reveals the molecular pathogenesis of congenital cataracts induced by Crim1, and this information is expected to facilitate clinical genetic testing, molecular diagnosis, prognosis, and individualized chemotherapy for congenital cataracts (CC).',\n",
       "  'subjects': ['Biomedicine',\n",
       "   'Human Genetics',\n",
       "   'Biochemistry, general',\n",
       "   'Zoology',\n",
       "   'Medical Microbiology']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s41870-022-01148-9',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s41870-022-01148-9'}],\n",
       "  'title': 'An energy efficient robust resource provisioning based on improved PSO-ANN',\n",
       "  'creators': [{'creator': 'Srivastava, Ankita'},\n",
       "   {'ORCID': '0000-0003-4633-1879', 'creator': 'Kumar, Narander'}],\n",
       "  'publicationName': 'International Journal of Information Technology',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s41870-022-01148-9',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2511-2112',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Research'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '11',\n",
       "  'journalId': '41870',\n",
       "  'copyright': \"©2022 The Author(s), under exclusive licence to Bharati Vidyapeeth's Institute of Computer Applications and Management\",\n",
       "  'abstract': \"High-performance cloud computing has revolutionized computing. Resource provisioning has become an indispensable barrier in serving the high-end service facilitating computing paradigm. Energy efficient provisioning has become the major obstacle in the cloud as it downsizes the cost of the cloud system. This paper has proposed a technique to resolve the provisioning problem by scheduling the tasks to virtual machines (VMs) along with special consideration for energy usage. The technique aimed to elevate the potency of the scheduling algorithm in the real-time cloud application by curtailing down the system's energy consumption, makespan, and average start time. The technique has been inspired by the Artificial Neural Network (ANN), utilizing the Particle Swarm Optimization (PSO) algorithm to train the datasets. The technique has been implemented on the Cloudsim tool and compared with state of the art. The result obtained has shown that the technique has turned out to be performing better by improving the energy efficiency by 12%, makespan by an average of 20%, and the average start time by more than 50%.\",\n",
       "  'subjects': ['Computer Science',\n",
       "   'Computer Science, general',\n",
       "   'Artificial Intelligence',\n",
       "   'Machine Learning',\n",
       "   'Image Processing and Computer Vision',\n",
       "   'Software Engineering',\n",
       "   'Computer Imaging, Vision, Pattern Recognition and Graphics']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s13278-022-01019-y',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s13278-022-01019-y'}],\n",
       "  'title': 'Early multi-class ensemble-based fake news detection using content features',\n",
       "  'creators': [{'creator': 'Rezaei, Sajjad'},\n",
       "   {'creator': 'Kahani, Mohsen'},\n",
       "   {'creator': 'Behkamal, Behshid'},\n",
       "   {'creator': 'Jalayer, Abdulrahman'}],\n",
       "  'publicationName': 'Social Network Analysis and Mining',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s13278-022-01019-y',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1869-5469',\n",
       "  'volume': '13',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Original Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '11',\n",
       "  'journalId': '13278',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer-Verlag GmbH Austria, part of Springer Nature',\n",
       "  'abstract': 'Nowadays, social media plays an essential role in spreading the news with low cost and high speed in publishing, and easy availability. Given that, anyone can publish any news on social networks, with some of them to be fake. These fake stories should be detected as soon as possible since they might have negative impacts on the society. To address this issue, most researches consider fake news detection as a binary classification problem. However, as some news are half-true, recently, multi-class detection has gained more attention. This paper investigates an early detection of fake news using multi-class classification. This is achieved by extracting the content features, such as sentiment and semantic features, from the news. The proposed model employs five classifiers (Random Forest, Support Vector Machine, Decision Tree, LightGBM, and XGBoost) as primary classifiers. Furthermore, AdaBoost is used for the meta-learning algorithm to develop a stacking generalization model. Stacking generalization is an ensemble learning method that uses all data produced by the first-level algorithms. We trained our model with PolitiFact data for the evaluation, and the model performance was evaluated by Accuracy, Precision, Recall, and F1 score. Excremental evaluation of the real-world datasets showed that our proposed model outperformed all previous works in both binary and multi-class classifications.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Data Mining and Knowledge Discovery',\n",
       "   'Applications of Graph Theory and Complex Networks',\n",
       "   'Game Theory, Economics, Social and Behav. Sciences',\n",
       "   'Statistics for Social Sciences, Humanities, Law',\n",
       "   'Methodology of the Social Sciences']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11276-022-03215-2',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11276-022-03215-2'}],\n",
       "  'title': 'A mathematical analysis about the geo-temporal characterization of the multi-class maliciousness of an IP address',\n",
       "  'creators': [{'ORCID': '0000-0002-5610-0153',\n",
       "    'creator': 'DeCastro-García, Noemí'},\n",
       "   {'creator': 'Escudero García, David'},\n",
       "   {'creator': 'Carriegos, Miguel V.'}],\n",
       "  'publicationName': 'Wireless Networks',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11276-022-03215-2',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1572-8196',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '16',\n",
       "  'journalId': '11276',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer Science+Business Media, LLC, part of Springer Nature',\n",
       "  'abstract': 'The degree of severity of a cybersecurity event of potentially malicious activity is crucial to determine an appropriate response. Machine learning techniques are used to obtain models to achieve a correct characterization of events as well as an accurate maliciousness measure. A model is constructed by setting up an effective collection of features that are able to characterize the maliciousness of IP addresses. As a first approach, a set of contextual features has been selected. These features are simple to extract and require no high costs. That set contains 23 features: 18 obtained through time series analysis, and the other 5 extracted directly related with the spatial (geolocation) and temporal (time of occurrence) similarity of the events. The test of the feature set is conducted with statistical analyses to determine which features are the most effective and the impact of hyperparameter selection in the construction of a time series of features. In addition, the effect of IP geo-localization shifts on model performance is studied through concept drift theory. The results conclude that, overall, the feature set provides adequate performance for our task, although more complex features may be required to achieve better performance. The best results are obtained with the geolocation and time occurrence features.',\n",
       "  'subjects': ['Engineering',\n",
       "   'Communications Engineering, Networks',\n",
       "   'Computer Communication Networks',\n",
       "   'Electrical Engineering',\n",
       "   'IT in Business']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s43032-022-01161-0',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s43032-022-01161-0'}],\n",
       "  'title': 'Identification of Common and Specific Genes Involved in Mouse Models of Age-Related and Cyclophosphamide-Induced Diminished Ovarian Reserve',\n",
       "  'creators': [{'creator': 'He, Ruifen'},\n",
       "   {'creator': 'Fan, Qigang'},\n",
       "   {'creator': 'Li, Yi'},\n",
       "   {'creator': 'Zhu, Qinying'},\n",
       "   {'creator': 'Hu, Dan'},\n",
       "   {'creator': 'Du, Junhong'},\n",
       "   {'creator': 'Xing, Yijuan'},\n",
       "   {'creator': 'Li, Hongli'},\n",
       "   {'creator': 'Liang, Xiaolei'},\n",
       "   {'ORCID': '0000-0002-1449-4326', 'creator': 'Yang, Yongxiu'}],\n",
       "  'publicationName': 'Reproductive Sciences',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s43032-022-01161-0',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1933-7205',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Reproductive Endocrinology: Original Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '14',\n",
       "  'journalId': '43032',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Society for Reproductive Investigation',\n",
       "  'abstract': 'Diminished ovarian reserve (DOR) is an etiologically heterogeneous disorder that usually leads to poor reproductive outcomes. Does a specific or common pathogenesis exist for DOR subtypes with different etiologies? Two frequently used mouse models, age-related DOR (AR-DOR) and cyclophosphamide (CTX)-induced DOR (CTX-DOR), were successfully established, and RNA sequencing was performed on ovarian tissue samples. Differentially expressed genes (DEGs) in each subtype and common DEGs (co-DEGs) in the two subtypes were identified. Subsequently, we performed comprehensive bioinformatics analyses, including an evaluation of immune cell infiltration. Finally, the genes of interest were further validated by performing RT-qPCR and immunohistochemistry. In AR-DOR mice, functional enrichment analyses showed that upregulated DEGs were mainly involved in the inflammatory/immune response, while downregulated DEGs were involved in DNA damage repair. In CTX-DOR mice, the inflammatory/immune response and cell apoptosis played significant roles. Meanwhile, 406 co-DEGs were identified from the two models. The biological functions of these co-DEGs were associated with inflammatory/immune responses. The analysis of immune cell infiltration showed reduced infiltration of Treg cells, as well as increased infiltration of M0 macrophages, NK resting, and T cells CD4 follicular in both DOR subtypes. The results of the validation experiments were consistent with the RNA sequencing data. In conclusion, the inflammatory/immune response might be the common pathogenesis for the two DOR subtypes, while DNA repair and cell apoptosis may have different roles in the two subtypes. These results may provide potential insights for mechanistic research and therapeutic targets of DOR.',\n",
       "  'subjects': ['Medicine & Public Health',\n",
       "   'Reproductive Medicine',\n",
       "   'Embryology',\n",
       "   'Obstetrics/Perinatology/Midwifery']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s10971-022-06012-7',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s10971-022-06012-7'}],\n",
       "  'title': 'In-vitro degradation behavior of hybrid epoxy-alkyl sol–gel/anodized composite coating on AZ31B Mg alloy',\n",
       "  'creators': [{'ORCID': '0000-0002-8936-0533', 'creator': 'Merino, E.'},\n",
       "   {'creator': 'El Tawil, M.'},\n",
       "   {'creator': 'Sobrados, I.'},\n",
       "   {'creator': 'Durán, A.'},\n",
       "   {'creator': 'Castro, Y.'}],\n",
       "  'publicationName': 'Journal of Sol-Gel Science and Technology',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1007/s10971-022-06012-7',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1573-4846',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper',\n",
       "   'Original Paper: Sol-gel and hybrid materials with surface modification for applications'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '9',\n",
       "  'journalId': '10971',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Mg alloys are considered a promising alternative to use as biodegradable implants. However, the main limitations of these alloys are the high susceptibility to hydrogen evolution and corrosion. Consequently, it is necessary to control the corrosion rate through surface treatment techniques. In this regard, a double layer system, based on the first deposition of a MgO coating and the subsequent deposition of an organo-silane sol-gel coating (anodized/SGMI) was proposed to increase the biocompatibility and reduce the corrosion rate of Mg alloy. Sol-gel coatings were prepared using methyltriethoxysilane (MTES), (glycidyloxypropyl) trimethoxysilane (GPTMS), SiO 2 nanoparticles, and 1-methyl imidazole (1-MI). 29 Si and 13 C NMR measurements were performed to investigate the inorganic-organic network formation during the sol-gel synthesis, and to characterize the structure of the cured silane coating. According to the results, the addition of 1-MI mainly promotes the condensation reaction of GPTMS and MTES. After sol-gel deposition, a higher cross-linking coating network was obtained and the degradation of AZ31B Mg alloy in simulated body fluid (SBF) was slowed down. The control of the hydrogen and pH evolution over 450\\u2009h suggest that the anodized/SGMI coating system may be suitable for short-term implants applications; particularly for applications that are expected to degrade after 10-12 days such as surgical skin staples. Graphical Abstract To control the corrosion of AZ31B Mg alloy in simulated body fluid (SBF) using a double layer system based on anodized and sol-gel coating. The corrosion rate of AZ31B Mg alloy in simulated body fluid (SBF) solution was controlled by using anodizing and sol-gel technique. A highly cross-linked silica network was successfully prepared using methyltriethoxysilane (MTES), (glycidyloxypropyl) trimethoxysilane (GPTMS), SiO 2 nanoparticles, and 1-methyl imidazole (1-MI). The crack-free epoxy-alkyl silane coating deposited on the top of the anodized substrates allows the evolution of hydrogen and pH to be controlled. The hydrophilic nature and in vitro results of the coated AZ31B Mg alloy substrates suggest that the proposed coating system is suitable for short-term implant applications.',\n",
       "  'subjects': ['Materials Science',\n",
       "   'Ceramics, Glass, Composites, Natural Materials',\n",
       "   'Inorganic Chemistry',\n",
       "   'Optical and Electronic Materials',\n",
       "   'Nanotechnology']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11334-022-00522-x',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11334-022-00522-x'}],\n",
       "  'title': 'Automated credit assessment framework using ETL process and machine learning',\n",
       "  'creators': [{'creator': 'Biswas, Neepa'},\n",
       "   {'creator': 'Mondal, Anindita Sarkar'},\n",
       "   {'creator': 'Kusumastuti, Ari'},\n",
       "   {'creator': 'Saha, Swati'},\n",
       "   {'ORCID': '0000-0003-3647-5799', 'creator': 'Mondal, Kartick Chandra'}],\n",
       "  'publicationName': 'Innovations in Systems and Software Engineering',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11334-022-00522-x',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1614-5054',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper',\n",
       "   'S.I. : Low Resource Machine Learning Algorithms (LR-MLA)'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '14',\n",
       "  'journalId': '11334',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer-Verlag London Ltd., part of Springer Nature',\n",
       "  'abstract': 'In the current business scenario, real-time analysis of enterprise data through Business Intelligence (BI) is crucial for supporting operational activities and taking any strategic decision. The automated ETL (extraction, transformation, and load) process ensures data ingestion into the data warehouse in near real-time, and insights are generated through the BI process based on real-time data. In this paper, we have concentrated on automated credit risk assessment in the financial domain based on the machine learning approach. The machine learning-based classification techniques can furnish a self-regulating process to categorize data. Establishing an automated credit decision-making system helps the lending institution to manage the risks, increase operational efficiency and comply with regulators. In this paper, an empirical approach is taken for credit risk assessment using logistic regression and neural network classification method in compliance with Basel II standards. Here, Basel II standards are adopted to calculate the expected loss. The required data integration for building machine learning models is done through an automated ETL process. We have concluded this research work by evaluating this new methodology for credit risk assessment.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Software Engineering',\n",
       "   'Artificial Intelligence',\n",
       "   'Computer Applications']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s00429-022-02608-5',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s00429-022-02608-5'}],\n",
       "  'title': 'How processing emotion affects language control in bilinguals',\n",
       "  'creators': [{'creator': 'Liu, Huanhuan'},\n",
       "   {'creator': 'Liu, Wenxin'},\n",
       "   {'creator': 'Schwieter, John W.'},\n",
       "   {'creator': 'Wu, Yan Jing'}],\n",
       "  'publicationName': 'Brain Structure and Function',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s00429-022-02608-5',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1863-2661',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '15',\n",
       "  'journalId': '429',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer-Verlag GmbH Germany, part of Springer Nature',\n",
       "  'abstract': 'Research has shown that several variables affect language control among bilingual speakers but the effect of affective processing remains unexplored. Chinese–English bilinguals participated in a novel prime-target language switching experiment in which they first judged the affective valence (i.e., positive or negative) of auditorily presented words and then named pictures with neutral emotional valence in either the same (non-switch trial) or different language (switch trial). Brain activity was monitored using functional magnetic resonance imaging (fMRI). The behavioral performance showed that the typical switch cost (i.e., the calculated difference between switch and non-switch trials) emerged after processing positive words but not after negative words. Brain imaging demonstrated that processing negative words immediately before non-switch picturing naming trials (but not for switch trials) increased activation in brain areas associated with domain-general cognitive control. The opposite patterns were found after processing positive words. These findings suggest that an (emotional) negative priming effect is induced by spontaneous exposure to negative words and that these priming effects may be triggered by reactive emotional processing and that they may interact with higher level cognitive functions.',\n",
       "  'subjects': ['Biomedicine', 'Neurosciences', 'Cell Biology', 'Neurology']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s10098-022-02462-3',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s10098-022-02462-3'}],\n",
       "  'title': 'Economic and environmental analysis of composite recycling techniques for the design of future waste management supply chains',\n",
       "  'creators': [{'creator': 'Jiang, Xiaoli'},\n",
       "   {'creator': 'Shao, Minghui'},\n",
       "   {'creator': 'Bateer, Buhe'}],\n",
       "  'publicationName': 'Clean Technologies and Environmental Policy',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s10098-022-02462-3',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1618-9558',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Original Paper'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '13',\n",
       "  'journalId': '10098',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer-Verlag GmbH Germany, part of Springer Nature',\n",
       "  'abstract': 'Carbon fiber-reinforced polymer composites have permeated every aspect of modern life. Significant progress has been made in engineering communities to recycle composite wastes. While emerging recycling techniques can recover both fibers and resins at mild processing conditions, they are currently limited on the laboratory scale. It remains unclear if they could lead to market viability and environmental sustainability when they are used for industrial applications; if not, what the limiting factors are and how to improve them. In this study, the environmental and economic impacts of existing recycling techniques are studied. The average recycling costs are analyzed, and the environmental impacts are examined. It reveals that while the emerging techniques can recycle composites with extremely low environmental impact, their costs are significant compared to conventional techniques, primarily due to the expensive solvents and catalysts used to enable recycling. However, the study shows that it is possible to combine conventional waste processing techniques with emerging ones to enable recycling scenarios that are both cost-effective and environmentally beneficial. Overall, the study in this paper provides valuable insights for the design of future waste management supply chains and appropriate business models for the sustainable development of society. Graphical abstract ',\n",
       "  'subjects': ['Environment',\n",
       "   'Sustainable Development',\n",
       "   'Industrial Chemistry/Chemical Engineering',\n",
       "   'Industrial and Production Engineering',\n",
       "   'Environmental Engineering/Biotechnology',\n",
       "   'Environmental Economics']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s42600-022-00250-y',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s42600-022-00250-y'}],\n",
       "  'title': 'Quantitative analysis of second harmonic generated images of collagen fibers: a review',\n",
       "  'creators': [{'ORCID': '0000-0001-5135-6320', 'creator': 'Nejim, Zeineb'},\n",
       "   {'creator': 'Navarro, Laurent'},\n",
       "   {'creator': 'Morin, Claire'},\n",
       "   {'creator': 'Badel, Pierre'}],\n",
       "  'publicationName': 'Research on Biomedical Engineering',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1007/s42600-022-00250-y',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2446-4740',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['ReviewPaper', 'Review'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '23',\n",
       "  'journalId': '42600',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Purpose The human body is a complex structure. Its strength is ensured by the collagen protein which exists under the form of fibers. The quantitative analysis of these fibers in biological tissues can be very interesting to establish a relationship between the microstructure and their functions. This analysis is usually performed using two-photon microscopy and second harmonic generated (SHG) images. Lately, more and more researchers focused on the use of SHG images since it is a non-invasive technique and allows the capture of collagen fibers only. Many image-processing techniques can be used to extract quantitative information from those images such as fiber orientations, dimensions, and density. Therefore, accurate measure extraction depends mainly on the used image processing methods and, thus, it is necessary to know what processing technique to use. Methods The main purpose of this article is to exhibit the most used techniques in collagen fiber quantitative analysis then categorize them according to the information to extract. A comparison of three most used methods in fiber orientation’s estimation is carried out. Result and conclusion Despite the considerable number of papers aiming to quantitatively analyze collagen fibers from SHG images, two main aspects were not deeply covered. First, the use of deep learning algorithms is still limited even for segmentation and denoizing applications. Second, most of the studies processed in this review focused on two-dimensional SHG images and did not take into consideration collagen fibers as a three-dimensional volume.',\n",
       "  'subjects': ['Engineering',\n",
       "   'Biomedical Engineering and Bioengineering',\n",
       "   'Biomaterials',\n",
       "   'Biomedical Engineering/Biotechnology']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s12094-022-03034-z',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s12094-022-03034-z'}],\n",
       "  'title': 'Predicted cardiac and second cancer risks for patients undergoing VMAT for mediastinal Hodgkin lymphoma',\n",
       "  'creators': [{'ORCID': '0000-0001-9199-460X',\n",
       "    'creator': 'Houlihan, Orla A.'},\n",
       "   {'creator': 'Ntentas, Georgios'},\n",
       "   {'creator': 'Cutter, David J.'},\n",
       "   {'creator': 'Daly, Patricia'},\n",
       "   {'creator': 'Gillham, Charles'},\n",
       "   {'creator': 'McArdle, Orla'},\n",
       "   {'creator': 'Duane, Frances K.'}],\n",
       "  'publicationName': 'Clinical and Translational Oncology',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1007/s12094-022-03034-z',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1699-3055',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'RESEARCH ARTICLE'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '10',\n",
       "  'journalId': '12094',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Background and purpose To predict treatment-related cardiovascular disease (CVD) and second cancer 30-year absolute mortality risks (AMR 30 ) for patients with mediastinal Hodgkin lymphoma in a large multicentre radiation oncology network in Ireland. Material and methods This study includes consecutive patients treated for mediastinal lymphoma using chemotherapy and involved site radiotherapy (RT) 2016–2019. Radiation doses to heart, left ventricle, cardiac valves, lungs, oesophagus, carotid arteries and female breasts were calculated. Individual CVD and second cancer AMR 30 were predicted using Irish background population rates and dose–response relationships. Results Forty-four patients with Hodgkin lymphoma were identified, 23 females, median age 28 years. Ninety-eight percent received anthracycline, 80% received 4–6 cycles ABVD. Volumetric modulated arc therapy (VMAT)\\u2009±\\u2009deep inspiration breath hold (DIBH) was delivered, median total prescribed dose 30\\xa0Gy. Average mean heart dose 9.8\\xa0Gy (range 0.2–23.8\\xa0Gy). Excess treatment-related mean AMR 30 from CVD was 2.18% (0.79, 0.90, 0.01, 0.13 and 0.35% for coronary disease, heart failure, valvular disease, stroke and other cardiac diseases), 1.07% due to chemotherapy and a further 1.11% from RT. Excess mean AMR 30 for second cancers following RT were: lung cancer 2.20%, breast cancer in females 0.34%, and oesophageal cancer 0.28%. Conclusion For patients with mediastinal lymphoma excess mortality risks from CVD and second cancers remain clinically significant despite contemporary chemotherapy and photon-RT. Efforts to reduce the toxicity of combined modality treatment, for example, using DIBH, reduced margins and advanced RT, e.g. proton beam therapy, should be continued to further reduce potentially fatal treatment effects.',\n",
       "  'subjects': ['Medicine & Public Health', 'Oncology']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1038/s41598-022-27209-4',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1038/s41598-022-27209-4'}],\n",
       "  'title': 'Design and analysis of welding inspection robot',\n",
       "  'creators': [{'creator': 'Zhang, Pengyu'},\n",
       "   {'creator': 'Wang, Ji'},\n",
       "   {'creator': 'Zhang, Feng'},\n",
       "   {'creator': 'Xu, Peiquan'},\n",
       "   {'creator': 'Li, Leijun'},\n",
       "   {'creator': 'Li, Baoming'}],\n",
       "  'publicationName': 'Scientific Reports',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1038/s41598-022-27209-4',\n",
       "  'publisher': 'Nature',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2045-2322',\n",
       "  'volume': '12',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '16',\n",
       "  'journalId': '41598',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Periodic inspection, commonly performed by a technician, of weld seam quality is important for assessing equipment reliability. To save labor costs and improve efficiency, an autonomous navigation and inspection robot is developed. The development process involves the design of chassis damping, target detection mechanism, control system, and algorithms. For performing weld inspection in complex, outdoor, environments, an algorithm is developed for the robot to avoid any obstacles. This algorithm for planning the inspection route is based on an improved timed-elastic-band (TEB) algorithm. The developed robot is capable of conducting inspection tasks in complex and dangerous environments efficiently and autonomously.',\n",
       "  'subjects': ['Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, multidisciplinary']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1038/s41598-022-27140-8',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1038/s41598-022-27140-8'}],\n",
       "  'title': 'Reverse engineering for reconstructing baseline features of dry age-related macular degeneration in optical coherence tomography',\n",
       "  'creators': [{'creator': 'Wang, Shuxian'},\n",
       "   {'creator': 'Wang, Ziyuan'},\n",
       "   {'creator': 'Vejalla, Srimanasa'},\n",
       "   {'creator': 'Ganegoda, Anushika'},\n",
       "   {'creator': 'Nittala, Muneeswar Gupta'},\n",
       "   {'creator': 'Sadda, SriniVas Reddy'},\n",
       "   {'creator': 'Hu, Zhihong Jewel'}],\n",
       "  'publicationName': 'Scientific Reports',\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1038/s41598-022-27140-8',\n",
       "  'publisher': 'Nature',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '2045-2322',\n",
       "  'volume': '12',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '10',\n",
       "  'journalId': '41598',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Age-related macular degeneration (AMD) is the most widespread cause of blindness and the identification of baseline AMD features or biomarkers is critical for early intervention. Optical coherence tomography (OCT) imaging produces a 3D volume consisting of cross sections of retinal tissue while fundus fluorescence (FAF) imaging produces a 2D mapping of retina. FAF has been a good standard for assessing dry AMD late-stage geographic atrophy (GA) while OCT has been used for assessing early AMD biomarkers beyond as well. However, previous approaches in large extent defined AMD features subjectively based on clinicians’ observation. Deep learning—an objective artificial intelligence approach, may enable to discover ’true’ salient AMD features. We develop a novel reverse engineering approach which bases on the backbone of a fully convolutional neural network to objectively identify and visualize AMD early biomarkers in OCT from baseline exams before significant atrophy occurs. Utilizing manually annotated GA regions on FAF from a follow-up visit as ground truth, we segment GA regions and reconstruct early AMD features in baseline OCT volumes. In this preliminary exploration, compared with ground truth, we achieve baseline GA segmentation accuracy of 0.95 and overlapping ratio of 0.65. The reconstructions consistently highlight that large druse and druse clusters with or without mixed hyper-reflective focus lesion on baseline OCT cause the conversion of GA after 12\\xa0months. However, hyper-reflective focus lesions and subretinal drusenoid deposit lesions alone are not seen such conversion after 12\\xa0months. Further research with larger dataset would be needed to verify these findings.',\n",
       "  'subjects': ['Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, Humanities and Social Sciences, multidisciplinary',\n",
       "   'Science, multidisciplinary']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s11192-022-04627-9',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s11192-022-04627-9'}],\n",
       "  'title': 'An artificial intelligence-based framework for data-driven categorization of computer scientists: a case study of world’s Top 10 computing departments',\n",
       "  'creators': [{'creator': 'Ali, Nisar'},\n",
       "   {'ORCID': '0000-0003-3094-3483', 'creator': 'Halim, Zahid'},\n",
       "   {'creator': 'Hussain, Syed Fawad'}],\n",
       "  'publicationName': 'Scientometrics',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s11192-022-04627-9',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1588-2861',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': 'OriginalPaper',\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '33',\n",
       "  'journalId': '11192',\n",
       "  'copyright': '©2022 Akadémiai Kiadó, Budapest, Hungary',\n",
       "  'abstract': 'The total number of published articles and the resulting citations are generally acknowledged as suitable criteria of the scientist’s evaluation. However, it is challenging to determine the ranking of scientists as the value of their scientific work (at times) is not directly reflective of the abovementioned aspects. In this regard, multiple other elements needs to be examined in combination for better evaluating the scientific worth of an individual. This work presents a learning-based technique, i.e., an Artificial Intelligence (AI)-based solution towards categorizing scientists utilizing a multifaceted criteria. In this context, a novel ranking metric is proposed which is grounded on authorship, experience, publications count, total citations, i10-index, and h -index. To assess the proposed framework’s performance, a dataset is collected considering the world’s top ten computing departments and ten domestic ones. This results in a data of 1000 computer scientists. The dataset is preprocessed and afterwards three techniques for feature selection are employed, i.e., Mutual Information (MI), Chi-Square ( X 2 ), and Fisher-Test (F-Test) to rank the features in the data. To validate the collected data, the framework has three clustering techniques as well, namely, k -medoids, k -means, and spectral clustering to identify the optimum number of heterogeneous groups. Three cluster validity indices are used to evaluate the clustering outcomes, namely, Calinski-Harabasz Index (CHI), Davies Bouldin Index (DBI), and Silhouette Coefficient (SC). Once the optimum clusters are obtained, five classification procedures are used, including, Artificial Neural Network (ANN), k -Nearest Neighbor ( k -NN), Decision Tree (DT), Gaussian Naive Bayes (GNB), and Linear Regression Classifier (LRC) to predict the category of a previously unknown scientist. Among all classifiers, an average accuracy of 94.44% is shown by the ANN to predict an unknown/new scientist category. The current proposal is also compared with closely related past works. The proposed framework offers the possibility to independently classify scientists based on AI techniques.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Information Storage and Retrieval',\n",
       "   'Library Science']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s12274-022-5368-1',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s12274-022-5368-1'}],\n",
       "  'title': 'Electrospun Fe0.64Ni0.36/MXene/CNFs nanofibrous membranes with multicomponent heterostructures as flexible electromagnetic wave absorbers',\n",
       "  'creators': [{'creator': 'Zhang, Shuo'},\n",
       "   {'creator': 'Jia, Zirui'},\n",
       "   {'creator': 'Zhang, Yan'},\n",
       "   {'creator': 'Wu, Guanglei'}],\n",
       "  'publicationName': 'Nano Research',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s12274-022-5368-1',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1998-0000',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Research Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '13',\n",
       "  'journalId': '12274',\n",
       "  'copyright': '©2022 Tsinghua University Press',\n",
       "  'abstract': 'Two-dimensional metal carbide or nitride materials (MXenes) are widely used in electromagnetic wave absorption because of their unique structure. Herein, a novel composite preparation strategy has been proposed to design dendritic nanofibers based on the electrostatic spinning methods. The multifunctional MXene nanosheets are used as the dendritic matrix, and magnetic nanoparticles are embedded in the nanosheets as magnetic loss units. Multidimensional nanocomposites have interlaced carbon fiber networks, large-scale magnetically coupled networks, and a lot of multi-heterojunction interface structures, which endow the composites with extraordinary conduction loss, magnetic loss, and polarization loss capabilities, respectively. The impedance matching and loss mechanisms of the composites are improved by optimizing the synergistic relationship between the components and building a suitable structure. The optimum reflection loss (RL) of −54.1 dB is achieved at 2.7 mm and a wide effective absorption bandwidth (EAB, RL below −10 dB) of 7.76 GHz is obtained at a small thickness of 2.1 mm for the nanocomposites. The distinctive microstructures of the nanofibrous membranes give rise to their flexibility, waterproof, and electromagnetic wave absorption performance and endow the nanofibrous membranes potential to be utilized as lightweight, efficient electromagnetic wave protective fabric in harsh environment.',\n",
       "  'subjects': ['Materials Science',\n",
       "   'Nanotechnology',\n",
       "   'Materials Science, general',\n",
       "   'Atomic/Molecular Structure and Spectra',\n",
       "   'Condensed Matter Physics',\n",
       "   'Biotechnology',\n",
       "   'Biomedicine, general']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s00138-022-01363-1',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s00138-022-01363-1'}],\n",
       "  'title': 'Superpixel-based foreground-preserving image stitching',\n",
       "  'creators': [{'creator': 'Miao, Xinpeng'},\n",
       "   {'ORCID': '0000-0002-4000-2380', 'creator': 'Qu, Tao'},\n",
       "   {'creator': 'Chen, Xi'},\n",
       "   {'creator': 'He, Chu'}],\n",
       "  'publicationName': 'Machine Vision and Applications',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s00138-022-01363-1',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1432-1769',\n",
       "  'volume': '34',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Original Paper'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '13',\n",
       "  'journalId': '138',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to Springer-Verlag GmbH Germany, part of Springer Nature',\n",
       "  'abstract': 'Image stitching aims to stitch multiple images with overlapping areas into a high-resolution image with natural appearance, no ghosting and no seam, so as to obtain a wide field of vision. And the stitching is required to be as fast as possible. In this article, we discuss the loss of foreground object caused by the seam-cut method for image stitching. Moreover, we propose an improved seam-cut method based on superpixel to solve this problem. The proposed method uses the matching information of feature points and superpixel segmentation to divide the scene into foreground and background. By adjusting the penalty value of foreground superpixels in the superpixel-based seam-cut method, the foreground object can avoid the cutting of optimal seam. The experimental results demonstrate that our method will not cause the ghosting of foreground object or the loss of foreground object in the stitching result in the scene with large parallax. In this method, the foreground object will be preserved completely.',\n",
       "  'subjects': ['Computer Science',\n",
       "   'Pattern Recognition',\n",
       "   'Image Processing and Computer Vision',\n",
       "   'Communications Engineering, Networks']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1186/s13195-022-01152-y',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1186/s13195-022-01152-y'}],\n",
       "  'title': 'Effects of non-modifiable risk factors of Alzheimer’s disease on intracortical myelin content',\n",
       "  'creators': [{'creator': 'Fernandez-Alvarez, Marina'},\n",
       "   {'creator': 'Atienza, Mercedes'},\n",
       "   {'creator': 'Cantero, Jose L.'}],\n",
       "  'publicationName': \"Alzheimer's Research & Therapy\",\n",
       "  'openaccess': 'true',\n",
       "  'doi': '10.1186/s13195-022-01152-y',\n",
       "  'publisher': 'BioMed Central',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '1758-9193',\n",
       "  'volume': '14',\n",
       "  'number': '1',\n",
       "  'genre': ['OriginalPaper', 'Research'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '13',\n",
       "  'journalId': '13195',\n",
       "  'copyright': '©2022 The Author(s)',\n",
       "  'abstract': 'Background Non-modifiable risk factors of Alzheimer’s disease (AD) have lifelong effects on cortical integrity that could be mitigated if identified at early stages. However, it remains unknown whether cortical microstructure is affected in older individuals with non-modifiable AD risk factors and whether altered cortical tissue integrity produces abnormalities in brain functional networks in this AD-risk population. Methods Using relative T1w/T2w (rT1w/T2w) ratio maps, we have compared tissue integrity of normal-appearing cortical GM between controls and cognitively normal older adults with either APOE4 ( N = 50), with a first-degree family history (FH) of AD ( N = 52), or with the co-occurrence of both AD risk factors (APOE4+FH) ( N = 35). Additionally, individuals with only one risk factor (APOE4 or FH) were combined into one group ( N = 102) and compared with controls. The same number of controls matched in age, sex, and years of education was employed for each of these comparisons. Group differences in resting state functional connectivity (rs-FC) patterns were also investigated, using as FC seeds those cortical regions showing significant changes in rT1w/T2w ratios. Results Overall, individuals with non-modifiable AD risk factors exhibited significant variations in rT1w/T2w ratios compared to controls, being APOE4 and APOE4+FH at opposite ends of a continuum. The co-occurrence of APOE4 and FH was further accompanied by altered patterns of rs-FC. Conclusions These findings may have practical implications for early detection of cortical abnormalities in older populations with APOE4 and/or FH\\xa0of AD and open new avenues to monitor changes in cortical tissue integrity associated with non-modifiable AD risk factors.',\n",
       "  'subjects': ['Biomedicine',\n",
       "   'Neurosciences',\n",
       "   'Neurology',\n",
       "   'Geriatrics/Gerontology',\n",
       "   'Geriatric Psychiatry']},\n",
       " {'contentType': 'Article',\n",
       "  'identifier': 'doi:10.1007/s12596-022-01072-y',\n",
       "  'language': 'en',\n",
       "  'url': [{'format': '',\n",
       "    'platform': '',\n",
       "    'value': 'http://dx.doi.org/10.1007/s12596-022-01072-y'}],\n",
       "  'title': 'Design of 16\\u2009×\\u2009100\\xa0Gbps free-space optical system using advanced modulation techniques',\n",
       "  'creators': [{'creator': 'Kathpal, Namita'},\n",
       "   {'creator': 'Garg, Amit Kumar'},\n",
       "   {'creator': 'Goyal, Priyanka'},\n",
       "   {'creator': 'Awasthi, Yogendra Kumar'}],\n",
       "  'publicationName': 'Journal of Optics',\n",
       "  'openaccess': 'false',\n",
       "  'doi': '10.1007/s12596-022-01072-y',\n",
       "  'publisher': 'Springer',\n",
       "  'publicationDate': '2022-12-31',\n",
       "  'publicationType': 'Journal',\n",
       "  'issn': '0974-6900',\n",
       "  'volume': '',\n",
       "  'number': '',\n",
       "  'genre': ['OriginalPaper', 'Research Article'],\n",
       "  'startingPage': '1',\n",
       "  'endingPage': '18',\n",
       "  'journalId': '12596',\n",
       "  'copyright': '©2022 The Author(s), under exclusive licence to The Optical Society of India',\n",
       "  'abstract': 'In this paper, advanced modulation techniques based on external modulation, such as I-AMI, AMI, and DBRZ, are designed to improve long-haul transmission and lower high power requirements in the FSO communication system. For long-haul 16\\u2009×\\u2009100\\xa0Gbps FSO systems, an alternate mark inversion modulation approach with high-data rate distortion less transmission is possible. The results demonstrate improved performance in terms of performance indicators such as Q-factor (29.62) and BER (3.8e −193 ), as well as improved data transmission capabilities of up to 100\\xa0Gbps over a distance of 9\\xa0km in clear weather. Because here, AMI technique uses a low input power of 5\\xa0dBm instead of a conventional power of 10\\xa0dBm, the achieved findings demonstrate a reasonable reduction in signal degradation, especially nonlinear effects.',\n",
       "  'subjects': ['Physics',\n",
       "   'Physics, general',\n",
       "   'Optics, Lasers, Photonics, Optical Devices']}]"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record = d_springer[\"records\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "82c96f6f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['contentType', 'identifier', 'language', 'url', 'title', 'publicationName', 'openaccess', 'doi', 'publisher', 'publicationDate', 'publicationType', 'issn', 'volume', 'number', 'genre', 'startingPage', 'endingPage', 'journalId', 'copyright', 'abstract', 'subjects'])"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record[1].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "5eecd589",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'creator': 'Hosseinzadeh, Mohammad Mehdi'}, {'creator': 'Cannataro, Mario'}, {'ORCID': '0000-0001-5542-2997', 'creator': 'Guzzi, Pietro Hiram'}, {'creator': 'Dondi, Riccardo'}]\n",
      "[{'creator': 'Hara, Takanori'}, {'creator': 'Sasabe, Masahiro'}]\n",
      "[{'creator': 'Yang, Fan'}, {'creator': 'Comaniciu, Cristina'}, {'creator': 'Krishnan, P.'}, {'creator': 'Sabnani, Krishan'}, {'creator': 'Greenwald, Lloyd'}]\n",
      "[{'ORCID': '0000-0002-8321-2070', 'creator': 'Acar, Vedat'}, {'creator': 'Eksioglu, Ender M.'}]\n",
      "[{'creator': 'Alharby, Maher'}]\n",
      "[{'ORCID': '0000-0001-9717-4354', 'creator': 'Maass, Alejandro I.'}, {'creator': 'Nešić, Dragan'}, {'creator': 'Postoyan, Romain'}, {'creator': 'Tan, Ying'}]\n",
      "[{'ORCID': '0000-0002-8387-3565', 'creator': 'Liu, Kangcheng'}]\n",
      "[{'ORCID': '0000-0002-3726-3416', 'creator': 'Gao, Sheng'}, {'ORCID': '0000-0003-1074-7776', 'creator': 'Sun, Hongchun'}, {'creator': 'Ma, Sihan'}]\n",
      "[{'creator': 'Wan, Honglin'}, {'creator': 'Zhong, Zongfeng'}, {'creator': 'Li, Tianping'}, {'creator': 'Zhang, Huaxiang'}, {'ORCID': '0000-0001-6157-2051', 'creator': 'Sun, Jiande'}]\n",
      "[{'creator': 'Wang, Ying'}, {'creator': 'Zheng, Yunan'}, {'creator': 'Liu, Yiguang'}]\n",
      "[{'creator': 'Jinka, Sai Sagar'}, {'ORCID': '0000-0001-6600-722X', 'creator': 'Srivastava, Astitva'}, {'creator': 'Pokhariya, Chandradeep'}, {'creator': 'Sharma, Avinash'}, {'creator': 'Narayanan, P. J.'}]\n",
      "[{'ORCID': '0000-0002-2734-7747', 'creator': 'Santhadevi, D.'}, {'creator': 'Janet, B.'}]\n",
      "[{'creator': 'Amroune, Mohammed'}, {'creator': 'Boukaroura, Abdelkader'}, {'creator': 'Bourzami, Arif'}, {'creator': 'Slimani, Linda'}, {'creator': 'Musirin, Ismail'}]\n",
      "[{'creator': 'Zhang, Yipeng'}, {'creator': 'Hu, Bingliang'}, {'creator': 'Huang, Yingying'}, {'creator': 'Gao, Chi'}, {'creator': 'Wang, Quan'}]\n",
      "[{'creator': 'Moradi, M.'}, {'creator': 'Moradkhani, M. A.'}, {'creator': 'Hosseini, S. H.'}, {'creator': 'Olazar, M.'}]\n",
      "[{'ORCID': '0000-0002-7162-5038', 'creator': 'Buzzicotti, Michele'}, {'creator': 'Bonaccorso, Fabio'}]\n",
      "[{'creator': 'Patel, Eva'}, {'creator': 'Kushwaha, Dharmender Singh'}]\n",
      "[{'creator': 'Wharton, Ken'}, {'creator': 'Liu, Raylor'}]\n",
      "[{'ORCID': '0000-0001-9454-513X', 'creator': 'Sakthivel, Senthamizh Selvan'}, {'creator': 'Arunachalam, Venkadesan'}]\n",
      "[{'creator': 'Gu, Xinyi'}, {'creator': 'Wang, Shen'}, {'creator': 'Jin, Bo'}, {'creator': 'Qi, Zhidan'}, {'creator': 'Deng, Jin'}, {'creator': 'Huang, Chen'}, {'creator': 'Yin, Xiaofeng'}]\n",
      "[{'ORCID': '0000-0002-8851-0958', 'creator': 'Vinod, P. V.'}, {'ORCID': '0000-0003-1402-089X', 'creator': 'Trivedi, Shivam'}, {'ORCID': '0000-0003-3685-7295', 'creator': 'Hebbar, R.'}, {'ORCID': '0000-0002-3560-4993', 'creator': 'Jha, C. S.'}]\n",
      "[{'creator': 'Sun, Zhonghua'}, {'creator': 'Wang, Tianyi'}, {'creator': 'Dai, Meng'}]\n",
      "[{'ORCID': '0000-0003-1597-7327', 'creator': 'Moslehi, Samad'}, {'ORCID': '0000-0002-9375-3807', 'creator': 'Mahjub, Hossein'}, {'ORCID': '0000-0002-6054-9850', 'creator': 'Farhadian, Maryam'}, {'ORCID': '0000-0002-7483-3502', 'creator': 'Soltanian, Ali Reza'}, {'ORCID': '0000-0001-5802-6367', 'creator': 'Mamani, Mojgan'}]\n",
      "[{'ORCID': '0000-0003-4397-6196', 'creator': 'Uğuz, Sinan'}, {'creator': 'Şikaroğlu, Gulhan'}, {'creator': 'Yağız, Abdullah'}]\n",
      "[{'creator': 'Bordoloi, Kuntala Sarma'}, {'creator': 'Baruah, Pooja Moni'}, {'creator': 'Tanti, Bhaben'}, {'creator': 'Gill, Sarvajeet Singh'}, {'ORCID': '0000-0002-1144-7743', 'creator': 'Agarwala, Niraj'}]\n",
      "[{'creator': 'Han, Cailing'}, {'creator': 'Huang, Xi'}, {'creator': 'Zhang, Yimin'}, {'creator': 'Wang, Minghui'}]\n",
      "[{'creator': 'Libiseller-Egger, Julian'}, {'creator': 'Phelan, Jody E.'}, {'creator': 'Attia, Zachi I.'}, {'creator': 'Benavente, Ernest Diez'}, {'creator': 'Campino, Susana'}, {'creator': 'Friedman, Paul A.'}, {'creator': 'Lopez-Jimenez, Francisco'}, {'creator': 'Leon, David A.'}, {'creator': 'Clark, Taane G.'}]\n",
      "[{'ORCID': '0000-0002-7260-5063', 'creator': 'Carmo, Gisleine'}, {'creator': 'Felizardo, Luiz Flávio'}, {'creator': 'Castro Alcântara, Valderí'}, {'creator': 'Silva, Cristiane Aparecida'}, {'creator': 'Prado, José Willer'}]\n",
      "[{'ORCID': '0000-0001-7479-2803', 'creator': 'Kumari, Maya'}, {'creator': 'Sarma, Kiranmay'}, {'creator': 'Sharma, Richa'}]\n",
      "[{'creator': 'Zhang, Liu-Xia'}, {'creator': 'Dong, Jie'}, {'creator': 'Wei, Hui'}, {'creator': 'Shi, Shao-Hua'}, {'creator': 'Lu, Ai-Ping'}, {'creator': 'Deng, Gui-Ming'}, {'creator': 'Cao, Dong-Sheng'}]\n",
      "[{'ORCID': '0000-0001-6402-5117', 'creator': 'Peker, Iskender'}, {'creator': 'AR, Ilker Murat'}, {'creator': 'Erol, Ismail'}, {'creator': 'Searcy, Cory'}]\n",
      "[{'creator': 'Zhang, Ziran'}, {'creator': 'Meng, Fanke'}, {'creator': 'Zhou, Jing'}, {'creator': 'Zhang, Zhihan'}, {'creator': 'Liang, Xiaotian'}, {'creator': 'Meng, Meijun'}, {'creator': 'Yi, Guoguo'}, {'creator': 'Fu, Min'}]\n",
      "[{'creator': 'Srivastava, Ankita'}, {'ORCID': '0000-0003-4633-1879', 'creator': 'Kumar, Narander'}]\n",
      "[{'creator': 'Rezaei, Sajjad'}, {'creator': 'Kahani, Mohsen'}, {'creator': 'Behkamal, Behshid'}, {'creator': 'Jalayer, Abdulrahman'}]\n",
      "[{'ORCID': '0000-0002-5610-0153', 'creator': 'DeCastro-García, Noemí'}, {'creator': 'Escudero García, David'}, {'creator': 'Carriegos, Miguel V.'}]\n",
      "[{'creator': 'He, Ruifen'}, {'creator': 'Fan, Qigang'}, {'creator': 'Li, Yi'}, {'creator': 'Zhu, Qinying'}, {'creator': 'Hu, Dan'}, {'creator': 'Du, Junhong'}, {'creator': 'Xing, Yijuan'}, {'creator': 'Li, Hongli'}, {'creator': 'Liang, Xiaolei'}, {'ORCID': '0000-0002-1449-4326', 'creator': 'Yang, Yongxiu'}]\n",
      "[{'ORCID': '0000-0002-8936-0533', 'creator': 'Merino, E.'}, {'creator': 'El Tawil, M.'}, {'creator': 'Sobrados, I.'}, {'creator': 'Durán, A.'}, {'creator': 'Castro, Y.'}]\n",
      "[{'creator': 'Biswas, Neepa'}, {'creator': 'Mondal, Anindita Sarkar'}, {'creator': 'Kusumastuti, Ari'}, {'creator': 'Saha, Swati'}, {'ORCID': '0000-0003-3647-5799', 'creator': 'Mondal, Kartick Chandra'}]\n",
      "[{'creator': 'Liu, Huanhuan'}, {'creator': 'Liu, Wenxin'}, {'creator': 'Schwieter, John W.'}, {'creator': 'Wu, Yan Jing'}]\n",
      "[{'creator': 'Jiang, Xiaoli'}, {'creator': 'Shao, Minghui'}, {'creator': 'Bateer, Buhe'}]\n",
      "[{'ORCID': '0000-0001-5135-6320', 'creator': 'Nejim, Zeineb'}, {'creator': 'Navarro, Laurent'}, {'creator': 'Morin, Claire'}, {'creator': 'Badel, Pierre'}]\n",
      "[{'ORCID': '0000-0001-9199-460X', 'creator': 'Houlihan, Orla A.'}, {'creator': 'Ntentas, Georgios'}, {'creator': 'Cutter, David J.'}, {'creator': 'Daly, Patricia'}, {'creator': 'Gillham, Charles'}, {'creator': 'McArdle, Orla'}, {'creator': 'Duane, Frances K.'}]\n",
      "[{'creator': 'Zhang, Pengyu'}, {'creator': 'Wang, Ji'}, {'creator': 'Zhang, Feng'}, {'creator': 'Xu, Peiquan'}, {'creator': 'Li, Leijun'}, {'creator': 'Li, Baoming'}]\n",
      "[{'creator': 'Wang, Shuxian'}, {'creator': 'Wang, Ziyuan'}, {'creator': 'Vejalla, Srimanasa'}, {'creator': 'Ganegoda, Anushika'}, {'creator': 'Nittala, Muneeswar Gupta'}, {'creator': 'Sadda, SriniVas Reddy'}, {'creator': 'Hu, Zhihong Jewel'}]\n",
      "[{'creator': 'Ali, Nisar'}, {'ORCID': '0000-0003-3094-3483', 'creator': 'Halim, Zahid'}, {'creator': 'Hussain, Syed Fawad'}]\n",
      "[{'creator': 'Zhang, Shuo'}, {'creator': 'Jia, Zirui'}, {'creator': 'Zhang, Yan'}, {'creator': 'Wu, Guanglei'}]\n",
      "[{'creator': 'Miao, Xinpeng'}, {'ORCID': '0000-0002-4000-2380', 'creator': 'Qu, Tao'}, {'creator': 'Chen, Xi'}, {'creator': 'He, Chu'}]\n",
      "[{'creator': 'Fernandez-Alvarez, Marina'}, {'creator': 'Atienza, Mercedes'}, {'creator': 'Cantero, Jose L.'}]\n",
      "[{'creator': 'Kathpal, Namita'}, {'creator': 'Garg, Amit Kumar'}, {'creator': 'Goyal, Priyanka'}, {'creator': 'Awasthi, Yogendra Kumar'}]\n"
     ]
    }
   ],
   "source": [
    "for data in record:\n",
    "    if \"creators\" in data.keys():\n",
    "        print(data[\"creators\"])\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fba6a34a",
   "metadata": {},
   "source": [
    "## Scopus dataset\n",
    "\n",
    "I forget the details of the filter settings, which are roughly as follows\n",
    "\n",
    "|Search within|Search document|\n",
    "|-----|-----|\n",
    "|Language|english|\n",
    "|Affiliation country|netherlands|\n",
    "|Keywords|social AND network AND analysis|\n",
    "|Published from|2018|"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "09c08e8d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Authors</th>\n",
       "      <th>Author(s) ID</th>\n",
       "      <th>Title</th>\n",
       "      <th>Year</th>\n",
       "      <th>Cited by</th>\n",
       "      <th>Link</th>\n",
       "      <th>Affiliations</th>\n",
       "      <th>Authors with affiliations</th>\n",
       "      <th>Abstract</th>\n",
       "      <th>Author Keywords</th>\n",
       "      <th>Index Keywords</th>\n",
       "      <th>References</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Mühlbӓck A., van Walsem M., Nance M., Arnesen ...</td>\n",
       "      <td>58081570200;36163594500;58082093700;5780893390...</td>\n",
       "      <td>What we don't need to prove but need to do in ...</td>\n",
       "      <td>2023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>Department of Neurology, University Ulm, Ulm, ...</td>\n",
       "      <td>Mühlbӓck, A., Department of Neurology, Univers...</td>\n",
       "      <td>Background: Huntington’s disease is a complex ...</td>\n",
       "      <td>Care; Case manager; Guidelines; Huntington’s d...</td>\n",
       "      <td>consensus; health care delivery; health care p...</td>\n",
       "      <td>Simpson, S.A., Rae, D., A standard of care for...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Yang Q., Coutino M., Leus G., Giannakis G.B.</td>\n",
       "      <td>57212307366;57189591807;35474798400;36049280600;</td>\n",
       "      <td>Autoregressive graph Volterra models and appli...</td>\n",
       "      <td>2023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>Department of Mechanical Engineering, Universi...</td>\n",
       "      <td>Yang, Q., Department of Mechanical Engineering...</td>\n",
       "      <td>Graph-based learning and estimation are fundam...</td>\n",
       "      <td>Graph inference; Higher-order interactions; Li...</td>\n",
       "      <td>Electric power distribution; Graph theory; Reg...</td>\n",
       "      <td>Liben-Nowell, D., Kleinberg, J., The link-pred...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>van de Kuilen L., Leseman P.P.M., de Wolf I.M.</td>\n",
       "      <td>58033343700;6602363622;58032496800;</td>\n",
       "      <td>Influence of local ECEC policy on the quality ...</td>\n",
       "      <td>2023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>Inspectorate of Education, St.-Jacobsstraat 20...</td>\n",
       "      <td>van de Kuilen, L., Inspectorate of Education, ...</td>\n",
       "      <td>This study investigated the influence of local...</td>\n",
       "      <td>Early childhood education and care; Emotional ...</td>\n",
       "      <td>article; child; childhood; education; expectat...</td>\n",
       "      <td>Akgündüz, Y.E., Jongen, E.L.W., Leseman, P.P.M...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fathima Mafaziya Nijamdeen T.W.G., Ratsimbazaf...</td>\n",
       "      <td>58032052100;56147008300;36455764700;5803188530...</td>\n",
       "      <td>Mangrove management in Sri Lanka and stakehold...</td>\n",
       "      <td>2023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>Systems Ecology and Resource Management Resear...</td>\n",
       "      <td>Fathima Mafaziya Nijamdeen, T.W.G., Systems Ec...</td>\n",
       "      <td>Understanding the extent of collaboration amon...</td>\n",
       "      <td>Conservation; Environmental policy; Forest man...</td>\n",
       "      <td>conservation management; environmental policy;...</td>\n",
       "      <td>Abeykoon, L.C.K., Thilakarathne, E., Abeygunaw...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Hickey G.M., Roozee E., Voogd R., de Vries J.R...</td>\n",
       "      <td>8875857500;57810943000;57203950202;37664657200...</td>\n",
       "      <td>On the architecture of collaboration in inter-...</td>\n",
       "      <td>2023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>McGill University, Faculty of Agricultural and...</td>\n",
       "      <td>Hickey, G.M., McGill University, Faculty of Ag...</td>\n",
       "      <td>This paper reviews the architecture of collabo...</td>\n",
       "      <td>Collaborative governance; Collective action; C...</td>\n",
       "      <td>collective action; conceptual framework; fishe...</td>\n",
       "      <td>Anderson, S.W., Christ, M.H., Dekker, H.C., Se...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>Brooks L., Janssen M., Papazafeiropoulou A.</td>\n",
       "      <td>14019098600;16199813000;6506379075;</td>\n",
       "      <td>eGovPoliNet: Experiences from building a polic...</td>\n",
       "      <td>2018</td>\n",
       "      <td>1.0</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>Centre for Computing and Social Responsibility...</td>\n",
       "      <td>Brooks, L., Centre for Computing and Social Re...</td>\n",
       "      <td>Policy-making in the digital age is an area wh...</td>\n",
       "      <td>eGovernment; Policy informatics; Research comm...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auyang, S.Y., (1998) Foundations of Complex-Sy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>Ghaffar F., Peirce N., Serlie A.</td>\n",
       "      <td>57203850132;23398182000;16944298700;</td>\n",
       "      <td>Assessing Leadership Competencies Through Soci...</td>\n",
       "      <td>2018</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>IBM Ireland, Dublin, Ireland; School of Comput...</td>\n",
       "      <td>Ghaffar, F., IBM Ireland, Dublin, Ireland; Pei...</td>\n",
       "      <td>Leadership competencies are regularly identifi...</td>\n",
       "      <td>Assessment; Competencies; Great Eight; Leaders...</td>\n",
       "      <td>Learning systems; Social networking (online); ...</td>\n",
       "      <td>Amichai-Hamburger, Y., Vinitzky, G., Social ne...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>Ooms W., Werker C., Caniëls M.</td>\n",
       "      <td>56464012500;7801373455;6506807289;</td>\n",
       "      <td>Personal and social proximity empowering colla...</td>\n",
       "      <td>2018</td>\n",
       "      <td>15.0</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>Faculty of Management, Science, and Technology...</td>\n",
       "      <td>Ooms, W., Faculty of Management, Science, and ...</td>\n",
       "      <td>The proximity framework serves to analyse and ...</td>\n",
       "      <td>Collaboration; Innovation; Personal proximity;...</td>\n",
       "      <td>empirical analysis; empowerment; innovation; k...</td>\n",
       "      <td>Balland, P.A., Boschma, R., Frenken, K., Proxi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>Akkermans H.</td>\n",
       "      <td>7003996949;</td>\n",
       "      <td>The emergence of trust: Social figuration in s...</td>\n",
       "      <td>2018</td>\n",
       "      <td>1.0</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>Tilburg University, Netherlands</td>\n",
       "      <td>Akkermans, H., Tilburg University, Netherlands</td>\n",
       "      <td>Trust is an essential governance mechanism in ...</td>\n",
       "      <td>Complexity theory; Interorganizational collabo...</td>\n",
       "      <td>simulation; social theory</td>\n",
       "      <td>Akkermans, H.A., Emergent supply networks. Sys...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>Paci F., Squicciarini A., Zannone N.</td>\n",
       "      <td>15049772600;6506303417;9132349400;</td>\n",
       "      <td>Survey on access control for community-centere...</td>\n",
       "      <td>2018</td>\n",
       "      <td>56.0</td>\n",
       "      <td>https://www.scopus.com/inward/record.uri?eid=2...</td>\n",
       "      <td>Department of Electronics and Computer Science...</td>\n",
       "      <td>Paci, F., Department of Electronics and Comput...</td>\n",
       "      <td>The last decades have seen a growing interest ...</td>\n",
       "      <td>Collaborative access control; Data governance;...</td>\n",
       "      <td>Computer supported cooperative work; Distribut...</td>\n",
       "      <td>Ahn, G.-J., Jin, J., Shehab, M., Policy-driven...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>215 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               Authors  \\\n",
       "0    Mühlbӓck A., van Walsem M., Nance M., Arnesen ...   \n",
       "1         Yang Q., Coutino M., Leus G., Giannakis G.B.   \n",
       "2       van de Kuilen L., Leseman P.P.M., de Wolf I.M.   \n",
       "3    Fathima Mafaziya Nijamdeen T.W.G., Ratsimbazaf...   \n",
       "4    Hickey G.M., Roozee E., Voogd R., de Vries J.R...   \n",
       "..                                                 ...   \n",
       "210        Brooks L., Janssen M., Papazafeiropoulou A.   \n",
       "211                   Ghaffar F., Peirce N., Serlie A.   \n",
       "212                     Ooms W., Werker C., Caniëls M.   \n",
       "213                                       Akkermans H.   \n",
       "214               Paci F., Squicciarini A., Zannone N.   \n",
       "\n",
       "                                          Author(s) ID  \\\n",
       "0    58081570200;36163594500;58082093700;5780893390...   \n",
       "1     57212307366;57189591807;35474798400;36049280600;   \n",
       "2                  58033343700;6602363622;58032496800;   \n",
       "3    58032052100;56147008300;36455764700;5803188530...   \n",
       "4    8875857500;57810943000;57203950202;37664657200...   \n",
       "..                                                 ...   \n",
       "210                14019098600;16199813000;6506379075;   \n",
       "211               57203850132;23398182000;16944298700;   \n",
       "212                 56464012500;7801373455;6506807289;   \n",
       "213                                        7003996949;   \n",
       "214                 15049772600;6506303417;9132349400;   \n",
       "\n",
       "                                                 Title  Year  Cited by  \\\n",
       "0    What we don't need to prove but need to do in ...  2023       NaN   \n",
       "1    Autoregressive graph Volterra models and appli...  2023       NaN   \n",
       "2    Influence of local ECEC policy on the quality ...  2023       NaN   \n",
       "3    Mangrove management in Sri Lanka and stakehold...  2023       NaN   \n",
       "4    On the architecture of collaboration in inter-...  2023       NaN   \n",
       "..                                                 ...   ...       ...   \n",
       "210  eGovPoliNet: Experiences from building a polic...  2018       1.0   \n",
       "211  Assessing Leadership Competencies Through Soci...  2018       NaN   \n",
       "212  Personal and social proximity empowering colla...  2018      15.0   \n",
       "213  The emergence of trust: Social figuration in s...  2018       1.0   \n",
       "214  Survey on access control for community-centere...  2018      56.0   \n",
       "\n",
       "                                                  Link  \\\n",
       "0    https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "1    https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "2    https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "3    https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "4    https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "..                                                 ...   \n",
       "210  https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "211  https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "212  https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "213  https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "214  https://www.scopus.com/inward/record.uri?eid=2...   \n",
       "\n",
       "                                          Affiliations  \\\n",
       "0    Department of Neurology, University Ulm, Ulm, ...   \n",
       "1    Department of Mechanical Engineering, Universi...   \n",
       "2    Inspectorate of Education, St.-Jacobsstraat 20...   \n",
       "3    Systems Ecology and Resource Management Resear...   \n",
       "4    McGill University, Faculty of Agricultural and...   \n",
       "..                                                 ...   \n",
       "210  Centre for Computing and Social Responsibility...   \n",
       "211  IBM Ireland, Dublin, Ireland; School of Comput...   \n",
       "212  Faculty of Management, Science, and Technology...   \n",
       "213                    Tilburg University, Netherlands   \n",
       "214  Department of Electronics and Computer Science...   \n",
       "\n",
       "                             Authors with affiliations  \\\n",
       "0    Mühlbӓck, A., Department of Neurology, Univers...   \n",
       "1    Yang, Q., Department of Mechanical Engineering...   \n",
       "2    van de Kuilen, L., Inspectorate of Education, ...   \n",
       "3    Fathima Mafaziya Nijamdeen, T.W.G., Systems Ec...   \n",
       "4    Hickey, G.M., McGill University, Faculty of Ag...   \n",
       "..                                                 ...   \n",
       "210  Brooks, L., Centre for Computing and Social Re...   \n",
       "211  Ghaffar, F., IBM Ireland, Dublin, Ireland; Pei...   \n",
       "212  Ooms, W., Faculty of Management, Science, and ...   \n",
       "213     Akkermans, H., Tilburg University, Netherlands   \n",
       "214  Paci, F., Department of Electronics and Comput...   \n",
       "\n",
       "                                              Abstract  \\\n",
       "0    Background: Huntington’s disease is a complex ...   \n",
       "1    Graph-based learning and estimation are fundam...   \n",
       "2    This study investigated the influence of local...   \n",
       "3    Understanding the extent of collaboration amon...   \n",
       "4    This paper reviews the architecture of collabo...   \n",
       "..                                                 ...   \n",
       "210  Policy-making in the digital age is an area wh...   \n",
       "211  Leadership competencies are regularly identifi...   \n",
       "212  The proximity framework serves to analyse and ...   \n",
       "213  Trust is an essential governance mechanism in ...   \n",
       "214  The last decades have seen a growing interest ...   \n",
       "\n",
       "                                       Author Keywords  \\\n",
       "0    Care; Case manager; Guidelines; Huntington’s d...   \n",
       "1    Graph inference; Higher-order interactions; Li...   \n",
       "2    Early childhood education and care; Emotional ...   \n",
       "3    Conservation; Environmental policy; Forest man...   \n",
       "4    Collaborative governance; Collective action; C...   \n",
       "..                                                 ...   \n",
       "210  eGovernment; Policy informatics; Research comm...   \n",
       "211  Assessment; Competencies; Great Eight; Leaders...   \n",
       "212  Collaboration; Innovation; Personal proximity;...   \n",
       "213  Complexity theory; Interorganizational collabo...   \n",
       "214  Collaborative access control; Data governance;...   \n",
       "\n",
       "                                        Index Keywords  \\\n",
       "0    consensus; health care delivery; health care p...   \n",
       "1    Electric power distribution; Graph theory; Reg...   \n",
       "2    article; child; childhood; education; expectat...   \n",
       "3    conservation management; environmental policy;...   \n",
       "4    collective action; conceptual framework; fishe...   \n",
       "..                                                 ...   \n",
       "210                                                NaN   \n",
       "211  Learning systems; Social networking (online); ...   \n",
       "212  empirical analysis; empowerment; innovation; k...   \n",
       "213                          simulation; social theory   \n",
       "214  Computer supported cooperative work; Distribut...   \n",
       "\n",
       "                                            References  \n",
       "0    Simpson, S.A., Rae, D., A standard of care for...  \n",
       "1    Liben-Nowell, D., Kleinberg, J., The link-pred...  \n",
       "2    Akgündüz, Y.E., Jongen, E.L.W., Leseman, P.P.M...  \n",
       "3    Abeykoon, L.C.K., Thilakarathne, E., Abeygunaw...  \n",
       "4    Anderson, S.W., Christ, M.H., Dekker, H.C., Se...  \n",
       "..                                                 ...  \n",
       "210  Auyang, S.Y., (1998) Foundations of Complex-Sy...  \n",
       "211  Amichai-Hamburger, Y., Vinitzky, G., Social ne...  \n",
       "212  Balland, P.A., Boschma, R., Frenken, K., Proxi...  \n",
       "213  Akkermans, H.A., Emergent supply networks. Sys...  \n",
       "214  Ahn, G.-J., Jin, J., Shehab, M., Policy-driven...  \n",
       "\n",
       "[215 rows x 12 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata = pd.read_csv(\"./scopus.csv\")\n",
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "cbb1ec92",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"Simpson, S.A., Rae, D., A standard of care for Huntington’s disease: who, what and why (2012) Neurodegener Dis Manag, 2 (1), pp. 1-5; Thomas, A., Managed care: the principles approach (1996) HEC Forum, 8 (2), pp. 109-125. , COI: 1:STN:280:DyaK283ktFCrsA%3D%3D; A novel gene containing a trinucleotide repeat that is expanded and unstable on Huntington's disease chromosomes (1993) Cell, 72 (6), pp. 971-983; Roos, R.A.C., Huntington's disease: a clinical review (2010) Orphanet J Rare Dis., 5 (1), p. 40; Cognitive and Behavioral Changes in Huntington Disease before Diagnosis, pp. 69-91; Andrich, J., Saft, C., Ostholt, N., Müller, T., Complex movement behaviour and progression of Huntington's disease (2007) Neurosci Lett, 416 (3), pp. 272-274. , COI: 1:CAS:528:DC%2BD2sXjslSqtb8%3D; (2001) J Neurol Neurosurg Psychiatry, 71 (3), pp. 310-314; Thompson, J.C., Harris, J., Sollom, A.C., Stopford, C.L., Howard, E., Snowden, J.S., Longitudinal evaluation of neuropsychiatric symptoms in Huntington's disease (2012) J Neuropsychiatry Clin Neurosci, 24 (1), pp. 53-60; Paoli, R., Botturi, A., Ciammola, A., Silani, V., Prunas, C., Lucchiari, C., Neuropsychiatric burden in Huntington’s disease (2017) Brain Sci, 7 (12), p. 67; Boll, T., Heaton, R., Reitan, R., Neuropsychological and emotional correlates of Huntington's chorea (1974) J Nerv Ment Dis, 158 (1), pp. 61-69. , COI: 1:STN:280:DyaE2c%2FntVOmtQ%3D%3D; Røthing, M., Malterud, K., Frich, J.C., Family caregivers' views on coordination of care in Huntington's disease: a qualitative study (2015) Scand J Caring Sci, 29 (4), pp. 803-809; van Walsem, M.R., Howe, E.I., Iversen, K., Frich, J.C., Andelic, N., Unmet needs for healthcare and social support services in patients with Huntington's disease: a cross-sectional population-based study (2015) Orphanet J Rare Dis, 10, p. 124; Siminoff, L.A., Incorporating patient and family preferences into evidence-based medicine (2013) BMC Med Inform Decis Mak., 13, p. S6; World Health Organisation (2022) Quality of Care Definition, , https://www.who.int/health-topics/quality-of-care#tab=tab_1, Available from; Masic, I., Miokovic, M., Muhamedagic, B., Evidence based medicine—new approaches and challenges (2008) Acta Inform Med, 16 (4), pp. 219-225; (2011), https://vsop.nl/media/uploads/file/EUCERD%20recommendations.pdf, [Internet]., Available from; (2022) Patient Journey&#x2014;Huntington's Disease European Reference Network for Rare Neurological Diseases, , https://www.ern-rnd.eu/patient-journey-huntingtons-disease/, Available from; Bachoud-Lévi, A.C., Ferreira, J., Massart, R., Youssov, K., Rosser, A., Busse, M., International guidelines for the treatment of Huntington's disease (2019) Front Neurol, 10, p. 710; Eddy, C.M., Parkinson, E.G., Rickards, H.E., Changes in mental state and behaviour in Huntington's disease (2016) Lancet Psychiatry, 3 (11), pp. 1079-1086; van Duijn, E., Craufurd, D., Hubers, A.A., Giltay, E.J., Bonelli, R., Rickards, H., Neuropsychiatric symptoms in a European Huntington's disease cohort (REGISTRY) (2014) J Neurol Neurosurg Psychiatry, 85 (12), pp. 1411-1418; Anderson, K.E., Van Duijn, E., Craufurd, D., Drazinic, C., Edmondson, M., Goodman, N., Clinical management of neuropsychiatric symptoms of Huntington disease: expert-based consensus guidelines on agitation, anxiety, apathy, psychosis and sleep disorders (2018) J Huntington's Dis, 7 (4), pp. 355-366; Quigley, J., Juvenile Huntington’s disease: diagnostic and treatment considerations for the psychiatrist (2017) Curr Psychiatry Rep, 19 (2), pp. 1-4; Gibson, J.S., Isaacs, D.A., Claassen, D.O., Stovall, J.G., Lifetime neuropsychiatric symptoms in Huntington's disease: implications for psychiatric nursing (2021) Arch Psychiatr Nurs, 35 (3), pp. 284-289; Simpson, J., Eccles, F., Zarotti, N., (2021) Extended evidence-based guidance on psychological interventions for psychological difficulties in individuals with Huntington’s Disease.; Zarotti, N., Dale, M., Eccles, F., Simpson, J., Psychological interventions for people with Huntington’s disease: a call to arms (2020) J Huntington's Dis, 9, pp. 231-243; Snowden, J.S., The neuropsychology of Huntington's disease (2017) Arch Clin Neuropsychol, 32 (7), pp. 876-887; Stoker, T.B., Mason, S.L., Greenland, J.C., Holden, S.T., Santini, H., Barker, R.A., Huntington's disease: diagnosis and management (2022) Pract Neurol, 22 (1), pp. 32-41; Mestre, T.A., Bachoud-Lévi, A.C., Marinus, J., Stout, J.C., Paulsen, J.S., Como, P., Rating scales for cognition in Huntington's disease: critique and recommendations (2018) Mov Disord, 33 (2), pp. 187-195; Clare, L., Teale, J.C., Toms, G., Kudlicka, A., Evans, I., Abrahams, S., Cognitive rehabilitation, self-management, psychotherapeutic and caregiver support interventions in progressive neurodegenerative conditions: a scoping review (2018) Neuro Rehab, 43 (4), pp. 443-471; Bartlett, D.M., Govus, A., Rankin, T., Lampit, A., Feindel, K., Poudel, G., The effects of multidisciplinary rehabilitation on neuroimaging, biological, cognitive and motor outcomes in individuals with premanifest Huntington's disease (2020) J Neurol Sci, 416; Cruickshank, T.M., Thompson, J.A., Domínguez, D.J., Reyes, A.P., Bynevelt, M., Georgiou-Karistianis, N., The effect of multidisciplinary rehabilitation on brain structure and cognition in Huntington's disease: an exploratory study (2015) Brain Behav, 5 (2); Yhnell, E., Furby, H., Lowe, R.S., Brookes-Howell, L.C., Drew, C.J.G., Playle, R., A randomised feasibility study of computerised cognitive training as a therapeutic intervention for people with Huntington’s disease (CogTrainHD) (2020) Pilot Feasib Stud, 6 (1), p. 88; El-Nimr, G., Barrett, K., Huntington's disease: GP guide to clinical management (2006) Prescriber, 17 (10), pp. 23-31; Nance, M., Paulsen, J.S., Rosenblatt, A., Wheelock, V., A physician’s guide to the management of Huntington disease, 3rd edn Canada: The Huntington Society of Canada (HSC), 2013, p. p. 120.; Nance, M.A., Huntington disease: clinical, genetic, and social aspects (1998) J Geriatr Psychiatry Neurol, 11 (2), pp. 61-70. , COI: 1:STN:280:DyaK1M%2Fptlemtg%3D%3D; Macleod, R., Tibben, A., Frontali, M., Evers-Kiebooms, G., Jones, A., Martinez-Descales, A., Recommendations for the predictive genetic test in Huntington's disease (2013) Clin Genet, 83 (3), pp. 221-231. , COI: 1:STN:280:DC%2BC38nntVeqsA%3D%3D; Craufurd, D., Macleod, R., Frontali, M., Quarrell, O., Bijlsma, E.K., Davis, M., Diagnostic genetic testing for Huntington's disease (2015) Pract Neurol, 15 (1), pp. 80-84; Nance, M.A., Comprehensive care in Huntington's disease: a physician's perspective (2007) Brain Res Bull, 72 (2-3), pp. 175-178; Kenny, R., Wilson, E., Successful multidisciplinary team working: an evaluation of a Huntington's disease service (2012) Br J Neurosci Nurs, 8 (3), pp. 137-142; Baker, M., McLaren, S., Crichton, N., A review of the literature on nursing in Huntington's disease: a need for specialist knowledge (2009) Br J Neurosci Nurs, 5 (8), pp. 360-366; Hamric, A.B.T., (2019) Advanced Practice Nursing: An Integrative Approach, , 6th edn. St. Louis, Missourie: Elsevier; Yale, R., Martindale, B., Social work with Huntington's Chorea (1984) Br J Soc Work, 14 (1), pp. 157-171; Quinn, L., Kegelmeyer, D., Kloos, A., Rao, A.K., Busse, M., Fritz, N.E., Clinical recommendations to guide physical therapy practice for Huntington disease (2020) Neurology, 94 (5), pp. 217-228; Nance, M.A., Therapy in Huntington’s disease: where are we? (2012) Curr Neurol Neurosci Rep, 12 (4), pp. 359-366. , COI: 1:CAS:528:DC%2BC38Xpsl2ju7Y%3D; Cook, C., Page, K., Wagstaff, A., Simpson, S.A., Rae, D., Development of guidelines for occupational therapy in Huntington’s disease (2012) Neurodegener Dis Manag, 2 (1), pp. 79-87. , COI: 1:CAS:528:DC%2BC38XhvFGls7g%3D; Brotherton, A., Campos, L., Rowell, A., Zoia, V., Simpson, S.A., Rae, D., Nutritional management of individuals with Huntington’s disease: nutritional guidelines (2012) Neurodegener Dis Manag, 2 (1), pp. 33-43; Hamilton, A., Ferm, U., Heemskerk, A.-W., Twiston-Davies, R., Matheson, K.Y., Simpson, S.A., Management of speech, language and communication difficulties in Huntington’s disease (2012) Neurodegener Dis Manag, 2 (1), pp. 67-77; Hamilton, A., Heemskerk, A.-W., Loucas, M., Twiston-Davies, R., Matheson, K.Y., Simpson, S.A., Oral feeding in Huntington’s disease: a guideline document for speech and language therapists (2012) Neurodegener Dis Manag, 2 (1), pp. 45-53; Rae, D., Manley, G., Oral health care in people with Huntington’s disease (2015) Diet and nutrition in dementia and cognitive decline, pp. 1125-1133. , Elsevier;,. p; Munhoz, L., Jabbar, A.Q., Silva Filho, W., Nagai, A.Y., Arita, E.S., The oral manifestations of Huntington's disease: A systematic review of prevalence (2021) Oral Dis; Manley, G., Lane, H., Carlsson, A., Ahlborg, B., Mårtensson, Å., Nilsson, M.B., Guideline for oral healthcare of adults with Huntington’s disease (2012) Neurodegener Dis Manag, 2 (1), pp. 55-65; Devlin, K., Alshaikh, J.T., Pantelyat, A., Music therapy and music-based interventions for movement disorders (2019) Curr Neurol Neurosci Rep, 19 (11), p. 83; Herzog-Krzywoszanska, R., Krzywoszanski, L., Sleep disorders in Huntington's disease (2019) Front Psychiatry, 10, p. 221; Killoran, A., Biglan, K.M., Current therapeutic options for Huntington's disease: good clinical practice versus evidence-based approaches? (2014) Mov Disord, 29 (11), pp. 1404-1413\""
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata[\"References\"][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "d9ca5b48",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Simpson, S.A., Rae, D., A standard of care for Huntington’s disease: who, what and why (2012) Neurodegener Dis Manag, 2 (1), pp. 1-5',\n",
       " 'Thomas, A., Managed care: the principles approach (1996) HEC Forum, 8 (2), pp. 109-125. , COI: 1:STN:280:DyaK283ktFCrsA%3D%3D',\n",
       " \"A novel gene containing a trinucleotide repeat that is expanded and unstable on Huntington's disease chromosomes (1993) Cell, 72 (6), pp. 971-983\",\n",
       " \"Roos, R.A.C., Huntington's disease: a clinical review (2010) Orphanet J Rare Dis., 5 (1), p. 40\",\n",
       " 'Cognitive and Behavioral Changes in Huntington Disease before Diagnosis, pp. 69-91',\n",
       " \"Andrich, J., Saft, C., Ostholt, N., Müller, T., Complex movement behaviour and progression of Huntington's disease (2007) Neurosci Lett, 416 (3), pp. 272-274. , COI: 1:CAS:528:DC%2BD2sXjslSqtb8%3D\",\n",
       " '(2001) J Neurol Neurosurg Psychiatry, 71 (3), pp. 310-314',\n",
       " \"Thompson, J.C., Harris, J., Sollom, A.C., Stopford, C.L., Howard, E., Snowden, J.S., Longitudinal evaluation of neuropsychiatric symptoms in Huntington's disease (2012) J Neuropsychiatry Clin Neurosci, 24 (1), pp. 53-60\",\n",
       " 'Paoli, R., Botturi, A., Ciammola, A., Silani, V., Prunas, C., Lucchiari, C., Neuropsychiatric burden in Huntington’s disease (2017) Brain Sci, 7 (12), p. 67',\n",
       " \"Boll, T., Heaton, R., Reitan, R., Neuropsychological and emotional correlates of Huntington's chorea (1974) J Nerv Ment Dis, 158 (1), pp. 61-69. , COI: 1:STN:280:DyaE2c%2FntVOmtQ%3D%3D\",\n",
       " \"Røthing, M., Malterud, K., Frich, J.C., Family caregivers' views on coordination of care in Huntington's disease: a qualitative study (2015) Scand J Caring Sci, 29 (4), pp. 803-809\",\n",
       " \"van Walsem, M.R., Howe, E.I., Iversen, K., Frich, J.C., Andelic, N., Unmet needs for healthcare and social support services in patients with Huntington's disease: a cross-sectional population-based study (2015) Orphanet J Rare Dis, 10, p. 124\",\n",
       " 'Siminoff, L.A., Incorporating patient and family preferences into evidence-based medicine (2013) BMC Med Inform Decis Mak., 13, p. S6',\n",
       " 'World Health Organisation (2022) Quality of Care Definition, , https://www.who.int/health-topics/quality-of-care#tab=tab_1, Available from',\n",
       " 'Masic, I., Miokovic, M., Muhamedagic, B., Evidence based medicine—new approaches and challenges (2008) Acta Inform Med, 16 (4), pp. 219-225',\n",
       " '(2011), https://vsop.nl/media/uploads/file/EUCERD%20recommendations.pdf, [Internet]., Available from',\n",
       " \"(2022) Patient Journey&#x2014;Huntington's Disease European Reference Network for Rare Neurological Diseases, , https://www.ern-rnd.eu/patient-journey-huntingtons-disease/, Available from\",\n",
       " \"Bachoud-Lévi, A.C., Ferreira, J., Massart, R., Youssov, K., Rosser, A., Busse, M., International guidelines for the treatment of Huntington's disease (2019) Front Neurol, 10, p. 710\",\n",
       " \"Eddy, C.M., Parkinson, E.G., Rickards, H.E., Changes in mental state and behaviour in Huntington's disease (2016) Lancet Psychiatry, 3 (11), pp. 1079-1086\",\n",
       " \"van Duijn, E., Craufurd, D., Hubers, A.A., Giltay, E.J., Bonelli, R., Rickards, H., Neuropsychiatric symptoms in a European Huntington's disease cohort (REGISTRY) (2014) J Neurol Neurosurg Psychiatry, 85 (12), pp. 1411-1418\",\n",
       " \"Anderson, K.E., Van Duijn, E., Craufurd, D., Drazinic, C., Edmondson, M., Goodman, N., Clinical management of neuropsychiatric symptoms of Huntington disease: expert-based consensus guidelines on agitation, anxiety, apathy, psychosis and sleep disorders (2018) J Huntington's Dis, 7 (4), pp. 355-366\",\n",
       " 'Quigley, J., Juvenile Huntington’s disease: diagnostic and treatment considerations for the psychiatrist (2017) Curr Psychiatry Rep, 19 (2), pp. 1-4',\n",
       " \"Gibson, J.S., Isaacs, D.A., Claassen, D.O., Stovall, J.G., Lifetime neuropsychiatric symptoms in Huntington's disease: implications for psychiatric nursing (2021) Arch Psychiatr Nurs, 35 (3), pp. 284-289\",\n",
       " 'Simpson, J., Eccles, F., Zarotti, N., (2021) Extended evidence-based guidance on psychological interventions for psychological difficulties in individuals with Huntington’s Disease.',\n",
       " \"Zarotti, N., Dale, M., Eccles, F., Simpson, J., Psychological interventions for people with Huntington’s disease: a call to arms (2020) J Huntington's Dis, 9, pp. 231-243\",\n",
       " \"Snowden, J.S., The neuropsychology of Huntington's disease (2017) Arch Clin Neuropsychol, 32 (7), pp. 876-887\",\n",
       " \"Stoker, T.B., Mason, S.L., Greenland, J.C., Holden, S.T., Santini, H., Barker, R.A., Huntington's disease: diagnosis and management (2022) Pract Neurol, 22 (1), pp. 32-41\",\n",
       " \"Mestre, T.A., Bachoud-Lévi, A.C., Marinus, J., Stout, J.C., Paulsen, J.S., Como, P., Rating scales for cognition in Huntington's disease: critique and recommendations (2018) Mov Disord, 33 (2), pp. 187-195\",\n",
       " 'Clare, L., Teale, J.C., Toms, G., Kudlicka, A., Evans, I., Abrahams, S., Cognitive rehabilitation, self-management, psychotherapeutic and caregiver support interventions in progressive neurodegenerative conditions: a scoping review (2018) Neuro Rehab, 43 (4), pp. 443-471',\n",
       " \"Bartlett, D.M., Govus, A., Rankin, T., Lampit, A., Feindel, K., Poudel, G., The effects of multidisciplinary rehabilitation on neuroimaging, biological, cognitive and motor outcomes in individuals with premanifest Huntington's disease (2020) J Neurol Sci, 416\",\n",
       " \"Cruickshank, T.M., Thompson, J.A., Domínguez, D.J., Reyes, A.P., Bynevelt, M., Georgiou-Karistianis, N., The effect of multidisciplinary rehabilitation on brain structure and cognition in Huntington's disease: an exploratory study (2015) Brain Behav, 5 (2)\",\n",
       " 'Yhnell, E., Furby, H., Lowe, R.S., Brookes-Howell, L.C., Drew, C.J.G., Playle, R., A randomised feasibility study of computerised cognitive training as a therapeutic intervention for people with Huntington’s disease (CogTrainHD) (2020) Pilot Feasib Stud, 6 (1), p. 88',\n",
       " \"El-Nimr, G., Barrett, K., Huntington's disease: GP guide to clinical management (2006) Prescriber, 17 (10), pp. 23-31\",\n",
       " 'Nance, M., Paulsen, J.S., Rosenblatt, A., Wheelock, V., A physician’s guide to the management of Huntington disease, 3rd edn Canada: The Huntington Society of Canada (HSC), 2013, p. p. 120.',\n",
       " 'Nance, M.A., Huntington disease: clinical, genetic, and social aspects (1998) J Geriatr Psychiatry Neurol, 11 (2), pp. 61-70. , COI: 1:STN:280:DyaK1M%2Fptlemtg%3D%3D',\n",
       " \"Macleod, R., Tibben, A., Frontali, M., Evers-Kiebooms, G., Jones, A., Martinez-Descales, A., Recommendations for the predictive genetic test in Huntington's disease (2013) Clin Genet, 83 (3), pp. 221-231. , COI: 1:STN:280:DC%2BC38nntVeqsA%3D%3D\",\n",
       " \"Craufurd, D., Macleod, R., Frontali, M., Quarrell, O., Bijlsma, E.K., Davis, M., Diagnostic genetic testing for Huntington's disease (2015) Pract Neurol, 15 (1), pp. 80-84\",\n",
       " \"Nance, M.A., Comprehensive care in Huntington's disease: a physician's perspective (2007) Brain Res Bull, 72 (2-3), pp. 175-178\",\n",
       " \"Kenny, R., Wilson, E., Successful multidisciplinary team working: an evaluation of a Huntington's disease service (2012) Br J Neurosci Nurs, 8 (3), pp. 137-142\",\n",
       " \"Baker, M., McLaren, S., Crichton, N., A review of the literature on nursing in Huntington's disease: a need for specialist knowledge (2009) Br J Neurosci Nurs, 5 (8), pp. 360-366\",\n",
       " 'Hamric, A.B.T., (2019) Advanced Practice Nursing: An Integrative Approach, , 6th edn. St. Louis, Missourie: Elsevier',\n",
       " \"Yale, R., Martindale, B., Social work with Huntington's Chorea (1984) Br J Soc Work, 14 (1), pp. 157-171\",\n",
       " 'Quinn, L., Kegelmeyer, D., Kloos, A., Rao, A.K., Busse, M., Fritz, N.E., Clinical recommendations to guide physical therapy practice for Huntington disease (2020) Neurology, 94 (5), pp. 217-228',\n",
       " 'Nance, M.A., Therapy in Huntington’s disease: where are we? (2012) Curr Neurol Neurosci Rep, 12 (4), pp. 359-366. , COI: 1:CAS:528:DC%2BC38Xpsl2ju7Y%3D',\n",
       " 'Cook, C., Page, K., Wagstaff, A., Simpson, S.A., Rae, D., Development of guidelines for occupational therapy in Huntington’s disease (2012) Neurodegener Dis Manag, 2 (1), pp. 79-87. , COI: 1:CAS:528:DC%2BC38XhvFGls7g%3D',\n",
       " 'Brotherton, A., Campos, L., Rowell, A., Zoia, V., Simpson, S.A., Rae, D., Nutritional management of individuals with Huntington’s disease: nutritional guidelines (2012) Neurodegener Dis Manag, 2 (1), pp. 33-43',\n",
       " 'Hamilton, A., Ferm, U., Heemskerk, A.-W., Twiston-Davies, R., Matheson, K.Y., Simpson, S.A., Management of speech, language and communication difficulties in Huntington’s disease (2012) Neurodegener Dis Manag, 2 (1), pp. 67-77',\n",
       " 'Hamilton, A., Heemskerk, A.-W., Loucas, M., Twiston-Davies, R., Matheson, K.Y., Simpson, S.A., Oral feeding in Huntington’s disease: a guideline document for speech and language therapists (2012) Neurodegener Dis Manag, 2 (1), pp. 45-53',\n",
       " 'Rae, D., Manley, G., Oral health care in people with Huntington’s disease (2015) Diet and nutrition in dementia and cognitive decline, pp. 1125-1133. , Elsevier;,. p',\n",
       " \"Munhoz, L., Jabbar, A.Q., Silva Filho, W., Nagai, A.Y., Arita, E.S., The oral manifestations of Huntington's disease: A systematic review of prevalence (2021) Oral Dis\",\n",
       " 'Manley, G., Lane, H., Carlsson, A., Ahlborg, B., Mårtensson, Å., Nilsson, M.B., Guideline for oral healthcare of adults with Huntington’s disease (2012) Neurodegener Dis Manag, 2 (1), pp. 55-65',\n",
       " 'Devlin, K., Alshaikh, J.T., Pantelyat, A., Music therapy and music-based interventions for movement disorders (2019) Curr Neurol Neurosci Rep, 19 (11), p. 83',\n",
       " \"Herzog-Krzywoszanska, R., Krzywoszanski, L., Sleep disorders in Huntington's disease (2019) Front Psychiatry, 10, p. 221\",\n",
       " \"Killoran, A., Biglan, K.M., Current therapeutic options for Huntington's disease: good clinical practice versus evidence-based approaches? (2014) Mov Disord, 29 (11), pp. 1404-1413\"]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cite = metadata['References'][0]\n",
    "refs = cite.split(\"; \")\n",
    "refs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "c30e254d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['58081570200',\n",
       " '36163594500',\n",
       " '58082093700',\n",
       " '57808933900',\n",
       " '58082444500',\n",
       " '58017352200',\n",
       " '58081399200',\n",
       " '57222279300',\n",
       " '58081570300',\n",
       " '57197869642',\n",
       " '55809150500',\n",
       " '36962443700']"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test1 = metadata['Author(s) ID'][0]\n",
    "currentline = test1.split(\";\")\n",
    "currentline.pop()\n",
    "currentline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "78548f92",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Mühlbӓck A.',\n",
       " 'van Walsem M.',\n",
       " 'Nance M.',\n",
       " 'Arnesen A.',\n",
       " 'Page K.',\n",
       " 'Fisher A.',\n",
       " 'van Kampen M.',\n",
       " 'Nuzzi A.',\n",
       " 'Limpert R.',\n",
       " 'Fossmo H.L.',\n",
       " 'Cruickshank T.',\n",
       " 'Veenhuizen R.',\n",
       " \"the Multidisciplinary Care and Treatment Working Group of the European Huntington's Disease Network\"]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test2 = metadata['Authors'][0]\n",
    "currentline2 = test2.split(\", \")\n",
    "#currentline2.pop()\n",
    "currentline2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "80ae553c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mühlbӓck A., van Walsem M., Nance M., Arnesen A., Page K., Fisher A., van Kampen M., Nuzzi A., Limpert R., Fossmo H.L., Cruickshank T., Veenhuizen R., the Multidisciplinary Care and Treatment Working Group of the European Huntington's Disease Network\n",
      "Yang Q., Coutino M., Leus G., Giannakis G.B.\n",
      "van de Kuilen L., Leseman P.P.M., de Wolf I.M.\n",
      "Fathima Mafaziya Nijamdeen T.W.G., Ratsimbazafy H.A., Kodikara K.A.S., Ashara Nijamdeen T.W.G.F., Thahira T., Peruzzo S., Dahdouh-Guebas F., Hugé J.\n",
      "Hickey G.M., Roozee E., Voogd R., de Vries J.R., Sohns A., Kim D., Temby O.\n",
      "Sormani E., Rossano-Rivero S.\n",
      "Betts K., Creechan L., Cawkwell R., Finn-kelcey I., Griffin C.J., Hagopian A., Hartley D., Manalili M.A.R., Murkumbi I., O’donoghue S., Shanahan C., Stenning A., Zisk A.H.\n",
      "Zhou Y., Bos N.A., Diemers A.D., Brouwer J.\n",
      "Lee J., Babcock J., Pham T.S., Bui T.H., Kang M.\n",
      "van Venrooij L.T., Barnhoorn P.C., Barnhoorn-Bos A.M., Vermeiren R.R.J.M., Crone M.R.\n",
      "van der Veen D.J., Jellema S., van der Wees P.J., Graff M.J.L., de Swart B.J.M., Steultjens E.M.J.\n",
      "Rice W.S.\n",
      "Brunnekreef J.J., Feleus A., Miedema H.S., Staal J.B., Hutting N.\n",
      "Kools N., Dekker G.G., Kaijen B.A.P., Meijboom B.R., Bovens R.H.L.M., Rozema A.D.\n",
      "Klop H.T., de Veer A.J.E., Gootjes J.R.G., van de Mheen D., van Laere I.R., Slockers M.T., Onwuteaka-Philipsen B.D.\n",
      "Garrido-Skurkowicz N., Steglich C.\n",
      "Blanken M., Mathijssen J., van Nieuwenhuizen C., Raab J., van Oers H.\n",
      "Orrell M., Oliveira D., McDermott O., Verhey F.R.J., Dassen F.C.M., Dröes R.-M.\n",
      "Houser R.S., Pramana K.E.R., Ertsen M.W.\n",
      "Peisah C., de Mendonça Lima C., Verbeek H., Rabheru K.\n",
      "Reyes-Díaz M., Celly A., Folch C., Lorente N., Stuardo V., Veras M.A., Barros H., Meireles P., Ramírez D., Jonas K.J., Marcus U., Schmidt A.J., Caceres C.F., Casabona J.\n",
      "Poppelaars F., da Costa M.G., Inkeri Lokki A., Mallah K., Nord D., Reddaway J., Schäfer N.\n",
      "Lamper C., Huijnen I.P.J., Kroese M.E.A.L., Köke A.J., Brouwer G., Ruwaard D., Verbunt J.A.M.C.F.\n",
      "Gebauer J., Skinner R., Haupt R., Kremer L., van der Pal H., Michel G., Armstrong G.T., Hudson M.M., Hjorth L., Lehnert H., Langer T.\n",
      "Mafaziya Nijamdeen T.W.G.F., Hugé J., Ratsimbazafy H.A., Kodikara K.A.S., Dahdouh-Guebas F.\n",
      "van Iersel M., de Vos R., van Rijn M., Latour C.H.M., Kirschner P.A., Scholte op Reimer W.J.M.\n",
      "Hufsky F., Abecasis A., Agudelo-Romero P., Bletsa M., Brown K., Claus C., Deinhardt-Emmer S., Deng L., Friedel C.C., Gismondi M.I., Kostaki E.G., Kühnert D., Kulkarni-Kale U., Metzner K.J., Meyer I.M., Miozzi L., Nishimura L., Paraskevopoulou S., Pérez-Cataluña A., Rahlff J., Thomson E., Tumescheit C., van der Hoek L., Van Espen L., Vandamme A.-M., Zaheri M., Zuckerman N., Marz M.\n",
      "Vermond D., de Groot E., Sills V.A., Lyratzopoulos G., Walter F.M., de Wit N.J., Rubin G.\n",
      "Bellotti E., Czerniawska D., Everett M.G., Guadalupi L.\n",
      "Darnaude A.M., Teff-Seker Y., Tanner S., López-López L., Hidalgo M., Beger M., Türkmen A., Volkaert F., Hunter E.\n",
      "Husson O., Drabbe C., Schuster K., van Kampen P., Koops C., Weidema M., Davidson R., Wartenberg M., Artzner E., Gonzato O., Fernandez N., Kasper B., Pilgermann K., Wilson R., van der Graaf W.T.A., van Oortmerssen G.\n",
      "Pigford A.-A.E., Hickey G.M., Klerkx L.\n",
      "Maggio L.A., Haustein S., Costello J.A., Driessen E.W., Artino Jr A.R.\n",
      "Esteve Del Valle M., Broersma M., Ponsioen A.\n",
      "Ferreira F., Duarte J., Ugulino W.\n",
      "Lim S.M., Allard N.L., Devereux J., Cowie B.C., Tydeman M., Miller A., Ho K., Cleveland B., Singleton L., Aarons K., Eleftheriou P., Chan T., Braitberg G., Maier A.\n",
      "van Marrewijk A., den Ende L.V.\n",
      "Jasny L., Sayles J., Hamilton M., Roldan Gomez L., Jacobs D., Prell C., Matous P., Schiffer E., Guererro A.M., Barnes M.L.\n",
      "Marchegiani L., Brunetta F., Annosi M.C.\n",
      "Yuan Y., van Knippenberg D.\n",
      "De Groot E., Vermond D., Sills V.A., Mol S.S.L., Walter F.M., Rubin G., De Wit N.J.\n",
      "Jansen J., Boot C.R.L., Alma M.A., Brouwer S.\n",
      "Capponi G., Corrocher N.\n",
      "Noben I., Brouwer J., Deinum J.F., Hofman W.H.A.\n",
      "Belderbos R., Lokshin B., Boone C., Jacob J.\n",
      "Mierau K.\n",
      "Miller R., Lambert M.L., Frohnwieser A., Brecht K.F., Bugnyar T., Crampton I., Garcia-Pelegrin E., Gould K., Greggor A.L., Izawa E.-I., Kelly D.M., Li Z., Luo Y., Luong L.B., Massen J.J.M., Nieder A., Reber S.A., Schiestl M., Seguchi A., Sepehri P., Stevens J.R., Taylor A.H., Wang L., Wolff L.M., Zhang Y., Clayton N.S.\n",
      "Meilianti S., Oliveira R., Okoya F., Ercolin L., Mill D., Obidiegwu O., Uzman N., Lim R.\n",
      "Feng K., Karreman B., Zeng D., Pennings E.\n",
      "Kroeze E., Huijsman R.\n",
      "Achieng A.O., Lawrence T.J., Flavin B., Migeni A.Z., Coffey T.J., Otieno M.J., Stein H., Irvine K., Opaa B.\n",
      "Lambrechts F., Huybrechts J., De Massis A., Lehmann E.E.\n",
      "Chrysanthopoulou F., Lameris M., Greil G., Vudragovic D., Flynn K.\n",
      "Van Den Heuvel R., Jansens R., Littler B., Huijnen C., Di Nuovo A., Bonarini A., Desideri L., Encarnação P., Lekova A., De Witte L.\n",
      "Marsh J., Boszhard I., Contargyris A., Cullen J., Junge K., Molinari F., Osella M., Raspanti C.\n",
      "Blanken M., Mathijssen J., van Nieuwenhuizen C., Raab J., van Oers H.\n",
      "van Tongeren P.\n",
      "van den Dries M.H., Kerkhof M.J.H., Homme S.T.\n",
      "Calderón-Contreras R., Balvanera P., Trimble M., Langle-Flores A., Jobbágy E., Maass Moreno M., Marcone J., Mazzeo N., Muñoz Anaya M.M., Ortiz-Rodríguez I.A., Perevochtchikova M., Avila-Foucat S., Bonilla-Moheno M., Beth Clark L., Equihua M., Ayala-Orozco B., Bueno I., Hensler L., Leyva Aguilera J.C., Martínez Ramos M., Merçon J., Mesa-Jurado M.A., Österblom H., Pacheco-Vega R., Pérez Alcántara B., Pérez-Maqueo O., Porter-Bolland L., Quijas S., Quiroz Rosas L.E., Rios Patron E., Rocha-Gordo J.C., Rojo Negrete I.A., Romero-Duque L.P., Rosell J.A., Scheffer M., Vázquez L.-B., Villada Canela M., Velázquez M.\n",
      "Cohen J., Patterson P., Noke M., Clarke K., Husson O.\n",
      "Filimonau V., Ermolaev V.A.\n",
      "Niemi M.E.K., Karjalainen J., Liao R.G., Neale B.M., Daly M., Ganna A., Pathak G.A., Andrews S.J., Kanai M., Veerapen K., Fernandez-Cadenas I., Schulte E.C., Striano P., Marttila M., Minica C., Marouli E., Karim M.A., Wendt F.R., Savage J., Sloofman L., Butler-Laporte G., Kim H.-N., Kanoni S., Okada Y., Byun J., Han Y., Uddin M.J., Smith G.D., Willer C.J., Buxbaum J.D., Mehtonen J., Finucane H., Cordioli M., Martin A.R., Zhou W., Pasaniuc B., Julienne H., Aschard H., Shi H., Yengo L., Polimanti R., Ghoussaini M., Schwartzentruber J., Dunham I., Chwialkowska K., Francescatto M., Trankiem A., Balaconis M.K., Davis L., Lee S., Priest J., Renieri A., Sankaran V.G., van Heel D., Deelen P., Brent Richards J., Nakanishi T., Biesecker L., Eric Kerchberger V., Kenneth Baillie J., Mari F., Bernasconi A., Ceri S., Canakoglu A., Wolford B., Faucon A., Dutta A.K., Schurmann C., Harry E., Birney E., Nguyen H., Nasir J., Kaunisto M., Solomonson M., Dueker N., Vadgama N., Limou S., Rahmouni S., Mbarek H., Darwish D., Uddin M.M., Albertos R., Pérez-Tur J., Li R., Folkersen L., Moltke I., Koelling N., Teumer A., Kousathanas A., Utrilla A., Verdugo R.A., Zárate R., Medina-Gómez C., Gómez-Cabrero D., Carnero-Montoro E., Cadilla C.L., Moreno-Estrada A., Garmendia A., Moya L., Sedaghati-Khayat B., Boua P.R., Favé M.-J., Francioli L., Lemaçon A., Migeotte I., Patel S., Varnai R., Szentpeteri J.L., Sipeky C., Colombo F., von Hohenstaufen K., Lio P., Vallerga C., Wang Q., Tanigawa Y., Im H., Han C., Song H., Lim J., Lee Y., Kim S., Im S., Atanasovska B., Ahmad H.F., Boer C., Jansen P., Kaja E., Pasko D., Kennis-Szilagyi I., Kornilov S.A., Prijatelj V., Prokic I., Sivanadhan I., Perumal S., Esmaeeli S., Pearson N.M., Auton A., Shelton J.F., Shastri A.J., Filshtein-Sonmez T., Coker D., Symons A., Esparza-Gordillo J., Aslibekyan S., O’Connell J., Ye C., Weldon C.H., Perera M., O’Leary K., Tuck M., O’Brien T., Meltzer D., O’Donnell P., Nutescu E., Yang G., Alarcon C., Herrmann S., Mazurek S., Banagan J., Hamidi Z., Barbour A., Raffat N., Moreno D., Friedman P., Ferwerda B., van de Beek D., Brouwer M.C., Vlaar A.P.J., Joost Wiersinga W., Posthuma D., Tissink E., Koos Zwinderman A.H., Uffelmann E., van Agtmael M., Algera A.G., van Baarle F., Bax D., Beudel M., Bogaard H.J., Bomers M., Bonta P.I., Bos L., Botta M., de Brabander J., de Bree G., de Bruin S., Bugiani M., Bulle E., Chouchane O., Cloherty A., Dongelmans D., Elbers P., Fleuren L., Geerlings S., Geerts B., Geijtenbeek T., Girbes A., Goorhuis B., Grobusch M.P., Hafkamp F., Hagens L., Hamann J., Harris V., Hemke R., Hermans S.M., Heunks L., Hollmann M., Horn J., Hovius J.W., de Jong M.D., Koning R., van Mourik N., Nellen J., Nossent E.J., Paulus F., Peters E., van der Poll T., Preckel B., Prins J.M., Raasveld J., Reijnders T., Schinkel M., Schultz M.J., Schuurman A., Sigaloff K., Smit M., Stijnis C.S., Stilma W., Teunissen C., Thoral P., Tsonas A., van der Valk M., Veelo D., de Vries H., van Vugt M., Wouters D., Minnaar R.P., Kromhout A., van Uffelen K.W.J., Wolterman R.A., Roberts G., Park D., Ball C.A., Coignet M., McCurdy S., Knight S., Partha R., Rhead B., Zhang M., Berkowitz N., Gaddis M., Noto K., Ruiz L., Pavlovic M., Hong E.L., Rand K., Girshick A., Guturu H., Baltzell A.H., Guntz J., Beguin Y., Pigazzini S., Nkambule L., Bouysran Y., Busson A., Peyrassol X., Wilkin F., Pichon B., Smits G., Vandernoot I., Goffard J.-C., Georges M., Moutschen M., Misset B., Darcis G., Guiot J., Jadot L., Azarzar S., Dellot P., Gofflot S., Claassen S., Bertrand A., Parzibut G., Clarinval M., Moermans C., Malaise O., El Kandoussi K., Thonon R., Huynen P., Mesdagh A., Melo S., Jacques N., Di Valentin E., Giroule F., Collignon A., Radermecker C., Lebrun M., Perée H., Latour S., Barada O., Sanchez J., Josse C., Boujemla B., Meunier M., Mariavelle E., Anania S., Gazon H., Juszczak D., Fadeur M., Camby S., Meuris C., Thys M., Jacques J., Henket M., Léonard P., Frippiat F., Giot J.-B., Sauvage A.-S., Von Frenckell C., Mni M., Wéry M., Staderoli A., Belhaj Y., Lambermont B., Morrison D.R., Mooser V., Brent Richards J., Forgetta V., Li R., Ghosh B., Laurent L., Belisle A., Henry D., Abdullah T., Adeleye O., Mamlouk N., Kimchi N., Afrasiabi Z., Rezk N., Vulesevic B., Bouab M., Guzman C., Petitjean L., Tselios C., Xue X., Afilalo J., Afilalo M., Oliveira M., Brenner B., Brassard N., Durand M., Schurr E., Lepage P., Ragoussis J., Auld D., Chassé M., Kaufmann D.E., Mark Lathrop G., Adra D., Davis L.K., Cox N.J., Below J.E., Sealock J.M., Faucon A.B., Shuey M.M., Polikowsky H.G., Petty L.E., Shaw D.M., Chen H.-H., Zhu W., Ludwig K.U., Schröder J., Maj C., Rolker S., Nöthen M.M., Fazaal J., Keitel V., Jensen B.-E.O., Feldt T., Kurth I., Marx N., Dreher M., Pink I., Cornberg M., Illig T., Lehmann C., Schommers P., Augustin M., Knopp L., Eggermann T., Volland S., Altmüller J., Berger M.M., Brenner T., Hinney A., Witzke O., Bals R., Herr C., Ludwig N., Walter J., Fuchsberger C., Pattaro C., De Grandi A., Pramstaller P., Emmert D., Melotti R., Foco L., Mascalzoni D., Gögele M., Domingues F., Hicks A., Gignoux C.R., Wicks S.J., Crooks K., Barnes K.C., Daya M., Shortt J., Rafaels N., Chavan S., Goldstein D.B., Kiryluk K., Sengupta S., Chung W., Reilly M.P., Khan A., Wang C., Povysil G., Bhardwaj N., Gharavi A.G., Ionita-Laza I., Shang N., O’Byrne S.M., Nandakumar R., Menon A., So Y.S., Hod E., Pendrick D., Park S.-K., Kim H.-L., Kang C.K., Lee H.-J., Song K.-H., Yoon K.J., Paik N.-J., Seok W., Yoon H., Joo E.-J., Chang Y., Ryu S., Park W.B., Park J.S., Park K.U., Ham S.Y., Jung J., Kim E.S., Kim H.B., Ellinghaus D., Degenhardt F., Cáceres M., Juzenas S., Lenz T.L., Albillos A., Julià A., Heidecker B., Garcia F., Kurth F., Tran F., Hanses F., Zoller H., Holter J.C., Fernández J., Sander L.E., Rosenstiel P., Koehler P., de Cid R., Asselta R., Schreiber S., Hehr U., Prati D., Baselli G., Valenti L., Bujanda L., Banales J.M., D’Amato M., Romero-Gómez M., Buti M., Invernizzi P., Franke A., Hov J.R., Karlsen T.H., Folseraas T., Maya-Miles D., Teles A., Azuure C., Wacker E.M., Uellendahl-Werth F., ElAbd H., Arora J., Lerga-Jaso J., Wienbrandt L., Rühlemann M.C., Wendorff M., Vadla M.S., Lenning O.B., Özer O., Myhre R., Raychaudhuri S., Tanck A., Gassner C., Hemmrich-Stanisak G., Kässens J., Figuera Basso M.E., Schulzky M., Wittig M., Braun N., Wesse T., Albrecht W., Yi X., Ortiz A.B., Chercoles A.G., Ruiz A., Mantovani A., Holten A.R., Mayer A., Cherubini A., Protti A., Aghemo A., Gerussi A., Ramirez A., Braun A., Barreira A., Lleo A., Kildal A.B., Glück A., Nolla A.C., Latiano A., Dyrhol-Riise A.M., Muscatello A., Voza A., Rando-Segura A., Solier A., Karina B., Cortes B., Mateos B., Nafria-Jimenez B., Schaefer B., Bellinghausen C., Ferrando C., Quereda C., Skurk C., Thibeault C., Spinner C.D., Lange C., Hu C., Cappadona C., Bianco C., Sancho C., Hoff D.A.L., Galimberti D., Jiménez D., Pestaña D., Toapanta D., Azzolini E., Scarpini E., Helbig E.T., Urrechaga E., Paraboschi E.M., Pontali E., Reverter E., Navas E., Arana E., Sánchez F.G., Ceriotti F., Malvestiti F., Mesonero F., Pezzoli G., Lamorte G., Neb H., My I., Hernández I., de Rojas I., Galván-Femenia I., Heyckendorf J., Rybniker J., Badia J.R., Schneider J., Goikoetxea J., Kraft J., Müller K.E., Gaede K.I., Garcia-Etxebarria K., Tonby K., Heggelund L., Izquierdo-Sanchez L., Sumoy L., Lippert L.J., Terranova L., Garbarino L., Téllez L., Roade L., Ostadreza M., Intxausti M., Kogevinas M., Gutiérrez-Stampa M.A., Vehreschild M.J.G.T., Marquié M., Castoldi M., Cecconi M., Boada M., Seilmaier M.J., Mazzocco M., Rodríguez-Gandía M., Ayo N.I., Blay N., Martínez N., Cornely O.A., Palmieri O., Tentorio P., Rodrigues P.M., España P.P., Hoffmann P., Bacher P., Suwalski P., de Pablo R., Nieto R., Badalamenti S., Ciesek S., Bombace S., Wilfling S., Brunak S., Heilmann-Heimbach S., Ripke S., Bahmer T., Landmesser U., Protzer U., Rimoldi V., Skogen V., Andrade V., Moreno V., Poller W., Farre X., Wang X., Khodamoradi Y., Karadeniz Z., de Salazar A., Palom A., Garcia-Fernandez A.-E., Blanco-Grau A., Zanella A., Bandera A., Nebel A., Biondi A., Caballero-Garralda A., Gori A., Lind A., Fracanzani A.L., Peschuck A., Pesenti A., de la Horra C., Milani C., Paccapelo C., Angelini C., Cea C., Muñiz-Diaz E., Sandoval E., Calderón E.J., Solligård E., Aziz F., Martinelli-Boneschi F., Peyvandi F., Blasi F., Medrano F.J., Rodriguez-Frias F., Müller F., Grasselli G., Costantino G., Cardamone G., Foti G., Matullo G., Kurihara H., Afset J.E., Damås J.K., Ampuero J., Martín J., Erdmann J., Bergan J., Goerg S., Ferrusquía-Acosta J., Quero J.H., Delgado J., Guerrero J.M., Risnes K., Bettini L.R., Moreira L., Gustad L.T., Santoro L., Scudeller L., Riveiro-Barciela M., Schaefer M., Carrabba M., Valsecchi M.G., Hernandez-Tejero M., Acosta-Herrera M., D’Angiò M., Baldini M., Cazzaniga M., Ciccarelli M., Bocciolone M., Miozzo M., Chueca N., Montano N., Faverio P., Preatoni P., Bonfanti P., Omodei P., Castro P., Ferrer R., Gualtierotti R., Gallego-Durán R., Morilla R., Haider S., Marsal S., Aneli S., Pelusi S., Bosari S., Aliberti S., Dudman S., Zheng T., Pumarola T., Cejudo T.G., Monzani V., Friaza V., Peter W., Dopazo X., Duga S., May S., Grimsrud M.M., Gudbjartsson D.F., Stefansson K., Sulem P., Sveinbjornsson G., Melsted P., Norddahl G., Moore K.H.S., Thorsteinsdottir U., Holm H., Alarcón-Riquelme M.E., Bernardo D., Martínez-Bueno M., Rello S.R., Mägi R., Milani L., Metspalu A., Laisk T., Läll K., Lepamets M., Esko T., Reimann E., Naaber P., Laane E., Pesukova J., Peterson P., Kisand K., Tabri J., Allos R., Hensen K., Starkopf J., Ringmets I., Tamm A., Kallaste A., Alavere H., Metsalu K., Puusepp M., Kristiansson K., Koskelainen S., Perola M., Donner K., Kivinen K., Palotie A., Rivolta C., Bochud P.-Y., Bibert S., Boillat N., Nussle S.G., Albrich W., Quinodoz M., Kamdar D., Suh N., Neofytos D., Erard V., Voide C., Bochud P.Y., Bibert S., Quinodoz M., Kamdar D., Neofytos D., Erard V., Voide C., Friolet R., Vollenweider P., Pagani J.L., Oddo M., Meyer zu Bentrup F., Conen A., Clerc O., Marchetti O., Guillet A., Guyat-Jacques C., Foucras S., Rime M., Chassot J., Jaquet M., Merlet Viollet R., Lannepoudenx Y., Portopena L., Bochud P.Y., Vollenweider P., Pagani J.L., Desgranges F., Filippidis P., Guéry B., Haefliger D., Kampouri E.E., Manuel O., Munting A., Papadimitriou-Olivgeris M., Regina J., Rochat-Stettler L., Suttels V., Tadini E., Tschopp J., Van Singer M., Viala B., Boillat-Blanco N., Brahier T., Hügli O., Meuwly J.Y., Pantet O., Gonseth Nussle S., Bochud M., D’Acremont V., Estoppey Younes S., Albrich W.C., Suh N., Cerny A., O’Mahony L., von Mering C., Bochud P.Y., Frischknecht M., Kleger G.-R., Filipovic M., Kahlert C.R., Wozniak H., Rochat Negro T., Pugin J., Bouras K., Knapp C., Egger T., Perret A., Montillier P., di Bartolomeo C., Barda B., Carreras A., Galván-Femenía I., Farré X., Cortés B., Mercader J.M., Guindo-Martinez M., Torrents D., Garcia-Aymerich J., Castaño-Vinyals G., Dobaño C., Gori M., Mondelli M.U., Castelli F., Vaghi M., Rusconi S., Montagnani F., Bargagli E., Franchi F., Mazzei M.A., Cantarini L., Tacconi D., Feri M., Scala R., Spargi G., Nencioni C., Bandini M., Caldarelli G.P., Spagnesi M., Canaccini A., Ognibene A., Monforte A.D., Girardis M., Antinori A., Francisci D., Schiaroli E., Scotton P.G., Panese S., Scaggiante R., Monica M.D., Capasso M., Fiorentino G., Castori M., Aucella F., Di Biagio A., Masucci L., Valente S., Mandalà M., Zucchi P., Giannattasio F., Coviello D.A., Mussini C., Bosio G., Tavecchia L., Crotti L., Rizzi M., La Rovere M.T., Sarzi-Braga S., Bussotti M., Ravaglia S., Artuso R., Perrella A., Romani D., Bergomi P., Catena E., Vincenti A., Ferri C., Grassi D., Pessina G., Tumbarello M., Di Pietro M., Sabrina R., Luchi S., Barbieri C., Acquilini D., Andreucci E., Paciosi F., Segala F.V., Tiseo G., Lista M., Poscente M., De Vivo O., Petrocelli P., Guarnaccia A., Baroni S., Perticaroli V., Furini S., Dei S., Benetti E., Picchiotti N., Sanarico M., Pinoli P., Raimondi F., Biscarini F., Stella A., Bergomi M., Zguro K., Capitani K., Tanfoni M., Fallerini C., Daga S., Baldassarri M., Fava F., Frullanti E., Valentino F., Doddato G., Giliberti A., Tita R., Amitrano S., Bruttini M., Croci S., Meloni I., Mencarelli M.A., Lo Rizzo C., Pinto A.M., Beligni G., Tommasi A., Di Sarno L., Palmieri M., Carriero M.L., Alaverdian D., Iuso N., Inchingolo G., Busani S., Bruno R., Vecchia M., Ann Belli M., Mantovani S., Ludovisi S., Quiros-Roldan E., Antoni M.D., Zanella I., Siano M., Emiliozzi A., Fabbiani M., Rossetti B., Zanelli G., Bergantini L., D’Alessandro M., Cameli P., Bennet D., Anedda F., Marcantonio S., Scolletta S., Guerrini S., Conticini E., Frediani B., Spertilli C., Donati A., Guidelli L., Corridi M., Croci L., Piacentini P., Desanctis E., Cappelli S., Verzuri A., Anemoli V., Pancrazi A., Lorubbio M., Merlini E., Miraglia F.G., Venturelli S., Cossarizza A., Vergori A., Gabrieli A., Riva A., Andretta F., Gatti F., Parisi S.G., Baratti S., Piscopo C., Russo R., Andolfo I., Iolascon A., Carella M., Merla G., Squeo G.M., Raggi P., Marciano C., Perna R., Bassetti M., Sanguinetti M., Giorli A., Salerni L., Parravicini P., Menatti E., Trotta T., Coiro G., Lena F., Martinelli E., Mancarella S., Gabbi C., Maggiolo F., Ripamonti D., Bachetti T., Suardi C., Parati G., Bottà G., Di Domenico P., Rancan I., Bianchi F., Colombo R., Falcone M., Hunt K.A., Trembath R.C., Martin H.C., Mason D., Trivedi B., Wright J., Finer S., Griffiths C.J., Akhtar S., Anwar M., Arciero E., Ashraf S., Breen G., Chung R., Curtis C.J., Chowdhury M., Colligan G., Deloukas P., Durham C., Griffiths C., Huang Q.Q., Hurles M., Hussain S., Islam K., Khan A., Khan A., Lavery C., Lee S.H., Lerner R., MacArthur D., MacLaughlin B., Martin H., Miah S., Newman B., Safa N., Tahmasebi F., Smith A.V., Boughton A.P., Li K.W., LeFaive J., Annis A., Jannes C.E., Krieger J.E., Pereira A.C., Velho M., Marques E., Lima I.R., Tada M.T., Valino K., McCarthy M., Rosenberger C., Lee J.E., Chang D., Hammer C., Hunkapiller J., Mahajan A., Pendergrass S., Sucheston-Campbell L., Yaspan B., Lee H.S., Shin E., Jang H.Y., Kim S., Kym S., Kim Y.-S., Jeong H., Kwon K.T., Kim S.-W., Kim J.Y., Jang Y.R., Kim H., Lee J.Y., Lee S., Choe K.-W., Kang Y.M., Jee S.H., Jung K.J., Parikh V., Ashley E., Wheeler M., Rivas M., Bustamante C., Pinksy B., Febbo P., Farh K., Schroth G.P., deSouza F., Dalton K., Christle J., Deboever C., Szalma S., Rubinacci S., Delaneau O., Gorzynski J., de Jong H., Sutton S., Youlton N., Joshi R., Jimenez-Morales D., Hughes C., Amar D., Ioannidis A., Hershman S., Kirillova A., Seo K., Huang Y., Shoura M., Hammond N., Watson N., Raja A., Huang C., Sahoo M., Wang H., Zhen J., Rakitko A., Ilinsky V., Yermakovich D., Popov I., Chernitsov A., Kovalenko E., Krasnenko A., Plotnikov N., Stetsenko I., Kim A., Cirulli E.T., Schiabor Barrett K.M., Bolze A., White S., Washington N.L., Lu J.T., Riffle S., Tanudjaja F., Wang X., Ramirez J.M., III, Leonetti N., Sandoval E., Neveux I., Dabe S., Grzymski J.J., Miñano J.I.E., Aguirre L.A., López-Collazo E., de la Mata Pazos M., Cerrato L., Lozano-Rodríguez R., Avendaño-Ortiz J., Arcos V.T., Montalbán-Hernández K.M., Quiroga J.V., Pascual-Iglesias A., Maroun-Eid C., Martín-Quirós A., Namkoong H., Imoto S., Katayama K., Fukunaga K., Kitagawa Y., Sato T., Hasegawa N., Kumanogoh A., Kimura A., Ai M., Tokunaga K., Kanai T., Miyano S., Ogawa S., Edahiro R., Sonehara K., Shirai Y., Ishii M., Kabata H., Masaki K., Kamata H., Ikemura S., Chubachi S., Okamori S., Terai H., Tanaka H., Morita A., Lee H., Asakura T., Sasaki J., Morisaki H., Uwamino Y., Nanki K., Mikami Y., Tomono K., Kato K., Matsuda F., Takahashi M., Hizawa N., Takeda Y., Hirata H., Shiroyama T., Miyawaki S., Suzuki K., Maeda Y., Nii T., Noda Y., Niitsu T., Adachi Y., Enomoto T., Amiya S., Hara R., Takahashi K., Anzai T., Hasegawa T., Ito S., Koike R., Endo A., Uchimura Y., Miyazaki Y., Honda T., Tateishi T., Tohda S., Ichimura N., Sonobe K., Sassa C., Nakajima J., Nannya Y., Omae Y., Takahashi K., Harada N., Hiki M., Takagi H., Nakamura A., Tagaya E., Kawana M., Arimura K., Ishiguro T., Takayanagi N., Isono T., Takaku Y., Takano K., Anan R., Nakajima Y., Nakano Y., Nishio K., Ueda S., Hayashi R., Tateno H., Hase I., Yoshida S., Suzuki S., Mitamura K., Saito F., Ueda T., Azuma M., Nagasaki T., Yasui Y., Hasegawa Y., Mutoh Y., Yoshiyama T., Shoko T., Kojima M., Adachi T., Ishikawa M., Takahashi K., Watanabe K., Manabe T., Ito F., Fukui T., Funatsu Y., Koh H., Hirai Y., Kawashima H., Narita A., Niwa K., Sekikawa Y., Saito F., Yoshiya K., Yoshihara T., Suzuki Y., Nakayama S., Masuzawa K., Nishi K., Nishitsuji M., Tani M., Inoue T., Hirano T., Kobayashi K., Miyazawa N., Kimura Y., Sado R., Ogura T., Kitamura H., Murohashi K., Nakachi I., Baba R., Arai D., Fuke S., Saito H., Kuwahara N., Fujiwara A., Okada T., Baba T., Noda J., Mashimo S., Yagi K., Shiomi T., Hashiguchi M., Odani T., Mochimaru T., Oyamada Y., Mori N., Izumi N., Nagata K., Taki R., Murakami K., Yamada M., Sugiura H., Hayashi K., Shimizu T., Gon Y., Fujitani S., Tsuchida T., Yoshida T., Kagaya T., Kita T., Sakagami S., Kimizuka Y., Kawana A., Nakamura Y., Ishikura H., Takata T., Kikuchi T., Taniyama D., Nakamura M., Kodama N., Kaneyama Y., Maeda S., Nagasaki Y., Okamoto M., Ishihara S., Ito A., Chihara Y., Takeuchi M., Onoi K., Hashimoto N., Wakahara K., Ando A., Masuda M., Wakabayashi A., Watanabe H., Sageshima H., Nakada T.-A., Abe R., Shimada T., Kawamura K., Ichikado K., Nishiyama K., Yamasaki M., Hashimoto S., Kusaka Y., Ohba T., Isogai S., Takada M., Kanda H., Komase Y., Sano F., Asano K., Oguma T., Harada M., Takahashi T., Shibusawa T., Abe S., Kono Y., Togashi Y., Izumo T., Inomata M., Awano N., Ogawa S., Ogata T., Ishihara S., Kanehiro A., Ozaki S., Fuchimoto Y., Kitagawa Y., Yoshida S., Ogura S., Nishiyama K., Yoshida K., Beppu S., Fukuyama S., Eriguchi Y., Yonekawa A., Inoue Y., Yamagata K., Chiba S., Narumoto O., Nagai H., Ooshima N., Motegi M., Sagara H., Tanaka A., Ohta S., Shibata Y., Tanino Y., Sato Y., Yamada Y., Hashino T., Shinoki M., Iwagoe H., Imamura T., Umeda A., Shimada H., Endo M., Hayashi S., Takahashi M., Nakano S., Yatomi M., Maeno T., Ishii T., Utsugi M., Ono A., Kanaoka K., Ihara S., Komuta K., Franke L., Boezen M., Claringbould A., Lopera E., Warmerdam R., Vonk J.M., van Blokland I., Lanting P., Ori A.P.S., Obeidat M., Hernández Cordero A.I., Sin D.D., Bossé Y., Joubert P., Hao K., Nickle D., Timens W., van den Berge M., Feng Y.-C.A., Mercader J., Weiss S.T., Karlson E.W., Smoller J.W., Murphy S.N., Meigs J.B., Woolley A.E., Green R.C., Perez E.F., Zöllner S., Wang J., Beck A., Sloofman L.G., Ascolillo S., Sebra R.P., Collins B.L., Levy T., Sealfon S.C., Jordan D.M., Thompson R.C., Gettler K., Chaudhary K., Belbin G.M., Preuss M., Hoggart C., Choi S., Underwood S.J., Salib I., Britvan B., Keller K., Tang L., Peruggia M., Hiester L.L., Niblo K., Aksentijevich A., Labkowsky A., Karp A., Zlatopolsky M., Zyndorf M., Charney A.W., Beckmann N.D., Schadt E.E., Abul-Husn N.S., Cho J.H., Itan Y., Kenny E.E., Loos R.J.F., Nadkarni G.N., Do R., O’Reilly P., Huckins L.M., Ferreira M.A.R., Abecasis G.R., Leader J.B., Cantor M.N., Justice A.E., Carey D.J., Chittoor G., Josyula N.S., Kosmicki J.A., Horowitz J.E., Baras A., Gass M.C., Yadav A., Mirshahi T., Hottenga J.J., Bartels M., de Geus E.J.C., Nivard M.G., Verma A., Ritchie M.D., Rader D., Li B., Verma S.S., Lucas A., Bradford Y., Zara F., Salpietro V., Scala M., Iacomino M., Scudieri P., Bocciardi R., Minetti C., Riva A., Vari M.S., Rahier J.-F., Giorgio E., Carli D., Louis E., Bulik C.M., Landén M., Brusco A., Ferrero G.B., Madia F., Fundín B., Ismail S.I., Saad C., Al-Sarraj Y., Badji R.M., Al-Muftah W., Al Thani A., Afifi N., Klovins J., Rovite V., Rescenko R., Peculis R., Ustinova M., Zeberg H., Frithiof R., Hultström M., Lipcsey M., Johnson R., Geschwind D.H., Freimer N., Butte M.J., Ding Y., Chiu A., Chang T.S., Boutros P., Moutsianas L., Caulfield M.J., Scott R.H., Walker S., Stuckey A., Odhams C.A., Rhodes D., Fowler T., Rendon A., Chan G., Arumugam P., Karczewski K.J., Wilson D.J., Spencer C.A., Crook D.W., Wyllie D.H., O’Connell A.M., Brent Richards J., Atkinson E.G., Tsuo K., Baya N., Turley P., Gupta R., Walters R.K., Palmer D.S., Sarma G., Cheng N., Lu W., Churchhouse C., Goldstein J.I., King D., Seed C., Bryant S., Kyle Satterstrom F., Band G., Earle S.G., Lin S.-K., Arning N., Armstrong J., Rudkin J.K., Callier S., Cusick C., Soranzo N., Zhao J.H., Danesh J., Di Angelantonio E., Butterworth A.S., Sun Y.V., Huffman J.E., Cho K., O’Donnell C.J., Tsao P., Michael Gaziano J., Peloso G., Ho Y.-L., Mian M., Scaggiante F., Chang X., Glessner J.R., Hakonarson H., Kenneth Baillie J., McGuigan P.J., Moore L.S.P., Vizcaychipi M.P., Hall K., Campbell A., Nichol A., Ward G., Page V.J., Adeniji K., Agranoff D., Agwuh K., Ail D., Aldera E.L., Alegria A., Angus B., Ashish A., Atkinson D., Bari S., Barlow G., Barnass S., Barrett N., Bassford C., Basude S., Baxter D., Beadsworth M., Bernatoniene J., Berridge J., Best N., Bothma P., Chadwick D., Brittain-Long R., Bulteel N., Burden T., Caruth V., Chambler D., Chee N., Child J., Clark T., Collini P., Cosgrove C., Cupitt J., Cutino-Moguel M.-T., Dark P., Dawson C., Dervisevic S., Donnison P., Douthwaite S., Drummond A., DuRand I., Dushianthan A., Dyer T., Evans C., Eziefula C., Fegan C., Finn A., Fullerton D., Garg S., Garg A., Gkrania-Klotsas E., Godden J., Goldsmith A., Graham C., Hardy E., Hartshorn S., Harvey D., Havalda P., Hawcutt D.B., Hobrok M., Jacobs M., Jain S., Jennings P., Kaliappan A., Kasipandian V., Kegg S., Kelsey M., Kendall J., Kerrison C., Kerslake I., Koch O., Koduri G., Koshy G., Laha S., Laird S., Larkin S., Leiner T., Lillie P., Limb J., Linnett V., Little J., Lyttle M., MacMahon M., MacNaughton E., Mankregod R., Masson H., Matovu E., McCullough K., McEwen R., Meda M., Minton J., Ward K., Mirfenderesky M., Mohandas K., Mok Q., Moon J., Moore E., Morgan P., Morris C., Mortimore K., Moses S., Mpenge M., Mulla R., Murphy M., Nagel M., Nagarajan T., Nelson M., O’Shea M.K., Otahal I., Ostermann M., Pais M., Panchatsharam S., Papakonstantinou D., Paraiso H., Patel B., Pattison N., Pepperell J., Peters M., Phull M., Pintus S., Pooni J.S., Post F., Price D., Prout R., Rae N., Reschreiter H., Reynolds T., Richardson N., Roberts M., Roberts D., Rose A., Rousseau G., Ryan B., Saluja T., Shah A., Shanmuga P., Sharma A., Shawcross A., Sizer J., Shankar-Hari M., Smith R., Snelson C., Spittle N., Staines N., Stambach T., Stewart R., Subudhi P., Tatham K., Thomas J., Thompson C., Thompson R., Tridente A., Tupper-Carey D., Twagira M., Ustianowski A., Vallotton N., Vincent-Smith L., Visuvanathan S., Vuylsteke A., Waddy S., Wake R., Walden A., Welters I., Whitehouse T., Whittaker P., Whittington A., Papineni P., Wijesinghe M., Williams M., Wilson L., Cole S., Winchester S., Wiselka M., Wolverson A., Wooton D.G., Workman A., Yates B., Young P., Kenneth Baillie J., Beale R., Bretherick A.D., Fourman M.H., Furniss J., Gountouna E., Grimes G., Haley C., Harrison D., Hayward C., Keating S., Klaric L., Klenerman P., Meynert A.M., Millar J., Pairo-Castineira E., Parkinson N., Ponting C.P., Porteous D.J., Rawlik K., Richmond A., Russell C.D., Shen X., Shih B., Tenesa A., Vitart V., Wang B., Wilson J.F., Wu Y., Yang J., Yang Z., Zechner M., Zhai R., Zheng C., Norman L., Pius R., Drake T.M., Fairfield C.J., Knight S.R., Mclean K.A., Murphy D., Shaw C.A., Dalton J., Girvan M., Saviciute E., Roberts S., Harrison J., Marsh L., Connor M., Halpin S., Jackson C., Gamble C., Leeming G., Law A., Wham M., Clohisey S., Hendry R., Scott-Brown J., Begg C., Hinds C., Ho A.Y.W., Horby P.W., Knight J., Ling L., Maslove D., McAuley D., Montgomery H., Openshaw P.J.M., Rowan K., Semple M.G., Summers C., Walsh T., Armstrong L., Bates H., Dooks E., Farquhar F., Hairsine B., McParland C., Packham S., Alldis Z., Astin-Chamberlain R., Bibi F., Biddle J., Blow S., Bolton M., Borra C., Bowles R., Burton M., Choudhury Y., Collier D., Cox A., Easthope A., Ebano P., Fotiadis S., Gurasashvili J., Halls R., Hartridge P., Kallon D., Kassam J., Lancoma-Malcolm I., Matharu M., May P., Mitchelmore O., Newman T., Patel M., Pheby J., Pinzuti I., Prime Z., Prysyazhna O., Shiel J., Tierney C., Wood S., Zak A., Zongo O., Forsey M., Nicholson A., Riches J., Vertue M., Wasson C., Finn S., Green J., Collins E., King B., Grauslyte L., Hussain M., Pogreban T., Rosaroso L., Salciute E., Franke G., Wong J., George A., Akeroyd L., Bano S., Bromley M., Gurr L., Lawton T., Morgan J., Sellick K., Warren D., Wilkinson B., McGowan J., Ledgard C., Stacey A., Pye K., Bellwood R., Bentley M., Loosley R., McGuinness H., Tench H., Wolf-Roberts R., Gibson S., Lyle A., McNeela F., Radhakrishnan J., Hughes A., Ali A., Brady M., Dale S., Dance A., Gledhill L., Greig J., Hanson K., Holdroyd K., Home M., Kelly D., Kitson R., Matapure L., Melia D., Mellor S., Nortcliffe T., Pinnell J., Robinson M., Shaw L., Shaw R., Thomis L., Wilson A., Wood T., Bayo L.-A., Merwaha E., Ishaq T., Hanley S., Antcliffe D., Banach D., Brett S., Coghlan P., Fernandez Z., Gordon A., Rojo R., Arias S.S., Templeton M., Jha R., Lim L., Bi R., Scholefield B., Ashton L., Williams A., Cheyne C., Saunderson A., Allan A., Anderson F., Kaye C., Liew J., Medhora J., Scott T., Trumper E., Botello A., Polgarova P., Stroud K., Meaney E., Jones M., Ng A., Agrawal S., Pathan N., White D., Daubney E., Elston K., Parker R., Reddy A., Turner-Bone I., Wilding L., Harding P., Jacob R., Jones C., Denmade C., Croft M., White I., Krishnamurthy V., Lim L., Griffin D., Muchenje N., Mupudzi M., Partridge R., Conyngham J.-A., Thomas R., Wright M., Corral M.A., Bastion V., Clarke D., David B., Kent H., Lorusso R., Lubimbi G., Murdoch S., Penacerrada M., Thomas A., Valentine J., Vochin A., Wulandari R., Djeugam B., Dawson J., Garrioch S., Tolson M., Aldridge J., de Almeida Martins L.G., Carungcong J., Beavis S., Dale K., Gascoyne R., Hawes J., Pritchard K., Stevenson L., Whileman A., Crawley R., Crew A., Cunningham M., Daniels A., Harrison L., Hope S., Inweregbu K., Jones S., Lancaster N., Matthews J., Nicholson A., Wray G., Benham L., Bradshaw Z., Brown J., Caswell M., Melling S., Preston S., Slawson N., Stoddard E., Warden S., Combes E., Joefield T., Monnery S., Beech V., Trotman S., Hopkins B., Thrasyvoulou L., Willis H., Anderson S., Birch J., Collins E., Hammerton K., O’Leary R., Abernathy C., Foster L., Gratrix A., Martinson V., Parkinson P., Stones E., Carbral-Ortega L., Kapoor R., Loader D., Castle K., Brandwood C., Smith L., Birchall K., Kolakaluri L., Baines D., Sukumaran A., Mapfunde I., Meredith M., Morris L., Ryan L., Clark A., Sampson J., Peters C., Dent M., Langley M., Wei S., Andrew A., Chablani M., Kirkby A., Netherton K., Bates M., Dasgin J., Gill J., Nilsson A., Apetri E., Basikolo C., Blackledge B., Catlow L., Charles B., Doonan R., Harris J., Harvey A., Horner D., Knowles K., Lee S., Lomas D., Lyons C., Marsden T., McLaughlan D., McMorrow L., Pendlebury J., Perez J., Poulaka M., Proudfoot N., Slaughter M., Slevin K., Taylor M., Thomas V., Walker D., Michael A., Collis M., Clark M., Coulding M., Jude E., McCormick J., Mercer O., Potla D., Rehman H., Savill H., Turner V., Davey M., Golden D., Seaman R., Hunt J., Dearden J., Dobson E., Mulcahy M., Munt S., O’Connor G., Philbin J., Rishton C., Tully R., Winnard S., Cagova L., Fofano A., Garner L., Holcombe H., Mepham S., Mitchell A.M., Mwaura L., Praman K., Vuylsteke A., Zamikula J., Bercades G., Hass I., MacCallum N., Martir G., Raith E., Reyes A., Smyth D., Taylor A., Hughes R.A., Thomas H., Rees A., Duskova M., Phipps J., Brooks S., Edwards M., Alexander P., Allen S., Bradley-Potts J., Brantwood C., Egan J., Felton T., Padden G., Ward L., Moss S., Glasgow S., Beesley K., Board S., Kubisz-Pudelko A., Lewis A., Perry J., Pippard L., Wood D., Buckley C., Brown A., Gregory J., O’Connell S., Belagodu Z., Fuller B., Gherman A., Olufuwa O., Paramsothy R., Stuart C., Oakley N., Kamundi C., Tyl D., Collins K., Silva P., Taylor J., King L., Coates C., Crowley M., Wakefield P., Beadle J., Johnson L., Sargeant J., Anderson M., Jardine C., Williams D., Parris V., Quaid S., Watson E., Melville J., Naisbitt J., Joseph R., Lazo M., Walton O., Neal A., Hill M., Kannan T., Allan E., Darlington K., Davies F., Easton J., Kumar S., Lean R., Menzies D., Pugh R., Qiu X., Davies L., Williams H., Scanlon J., Davies G., Mackay C., Lewis J., Rees S., Coetzee S., Gales A., Raj M., Sell C., Langton H., Watters M., Novis C., Arbane G., Bociek A., Campos S., Grau N., Jones T.O., Lim R., Marotti M., Whitton C., Barron A., Collins C., Kaul S., Passmore H., Prendergast C., Reed A., Rogers P., Shokkar R., Woodruff M., Middleton H., Polgar O., Nolan C., Mahay K., Sri-Chandana C., Scherewode J., Stephenson L., Marsh S., Bancroft H., Bellamy M., Carmody M., Daglish J., Moore F., Rhodes J., Sangombe M., Kadiri S., Scriven J., Ayers A., Harrison W., North J., Cavazza A., Cockrell M., Corcoran E., Depante M., Finney C., Jerome E., McPhail M., Nayak M., Noble H., O’Reilly K., Pappa E., Saha R., Saha S., Smith J., Knighton A., Gill M., Paul P., Ratnam V., Shelton S., Wynter I., Baptista D., Crowe R., Fernandes R., Herdman-Grant R., Joseph A., Loveridge A., McKenley I., Morino E., Naranjo A., Simms R., Sollesta K., Swain A., Venkatesh H., Khera J., Fox J., Barber R., Hewitt C., Hilldrith A., Jackson-Lawrence K., Shepardson S., Wills M., Butler S., Tavares S., Cunningham A., Hindale J., Arif S., George L., Twiss S., Wright D., Holland M., Keenan N., Lyons M., Wassall H., Marsh C., Mahenthran M., Carter E., Kong T., Adanini O., Bhatia N., Msiska M., Mew L., Mwaura E., Williams F., Wren L., Sutherland S.-B., Battle C., Brinkworth E., Harford R., Murphy C., Newey L., Rees T., Williams M., Arnold S., Brealey D., Hardy J., Houlden H., Moncur E., Tariq A., Tucci A., Convery K., Fottrell-Gould D., Hudig L., Keshet-Price J., Randell G., Stammers K., Abdelrazik M., Bakthavatsalam D., Elhassan M., Ganesan A., Haldeos A., Moreno-Cuesta J., Purohit D., Vincent R., Xavier K., Rohit K., Alasdair F., Saleem M., David C., Jenkins S., Lamond Z., Wall A., Reynolds J., Campbell H., Thompsom M., Dodds S., Duffy S., Butcher D., O’Sullivan S., Butterworth-Cowin N., Deacon B., Hibbert M., Pothecary C., Tetla D., Woodford C., Durga L., Kennard-Holden G., de Gordoa L.O.-R., Peasgood E., Phillips C., Skinner D., Gaylard J., Mullan D., Newman J., Davies E., Sathe S., Brimfield L., Daly Z., Pogson D., Rose S., Collins A., Khaliq W., Gude E.T., Allen L., Beranova E., Crisp N., Deery J., Hazelton T., Knight A., Price C., Tilbey S., Turki S., Turney S., Giles J., Booth S., Bell G., English K., Katary A., Wilcox L., Campbell R., Clarke N., Whiteside J., Mascarenhas M., Donaldson A., Matheson J., Barrett F., O’Hara M., O’Keefe L., Bradley C., Collier D., Hormis A., Walker R., Maynard V., Patel T., Smith M., Chukkambotla S., Kazi A., Hartley J., Dykes J., Hijazi M., Keith S., Khan M., Ryan-Smith J., Springle P., Thomas J., Truman N., Saad S., Coleman D., Fine C., Matt R., Gay B., Dalziel J., Ali S., Goodchild D., Harling R., Bhatterjee R., Goddard W., Davison C., Duberly S., Hargreaves J., Bolton R., Verlander M., Williams A., Blackman H., Creagh-Brown B., Donlon S., Michalak-Glinska N., Mtuwa S., Pristopan V., Salberg A., Smith E., Stone S., Piercy C., Verula J., Burda D., Montaser R., Harden L., Mayangao I., Marriott C., Bradley P., Harris C., Cooper J., Finch C., Liderth S., Quinn A., Waddington N., Fidler K., Tagliavini E., Donnelly K., Abel L., Brett M., Digby B., Gemmell L., Hornsby J., MacGoey P., O’Neil P., Price R., Rodden N., Rooney K., Sundaram R., Thomson N., Flanagan R., Hughes G., Latham S., McKenna E., Anderson J., Hull R., Rhead K., Branney D., Frankham J., Pitts S., White N., Cristiano D., Dormand N., Farzad Z., Gummadi M., Liyanage K., Patel B.V., Salmi S., Sloane G., Thwaites V., Varghese M., Zborowski A.C., Bean S., Burt K., Spivey M., Eastgate-Jackson C., Filipe H., Martin D., Maharajh A., Garcia S.M., De Neef M., Lynch C., Roche L., Howe G.S., Singh J., Turner K., Ellis H., Stroud N., Cherian S., Cutler S., Heron A.E., Roynon-Reed A., Szakmany T., Williams G., Richards O., Cheema Y., Ahmad N., Barker J., Bauchmuller K., Bird S., Cawthron K., Harrington K., Jackson Y., Kibutu F., Lenagh B., Masuko S., Mills G.H., Raithatha A., Wiles M., Willson J., Newell H., Lye A., Nwafor L., Jarman C., Rowland-Jones S., Foote D., Cole J., Thompson R., Watson J., Hesseldon L., Macharia I., Chetam L., Smith J., Ford A., Anderson S., Birchall K., Housley K., Walker S., Milner L., Hanratty H., Trower H., Phillips P., Oxspring S., Donne B., Bevan E., Martin J., Trodd D., Watson G., Brown C.W., Bunni L., Jennings C., Latif M., Marshall R., Subramanian G., Bandla N., Gellamucho M., Davies M., Thompson C., Trim F., Eapen B., Ahmed C., Baines B., Clamp S., Colley J., Haq R., Hayes A., Hulme J., Hussain S., Joseph S., Kumar R., Maqsood Z., Purewal M., Chandler B., Elliott K., Mallinson J., Turnbull A., Dent K., Horsley E., Akhtar M.N., Pearson S., Potoczna D., Spencer S., Blakemore H., Borislavova B., Faulkner B., Gendall E., Goff E., Hayes K., Thomas M., Worner R., Smith K., Stephens D., Delgado C.C., Dawson D., Ding L., Durrant G., Ezeobu O., Farnell-Ward S., Harrison A., Kanu R., Leaver S., Maccacari E., Manna S., Saluzzio R.P., Queiroz J., Samakomva T., Sicat C., Texeira J., Da Gloria E.F., Lisboa A., Rawlins J., Mathew J., Kinch A., Hurt W.J., Shah N., Clark V., Thanasi M., Yun N., Patel K., Crickmore V., Debreceni G., Wilkins J., Nicol L., Burn I., Hambrook G., Manso K., Penn R., Shanmugasundaram P., Tebbutt J., Thornton D., Rostron A., Roy A., Woods L., Cornell S., Wakinshaw F., Rogerson K., Jarmain J., Anderson P., Archer K., Austin K., Davis C., Durie A., Kelsall O., Thrush J., Vigurs C., Wood H.-L., Tranter H., Harrison A., Cowley N., McAlindon M., Burtenshaw A., Digby S., Low E., Morgan A., Cother N., Rankin T., Clayton S., McCurdy A., Allibone S., Mary-Genetu R., Patel A., Mac A., Murphy A., Mahjoob P., Nazari R., Worsley L., Fagan A., Ali I.A.M., Beaumont K., Blunt M., Coton Z., Curgenven H., Elsaadany M., Fernandes K., Ally S.M., Rangarajan H., Sarathy V., Selvanayagam S., Vedage D., White M., Fernandez-Roman J., Hamilton D.O., Johnson E., Johnston B., Lopez Martinez M., Mulla S., Shaw D., Waite A.A.C., Waugh V., Welters I.D., Williams K., Bemand T., Black E., Rosa A.D., Howle R., Jhanji S., Baikady R.R., Tatham K.C., Thomas B., Halkes M., Mercer P., Thornton L., West J., Baird T., Ruddy J., Reece-Anthony R., Birt M., Cowton A., Kay A., Kent M., Potts K., Wilkinson A., Naylor S., Brown E., Clark M., Purvis S., Cole J., Davies R., Duffin D., Hill H., Player B., Thomas E., Williams A., Beith C.M., Black K., Clements S., Morrison A., Strachan D., Taylor M., Clarkson M., D’Sylva S., Norman K., Coventry T., Fowler S., McGregor A., Brady A., Chan R., McIvor S., Prady H., Whittle H., Mathew B., Clapham M., Harper R., Poultney U., Rice P., Smith T., Mutch R., Baird Y., Butler A., Chadbourn I., Folkes L., Fox H., Gardner A., Gomez R., Hobden G., Hodgson L., King K., Margarson M., Martindale T., Meadows E., Raynard D., Thirlwall Y., Helm D., Margalef J., Greer S., Shuker K., Smuts S., Duffield J., Smith O., Mallon L., Claire W., Birkinshaw I., Carter J., Howard K., Ingham J., Joy R., Pearson H., Roche S., Scott Z., Knights E., Price A., Thomas A., Thorpe C., Abraheem A., Bamford P., Cawley K., Dunmore C., Faulkner M., Girach R., Jeffrey H., Jones R., London E., Nagra I., Nasir F., Sainsbury H., Smedley C., Khade R., Sundar A., Tsinaslanidis G., Behan T., Burnett C., Hatton J., Heeney E., Mitra A., Newton M., Pollard R., Stead R., Birch J., Bough L., Goodsell J., Tutton R., Williams P., Williams S., Winter-Goodwin B., Cowley A., Highgate J., Auld F., Donnachie J., Edmond I., Prentice L., Runciman N., Salutous D., Symon L., Todd A., Turner P., Short A., Sweeney L., Murdoch E., Senaratne D., Burns K., Higham A., Anderson T., Hawcutt D., O’Malley L., Rad L., Rogers N., Saunderson P., Allison K.S., Afolabi D., Whitbread J., Jones D., Dore R., Lankester L., Nikitas N., Wells C., Stowe B., Spencer K., Cathcart S., Duffy K., Puxty A., Puxty K., Turner L., Ireland J., Semple G., Barry P., Hilltout P., Evitts J., Tyler A., Waldron J., Irvine V., Shelley B., Akinkugbe O., Bamford A., Beech E., Belfield H., Bell M., Davies C., Jones G.A.L., McHugh T., Meghari H., O’Neill L., Peters M.J., Ray S., Tomas A.L., Gorman C., Gupta A., Timlick E., Brady R., Bonner S., Hugill K., Jones J., Liggett S., Bashyal A., Davidson N., Hutton P., McKechnie S., Wilson J., Flint N., Rekha P., Hales D., Cruz C., Gopal S., Harris N., Lake V., Metherell S., Radford E., Clement I., Patel B., Gulati A., Hays C., Webster K., Hudson A., Webster A., Stephenson E., McCormack L., Slater V., Nixon R., Hanson H., Fearby M., Kelly S., Bridgett V., Robinson P., Almaden-Boyle C., Austin P., Cabrelli L., Cole S., Casey M., Chapman S., Whyte C., Brayne A., Fisher E., Hunt J., Jackson P., Kaye D., Love N., Parkin J., Tuckey V., van Koutrik L., Carter S., Andrew B., Findlay L., Adams K., Bruce M., Connolly K., Duncan T., Michael H.T., Lindergard G., Hey S., Fox C., Alfonso J., Durrans L.J., Guerin J., Hruska M., Eltayeb A., Lamb T., Hodgkiss T., Cooper L., Rothwell J., Dennis C., McGregor A., Srikaran S., Sukha A., Davies K., O’Brien L., Omar Z., Perkins E., Lewis T., Sutherland I., Brooke H., Buckley S., Suarez J.C., Charlesworth R., Hansson K., Norris J., Poole A., Sandhu R., Sloan B., Smithson E., Thirumaran M., Wagstaff V., Metcalfe A., Camsooksai J., Humphrey C., Wadams B., DeAth Y., Adams C., Agasou A., Arden T., Bowes A., Boyle P., Beekes M., Button H., Capps N., Carnahan M., Carter A., Childs D., Donaldson D., Hard K., Hurford F., Hussain Y., Javaid A., Jones J., Jose S., Leigh M., Martin T., Millward H., Motherwell N., Rikunenko R., Stickley J., Summers J., Ting L., Tivenan H., Tonks L., Wilcox R., Bokhari M., Lucas R., McCormick W., Ritzema J., Sanderson A., Wild H., Baxter N., Henderson S., Kennedy-Hay S., McParland C., Rooney L., Sim M., McCreath G., Brunton M., Caterson J., Coles H., Frise M., Rai S.G., Keating L., Tilney E., Bartley S., Bhuie P., Downes C., Holding K., Riches K., Hilton M., Hayman M., Subramanian D., Daniel P., Zitter L., Benyon S., Marriott S., Park L., Keenan S., Gordon E., Quinn H., Baines K., Andrew G., Kenneth Baillie J., Barclay L., Callaghan M., Clark S., Hope D., Marshall L., McCulloch C., Briton K., Singleton J., Birch S., Simpson K., Craig J., Demetriou C., Eckbad C., Hierons S., Howie L., Mitchard S., Ramos L., Serrano-Ruiz A., White K., Kelly F., Amin V., Anastasescu E., Anumakonda V., Karthik K., Kausar R., Reid K., Smith J., Imeson-Wood J., Bellini A., Bryant J., Mayer A., Pickard A., Roe N., Sowter J., Howlett A., Criste K., Cusack R., Golder K., Golding H., Jones O., Leggett S., Male M., Marani M., Prager K., Williams T., Roberts B., Salmon K., Gondo P., Hadebe B., Kayani A., Masunda B., Ahmed A., Morris A., Jakkula S., Long K., Whiteley S., Wilby E., Ogg B., Moultrie S., Odam M., Bewley J., Garland Z., Grimmer L., Gumbrill B., Johnson R., Sweet K., Webster D., Efford G., Bennett S., Goodwin E., Jackson M., Kent A., Tibke C., Woodyatt W., Zaki A., Daniel A., Finn J., Saha R., Bremmer P., Allan J., Geary T., Houston G., Meikle A., O’Brien P., Bell D., Boyle R., Douglas K., Glass L., Lee E., Lennon L., Rattray A., Charnock R., McFarland D., Cosgrove D., Attwood B., Parsons P., Carmody S., Oblak M., Popescu M., Thankachen M., Baruah R., Morris S., Ferguson S., Shepherd A., Altabaibeh A., Alvaro A., Gilbert K., Ma L., Mostoles L., Parmar C., Simpson K., Jetha C., Booker L., Pratley A., Cosier T., Millen G., Schumacher N., Weston H., Rand J., Alex B., Bach B., Barclay W.S., Bogaert D., Chand M., Cooke G.S., Docherty A.B., Dunning J., da Silva Filipe A., Fletcher T., Green C.A., Harrison E.M., Hiscox J.A., Ijaz S., Khoo S., Lim W.S., Mentzer A.J., Merson L., Noursadeghi M., Palmarini M., Paxton W.A., Pollakis G., Price N., Rambaut A., Robertson D.L., Sancho-Shimizu V., Scott J.T., de Silva T., Sigfrid L., Solomon T., Sriskandan S., Stuart D., Tedder R.S., Thomson E.C., Roger Thompson A.A., Thwaites R.S., Gupta R.K., Palmieri C., Swann O.V., Zambon M., Dumas M.-E., Griffin J.L., Takats Z., Chechi K., Andrikopoulos P., Osagie A., Olanipekun M., Liggi S., Lewis M.R., dos Santos Correia G., Sands C.J., Takis P., Maslen L., Greenhalf W., Shaw V., McDonald S.E., Ahmed K.A., Armstrong J.A., Ashworth M., Asiimwe I.G., Bakshi S., Barlow S.L., Booth L., Brennan B., Bullock K., Catterall B.W.A., Clark J.J., Clarke E.A., Cooper L., Cox H., Davis C., Dincarslan O., Dunn C., Dyer P., Elliott A., Evans A., Finch L., Fisher L.W.S., Foster T., Garcia-Dorival I., Gunning P., Hartley C., Jensen R.L., Jones C.B., Jones T.R., Khandaker S., King K., Kiy R.T., Koukorava C., Lake A., Lant S., Latawiec D., Lavelle-Langham L., Lefteri D., Lett L., Livoti L.A., Mancini M., McDonald S., McEvoy L., McLauchlan J., Metelmann S., Miah N.S., Middleton J., Mitchell J., Moore S.C., Murphy E.G., Penrice-Randal R., Pilgrim J., Prince T., Reynolds W., Matthew Ridley P., Sales D., Shaw V.E., Shears R.K., Small B., Subramaniam K.S., Szemiel A., Taggart A., Tanianis-Hughes J., Thomas J., Trochu E., van Tonder L., Wilcock E., Eunice Zhang J., Flaherty L., Maziere N., Cass E., Carracedo A.D., Carlucci N., Holmes A., Massey H., Wrobel N., Morrice K., MacLean A., Armstrong R., Kenneth Baillie J., Boz C., Brown A., Clark R., Coutts A., Cullum L., Day N., Donnelly L., Duncan E., Fawkes A., Finernan P., Gilchrist T., Golightly A., Hafezi K., Law A., Law D., Law R., Law S., Macgillivray L., Maclean A., Mal H., McCafferty S., Mcmaster E., Meikle J., Murphy L., Oosthuyzen W., Paterson T., Stenhouse A., Swets M., Szoor-McElhinney H., Taneski F., Turtle L.C.W., Wackett T., Ward M., Weaver J., Coyle J., Gallagher B., Lidstone-Scott R., Hamilton D., Schon K., Furlong A., Biggs H., Griffiths F., Andrews E., Brickell K., Smyth M., Murphy L., Carson G., Hardwick H., Donohue C., COVID-19 Host Genetics Initiative\n",
      "Hewitt C., Bessembinder J., Buonocore M., Dunbar T., Garrett N., Kotova L., New S., Newton P., Parfitt R., Buontempo C., Doblas-Reyes F., Guglielmo F., Jacob D., Kjellström E., Krzic A., Martins H., Pietrosanti A., Terrado M.\n",
      "Hickey G.M., Snyder H.T., deVries J.R., Temby O.\n",
      "van Doorn M., Popma A., van Amelsvoort T., McEnery C., Gleeson J.F., Ory F.G., Jaspers M.W.M., Alvarez-Jimenez M., Nieman D.H.\n",
      "Mannucci I., Dang N.D.P., Huber H., Murry J.B., Abramson J., Althoff T., Banka S., Baynam G., Bearden D., Beleza-Meireles A., Benke P.J., Berland S., Bierhals T., Bilan F., Bindoff L.A., Braathen G.J., Busk Ø.L., Chenbhanich J., Denecke J., Escobar L.F., Estes C., Fleischer J., Groepper D., Haaxma C.A., Hempel M., Holler-Managan Y., Houge G., Jackson A., Kellogg L., Keren B., Kiraly-Borri C., Kraus C., Kubisch C., Le Guyader G., Ljungblad U.W., Brenman L.M., Martinez-Agosto J.A., Might M., Miller D.T., Minks K.Q., Moghaddam B., Nava C., Nelson S.F., Parant J.M., Prescott T., Rajabi F., Randrianaivo H., Reiter S.F., Schuurs-Hoeijmakers J., Shieh P.B., Slavotinek A., Smithson S., Stegmann A.P.A., Tomczak K., Tveten K., Wang J., Whitlock J.H., Zweier C., McWalter K., Juusola J., Quintero-Rivera F., Fischer U., Yeo N.C., Kreienkamp H.-J., Lessel D.\n",
      "Hopstaken J.S., van Dalen D., van der Kolk B.M., van Geenen E.J.M., Hermans J.J., Gootjes E.C., Schers H.J., van Dulmen A.M., van Laarhoven C.J.H.M., Stommel M.W.J.\n",
      "Smit L.C., Dikken J., Moolenaar N.M., Schuurmans M.J., de Wit N.J., Bleijenberg N.\n",
      "Vandenbroucke A.R.E., Crone E.A., Erp J.B.F.V., Güroğlu B., Hulshoff Pol H.E., de Kogel C.H., Krabbendam L., Jansen L.M.C., Brouwer A.-M.\n",
      "Bosworth G., Salemink K.\n",
      "Jansen N., Maple H., Forsberg A., Loud F., McColaugh L., Murphy M., Fuertes J., Gallego D., Paredes D., for ESOT Learning Workstream of the TLJ (Transplantation Learning Journey) project\n",
      "Nederhand J.\n",
      "Ferdiana A., Post M.W.M., Bültmann U., van der Klink J.J.L.\n",
      "Boekhout H.D., Traag V.A., Takes F.W.\n",
      "de Graaff N., Valeeva D.\n",
      "Bressan V., Hansen H., Koldby K., Andersen K.D., Snijder A., Tageo V., Allegretti N., Porcu F., Marsillas S., García A., Palese A.\n",
      "ForestPlots.net, Blundo C., Carilla J., Grau R., Malizia A., Malizia L., Osinaga-Acosta O., Bird M., Bradford M., Catchpole D., Ford A., Graham A., Hilbert D., Kemp J., Laurance S., Laurance W., Ishida F.Y., Marshall A., Waite C., Woell H., Bastin J.-F., Bauters M., Beeckman H., Boeckx P., Bogaert J., De Canniere C., de Haulleville T., Doucet J.-L., Hardy O., Hubau W., Kearsley E., Verbeeck H., Vleminckx J., Brewer S.W., Alarcón A., Araujo-Murakami A., Arets E., Arroyo L., Chavez E., Fredericksen T., Villaroel R.G., Sibauty G.G., Killeen T., Licona J.C., Lleigue J., Mendoza C., Murakami S., Gutierrez A.P., Pardo G., Peña-Claros M., Poorter L., Toledo M., Cayo J.V., Viscarra L.J., Vos V., Ahumada J., Almeida E., Almeida J., de Oliveira E.A., da Cruz W.A., de Oliveira A.A., Carvalho F.A., Obermuller F.A., Andrade A., Carvalho F.A., Vieira S.A., Aquino A.C., Aragão L., Araújo A.C., Assis M.A., Gomes J.A.M.A., Baccaro F., de Camargo P.B., Barni P., Barroso J., Bernacci L.C., Bordin K., de Medeiros M.B., Broggio I., Camargo J.L., Cardoso D., Carniello M.A., Rochelle A.L.C., Castilho C., Castro A.A.J.F., Castro W., Ribeiro S.C., Costa F., de Oliveira R.C., Coutinho I., Cunha J., da Costa L., da Costa Ferreira L., da Costa Silva R., da Graça Zacarias Simbine M., de Andrade Kamimura V., de Lima H.C., de Oliveira Melo L., de Queiroz L., de Sousa Lima J.R., do Espírito Santo M., Domingues T., dos Santos Prestes N.C., Carneiro S.E.S., Elias F., Eliseu G., Emilio T., Farrapo C.L., Fernandes L., Ferreira G., Ferreira J., Ferreira L., Ferreira S., Simon M.F., Freitas M.A., García Q.S., Manzatto A.G., Graça P., Guilherme F., Hase E., Higuchi N., Iguatemy M., Barbosa R.I., Jaramillo M., Joly C., Klipel J., do Amaral I.L., Levis C., Lima A.S., Dan M.L., Lopes A., Madeiros H., Magnusson W.E., dos Santos R.M., Marimon B., Junior B.H.M., Grillo R.M.M., Martinelli L., Reis S.M., Medeiros S., Meira-Junior M., Metzker T., Morandi P., do Nascimento N.M., Moura M., Müller S.C., Nagy L., Nascimento H., Nascimento M., Lima A.N., de Araújo R.O., Silva J.O., Pansonato M., Sabino G.P., de Abreu K.M.P., Rodrigues P.J.F.P., Piedade M., Rodrigues D., Rodrigues Pinto J.R., Quesada C., Ramos E., Ramos R., Rodrigues P., de Sousa T.R., Salomão R., Santana F., Scaranello M., Bergamin R.S., Schietti J., Schöngart J., Schwartz G., Silva N., Silveira M., Seixas C.S., Simbine M., Souza A.C., Souza P., Souza R., Sposito T., Junior E.S., do Vale J.D., Vieira I.C.G., Villela D., Vital M., Xaud H., Zanini K., Zartman C.E., Ideris N.K.H., Metali F.B.H., Salim K.A., Saparudin M.S., Serudin R.M., Sukri R.S., Begne S., Chuyong G., Djuikouo M.N., Gonmadje C., Simo-Droissart M., Sonké B., Taedoumg H., Zemagho L., Thomas S., Baya F., Saiz G., Espejo J.S., Chen D., Hamilton A., Li Y., Luo T., Niu S., Xu H., Zhou Z., Álvarez-Dávila E., Escobar J.C.A., Arellano-Peña H., Duarte J.C., Calderón J., Bravo L.M.C., Cuadrado B., Cuadros H., Duque A., Duque L.F., Espinosa S.M., Franke-Ante R., García H., Gómez A., González-M. R., Idárraga-Piedrahíta Á., Jimenez E., Jurado R., Oviedo W.L., López-Camacho R., Cruz O.A.M., Polo I.M., Paky E., Pérez K., Pijachi A., Pizano C., Prieto A., Ramos L., Correa Z.R., Richardson J., Rodríguez E., Rodriguez M. G.M., Rudas A., Stevenson P., Chudomelová M., Dancak M., Hédl R., Lhota S., Svatek M., Mukinzi J., Ewango C., Hart T., Yakusu E.K., Lisingo J., Makana J.-R., Mbayu F., Toirambe B., Mukendi J.T., Kvist L., Nebel G., Báez S., Céron C., Griffith D.M., Andino J.E.G., Neill D., Palacios W., Peñuela-Mora M.C., Rivas-Torres G., Villa G., Demissie S., Gole T., Gonfa T., Ruokolainen K., Baisie M., Bénédet F., Betian W., Bezard V., Bonal D., Chave J., Droissart V., Gourlet-Fleury S., Hladik A., Labrière N., Naisso P., Réjou-Méchain M., Sist P., Blanc L., Burban B., Derroire G., Dourdain A., Stahl C., Bengone N.N., Chezeaux E., Ondo F.E., Medjibe V., Mihindou V., White L., Culmsee H., Rangel C.D., Horna V., Wittmann F., Adu-Bredu S., Affum-Baffoe K., Foli E., Balinga M., Roopsind A., Singh J., Thomas R., Zagt R., Murthy I.K., Kartawinata K., Mirmanto E., Priyadi H., Samsoedin I., Sunderland T., Yassir I., Rovero F., Vinceti B., Hérault B., Aiba S.-I., Kitayama K., Daniels A., Tuagben D., Woods J.T., Fitriadi M., Karolus A., Khoon K.L., Majalap N., Maycock C., Nilus R., Tan S., Sitoe A., Coronado G. I., Ojo L., de Assis R., Poulsen A.D., Sheil D., Pezo K.A., Verde H.B., Moscoso V.C., Oroche J.C.C., Valverde F.C., Medina M.C., Cardozo N.D., de Rutte Corzo J., del Aguila Pasquel J., Llampazo G.F., Freitas L., Cabrera D.G., Villacorta R.G., Cabrera K.G., Soria D.G., Saboya L.G., Rios J.M.G., Pizango G.H., Coronado E.H., Huamantupa-Chuquimaco I., Huasco W.H., Aedo Y.T.H., Peña J.L.M., Mendoza A.M., Rodriguez V.M., Vargas P.N., Ramos S.C.P., Camacho N.P., Cruz A.P., Arevalo F.R., Huaymacari J.R., Rodriguez C.R., Paredes M.A.R., Bayona L.R., del Pilar Rojas Gonzales R., Peña M.E.R., Revilla N.S., Shareva Y.C.S., Trujillo R.T., Gamarra L.V., Martinez R.V., Arenas J.V., Amani C., Ifo S.A., Bocko Y., Boundja P., Ekoungoulou R., Hockemba M., Nzala D., Fofanah A., Taylor D., Bañares-de Dios G., Cayuela L., la Cerda Í.G.-D., Macía M., Stropp J., Playfair M., Wortel V., Gardner T., Muscarella R., Rutishauser E., Chao K.-J., Munishi P., Bánki O., Bongers F., Boot R., Fredriksson G., Reitsma J., ter Steege H., van Andel T., van de Meer P., van der Hout P., van Nieuwstadt M., van Ulft B., Veenendaal E., Vernimmen R., Zuidema P., Zwerts J., Akite P., Bitariho R., Chapman C., Gerald E., Leal M., Mucunguzi P., Abernethy K., Alexiades M., Baker T.R., Banda K., Banin L., Barlow J., Bennett A., Berenguer E., Berry N., Bird N.M., Blackburn G.A., Brearley F., Brienen R., Burslem D., Carvalho L., Cho P., Coelho F., Collins M., Coomes D., Cuni-Sanchez A., Dargie G., Dexter K., Disney M., Draper F., Duan M., Esquivel-Muelbert A., Ewers R., Fadrique B., Fauset S., Feldpausch T.R., França F., Galbraith D., Gilpin M., Gloor E., Grace J., Hamer K., Harris D., Jeffery K., Jucker T., Kalamandeen M., Klitgaard B., Levesley A., Lewis S.L., Lindsell J., Lopez-Gonzalez G., Lovett J., Malhi Y., Marthews T., McIntosh E., Melgaço K., Milliken W., Mitchard E., Moonlight P., Moore S., Morel A., Peacock J., Peh K.S.-H., Pendry C., Pennington R.T., de Oliveira Pereira L., Peres C., Phillips O.L., Pickavance G., Pugh T., Qie L., Riutta T., Roucoux K., Ryan C., Sarkinen T., Valeria C.S., Spracklen D., Stas S., Sullivan M., Swaine M., Talbot J., Taplin J., van der Heijden G., Vedovato L., Willcock S., Williams M., Alves L., Loayza P.A., Arellano G., Asa C., Ashton P., Asner G., Brncic T., Brown F., Burnham R., Clark C., Comiskey J., Damasco G., Davies S., Di Fiore T., Erwin T., Farfan-Rios W., Hall J., Kenfack D., Lovejoy T., Martin R., Montiel O.M., Pipoly J., Pitman N., Poulsen J., Primack R., Silman M., Steininger M., Swamy V., Terborgh J., Thomas D., Umunay P., Uriarte M., Torre E.V., Wang O., Young K., Aymard C. G.A., Hernández L., Fernández R.H., Ramírez-Angulo H., Salcedo P., Sanoja E., Serrano J., Torres-Lezama A., Le T.C., Le T.T., Tran H.D.\n",
      "Kapetanović M., Núñez A., van Oort N., Goverde R.M.P.\n",
      "Marthin J.K., Lucas J.S., Boon M., Casaulta C., Crowley S., Destouches D.M.S., Eber E., Escribano A., Haarman E., Hogg C., Maitre B., Marsh G., Martinu V., Moreno-Galdó A., Mussaffi H., Omran H., Pohunek P., Rindlisbacher B., Robinson P., Snijders D., Walker W.T., Yiallouros P., Johansen H.K., Nielsen K.G.\n",
      "Jasny L., Sayles J., Hamilton M., Roldan Gomez L., Jacobs D., Prell C., Matous P., Schiffer E., Guererro A.M., Barnes M.L.\n",
      "Smit M., Albanese A., Benson M., Edwards M.J., Graessner H., Hutchinson M., Jech R., Krauss J.K., Morgante F., Pérez Dueñas B., Reilly R.B., Tinazzi M., Contarino M.F., Tijssen M.A.J., The Collaborative Working Group\n",
      "Arroyave F.J., Goyeneche O.Y.R., Gore M., Heimeriks G., Jenkins J., Petersen A.M.\n",
      "Stanišić M., Lovrić M., Nedeljković J., Nonić D., Pezdevšek Malovrh Š.\n",
      "Lamper C., Beckers L., Kroese M., Verbunt J., Huijnen I.\n",
      "Atibila F., Hoor G.T., Donkoh E.T., Kok G.\n",
      "Catolino G., Palomba F., Tamburri D.A., Serebrenik A.\n",
      "Munthali N., van Paassen A., Leeuwis C., Lie R., van Lammeren R., Aguilar-Gallegos N., Oppong-Mensah B.\n",
      "van Dam L., Heijmans L., Stams G.J.\n",
      "Kieslinger B., Schaefer T., Fabian C.M., Biasin E., Bassi E., Freire R.R., Mowoh N., Arif N., Melis P.\n",
      "Radu R., Hernández-Ortega S., Borrega O., Palmeri A., Athanasiou D., Brooke N., Chapí I., Le Corvec A., Guglieri M., Perera-Lluna A., Garrido-Aguirre J., Ryll B., Escalera B.N.\n",
      "Ali S.R., Bryce J., Smythe C., Hytiris M., Priego A.L., Appelman-Dijkstra N.M., Ahmed S.F.\n",
      "Rosenberg L.N., Balouka N., Herer Y.T., Dani E., Gasparin P., Dobers K., Rüdiger D., Pättiniemi P., Portheine P., van Uden S.\n",
      "de Beurs D., Bockting C., Kerkhof A., Scheepers F., O’Connor R., Penninx B., van de Leemput I.\n",
      "Brown J., Murray D., Furlong K., Coco E., Dablander F.\n",
      "Milan S.\n",
      "Lykourentzou I., Robert L.P., Jr., Barlatier P.-J.\n",
      "Van Binsbergen L.T., Kebede M.G., Baugh J., Engers T.V., Van Vuurden D.G.\n",
      "Oostra D.L., Harmsen A., Nieuwboer M.S., Olde Rikkert M.G.M., Perry M.\n",
      "Miranda C.M., Escobar M.L.P., Guerrero J.E.E., Sánchez J.I.M.\n",
      "Diviák T., Dijkstra J.K., van der Wijk F., Oosting I., Wolters G.\n",
      "Sheikhalishahi M., Stork I., Zannone N.\n",
      "Antosz P., Jager W., Polhill G., Salt D.\n",
      "Verma H., Mlynář J., Pellaton C., Theler M., Widmer A., Evéquoz F.\n",
      "Argyrou F., Hirschler J., Karan F., Kugel R., Romancenca E., Neicun J.\n",
      "Nistelrooij I.V., Niemeijer A.\n",
      "Tjemkes B.\n",
      "Hengelaar A.H., Wittenberg Y., Kwekkeboom R., Van Hartingsveldt M., Verdonk P.\n",
      "Op ’t Land M., Krouwel M.R., Gort S.\n",
      "Meijer A., Thaens M.\n",
      "Iotova V., Schalin-Jäntti C., Bruegmann P., Broesamle M., Bratina N., Tillmann V., Hiort O., Pereira A.M.\n",
      "Fernandez de Arroyabe J.C., Schumann M., Sena V., Lucas P.\n",
      "Scharpff J., Schraven D., Volker L., Spaan M.T.J., de Weerdt M.M.\n",
      "de Groot E., van den Broek M., Fokkens J.T., Witte J.A.M., Damoiseaux R.A.M.J., Zwart D.L.M.\n",
      "Zhou Y., Diemers A.D., Brouwer J., Muntinghe F.L.H., Duvivier R.J., Pols J., Jaarsma A.D.C., Bos N.A.\n",
      "Lutkenhaus R.O., Jansz J., Bouman M.P.A.\n",
      "Arranz N., Arroyabe M.F., Schumann M.\n",
      "Van Der Ham A., Van Merode F., Ruwaard D., Van Raak A.\n",
      "Najm A., Kostine M., Pauling J.D., Ferreira A.C., Stevens K., Smith E., Eguiluz-Gracia I., Studenic P., Rodríguez-Carrio J., Ramiro S., Alunno A., Richez C., Nikiphorou E., Sepriano A.\n",
      "Arnold C., Hennrich P., Koetsenruijter J., Van Lieshout J., Peters-Klimm F., Wensing M.\n",
      "Wanzenböck I., Lata R., Ince D.\n",
      "Hoes M.F., te Riele A.S.J.M., Gladka M.M., Westenbrink B.D., van Hout G.P.J., van den Hoogenhof M.M.G., Ghigo A., Bollini S., Purcell N.H., Sohaib S.M.A., Kardys I., Kuster D.W.D.\n",
      "Clune C., O'Dwyer B.\n",
      "Sanz C.S., Alencar A.\n",
      "Bennett N.J., Finkbeiner E.M., Ban N.C., Belhabib D., Jupiter S.D., Kittinger J.N., Mangubhai S., Scholtens J., Gill D., Christie P.\n",
      "Fliedner M., Willener R.\n",
      "IJsbrandy C., van Harten W.H., Gerritsen W.R., Hermens R.P.M.G., Ottevanger P.B.\n",
      "Dekker N.R.P., Wanten G.J.A., Bleumink M., de Groot J.W., de Wijkerslooth L.R.H., Netters F.J.S.\n",
      "Aliannejadi M., Rafailidis D., Crestani F.\n",
      "De Vries D.H., Kinsman J., Takacs J., Tsolova S., Ciotti M.\n",
      "Fukkink R.G., van Verseveld M.\n",
      "Cockburn J., Cundill G., Shackleton S., Cele A., Cornelius S.F., Koopman V., le Roux J.-P., McLeod N., Rouget M., Schroder S., Van den Broeck D., Wright D.R., Zwinkels M.\n",
      "Confraria H., Wang L.\n",
      "Zhang S., Bakkes S., Roijers D., Spronck P.\n",
      "Purwitasari D., Fatichah C., Sumpeno S., Steglich C., Purnomo M.H.\n",
      "Faems D., Bos B., Noseleit F., Leten B.\n",
      "Deichmann D., Moser C., Birkholz J.M., Nerghes A., Groenewegen P., Wang S.\n",
      "Dekkers R., de Boer R., Gelsomino L.M., de Goeij C., Steeman M., Zhou Q., Sinclair S., Souter V.\n",
      "Dennissen M., Benschop Y., van den Brink M.\n",
      "Ebenau A., Dijkstra B., Ter Huurne C., Hasselaar J., Vissers K., Groot M.\n",
      "Buskens V., Corten R., Snijders C.\n",
      "Stark T.H., Rambaran J.A., McFarland D.A.\n",
      "Moreno-Miranda C., Pilamala A., Molina I., Cerda-Mejía L., Moreno-Miranda R., Rama D.\n",
      "van der Heiden T., Mirus F., van Hoof H.\n",
      "Morisson A., Bevilacqua C., Doussineau M.\n",
      "Cabral V., Van Winden W.\n",
      "Westen K.H., van Vugt M.D., Bähler M., Mulder C.L., Delespaul P.H., Kroon H.\n",
      "Wang X., van der Bij H., Dolfsma W.\n",
      "Petrea I., Shields-Zeeman L., Keet R., Nica R., Kraan K., Chihai J., Condrat V., Curocichin G., MENSANA Consortium\n",
      "Jaspers J.D.\n",
      "Fransen L., Schalk J., Auld G.\n",
      "Rossano-Rivero S., Wakkee I.\n",
      "El-Hussuna A., Myrelid P., Holubar S.D., Kotze P.G., Mackenzie G., Pellino G., Winter D., Davies J., Negoi I., Grewal P., Gallo G., Sahnan K., Rubio-Perez I., Clerc D., Demartines N., Glasbey J., Regueiro M., Sherif A.E., Neary P., Pata F., Silverberg M., Clermont S., Chadi S.A., Emile S., Buchs N., Millan M., Minaya-Bravo A., Elfeki H., de Simone V., Shalaby M., Gutierrez C., Ozen C., Yalçınkaya A., Rivadeneira D., Sturiale A., Yassin N., Spinelli A., Warusavitarne J., Ioannidis A., Wexner S., Mayol J., Open Source Research Collaborating Group\n",
      "Deljoo A., Koning R., Van Engers T., Gommans L., De Laat C.\n",
      "Zhu L., Zhu D., Wang X., Cunningham S.W., Wang Z.\n",
      "Contreras G., Bos J.W.B., Kleimeier S.\n",
      "Diviák T., Dijkstra J.K., Snijders T.A.B.\n",
      "Lee E.Q., Chukwueke U.N., Hervey-Jumper S.L., De Groot J.F., Leone J.P., Armstrong T.S., Chang S.M., Arons D., Oliver K., Verble K., Musella A., Willmarth N., Alexander B.M., Bates A., Doherty L., Galanis E., Gaffey S., Halkin T., Friday B.E., Fouladi M., Lin N.U., Macdonald D., Mehta M.P., Penas-Prado M., Vogelbaum M.A., Sahebjam S., Sandak D., Van Den Bent M., Weller M., Reardon D.A., Wen P.Y.\n",
      "Khayatzadeh-Mahani A., Ruckert A., Labonté R., Kenis P., Akbari-Javar M.R.\n",
      "Smale R., Spaargaren G., van Vliet B.\n",
      "Colles A., Ardeleanu E.-R., Candeias C., Ranzi A., Demeter Z., Hofer A., Kowalska M., Makris K.C., Arrebola J.P., Schoeters G., Hough R., Pérez-Carrascosa F.M., Iavarone I., Martin-Olmedo P., Kalantzi O.-I., Ancona C., Pasetto R., Fletcher T., Hoek G., de Hoogh K.\n",
      "Schepers J., de Vries J., van Weele A., Langerak F.\n",
      "Reis F., Sá-Moura B., Guardado D., Couceiro P., Catarino L., Mota-Pinto A., Veríssimo M.T., Teixeira A.M., Ferreira P.L., Lima M.P., Palavra F., Rama L., Santos L., van der Heijden R.A., Gonçalves C.E., Cunha A., Malva J.O.\n",
      "Dijkstra-Soudarissanane S., Assal K.E., Gunkel S., Haar F.T., Hindriks R., Kleinrouweler J.W., Niamut O.\n",
      "De Moor A.\n",
      "Babar M., Roos M.H., Nguyen P.H.\n",
      "Confraria H., Vargas F.\n",
      "Nikolaidis C., Ming C., Pedrazzani C., Van Der Horst T., Kaiser-Grolimund A., Ademi Z., Bührer-Landolt R., Bürki N., Caiata-Zufferey M., Champion V., Chappuis P.O., Kohler C., Erlanger T.E., Graffeo R., Hampel H., Heinimann K., Heinzelmann-Schwarz V., Kurzeder C., Monnerat C., Northouse L.L., Pagani O., Probst-Hensch N., Rabaglio M., Schoenau E., Sijbrands E.J.G., Taborelli M., Urech C., Viassolo V., Wieser S., Katapodi M.C.\n",
      "Angelstam P., Elbakidze M., Axelsson R., Khoroshev A., Pedroli B., Tysiachniouk M., Zabubenin E.\n",
      "Buijs A., Hansen R., Van der Jagt S., Ambrose-Oji B., Elands B., Lorance Rall E., Mattijssen T., Pauleit S., Runhaar H., Stahl Olafsson A., Steen Møller M.\n",
      "Hölscher K., Frantzeskaki N., Loorbach D.\n",
      "De Vries M., Kenis P., Kraaij-Dirkzwager M., Ruitenberg E.J., Raab J., Timen A.\n",
      "Loyens K.\n",
      "Doets I.J.E., Pettersson J.\n",
      "Civís M., Diaz-Gibson J., López S., Moolenaar N.\n",
      "Syed S., ní Aodha L., Scougal C., Spruit M.\n",
      "Cissen-Van Heugten W.H.J.M., Van Der Heijden P.T., Slaats I., Lagerveld S.E.\n",
      "Lutkenhaus R.O., Jansz J., Bouman M.P.A.\n",
      "Van Leth F., Brinkmann F., Cirillo D.M., Dheda K., Duarte R., Guglielmetti L., Kuksa L., Lange C., Mitnick C., Skrahina A., Zaman K., Bothamley G.\n",
      "Bakker Y.W., de Koning J., van Tatenhove J.\n",
      "Bergsten A., Jiren T.S., Leventon J., Dorresteijn I., Schultner J., Fischer J.\n",
      "Imbimbo M., Maury J.-M., Garassino M., Girard N., Hackl M., Eycken E.V., Henau K., Dimitrova N., Sekerija M., Dušek L., Mägi M., Malila N., Leinonen M., Velten M., Troussard X., Bouvier V., Guizard A.-V., Bouvier A.-M., Arveux P., Maynadié M., Woronoff A.-S., Robaszkiewicz M., Baldi I., Monnereau A., Tretarre B., Colonna M., Molinié F., Bara S., Schvartz C., Lapôtre-Ledoux B., Grosclaude P., Stabenow R., Luttmann S., Nennecke A., Engel J., Schubert-Fritschle G., Heidrich J., Holleczek B., Jónasson J.G., Clough-Gorr K., Comber H., Mazzoleni G., Giacomin A., Sardo A.S., Barchielli A., Serraino D., De Angelis R., Mallone S., Tavilla A., Pierannunzio D., Rossi S., Santaquilani M., Knijn A., Pannozzo F., Gennaro V., Benfatto L., Ricci P., Autelitano M., Spagnoli G., Fusco M., Usala M., Vitale F., Michiara M., Tumino R., Mangone L., Falcini F., Ferretti S., Angela Filiberti R., Marani E., Iannelli A., Sensi F., Piffer S., Gentilini M., Madeddu A., Ziino A., Maspero S., Candela P., Stracci F., Tagliabue G., Rugge M., Trama A., Gatta G., Botta L., Capocaccia R., Pildava S., Smailyte G., Calleja N., Johannesen T.B., Rachtan J., Góźdź S., Błaszczyk J., Kępska K., de Lacerda G.F., Bento M.J., Miranda A., Diba C.S., Almar E., Larrañaga N., de Munain A.L., Torrella-Ramos A., Díaz García J.M., Marcos-Gragera R., Sanchez M.J., Navarro C., Salmeron D., Moreno-Iribas C., Galceran J., Carulla M., Mousavi M., Bouchardy C., Ess S.M., Bordoni A., Konzelmann I., Rashbass J., Gavin A., Brewster D.H., Huws D.W., Visser O., Bielska-Lasota M., Primic-Zakelj M., Kunkler I., Benhamou E., RARECAREnet Working Group\n",
      "Orlandi E., Alfieri S., Simon C., Trama A., Licitra L., Hackl M., Eycken E.V., Henau K., Dimitrova N., Sekerija M., Dušek L., Mägi M., Malila N., Leinonen M., Velten M., Troussard X., Bouvier V., Guizard A.-V., Bouvier A.-M., Arveux P., Maynadié M., Woronoff A.-S., Robaszkiewic M., Baldi I., Monnereau A., Tretarre B., Colonna M., Molinié F., Bara S., Schvartz C., Lapôtre-Ledoux B., Grosclaude P., Stabenow R., Luttmann S., Nennecke A., Engel J., Schubert-Fritschle G., Heidrich J., Holleczek B., Jónasson J.G., Clough-Gorr K., Comber H., Mazzoleni G., Giacomin A., Sutera Sardo A., Barchielli A., Serraino D., De Angelis R., Mallone S., Tavilla A., Pierannunzio D., Rossi S., Santaquilani M., Knijn A., Pannozzo F., Gennaro V., Benfatto L., Ricci P., Autelitano M., Spagnoli G., Fusco M., Usala M., Vitale F., Michiara M., Tumino R., Mangone L., Falcini F., Ferretti S., Filiberti R.A., Marani E., Iannelli A., Sensi F., Piffer S., Gentilini M., Madeddu A., Ziino A., Maspero S., Candela P., Stracci F., Tagliabue G., Rugge M., Gatta G., Botta L., Capocaccia R., Pildava S., Smailyte G., Calleja N., Johannesen T.B., Rachtan J., Góźdź S., Błaszczyk J., Kępska K., de Lacerda G.F., Bento M.J., Miranda A., Diba C.S., Almar E., Larrañaga N., de Munain A.L., Torrella-Ramos A., Díaz García J.M., Marcos-Gragera R., Sanchez M.J., Navarro C., Salmeron D., Moreno-Iribas C., Galceran J., Carulla M., Mousavi M., Bouchardy C., Ess S.M., Bordoni A., Konzelmann I., Rashbass J., Gavin A., Brewster D.H., Huws D.W., Visser O., Bielska-Lasota M., Primic-Zakelj M., Kunkler I., Benhamou E., RARECAREnet Working Group\n",
      "Hogenstijn M., Meerman M., Zinsmeister J.\n",
      "Nupueng S., Oosterveer P., Mol A.P.J.\n",
      "Yang Q., Lu H., Scoglio C., de Jong M., Gruenbacher D.\n",
      "Jean S., Medema W., Adamowski J., Chew C., Delaney P., Wals A.\n",
      "Krops L.A., Hols D.H.J., Folkertsma N., Dijkstra P.U., Geertzen J.H.B., Dekker R.\n",
      "Deljoo A., Van Engers T., Koning R., Gommans L., De Laat C.\n",
      "Sol J., van der Wal M.M., Beers P.J., Wals A.E.J.\n",
      "Moreira S., Markus A., Laursen K.\n",
      "Ndeinoma A., Wiersum K.F., Arts B.\n",
      "Elmustapha H., Hoppe T., Bressers H.\n",
      "Richters A., Nieuwboer M.S., Olde Rikkert M.G.M., Melis R.J.F., Perry M., Van Der Marck M.A.\n",
      "Mei Z., Zhao X., Chen H., Chen W.\n",
      "Smyth M., Helgason I., Kresin F., Balestrini M., Unteidig A.B., Lawson S., Gaved M., Taylor N., Auger J., Hansen L.K., Schuler D.C., Woods M., Dourish P.\n",
      "Van Mierlo L.D., Chattat R., Evans S., Brooker D., Saibene F.L., Gamberini G., Farina E., Scorolli C., Szcześniak D., Urbańska K.M., Rymaszewska J., Dröes R.M., Meiland F.J.M.\n",
      "De Jong G., Schout G., Abma T.\n",
      "Hillebregt C.F., Scholten E.W.M., Ketelaar M., Post M.W.M., Visser-Meily J.M.A.\n",
      "Fonseca B.P.F., Albuquerque P.C., Noyons E., Zicker F.\n",
      "Meijer A., Thaens M.\n",
      "Vanlangendonck F., Takashima A., Willems R.M., Hagoort P.\n",
      "Rodger M., Langlois N., Middeldorp S., Kahn S., Sandset P.M., Brighton T., Huisman M., Meyer G., Konstantinides S., Ageno W., Morange P., Garcia D., Kreuziger L.B., Young L., Key N., Monreal M., Jiménez D.\n",
      "Tsilochristou O., Maggina P., Zolkipli Z., Sanchez Garcia S., Uysal P., Alvaro Lozano M., Atanaskovic-Markovic M., Baghdasaryan A., Beyer K., DuToit G., Gerth van Wijk R., Høst A., O'Hourihane J., Ingemann L., Kivistö J.E., Lopes dos Santos J.M., Melén E., Muraro A., Nieto A., Pajno G., Rose K., Réthy L.A., Sackesen C., Schmid Grendelmeier P., Valovirta E., Wickman M., Eigenmann P.A., Wahn U., van der Poel L.-A.\n",
      "Brehmer M., Podoynitsyna K., Langerak F.\n",
      "Cooper D.N., Mitropoulou C., Brand A., Dolzan V., Fortina P., Innocenti F., Lee M.T.M., Macek M., Jr., Mitropoulos K., Al-Mulla F., Prainsack B., van Schaik R.H., Squassina A., Taruscio D., Vayena E., Vozikis A., Williams M.S., Ylstra B., Patrinos G.P.\n",
      "Richards G., Duif L.\n",
      "Janssen K.L., Blazevic V., Lauche K.\n",
      "Brown P., Bocken N., Balkenende R.\n",
      "Cooper D.N., Mitropoulou C., Brand A., Dolzan V., Fortina P., Innocenti F., Lee M.T.M., Macek M., Mitropoulos K., Al-Mulla F., Prainsack B., van Schaik R.H., Squassina A., Taruscio D., Vayena E., Vozikis A., Williams M.S., Ylstra B., Patrinos G.P.\n",
      "Riemer K., Hafermalz E.\n",
      "Brooks L., Janssen M., Papazafeiropoulou A.\n",
      "Ghaffar F., Peirce N., Serlie A.\n",
      "Ooms W., Werker C., Caniëls M.\n",
      "Akkermans H.\n",
      "Paci F., Squicciarini A., Zannone N.\n"
     ]
    }
   ],
   "source": [
    "for i in metadata.index:\n",
    "    print(metadata[\"Authors\"][i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "7221b302",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('58081570200', '36163594500'),\n",
       " ('58081570200', '58082093700'),\n",
       " ('58081570200', '57808933900'),\n",
       " ('58081570200', '58082444500'),\n",
       " ('58081570200', '58017352200'),\n",
       " ('58081570200', '58081399200'),\n",
       " ('58081570200', '57222279300'),\n",
       " ('58081570200', '58081570300'),\n",
       " ('58081570200', '57197869642'),\n",
       " ('58081570200', '55809150500'),\n",
       " ('58081570200', '36962443700'),\n",
       " ('36163594500', '58082093700'),\n",
       " ('36163594500', '57808933900'),\n",
       " ('36163594500', '58082444500'),\n",
       " ('36163594500', '58017352200'),\n",
       " ('36163594500', '58081399200'),\n",
       " ('36163594500', '57222279300'),\n",
       " ('36163594500', '58081570300'),\n",
       " ('36163594500', '57197869642'),\n",
       " ('36163594500', '55809150500'),\n",
       " ('36163594500', '36962443700'),\n",
       " ('58082093700', '57808933900'),\n",
       " ('58082093700', '58082444500'),\n",
       " ('58082093700', '58017352200'),\n",
       " ('58082093700', '58081399200'),\n",
       " ('58082093700', '57222279300'),\n",
       " ('58082093700', '58081570300'),\n",
       " ('58082093700', '57197869642'),\n",
       " ('58082093700', '55809150500'),\n",
       " ('58082093700', '36962443700'),\n",
       " ('57808933900', '58082444500'),\n",
       " ('57808933900', '58017352200'),\n",
       " ('57808933900', '58081399200'),\n",
       " ('57808933900', '57222279300'),\n",
       " ('57808933900', '58081570300'),\n",
       " ('57808933900', '57197869642'),\n",
       " ('57808933900', '55809150500'),\n",
       " ('57808933900', '36962443700'),\n",
       " ('58082444500', '58017352200'),\n",
       " ('58082444500', '58081399200'),\n",
       " ('58082444500', '57222279300'),\n",
       " ('58082444500', '58081570300'),\n",
       " ('58082444500', '57197869642'),\n",
       " ('58082444500', '55809150500'),\n",
       " ('58082444500', '36962443700'),\n",
       " ('58017352200', '58081399200'),\n",
       " ('58017352200', '57222279300'),\n",
       " ('58017352200', '58081570300'),\n",
       " ('58017352200', '57197869642'),\n",
       " ('58017352200', '55809150500'),\n",
       " ('58017352200', '36962443700'),\n",
       " ('58081399200', '57222279300'),\n",
       " ('58081399200', '58081570300'),\n",
       " ('58081399200', '57197869642'),\n",
       " ('58081399200', '55809150500'),\n",
       " ('58081399200', '36962443700'),\n",
       " ('57222279300', '58081570300'),\n",
       " ('57222279300', '57197869642'),\n",
       " ('57222279300', '55809150500'),\n",
       " ('57222279300', '36962443700'),\n",
       " ('58081570300', '57197869642'),\n",
       " ('58081570300', '55809150500'),\n",
       " ('58081570300', '36962443700'),\n",
       " ('57197869642', '55809150500'),\n",
       " ('57197869642', '36962443700'),\n",
       " ('55809150500', '36962443700'),\n",
       " ('57212307366', '57189591807'),\n",
       " ('57212307366', '35474798400'),\n",
       " ('57212307366', '36049280600'),\n",
       " ('57189591807', '35474798400'),\n",
       " ('57189591807', '36049280600'),\n",
       " ('35474798400', '36049280600'),\n",
       " ('58033343700', '6602363622'),\n",
       " ('58033343700', '58032496800'),\n",
       " ('6602363622', '58032496800'),\n",
       " ('58032052100', '56147008300'),\n",
       " ('58032052100', '36455764700'),\n",
       " ('58032052100', '58031885300'),\n",
       " ('58032052100', '57995497400'),\n",
       " ('58032052100', '58031919700'),\n",
       " ('58032052100', '8964136400'),\n",
       " ('58032052100', '23980122300'),\n",
       " ('56147008300', '36455764700'),\n",
       " ('56147008300', '58031885300'),\n",
       " ('56147008300', '57995497400'),\n",
       " ('56147008300', '58031919700'),\n",
       " ('56147008300', '8964136400'),\n",
       " ('56147008300', '23980122300'),\n",
       " ('36455764700', '58031885300'),\n",
       " ('36455764700', '57995497400'),\n",
       " ('36455764700', '58031919700'),\n",
       " ('36455764700', '8964136400'),\n",
       " ('36455764700', '23980122300'),\n",
       " ('58031885300', '57995497400'),\n",
       " ('58031885300', '58031919700'),\n",
       " ('58031885300', '8964136400'),\n",
       " ('58031885300', '23980122300'),\n",
       " ('57995497400', '58031919700'),\n",
       " ('57995497400', '8964136400'),\n",
       " ('57995497400', '23980122300'),\n",
       " ('58031919700', '8964136400'),\n",
       " ('58031919700', '23980122300'),\n",
       " ('8964136400', '23980122300'),\n",
       " ('8875857500', '57810943000'),\n",
       " ('8875857500', '57203950202'),\n",
       " ('8875857500', '37664657200'),\n",
       " ('8875857500', '56736720000'),\n",
       " ('8875857500', '57192877578'),\n",
       " ('8875857500', '55531780200'),\n",
       " ('57810943000', '57203950202'),\n",
       " ('57810943000', '37664657200'),\n",
       " ('57810943000', '56736720000'),\n",
       " ('57810943000', '57192877578'),\n",
       " ('57810943000', '55531780200'),\n",
       " ('57203950202', '37664657200'),\n",
       " ('57203950202', '56736720000'),\n",
       " ('57203950202', '57192877578'),\n",
       " ('57203950202', '55531780200'),\n",
       " ('37664657200', '56736720000'),\n",
       " ('37664657200', '57192877578'),\n",
       " ('37664657200', '55531780200'),\n",
       " ('56736720000', '57192877578'),\n",
       " ('56736720000', '55531780200'),\n",
       " ('57192877578', '55531780200'),\n",
       " ('57222026375', '56925146300'),\n",
       " ('58066247700', '58066729400'),\n",
       " ('58066247700', '58067205300'),\n",
       " ('58066247700', '58067044700'),\n",
       " ('58066247700', '58066247800'),\n",
       " ('58066247700', '58066091400'),\n",
       " ('58066247700', '57946893500'),\n",
       " ('58066247700', '58066562800'),\n",
       " ('58066247700', '58066729500'),\n",
       " ('58066247700', '57942431600'),\n",
       " ('58066247700', '58066729600'),\n",
       " ('58066247700', '57192090631'),\n",
       " ('58066247700', '57211591615'),\n",
       " ('58066729400', '58067205300'),\n",
       " ('58066729400', '58067044700'),\n",
       " ('58066729400', '58066247800'),\n",
       " ('58066729400', '58066091400'),\n",
       " ('58066729400', '57946893500'),\n",
       " ('58066729400', '58066562800'),\n",
       " ('58066729400', '58066729500'),\n",
       " ('58066729400', '57942431600'),\n",
       " ('58066729400', '58066729600'),\n",
       " ('58066729400', '57192090631'),\n",
       " ('58066729400', '57211591615'),\n",
       " ('58067205300', '58067044700'),\n",
       " ('58067205300', '58066247800'),\n",
       " ('58067205300', '58066091400'),\n",
       " ('58067205300', '57946893500'),\n",
       " ('58067205300', '58066562800'),\n",
       " ('58067205300', '58066729500'),\n",
       " ('58067205300', '57942431600'),\n",
       " ('58067205300', '58066729600'),\n",
       " ('58067205300', '57192090631'),\n",
       " ('58067205300', '57211591615'),\n",
       " ('58067044700', '58066247800'),\n",
       " ('58067044700', '58066091400'),\n",
       " ('58067044700', '57946893500'),\n",
       " ('58067044700', '58066562800'),\n",
       " ('58067044700', '58066729500'),\n",
       " ('58067044700', '57942431600'),\n",
       " ('58067044700', '58066729600'),\n",
       " ('58067044700', '57192090631'),\n",
       " ('58067044700', '57211591615'),\n",
       " ('58066247800', '58066091400'),\n",
       " ('58066247800', '57946893500'),\n",
       " ('58066247800', '58066562800'),\n",
       " ('58066247800', '58066729500'),\n",
       " ('58066247800', '57942431600'),\n",
       " ('58066247800', '58066729600'),\n",
       " ('58066247800', '57192090631'),\n",
       " ('58066247800', '57211591615'),\n",
       " ('58066091400', '57946893500'),\n",
       " ('58066091400', '58066562800'),\n",
       " ('58066091400', '58066729500'),\n",
       " ('58066091400', '57942431600'),\n",
       " ('58066091400', '58066729600'),\n",
       " ('58066091400', '57192090631'),\n",
       " ('58066091400', '57211591615'),\n",
       " ('57946893500', '58066562800'),\n",
       " ('57946893500', '58066729500'),\n",
       " ('57946893500', '57942431600'),\n",
       " ('57946893500', '58066729600'),\n",
       " ('57946893500', '57192090631'),\n",
       " ('57946893500', '57211591615'),\n",
       " ('58066562800', '58066729500'),\n",
       " ('58066562800', '57942431600'),\n",
       " ('58066562800', '58066729600'),\n",
       " ('58066562800', '57192090631'),\n",
       " ('58066562800', '57211591615'),\n",
       " ('58066729500', '57942431600'),\n",
       " ('58066729500', '58066729600'),\n",
       " ('58066729500', '57192090631'),\n",
       " ('58066729500', '57211591615'),\n",
       " ('57942431600', '58066729600'),\n",
       " ('57942431600', '57192090631'),\n",
       " ('57942431600', '57211591615'),\n",
       " ('58066729600', '57192090631'),\n",
       " ('58066729600', '57211591615'),\n",
       " ('57192090631', '57211591615'),\n",
       " ('57219404780', '7006530227'),\n",
       " ('57219404780', '18934205300'),\n",
       " ('57219404780', '57191728473'),\n",
       " ('7006530227', '18934205300'),\n",
       " ('7006530227', '57191728473'),\n",
       " ('18934205300', '57191728473'),\n",
       " ('57202775286', '57716052300'),\n",
       " ('57202775286', '57716557700'),\n",
       " ('57202775286', '57716557800'),\n",
       " ('57202775286', '56106218300'),\n",
       " ('57716052300', '57716557700'),\n",
       " ('57716052300', '57716557800'),\n",
       " ('57716052300', '56106218300'),\n",
       " ('57716557700', '57716557800'),\n",
       " ('57716557700', '56106218300'),\n",
       " ('57716557800', '56106218300'),\n",
       " ('57191414226', '8046765200'),\n",
       " ('57191414226', '58035082900'),\n",
       " ('57191414226', '55877623600'),\n",
       " ('57191414226', '57565060800'),\n",
       " ('8046765200', '58035082900'),\n",
       " ('8046765200', '55877623600'),\n",
       " ('8046765200', '57565060800'),\n",
       " ('58035082900', '55877623600'),\n",
       " ('58035082900', '57565060800'),\n",
       " ('55877623600', '57565060800'),\n",
       " ('57193257015', '6505715160'),\n",
       " ('57193257015', '12785830300'),\n",
       " ('57193257015', '23488395100'),\n",
       " ('57193257015', '55911695800'),\n",
       " ('57193257015', '6603435766'),\n",
       " ('6505715160', '12785830300'),\n",
       " ('6505715160', '23488395100'),\n",
       " ('6505715160', '55911695800'),\n",
       " ('6505715160', '6603435766'),\n",
       " ('12785830300', '23488395100'),\n",
       " ('12785830300', '55911695800'),\n",
       " ('12785830300', '6603435766'),\n",
       " ('23488395100', '55911695800'),\n",
       " ('23488395100', '6603435766'),\n",
       " ('55911695800', '6603435766'),\n",
       " ('8226126100', '23484810500'),\n",
       " ('8226126100', '35451307600'),\n",
       " ('8226126100', '6701822947'),\n",
       " ('8226126100', '55443388800'),\n",
       " ('23484810500', '35451307600'),\n",
       " ('23484810500', '6701822947'),\n",
       " ('23484810500', '55443388800'),\n",
       " ('35451307600', '6701822947'),\n",
       " ('35451307600', '55443388800'),\n",
       " ('6701822947', '55443388800'),\n",
       " ('57321202100', '57841408200'),\n",
       " ('57321202100', '57841994500'),\n",
       " ('57321202100', '6602324876'),\n",
       " ('57321202100', '57211186721'),\n",
       " ('57321202100', '57192709811'),\n",
       " ('57841408200', '57841994500'),\n",
       " ('57841408200', '6602324876'),\n",
       " ('57841408200', '57211186721'),\n",
       " ('57841408200', '57192709811'),\n",
       " ('57841994500', '6602324876'),\n",
       " ('57841994500', '57211186721'),\n",
       " ('57841994500', '57192709811'),\n",
       " ('6602324876', '57211186721'),\n",
       " ('6602324876', '57192709811'),\n",
       " ('57211186721', '57192709811'),\n",
       " ('57201719234', '6602806074'),\n",
       " ('57201719234', '56467917100'),\n",
       " ('57201719234', '6602778404'),\n",
       " ('57201719234', '24280731000'),\n",
       " ('57201719234', '6504328323'),\n",
       " ('57201719234', '56843437800'),\n",
       " ('6602806074', '56467917100'),\n",
       " ('6602806074', '6602778404'),\n",
       " ('6602806074', '24280731000'),\n",
       " ('6602806074', '6504328323'),\n",
       " ('6602806074', '56843437800'),\n",
       " ('56467917100', '6602778404'),\n",
       " ('56467917100', '24280731000'),\n",
       " ('56467917100', '6504328323'),\n",
       " ('56467917100', '56843437800'),\n",
       " ('6602778404', '24280731000'),\n",
       " ('6602778404', '6504328323'),\n",
       " ('6602778404', '56843437800'),\n",
       " ('24280731000', '6504328323'),\n",
       " ('24280731000', '56843437800'),\n",
       " ('6504328323', '56843437800'),\n",
       " ('57639009800', '24367743300'),\n",
       " ('57477841700', '55348668300'),\n",
       " ('57477841700', '6602098289'),\n",
       " ('57477841700', '7004691405'),\n",
       " ('57477841700', '6701326090'),\n",
       " ('55348668300', '6602098289'),\n",
       " ('55348668300', '7004691405'),\n",
       " ('55348668300', '6701326090'),\n",
       " ('6602098289', '7004691405'),\n",
       " ('6602098289', '6701326090'),\n",
       " ('7004691405', '6701326090'),\n",
       " ('7006161600', '56005614600'),\n",
       " ('7006161600', '36340365300'),\n",
       " ('7006161600', '7007057244'),\n",
       " ('7006161600', '56590016100'),\n",
       " ('7006161600', '6602721437'),\n",
       " ('56005614600', '36340365300'),\n",
       " ('56005614600', '7007057244'),\n",
       " ('56005614600', '56590016100'),\n",
       " ('56005614600', '6602721437'),\n",
       " ('36340365300', '7007057244'),\n",
       " ('36340365300', '56590016100'),\n",
       " ('36340365300', '6602721437'),\n",
       " ('7007057244', '56590016100'),\n",
       " ('7007057244', '6602721437'),\n",
       " ('56590016100', '6602721437'),\n",
       " ('57998647900', '57191523209'),\n",
       " ('57998647900', '13605340400'),\n",
       " ('57191523209', '13605340400'),\n",
       " ('6602998405', '7004579422'),\n",
       " ('6602998405', '15849685800'),\n",
       " ('6602998405', '6602245692'),\n",
       " ('7004579422', '15849685800'),\n",
       " ('7004579422', '6602245692'),\n",
       " ('15849685800', '6602245692'),\n",
       " ('57222897193', '57220055355'),\n",
       " ('57222897193', '12446589100'),\n",
       " ('57222897193', '55111144200'),\n",
       " ('57222897193', '55441585900'),\n",
       " ('57222897193', '57653079000'),\n",
       " ('57222897193', '7005434514'),\n",
       " ('57222897193', '37110075500'),\n",
       " ('57222897193', '57195464324'),\n",
       " ('57222897193', '7005587150'),\n",
       " ('57222897193', '7003934779'),\n",
       " ('57222897193', '30567653800'),\n",
       " ('57222897193', '57205913586'),\n",
       " ('57222897193', '35227110600'),\n",
       " ('57220055355', '12446589100'),\n",
       " ('57220055355', '55111144200'),\n",
       " ('57220055355', '55441585900'),\n",
       " ('57220055355', '57653079000'),\n",
       " ('57220055355', '7005434514'),\n",
       " ('57220055355', '37110075500'),\n",
       " ('57220055355', '57195464324'),\n",
       " ('57220055355', '7005587150'),\n",
       " ('57220055355', '7003934779'),\n",
       " ('57220055355', '30567653800'),\n",
       " ('57220055355', '57205913586'),\n",
       " ('57220055355', '35227110600'),\n",
       " ('12446589100', '55111144200'),\n",
       " ('12446589100', '55441585900'),\n",
       " ('12446589100', '57653079000'),\n",
       " ('12446589100', '7005434514'),\n",
       " ('12446589100', '37110075500'),\n",
       " ('12446589100', '57195464324'),\n",
       " ('12446589100', '7005587150'),\n",
       " ('12446589100', '7003934779'),\n",
       " ('12446589100', '30567653800'),\n",
       " ('12446589100', '57205913586'),\n",
       " ('12446589100', '35227110600'),\n",
       " ('55111144200', '55441585900'),\n",
       " ('55111144200', '57653079000'),\n",
       " ('55111144200', '7005434514'),\n",
       " ('55111144200', '37110075500'),\n",
       " ('55111144200', '57195464324'),\n",
       " ('55111144200', '7005587150'),\n",
       " ('55111144200', '7003934779'),\n",
       " ('55111144200', '30567653800'),\n",
       " ('55111144200', '57205913586'),\n",
       " ('55111144200', '35227110600'),\n",
       " ('55441585900', '57653079000'),\n",
       " ('55441585900', '7005434514'),\n",
       " ('55441585900', '37110075500'),\n",
       " ('55441585900', '57195464324'),\n",
       " ('55441585900', '7005587150'),\n",
       " ('55441585900', '7003934779'),\n",
       " ('55441585900', '30567653800'),\n",
       " ('55441585900', '57205913586'),\n",
       " ('55441585900', '35227110600'),\n",
       " ('57653079000', '7005434514'),\n",
       " ('57653079000', '37110075500'),\n",
       " ('57653079000', '57195464324'),\n",
       " ('57653079000', '7005587150'),\n",
       " ('57653079000', '7003934779'),\n",
       " ('57653079000', '30567653800'),\n",
       " ('57653079000', '57205913586'),\n",
       " ('57653079000', '35227110600'),\n",
       " ('7005434514', '37110075500'),\n",
       " ('7005434514', '57195464324'),\n",
       " ('7005434514', '7005587150'),\n",
       " ('7005434514', '7003934779'),\n",
       " ('7005434514', '30567653800'),\n",
       " ('7005434514', '57205913586'),\n",
       " ('7005434514', '35227110600'),\n",
       " ('37110075500', '57195464324'),\n",
       " ('37110075500', '7005587150'),\n",
       " ('37110075500', '7003934779'),\n",
       " ('37110075500', '30567653800'),\n",
       " ('37110075500', '57205913586'),\n",
       " ('37110075500', '35227110600'),\n",
       " ('57195464324', '7005587150'),\n",
       " ('57195464324', '7003934779'),\n",
       " ('57195464324', '30567653800'),\n",
       " ('57195464324', '57205913586'),\n",
       " ('57195464324', '35227110600'),\n",
       " ('7005587150', '7003934779'),\n",
       " ('7005587150', '30567653800'),\n",
       " ('7005587150', '57205913586'),\n",
       " ('7005587150', '35227110600'),\n",
       " ('7003934779', '30567653800'),\n",
       " ('7003934779', '57205913586'),\n",
       " ('7003934779', '35227110600'),\n",
       " ('30567653800', '57205913586'),\n",
       " ('30567653800', '35227110600'),\n",
       " ('57205913586', '35227110600'),\n",
       " ('56465015800', '57200392935'),\n",
       " ('56465015800', '57896884600'),\n",
       " ('56465015800', '57220306572'),\n",
       " ('56465015800', '57360889600'),\n",
       " ('56465015800', '57188588366'),\n",
       " ('56465015800', '56861958600'),\n",
       " ('57200392935', '57896884600'),\n",
       " ('57200392935', '57220306572'),\n",
       " ('57200392935', '57360889600'),\n",
       " ('57200392935', '57188588366'),\n",
       " ('57200392935', '56861958600'),\n",
       " ('57896884600', '57220306572'),\n",
       " ('57896884600', '57360889600'),\n",
       " ('57896884600', '57188588366'),\n",
       " ('57896884600', '56861958600'),\n",
       " ('57220306572', '57360889600'),\n",
       " ('57220306572', '57188588366'),\n",
       " ('57220306572', '56861958600'),\n",
       " ('57360889600', '57188588366'),\n",
       " ('57360889600', '56861958600'),\n",
       " ('57188588366', '56861958600'),\n",
       " ('57209614945', '24338144900'),\n",
       " ('57209614945', '8296514700'),\n",
       " ('57209614945', '57201696502'),\n",
       " ('57209614945', '57786492600'),\n",
       " ('57209614945', '6602186413'),\n",
       " ('57209614945', '6701858537'),\n",
       " ('24338144900', '8296514700'),\n",
       " ('24338144900', '57201696502'),\n",
       " ('24338144900', '57786492600'),\n",
       " ('24338144900', '6602186413'),\n",
       " ('24338144900', '6701858537'),\n",
       " ('8296514700', '57201696502'),\n",
       " ('8296514700', '57786492600'),\n",
       " ('8296514700', '6602186413'),\n",
       " ('8296514700', '6701858537'),\n",
       " ('57201696502', '57786492600'),\n",
       " ('57201696502', '6602186413'),\n",
       " ('57201696502', '6701858537'),\n",
       " ('57786492600', '6602186413'),\n",
       " ('57786492600', '6701858537'),\n",
       " ('6602186413', '6701858537'),\n",
       " ('54937838500', '7202587597'),\n",
       " ('54937838500', '7102186256'),\n",
       " ('54937838500', '7005620037'),\n",
       " ('54937838500', '57204791050'),\n",
       " ('54937838500', '13807130100'),\n",
       " ('54937838500', '14832180400'),\n",
       " ('54937838500', '7202192005'),\n",
       " ('54937838500', '7003265353'),\n",
       " ('54937838500', '34570772200'),\n",
       " ('54937838500', '7102141909'),\n",
       " ('7202587597', '7102186256'),\n",
       " ('7202587597', '7005620037'),\n",
       " ('7202587597', '57204791050'),\n",
       " ('7202587597', '13807130100'),\n",
       " ('7202587597', '14832180400'),\n",
       " ('7202587597', '7202192005'),\n",
       " ('7202587597', '7003265353'),\n",
       " ('7202587597', '34570772200'),\n",
       " ('7202587597', '7102141909'),\n",
       " ('7102186256', '7005620037'),\n",
       " ('7102186256', '57204791050'),\n",
       " ('7102186256', '13807130100'),\n",
       " ('7102186256', '14832180400'),\n",
       " ('7102186256', '7202192005'),\n",
       " ('7102186256', '7003265353'),\n",
       " ('7102186256', '34570772200'),\n",
       " ('7102186256', '7102141909'),\n",
       " ('7005620037', '57204791050'),\n",
       " ('7005620037', '13807130100'),\n",
       " ('7005620037', '14832180400'),\n",
       " ('7005620037', '7202192005'),\n",
       " ('7005620037', '7003265353'),\n",
       " ('7005620037', '34570772200'),\n",
       " ('7005620037', '7102141909'),\n",
       " ('57204791050', '13807130100'),\n",
       " ('57204791050', '14832180400'),\n",
       " ('57204791050', '7202192005'),\n",
       " ('57204791050', '7003265353'),\n",
       " ('57204791050', '34570772200'),\n",
       " ('57204791050', '7102141909'),\n",
       " ('13807130100', '14832180400'),\n",
       " ('13807130100', '7202192005'),\n",
       " ('13807130100', '7003265353'),\n",
       " ('13807130100', '34570772200'),\n",
       " ('13807130100', '7102141909'),\n",
       " ('14832180400', '7202192005'),\n",
       " ('14832180400', '7003265353'),\n",
       " ('14832180400', '34570772200'),\n",
       " ('14832180400', '7102141909'),\n",
       " ('7202192005', '7003265353'),\n",
       " ('7202192005', '34570772200'),\n",
       " ('7202192005', '7102141909'),\n",
       " ('7003265353', '34570772200'),\n",
       " ('7003265353', '7102141909'),\n",
       " ('34570772200', '7102141909'),\n",
       " ('57221605651', '23980122300'),\n",
       " ('57221605651', '56147008300'),\n",
       " ('57221605651', '36455764700'),\n",
       " ('57221605651', '8964136400'),\n",
       " ('23980122300', '56147008300'),\n",
       " ('23980122300', '36455764700'),\n",
       " ('23980122300', '8964136400'),\n",
       " ('56147008300', '36455764700'),\n",
       " ('56147008300', '8964136400'),\n",
       " ('36455764700', '8964136400'),\n",
       " ('57189461786', '20333917800'),\n",
       " ('57189461786', '55225187000'),\n",
       " ('57189461786', '6603767582'),\n",
       " ('57189461786', '35243030300'),\n",
       " ('57189461786', '6603752469'),\n",
       " ('20333917800', '55225187000'),\n",
       " ('20333917800', '6603767582'),\n",
       " ('20333917800', '35243030300'),\n",
       " ('20333917800', '6603752469'),\n",
       " ('55225187000', '6603767582'),\n",
       " ('55225187000', '35243030300'),\n",
       " ('55225187000', '6603752469'),\n",
       " ('6603767582', '35243030300'),\n",
       " ('6603767582', '6603752469'),\n",
       " ('35243030300', '6603752469'),\n",
       " ('36611724000', '8835604200'),\n",
       " ('36611724000', '24576734900'),\n",
       " ('36611724000', '56580241600'),\n",
       " ('36611724000', '57209551521'),\n",
       " ('36611724000', '12761269700'),\n",
       " ('36611724000', '57190272868'),\n",
       " ('36611724000', '50760992300'),\n",
       " ('36611724000', '34569127200'),\n",
       " ('36611724000', '6602195572'),\n",
       " ('36611724000', '56463035500'),\n",
       " ('36611724000', '54789048000'),\n",
       " ('36611724000', '6507581306'),\n",
       " ('36611724000', '6603710238'),\n",
       " ('36611724000', '20735018600'),\n",
       " ('36611724000', '26434318500'),\n",
       " ('36611724000', '57219204911'),\n",
       " ('36611724000', '57209323859'),\n",
       " ('36611724000', '56480829300'),\n",
       " ('36611724000', '54986245000'),\n",
       " ('36611724000', '57213587752'),\n",
       " ('36611724000', '57204630901'),\n",
       " ('36611724000', '57208140398'),\n",
       " ('36611724000', '57202469544'),\n",
       " ('36611724000', '35380737400'),\n",
       " ('36611724000', '57220118235'),\n",
       " ('36611724000', '57205170885'),\n",
       " ('36611724000', '23134902300'),\n",
       " ('8835604200', '24576734900'),\n",
       " ('8835604200', '56580241600'),\n",
       " ('8835604200', '57209551521'),\n",
       " ('8835604200', '12761269700'),\n",
       " ('8835604200', '57190272868'),\n",
       " ('8835604200', '50760992300'),\n",
       " ('8835604200', '34569127200'),\n",
       " ('8835604200', '6602195572'),\n",
       " ('8835604200', '56463035500'),\n",
       " ('8835604200', '54789048000'),\n",
       " ('8835604200', '6507581306'),\n",
       " ('8835604200', '6603710238'),\n",
       " ('8835604200', '20735018600'),\n",
       " ('8835604200', '26434318500'),\n",
       " ('8835604200', '57219204911'),\n",
       " ('8835604200', '57209323859'),\n",
       " ('8835604200', '56480829300'),\n",
       " ('8835604200', '54986245000'),\n",
       " ('8835604200', '57213587752'),\n",
       " ('8835604200', '57204630901'),\n",
       " ('8835604200', '57208140398'),\n",
       " ('8835604200', '57202469544'),\n",
       " ('8835604200', '35380737400'),\n",
       " ('8835604200', '57220118235'),\n",
       " ('8835604200', '57205170885'),\n",
       " ('8835604200', '23134902300'),\n",
       " ('24576734900', '56580241600'),\n",
       " ('24576734900', '57209551521'),\n",
       " ('24576734900', '12761269700'),\n",
       " ('24576734900', '57190272868'),\n",
       " ('24576734900', '50760992300'),\n",
       " ('24576734900', '34569127200'),\n",
       " ('24576734900', '6602195572'),\n",
       " ('24576734900', '56463035500'),\n",
       " ('24576734900', '54789048000'),\n",
       " ('24576734900', '6507581306'),\n",
       " ('24576734900', '6603710238'),\n",
       " ('24576734900', '20735018600'),\n",
       " ('24576734900', '26434318500'),\n",
       " ('24576734900', '57219204911'),\n",
       " ('24576734900', '57209323859'),\n",
       " ('24576734900', '56480829300'),\n",
       " ('24576734900', '54986245000'),\n",
       " ('24576734900', '57213587752'),\n",
       " ('24576734900', '57204630901'),\n",
       " ('24576734900', '57208140398'),\n",
       " ('24576734900', '57202469544'),\n",
       " ('24576734900', '35380737400'),\n",
       " ('24576734900', '57220118235'),\n",
       " ('24576734900', '57205170885'),\n",
       " ('24576734900', '23134902300'),\n",
       " ('56580241600', '57209551521'),\n",
       " ('56580241600', '12761269700'),\n",
       " ('56580241600', '57190272868'),\n",
       " ('56580241600', '50760992300'),\n",
       " ('56580241600', '34569127200'),\n",
       " ('56580241600', '6602195572'),\n",
       " ('56580241600', '56463035500'),\n",
       " ('56580241600', '54789048000'),\n",
       " ('56580241600', '6507581306'),\n",
       " ('56580241600', '6603710238'),\n",
       " ('56580241600', '20735018600'),\n",
       " ('56580241600', '26434318500'),\n",
       " ('56580241600', '57219204911'),\n",
       " ('56580241600', '57209323859'),\n",
       " ('56580241600', '56480829300'),\n",
       " ('56580241600', '54986245000'),\n",
       " ('56580241600', '57213587752'),\n",
       " ('56580241600', '57204630901'),\n",
       " ('56580241600', '57208140398'),\n",
       " ('56580241600', '57202469544'),\n",
       " ('56580241600', '35380737400'),\n",
       " ('56580241600', '57220118235'),\n",
       " ('56580241600', '57205170885'),\n",
       " ('56580241600', '23134902300'),\n",
       " ('57209551521', '12761269700'),\n",
       " ('57209551521', '57190272868'),\n",
       " ('57209551521', '50760992300'),\n",
       " ('57209551521', '34569127200'),\n",
       " ('57209551521', '6602195572'),\n",
       " ('57209551521', '56463035500'),\n",
       " ('57209551521', '54789048000'),\n",
       " ('57209551521', '6507581306'),\n",
       " ('57209551521', '6603710238'),\n",
       " ('57209551521', '20735018600'),\n",
       " ('57209551521', '26434318500'),\n",
       " ('57209551521', '57219204911'),\n",
       " ('57209551521', '57209323859'),\n",
       " ('57209551521', '56480829300'),\n",
       " ('57209551521', '54986245000'),\n",
       " ('57209551521', '57213587752'),\n",
       " ('57209551521', '57204630901'),\n",
       " ('57209551521', '57208140398'),\n",
       " ('57209551521', '57202469544'),\n",
       " ('57209551521', '35380737400'),\n",
       " ('57209551521', '57220118235'),\n",
       " ('57209551521', '57205170885'),\n",
       " ('57209551521', '23134902300'),\n",
       " ('12761269700', '57190272868'),\n",
       " ('12761269700', '50760992300'),\n",
       " ('12761269700', '34569127200'),\n",
       " ('12761269700', '6602195572'),\n",
       " ('12761269700', '56463035500'),\n",
       " ('12761269700', '54789048000'),\n",
       " ('12761269700', '6507581306'),\n",
       " ('12761269700', '6603710238'),\n",
       " ('12761269700', '20735018600'),\n",
       " ('12761269700', '26434318500'),\n",
       " ('12761269700', '57219204911'),\n",
       " ('12761269700', '57209323859'),\n",
       " ('12761269700', '56480829300'),\n",
       " ('12761269700', '54986245000'),\n",
       " ('12761269700', '57213587752'),\n",
       " ('12761269700', '57204630901'),\n",
       " ('12761269700', '57208140398'),\n",
       " ('12761269700', '57202469544'),\n",
       " ('12761269700', '35380737400'),\n",
       " ('12761269700', '57220118235'),\n",
       " ('12761269700', '57205170885'),\n",
       " ('12761269700', '23134902300'),\n",
       " ('57190272868', '50760992300'),\n",
       " ('57190272868', '34569127200'),\n",
       " ('57190272868', '6602195572'),\n",
       " ('57190272868', '56463035500'),\n",
       " ('57190272868', '54789048000'),\n",
       " ('57190272868', '6507581306'),\n",
       " ('57190272868', '6603710238'),\n",
       " ('57190272868', '20735018600'),\n",
       " ('57190272868', '26434318500'),\n",
       " ('57190272868', '57219204911'),\n",
       " ('57190272868', '57209323859'),\n",
       " ('57190272868', '56480829300'),\n",
       " ('57190272868', '54986245000'),\n",
       " ('57190272868', '57213587752'),\n",
       " ('57190272868', '57204630901'),\n",
       " ('57190272868', '57208140398'),\n",
       " ('57190272868', '57202469544'),\n",
       " ('57190272868', '35380737400'),\n",
       " ('57190272868', '57220118235'),\n",
       " ('57190272868', '57205170885'),\n",
       " ('57190272868', '23134902300'),\n",
       " ('50760992300', '34569127200'),\n",
       " ('50760992300', '6602195572'),\n",
       " ('50760992300', '56463035500'),\n",
       " ('50760992300', '54789048000'),\n",
       " ('50760992300', '6507581306'),\n",
       " ('50760992300', '6603710238'),\n",
       " ('50760992300', '20735018600'),\n",
       " ('50760992300', '26434318500'),\n",
       " ('50760992300', '57219204911'),\n",
       " ('50760992300', '57209323859'),\n",
       " ('50760992300', '56480829300'),\n",
       " ('50760992300', '54986245000'),\n",
       " ('50760992300', '57213587752'),\n",
       " ('50760992300', '57204630901'),\n",
       " ('50760992300', '57208140398'),\n",
       " ('50760992300', '57202469544'),\n",
       " ('50760992300', '35380737400'),\n",
       " ('50760992300', '57220118235'),\n",
       " ('50760992300', '57205170885'),\n",
       " ('50760992300', '23134902300'),\n",
       " ('34569127200', '6602195572'),\n",
       " ('34569127200', '56463035500'),\n",
       " ('34569127200', '54789048000'),\n",
       " ('34569127200', '6507581306'),\n",
       " ('34569127200', '6603710238'),\n",
       " ('34569127200', '20735018600'),\n",
       " ('34569127200', '26434318500'),\n",
       " ('34569127200', '57219204911'),\n",
       " ('34569127200', '57209323859'),\n",
       " ('34569127200', '56480829300'),\n",
       " ('34569127200', '54986245000'),\n",
       " ('34569127200', '57213587752'),\n",
       " ('34569127200', '57204630901'),\n",
       " ('34569127200', '57208140398'),\n",
       " ('34569127200', '57202469544'),\n",
       " ('34569127200', '35380737400'),\n",
       " ('34569127200', '57220118235'),\n",
       " ('34569127200', '57205170885'),\n",
       " ('34569127200', '23134902300'),\n",
       " ('6602195572', '56463035500'),\n",
       " ('6602195572', '54789048000'),\n",
       " ('6602195572', '6507581306'),\n",
       " ('6602195572', '6603710238'),\n",
       " ('6602195572', '20735018600'),\n",
       " ('6602195572', '26434318500'),\n",
       " ('6602195572', '57219204911'),\n",
       " ('6602195572', '57209323859'),\n",
       " ('6602195572', '56480829300'),\n",
       " ('6602195572', '54986245000'),\n",
       " ('6602195572', '57213587752'),\n",
       " ('6602195572', '57204630901'),\n",
       " ('6602195572', '57208140398'),\n",
       " ('6602195572', '57202469544'),\n",
       " ('6602195572', '35380737400'),\n",
       " ('6602195572', '57220118235'),\n",
       " ('6602195572', '57205170885'),\n",
       " ('6602195572', '23134902300'),\n",
       " ('56463035500', '54789048000'),\n",
       " ('56463035500', '6507581306'),\n",
       " ('56463035500', '6603710238'),\n",
       " ('56463035500', '20735018600'),\n",
       " ('56463035500', '26434318500'),\n",
       " ('56463035500', '57219204911'),\n",
       " ('56463035500', '57209323859'),\n",
       " ('56463035500', '56480829300'),\n",
       " ('56463035500', '54986245000'),\n",
       " ('56463035500', '57213587752'),\n",
       " ('56463035500', '57204630901'),\n",
       " ('56463035500', '57208140398'),\n",
       " ('56463035500', '57202469544'),\n",
       " ('56463035500', '35380737400'),\n",
       " ('56463035500', '57220118235'),\n",
       " ('56463035500', '57205170885'),\n",
       " ('56463035500', '23134902300'),\n",
       " ('54789048000', '6507581306'),\n",
       " ('54789048000', '6603710238'),\n",
       " ('54789048000', '20735018600'),\n",
       " ('54789048000', '26434318500'),\n",
       " ('54789048000', '57219204911'),\n",
       " ('54789048000', '57209323859'),\n",
       " ('54789048000', '56480829300'),\n",
       " ('54789048000', '54986245000'),\n",
       " ('54789048000', '57213587752'),\n",
       " ('54789048000', '57204630901'),\n",
       " ('54789048000', '57208140398'),\n",
       " ('54789048000', '57202469544'),\n",
       " ('54789048000', '35380737400'),\n",
       " ('54789048000', '57220118235'),\n",
       " ('54789048000', '57205170885'),\n",
       " ('54789048000', '23134902300'),\n",
       " ('6507581306', '6603710238'),\n",
       " ('6507581306', '20735018600'),\n",
       " ('6507581306', '26434318500'),\n",
       " ('6507581306', '57219204911'),\n",
       " ('6507581306', '57209323859'),\n",
       " ('6507581306', '56480829300'),\n",
       " ('6507581306', '54986245000'),\n",
       " ('6507581306', '57213587752'),\n",
       " ('6507581306', '57204630901'),\n",
       " ('6507581306', '57208140398'),\n",
       " ('6507581306', '57202469544'),\n",
       " ('6507581306', '35380737400'),\n",
       " ('6507581306', '57220118235'),\n",
       " ('6507581306', '57205170885'),\n",
       " ('6507581306', '23134902300'),\n",
       " ('6603710238', '20735018600'),\n",
       " ('6603710238', '26434318500'),\n",
       " ('6603710238', '57219204911'),\n",
       " ('6603710238', '57209323859'),\n",
       " ('6603710238', '56480829300'),\n",
       " ('6603710238', '54986245000'),\n",
       " ('6603710238', '57213587752'),\n",
       " ('6603710238', '57204630901'),\n",
       " ('6603710238', '57208140398'),\n",
       " ('6603710238', '57202469544'),\n",
       " ('6603710238', '35380737400'),\n",
       " ('6603710238', '57220118235'),\n",
       " ('6603710238', '57205170885'),\n",
       " ('6603710238', '23134902300'),\n",
       " ('20735018600', '26434318500'),\n",
       " ('20735018600', '57219204911'),\n",
       " ('20735018600', '57209323859'),\n",
       " ('20735018600', '56480829300'),\n",
       " ('20735018600', '54986245000'),\n",
       " ('20735018600', '57213587752'),\n",
       " ('20735018600', '57204630901'),\n",
       " ('20735018600', '57208140398'),\n",
       " ('20735018600', '57202469544'),\n",
       " ('20735018600', '35380737400'),\n",
       " ('20735018600', '57220118235'),\n",
       " ('20735018600', '57205170885'),\n",
       " ('20735018600', '23134902300'),\n",
       " ('26434318500', '57219204911'),\n",
       " ('26434318500', '57209323859'),\n",
       " ('26434318500', '56480829300'),\n",
       " ('26434318500', '54986245000'),\n",
       " ('26434318500', '57213587752'),\n",
       " ('26434318500', '57204630901'),\n",
       " ('26434318500', '57208140398'),\n",
       " ('26434318500', '57202469544'),\n",
       " ('26434318500', '35380737400'),\n",
       " ('26434318500', '57220118235'),\n",
       " ('26434318500', '57205170885'),\n",
       " ('26434318500', '23134902300'),\n",
       " ('57219204911', '57209323859'),\n",
       " ('57219204911', '56480829300'),\n",
       " ('57219204911', '54986245000'),\n",
       " ('57219204911', '57213587752'),\n",
       " ('57219204911', '57204630901'),\n",
       " ('57219204911', '57208140398'),\n",
       " ('57219204911', '57202469544'),\n",
       " ('57219204911', '35380737400'),\n",
       " ('57219204911', '57220118235'),\n",
       " ('57219204911', '57205170885'),\n",
       " ('57219204911', '23134902300'),\n",
       " ('57209323859', '56480829300'),\n",
       " ('57209323859', '54986245000'),\n",
       " ('57209323859', '57213587752'),\n",
       " ('57209323859', '57204630901'),\n",
       " ('57209323859', '57208140398'),\n",
       " ('57209323859', '57202469544'),\n",
       " ('57209323859', '35380737400'),\n",
       " ('57209323859', '57220118235'),\n",
       " ('57209323859', '57205170885'),\n",
       " ('57209323859', '23134902300'),\n",
       " ('56480829300', '54986245000'),\n",
       " ('56480829300', '57213587752'),\n",
       " ('56480829300', '57204630901'),\n",
       " ('56480829300', '57208140398'),\n",
       " ('56480829300', '57202469544'),\n",
       " ('56480829300', '35380737400'),\n",
       " ('56480829300', '57220118235'),\n",
       " ('56480829300', '57205170885'),\n",
       " ('56480829300', '23134902300'),\n",
       " ('54986245000', '57213587752'),\n",
       " ('54986245000', '57204630901'),\n",
       " ('54986245000', '57208140398'),\n",
       " ('54986245000', '57202469544'),\n",
       " ('54986245000', '35380737400'),\n",
       " ('54986245000', '57220118235'),\n",
       " ('54986245000', '57205170885'),\n",
       " ('54986245000', '23134902300'),\n",
       " ('57213587752', '57204630901'),\n",
       " ('57213587752', '57208140398'),\n",
       " ('57213587752', '57202469544'),\n",
       " ('57213587752', '35380737400'),\n",
       " ('57213587752', '57220118235'),\n",
       " ('57213587752', '57205170885'),\n",
       " ('57213587752', '23134902300'),\n",
       " ('57204630901', '57208140398'),\n",
       " ('57204630901', '57202469544'),\n",
       " ('57204630901', '35380737400'),\n",
       " ('57204630901', '57220118235'),\n",
       " ('57204630901', '57205170885'),\n",
       " ('57204630901', '23134902300'),\n",
       " ('57208140398', '57202469544'),\n",
       " ('57208140398', '35380737400'),\n",
       " ('57208140398', '57220118235'),\n",
       " ('57208140398', '57205170885'),\n",
       " ('57208140398', '23134902300'),\n",
       " ('57202469544', '35380737400'),\n",
       " ('57202469544', '57220118235'),\n",
       " ('57202469544', '57205170885'),\n",
       " ('57202469544', '23134902300'),\n",
       " ('35380737400', '57220118235'),\n",
       " ('35380737400', '57205170885'),\n",
       " ('35380737400', '23134902300'),\n",
       " ('57220118235', '57205170885'),\n",
       " ('57220118235', '23134902300'),\n",
       " ('57205170885', '23134902300'),\n",
       " ('57195732410', '57189641697'),\n",
       " ('57195732410', '57489541100'),\n",
       " ('57195732410', '6603557582'),\n",
       " ('57195732410', '7103067740'),\n",
       " ('57195732410', '7003600497'),\n",
       " ('57195732410', '18040708300'),\n",
       " ('57189641697', '57489541100'),\n",
       " ('57189641697', '6603557582'),\n",
       " ('57189641697', '7103067740'),\n",
       " ('57189641697', '7003600497'),\n",
       " ('57189641697', '18040708300'),\n",
       " ('57489541100', '6603557582'),\n",
       " ('57489541100', '7103067740'),\n",
       " ('57489541100', '7003600497'),\n",
       " ('57489541100', '18040708300'),\n",
       " ('6603557582', '7103067740'),\n",
       " ('6603557582', '7003600497'),\n",
       " ('6603557582', '18040708300'),\n",
       " ('7103067740', '7003600497'),\n",
       " ('7103067740', '18040708300'),\n",
       " ('7003600497', '18040708300'),\n",
       " ('37053356400', '56411939100'),\n",
       " ('37053356400', '7102339373'),\n",
       " ('37053356400', '57190279338'),\n",
       " ('56411939100', '7102339373'),\n",
       " ('56411939100', '57190279338'),\n",
       " ('7102339373', '57190279338'),\n",
       " ('6507920875', '57115477800'),\n",
       " ('6507920875', '23135719900'),\n",
       " ('6507920875', '49863862000'),\n",
       " ('6507920875', '26534464000'),\n",
       " ('6507920875', '8530143500'),\n",
       " ('6507920875', '57830408100'),\n",
       " ('6507920875', '57830408200'),\n",
       " ('6507920875', '7202739109'),\n",
       " ('57115477800', '23135719900'),\n",
       " ('57115477800', '49863862000'),\n",
       " ('57115477800', '26534464000'),\n",
       " ('57115477800', '8530143500'),\n",
       " ('57115477800', '57830408100'),\n",
       " ('57115477800', '57830408200'),\n",
       " ('57115477800', '7202739109'),\n",
       " ('23135719900', '49863862000'),\n",
       " ('23135719900', '26534464000'),\n",
       " ('23135719900', '8530143500'),\n",
       " ('23135719900', '57830408100'),\n",
       " ('23135719900', '57830408200'),\n",
       " ('23135719900', '7202739109'),\n",
       " ('49863862000', '26534464000'),\n",
       " ('49863862000', '8530143500'),\n",
       " ('49863862000', '57830408100'),\n",
       " ('49863862000', '57830408200'),\n",
       " ('49863862000', '7202739109'),\n",
       " ('26534464000', '8530143500'),\n",
       " ('26534464000', '57830408100'),\n",
       " ('26534464000', '57830408200'),\n",
       " ('26534464000', '7202739109'),\n",
       " ('8530143500', '57830408100'),\n",
       " ('8530143500', '57830408200'),\n",
       " ('8530143500', '7202739109'),\n",
       " ('57830408100', '57830408200'),\n",
       " ('57830408100', '7202739109'),\n",
       " ('57830408200', '7202739109'),\n",
       " ('37052139500', '57210156037'),\n",
       " ('37052139500', '57222010397'),\n",
       " ('37052139500', '57777616000'),\n",
       " ('37052139500', '57776038100'),\n",
       " ('37052139500', '57195939196'),\n",
       " ('37052139500', '57777391000'),\n",
       " ('37052139500', '56430701900'),\n",
       " ('37052139500', '57777168300'),\n",
       " ('37052139500', '57189854493'),\n",
       " ('37052139500', '55111210500'),\n",
       " ('37052139500', '7005534467'),\n",
       " ('37052139500', '57777616100'),\n",
       " ('37052139500', '57199296355'),\n",
       " ('37052139500', '7005193770'),\n",
       " ('37052139500', '57205745674'),\n",
       " ('57210156037', '57222010397'),\n",
       " ('57210156037', '57777616000'),\n",
       " ('57210156037', '57776038100'),\n",
       " ('57210156037', '57195939196'),\n",
       " ('57210156037', '57777391000'),\n",
       " ('57210156037', '56430701900'),\n",
       " ...]"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name = []\n",
    "aid = []\n",
    "id_name = {}\n",
    "link = []\n",
    "\n",
    "def getCombinations(seq):\n",
    "    combinations = list()\n",
    "    seq = seq.sort()\n",
    "    for i in range(0,len(seq)):\n",
    "        for j in range(i+1,len(seq)):\n",
    "            combinations.append([seq[i],seq[j]])\n",
    "    return combinations\n",
    "\n",
    "for i in metadata.index:\n",
    "    name_temp = metadata[\"Authors\"][i].split(\", \")\n",
    "    aid_temp = metadata[\"Author(s) ID\"][i].split(\";\")\n",
    "    aid_temp.pop()\n",
    "    if len(name_temp) >= len(aid_temp):\n",
    "        for i in range(len(aid_temp)):\n",
    "            id_name[aid_temp[i]] = name_temp[i]\n",
    "\n",
    "    link = link+list(itertools.combinations(aid_temp, 2))\n",
    "#     else:\n",
    "#         for i in range(len(name_temp)):\n",
    "#             id_name[aid_temp[i]] = name_temp[i]\n",
    "        # print(len(name_temp) - len(aid_temp))\n",
    "        # print(\"----------\")\n",
    "\n",
    "link\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "f14242a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "66\n",
      "66\n"
     ]
    }
   ],
   "source": [
    "print(len(getCombinations(['58081570200',\n",
    " '36163594500',\n",
    " '58082093700',\n",
    " '57808933900',\n",
    " '58082444500',\n",
    " '58017352200',\n",
    " '58081399200',\n",
    " '57222279300',\n",
    " '58081570300',\n",
    " '57197869642',\n",
    " '55809150500',\n",
    " '36962443700'])))\n",
    "\n",
    "\n",
    "\n",
    "print(len(list(itertools.combinations(['58081570200',\n",
    " '36163594500',\n",
    " '58082093700',\n",
    " '57808933900',\n",
    " '58082444500',\n",
    " '58017352200',\n",
    " '58081399200',\n",
    " '57222279300',\n",
    " '58081570300',\n",
    " '57197869642',\n",
    " '55809150500',\n",
    " '36962443700'], 2))))\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "282ce8d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({('58081570200', '36163594500'): 1,\n",
       "         ('58081570200', '58082093700'): 1,\n",
       "         ('58081570200', '57808933900'): 1,\n",
       "         ('58081570200', '58082444500'): 1,\n",
       "         ('58081570200', '58017352200'): 1,\n",
       "         ('58081570200', '58081399200'): 1,\n",
       "         ('58081570200', '57222279300'): 1,\n",
       "         ('58081570200', '58081570300'): 1,\n",
       "         ('58081570200', '57197869642'): 1,\n",
       "         ('58081570200', '55809150500'): 1,\n",
       "         ('58081570200', '36962443700'): 1,\n",
       "         ('36163594500', '58082093700'): 1,\n",
       "         ('36163594500', '57808933900'): 1,\n",
       "         ('36163594500', '58082444500'): 1,\n",
       "         ('36163594500', '58017352200'): 1,\n",
       "         ('36163594500', '58081399200'): 1,\n",
       "         ('36163594500', '57222279300'): 1,\n",
       "         ('36163594500', '58081570300'): 1,\n",
       "         ('36163594500', '57197869642'): 1,\n",
       "         ('36163594500', '55809150500'): 1,\n",
       "         ('36163594500', '36962443700'): 1,\n",
       "         ('58082093700', '57808933900'): 1,\n",
       "         ('58082093700', '58082444500'): 1,\n",
       "         ('58082093700', '58017352200'): 1,\n",
       "         ('58082093700', '58081399200'): 1,\n",
       "         ('58082093700', '57222279300'): 1,\n",
       "         ('58082093700', '58081570300'): 1,\n",
       "         ('58082093700', '57197869642'): 1,\n",
       "         ('58082093700', '55809150500'): 1,\n",
       "         ('58082093700', '36962443700'): 1,\n",
       "         ('57808933900', '58082444500'): 1,\n",
       "         ('57808933900', '58017352200'): 1,\n",
       "         ('57808933900', '58081399200'): 1,\n",
       "         ('57808933900', '57222279300'): 1,\n",
       "         ('57808933900', '58081570300'): 1,\n",
       "         ('57808933900', '57197869642'): 1,\n",
       "         ('57808933900', '55809150500'): 1,\n",
       "         ('57808933900', '36962443700'): 1,\n",
       "         ('58082444500', '58017352200'): 1,\n",
       "         ('58082444500', '58081399200'): 1,\n",
       "         ('58082444500', '57222279300'): 1,\n",
       "         ('58082444500', '58081570300'): 1,\n",
       "         ('58082444500', '57197869642'): 1,\n",
       "         ('58082444500', '55809150500'): 1,\n",
       "         ('58082444500', '36962443700'): 1,\n",
       "         ('58017352200', '58081399200'): 1,\n",
       "         ('58017352200', '57222279300'): 1,\n",
       "         ('58017352200', '58081570300'): 1,\n",
       "         ('58017352200', '57197869642'): 1,\n",
       "         ('58017352200', '55809150500'): 1,\n",
       "         ('58017352200', '36962443700'): 1,\n",
       "         ('58081399200', '57222279300'): 1,\n",
       "         ('58081399200', '58081570300'): 1,\n",
       "         ('58081399200', '57197869642'): 1,\n",
       "         ('58081399200', '55809150500'): 1,\n",
       "         ('58081399200', '36962443700'): 1,\n",
       "         ('57222279300', '58081570300'): 1,\n",
       "         ('57222279300', '57197869642'): 1,\n",
       "         ('57222279300', '55809150500'): 1,\n",
       "         ('57222279300', '36962443700'): 1,\n",
       "         ('58081570300', '57197869642'): 1,\n",
       "         ('58081570300', '55809150500'): 1,\n",
       "         ('58081570300', '36962443700'): 1,\n",
       "         ('57197869642', '55809150500'): 1,\n",
       "         ('57197869642', '36962443700'): 1,\n",
       "         ('55809150500', '36962443700'): 1,\n",
       "         ('57212307366', '57189591807'): 1,\n",
       "         ('57212307366', '35474798400'): 1,\n",
       "         ('57212307366', '36049280600'): 1,\n",
       "         ('57189591807', '35474798400'): 1,\n",
       "         ('57189591807', '36049280600'): 1,\n",
       "         ('35474798400', '36049280600'): 1,\n",
       "         ('58033343700', '6602363622'): 1,\n",
       "         ('58033343700', '58032496800'): 1,\n",
       "         ('6602363622', '58032496800'): 1,\n",
       "         ('58032052100', '56147008300'): 1,\n",
       "         ('58032052100', '36455764700'): 1,\n",
       "         ('58032052100', '58031885300'): 1,\n",
       "         ('58032052100', '57995497400'): 1,\n",
       "         ('58032052100', '58031919700'): 1,\n",
       "         ('58032052100', '8964136400'): 1,\n",
       "         ('58032052100', '23980122300'): 1,\n",
       "         ('56147008300', '36455764700'): 2,\n",
       "         ('56147008300', '58031885300'): 1,\n",
       "         ('56147008300', '57995497400'): 1,\n",
       "         ('56147008300', '58031919700'): 1,\n",
       "         ('56147008300', '8964136400'): 2,\n",
       "         ('56147008300', '23980122300'): 1,\n",
       "         ('36455764700', '58031885300'): 1,\n",
       "         ('36455764700', '57995497400'): 1,\n",
       "         ('36455764700', '58031919700'): 1,\n",
       "         ('36455764700', '8964136400'): 2,\n",
       "         ('36455764700', '23980122300'): 1,\n",
       "         ('58031885300', '57995497400'): 1,\n",
       "         ('58031885300', '58031919700'): 1,\n",
       "         ('58031885300', '8964136400'): 1,\n",
       "         ('58031885300', '23980122300'): 1,\n",
       "         ('57995497400', '58031919700'): 1,\n",
       "         ('57995497400', '8964136400'): 1,\n",
       "         ('57995497400', '23980122300'): 1,\n",
       "         ('58031919700', '8964136400'): 1,\n",
       "         ('58031919700', '23980122300'): 1,\n",
       "         ('8964136400', '23980122300'): 1,\n",
       "         ('8875857500', '57810943000'): 1,\n",
       "         ('8875857500', '57203950202'): 1,\n",
       "         ('8875857500', '37664657200'): 1,\n",
       "         ('8875857500', '56736720000'): 1,\n",
       "         ('8875857500', '57192877578'): 1,\n",
       "         ('8875857500', '55531780200'): 2,\n",
       "         ('57810943000', '57203950202'): 1,\n",
       "         ('57810943000', '37664657200'): 1,\n",
       "         ('57810943000', '56736720000'): 1,\n",
       "         ('57810943000', '57192877578'): 1,\n",
       "         ('57810943000', '55531780200'): 1,\n",
       "         ('57203950202', '37664657200'): 1,\n",
       "         ('57203950202', '56736720000'): 1,\n",
       "         ('57203950202', '57192877578'): 1,\n",
       "         ('57203950202', '55531780200'): 1,\n",
       "         ('37664657200', '56736720000'): 1,\n",
       "         ('37664657200', '57192877578'): 1,\n",
       "         ('37664657200', '55531780200'): 1,\n",
       "         ('56736720000', '57192877578'): 1,\n",
       "         ('56736720000', '55531780200'): 1,\n",
       "         ('57192877578', '55531780200'): 1,\n",
       "         ('57222026375', '56925146300'): 1,\n",
       "         ('58066247700', '58066729400'): 1,\n",
       "         ('58066247700', '58067205300'): 1,\n",
       "         ('58066247700', '58067044700'): 1,\n",
       "         ('58066247700', '58066247800'): 1,\n",
       "         ('58066247700', '58066091400'): 1,\n",
       "         ('58066247700', '57946893500'): 1,\n",
       "         ('58066247700', '58066562800'): 1,\n",
       "         ('58066247700', '58066729500'): 1,\n",
       "         ('58066247700', '57942431600'): 1,\n",
       "         ('58066247700', '58066729600'): 1,\n",
       "         ('58066247700', '57192090631'): 1,\n",
       "         ('58066247700', '57211591615'): 1,\n",
       "         ('58066729400', '58067205300'): 1,\n",
       "         ('58066729400', '58067044700'): 1,\n",
       "         ('58066729400', '58066247800'): 1,\n",
       "         ('58066729400', '58066091400'): 1,\n",
       "         ('58066729400', '57946893500'): 1,\n",
       "         ('58066729400', '58066562800'): 1,\n",
       "         ('58066729400', '58066729500'): 1,\n",
       "         ('58066729400', '57942431600'): 1,\n",
       "         ('58066729400', '58066729600'): 1,\n",
       "         ('58066729400', '57192090631'): 1,\n",
       "         ('58066729400', '57211591615'): 1,\n",
       "         ('58067205300', '58067044700'): 1,\n",
       "         ('58067205300', '58066247800'): 1,\n",
       "         ('58067205300', '58066091400'): 1,\n",
       "         ('58067205300', '57946893500'): 1,\n",
       "         ('58067205300', '58066562800'): 1,\n",
       "         ('58067205300', '58066729500'): 1,\n",
       "         ('58067205300', '57942431600'): 1,\n",
       "         ('58067205300', '58066729600'): 1,\n",
       "         ('58067205300', '57192090631'): 1,\n",
       "         ('58067205300', '57211591615'): 1,\n",
       "         ('58067044700', '58066247800'): 1,\n",
       "         ('58067044700', '58066091400'): 1,\n",
       "         ('58067044700', '57946893500'): 1,\n",
       "         ('58067044700', '58066562800'): 1,\n",
       "         ('58067044700', '58066729500'): 1,\n",
       "         ('58067044700', '57942431600'): 1,\n",
       "         ('58067044700', '58066729600'): 1,\n",
       "         ('58067044700', '57192090631'): 1,\n",
       "         ('58067044700', '57211591615'): 1,\n",
       "         ('58066247800', '58066091400'): 1,\n",
       "         ('58066247800', '57946893500'): 1,\n",
       "         ('58066247800', '58066562800'): 1,\n",
       "         ('58066247800', '58066729500'): 1,\n",
       "         ('58066247800', '57942431600'): 1,\n",
       "         ('58066247800', '58066729600'): 1,\n",
       "         ('58066247800', '57192090631'): 1,\n",
       "         ('58066247800', '57211591615'): 1,\n",
       "         ('58066091400', '57946893500'): 1,\n",
       "         ('58066091400', '58066562800'): 1,\n",
       "         ('58066091400', '58066729500'): 1,\n",
       "         ('58066091400', '57942431600'): 1,\n",
       "         ('58066091400', '58066729600'): 1,\n",
       "         ('58066091400', '57192090631'): 1,\n",
       "         ('58066091400', '57211591615'): 1,\n",
       "         ('57946893500', '58066562800'): 1,\n",
       "         ('57946893500', '58066729500'): 1,\n",
       "         ('57946893500', '57942431600'): 1,\n",
       "         ('57946893500', '58066729600'): 1,\n",
       "         ('57946893500', '57192090631'): 1,\n",
       "         ('57946893500', '57211591615'): 1,\n",
       "         ('58066562800', '58066729500'): 1,\n",
       "         ('58066562800', '57942431600'): 1,\n",
       "         ('58066562800', '58066729600'): 1,\n",
       "         ('58066562800', '57192090631'): 1,\n",
       "         ('58066562800', '57211591615'): 1,\n",
       "         ('58066729500', '57942431600'): 1,\n",
       "         ('58066729500', '58066729600'): 1,\n",
       "         ('58066729500', '57192090631'): 1,\n",
       "         ('58066729500', '57211591615'): 1,\n",
       "         ('57942431600', '58066729600'): 1,\n",
       "         ('57942431600', '57192090631'): 1,\n",
       "         ('57942431600', '57211591615'): 1,\n",
       "         ('58066729600', '57192090631'): 1,\n",
       "         ('58066729600', '57211591615'): 1,\n",
       "         ('57192090631', '57211591615'): 1,\n",
       "         ('57219404780', '7006530227'): 2,\n",
       "         ('57219404780', '18934205300'): 2,\n",
       "         ('57219404780', '57191728473'): 2,\n",
       "         ('7006530227', '18934205300'): 1,\n",
       "         ('7006530227', '57191728473'): 1,\n",
       "         ('18934205300', '57191728473'): 2,\n",
       "         ('57202775286', '57716052300'): 1,\n",
       "         ('57202775286', '57716557700'): 1,\n",
       "         ('57202775286', '57716557800'): 1,\n",
       "         ('57202775286', '56106218300'): 1,\n",
       "         ('57716052300', '57716557700'): 1,\n",
       "         ('57716052300', '57716557800'): 1,\n",
       "         ('57716052300', '56106218300'): 1,\n",
       "         ('57716557700', '57716557800'): 1,\n",
       "         ('57716557700', '56106218300'): 1,\n",
       "         ('57716557800', '56106218300'): 1,\n",
       "         ('57191414226', '8046765200'): 1,\n",
       "         ('57191414226', '58035082900'): 1,\n",
       "         ('57191414226', '55877623600'): 1,\n",
       "         ('57191414226', '57565060800'): 1,\n",
       "         ('8046765200', '58035082900'): 1,\n",
       "         ('8046765200', '55877623600'): 1,\n",
       "         ('8046765200', '57565060800'): 1,\n",
       "         ('58035082900', '55877623600'): 1,\n",
       "         ('58035082900', '57565060800'): 1,\n",
       "         ('55877623600', '57565060800'): 1,\n",
       "         ('57193257015', '6505715160'): 1,\n",
       "         ('57193257015', '12785830300'): 1,\n",
       "         ('57193257015', '23488395100'): 1,\n",
       "         ('57193257015', '55911695800'): 1,\n",
       "         ('57193257015', '6603435766'): 1,\n",
       "         ('6505715160', '12785830300'): 1,\n",
       "         ('6505715160', '23488395100'): 1,\n",
       "         ('6505715160', '55911695800'): 1,\n",
       "         ('6505715160', '6603435766'): 1,\n",
       "         ('12785830300', '23488395100'): 1,\n",
       "         ('12785830300', '55911695800'): 1,\n",
       "         ('12785830300', '6603435766'): 1,\n",
       "         ('23488395100', '55911695800'): 1,\n",
       "         ('23488395100', '6603435766'): 1,\n",
       "         ('55911695800', '6603435766'): 1,\n",
       "         ('8226126100', '23484810500'): 1,\n",
       "         ('8226126100', '35451307600'): 1,\n",
       "         ('8226126100', '6701822947'): 1,\n",
       "         ('8226126100', '55443388800'): 1,\n",
       "         ('23484810500', '35451307600'): 1,\n",
       "         ('23484810500', '6701822947'): 1,\n",
       "         ('23484810500', '55443388800'): 1,\n",
       "         ('35451307600', '6701822947'): 1,\n",
       "         ('35451307600', '55443388800'): 1,\n",
       "         ('6701822947', '55443388800'): 1,\n",
       "         ('57321202100', '57841408200'): 1,\n",
       "         ('57321202100', '57841994500'): 1,\n",
       "         ('57321202100', '6602324876'): 1,\n",
       "         ('57321202100', '57211186721'): 1,\n",
       "         ('57321202100', '57192709811'): 1,\n",
       "         ('57841408200', '57841994500'): 1,\n",
       "         ('57841408200', '6602324876'): 1,\n",
       "         ('57841408200', '57211186721'): 1,\n",
       "         ('57841408200', '57192709811'): 1,\n",
       "         ('57841994500', '6602324876'): 1,\n",
       "         ('57841994500', '57211186721'): 1,\n",
       "         ('57841994500', '57192709811'): 1,\n",
       "         ('6602324876', '57211186721'): 1,\n",
       "         ('6602324876', '57192709811'): 1,\n",
       "         ('57211186721', '57192709811'): 1,\n",
       "         ('57201719234', '6602806074'): 1,\n",
       "         ('57201719234', '56467917100'): 1,\n",
       "         ('57201719234', '6602778404'): 1,\n",
       "         ('57201719234', '24280731000'): 1,\n",
       "         ('57201719234', '6504328323'): 1,\n",
       "         ('57201719234', '56843437800'): 1,\n",
       "         ('6602806074', '56467917100'): 1,\n",
       "         ('6602806074', '6602778404'): 1,\n",
       "         ('6602806074', '24280731000'): 1,\n",
       "         ('6602806074', '6504328323'): 1,\n",
       "         ('6602806074', '56843437800'): 1,\n",
       "         ('56467917100', '6602778404'): 1,\n",
       "         ('56467917100', '24280731000'): 1,\n",
       "         ('56467917100', '6504328323'): 1,\n",
       "         ('56467917100', '56843437800'): 1,\n",
       "         ('6602778404', '24280731000'): 1,\n",
       "         ('6602778404', '6504328323'): 1,\n",
       "         ('6602778404', '56843437800'): 1,\n",
       "         ('24280731000', '6504328323'): 1,\n",
       "         ('24280731000', '56843437800'): 1,\n",
       "         ('6504328323', '56843437800'): 1,\n",
       "         ('57639009800', '24367743300'): 1,\n",
       "         ('57477841700', '55348668300'): 2,\n",
       "         ('57477841700', '6602098289'): 2,\n",
       "         ('57477841700', '7004691405'): 2,\n",
       "         ('57477841700', '6701326090'): 2,\n",
       "         ('55348668300', '6602098289'): 2,\n",
       "         ('55348668300', '7004691405'): 2,\n",
       "         ('55348668300', '6701326090'): 2,\n",
       "         ('6602098289', '7004691405'): 2,\n",
       "         ('6602098289', '6701326090'): 2,\n",
       "         ('7004691405', '6701326090'): 2,\n",
       "         ('7006161600', '56005614600'): 1,\n",
       "         ('7006161600', '36340365300'): 1,\n",
       "         ('7006161600', '7007057244'): 1,\n",
       "         ('7006161600', '56590016100'): 1,\n",
       "         ('7006161600', '6602721437'): 1,\n",
       "         ('56005614600', '36340365300'): 1,\n",
       "         ('56005614600', '7007057244'): 1,\n",
       "         ('56005614600', '56590016100'): 1,\n",
       "         ('56005614600', '6602721437'): 1,\n",
       "         ('36340365300', '7007057244'): 1,\n",
       "         ('36340365300', '56590016100'): 1,\n",
       "         ('36340365300', '6602721437'): 1,\n",
       "         ('7007057244', '56590016100'): 1,\n",
       "         ('7007057244', '6602721437'): 1,\n",
       "         ('56590016100', '6602721437'): 1,\n",
       "         ('57998647900', '57191523209'): 1,\n",
       "         ('57998647900', '13605340400'): 1,\n",
       "         ('57191523209', '13605340400'): 1,\n",
       "         ('6602998405', '7004579422'): 1,\n",
       "         ('6602998405', '15849685800'): 1,\n",
       "         ('6602998405', '6602245692'): 1,\n",
       "         ('7004579422', '15849685800'): 1,\n",
       "         ('7004579422', '6602245692'): 1,\n",
       "         ('15849685800', '6602245692'): 1,\n",
       "         ('57222897193', '57220055355'): 1,\n",
       "         ('57222897193', '12446589100'): 1,\n",
       "         ('57222897193', '55111144200'): 1,\n",
       "         ('57222897193', '55441585900'): 1,\n",
       "         ('57222897193', '57653079000'): 1,\n",
       "         ('57222897193', '7005434514'): 1,\n",
       "         ('57222897193', '37110075500'): 1,\n",
       "         ('57222897193', '57195464324'): 1,\n",
       "         ('57222897193', '7005587150'): 1,\n",
       "         ('57222897193', '7003934779'): 1,\n",
       "         ('57222897193', '30567653800'): 1,\n",
       "         ('57222897193', '57205913586'): 1,\n",
       "         ('57222897193', '35227110600'): 1,\n",
       "         ('57220055355', '12446589100'): 1,\n",
       "         ('57220055355', '55111144200'): 1,\n",
       "         ('57220055355', '55441585900'): 1,\n",
       "         ('57220055355', '57653079000'): 1,\n",
       "         ('57220055355', '7005434514'): 1,\n",
       "         ('57220055355', '37110075500'): 1,\n",
       "         ('57220055355', '57195464324'): 1,\n",
       "         ('57220055355', '7005587150'): 1,\n",
       "         ('57220055355', '7003934779'): 1,\n",
       "         ('57220055355', '30567653800'): 1,\n",
       "         ('57220055355', '57205913586'): 1,\n",
       "         ('57220055355', '35227110600'): 1,\n",
       "         ('12446589100', '55111144200'): 1,\n",
       "         ('12446589100', '55441585900'): 1,\n",
       "         ('12446589100', '57653079000'): 1,\n",
       "         ('12446589100', '7005434514'): 1,\n",
       "         ('12446589100', '37110075500'): 1,\n",
       "         ('12446589100', '57195464324'): 1,\n",
       "         ('12446589100', '7005587150'): 1,\n",
       "         ('12446589100', '7003934779'): 1,\n",
       "         ('12446589100', '30567653800'): 1,\n",
       "         ('12446589100', '57205913586'): 1,\n",
       "         ('12446589100', '35227110600'): 1,\n",
       "         ('55111144200', '55441585900'): 1,\n",
       "         ('55111144200', '57653079000'): 1,\n",
       "         ('55111144200', '7005434514'): 1,\n",
       "         ('55111144200', '37110075500'): 1,\n",
       "         ('55111144200', '57195464324'): 1,\n",
       "         ('55111144200', '7005587150'): 1,\n",
       "         ('55111144200', '7003934779'): 1,\n",
       "         ('55111144200', '30567653800'): 1,\n",
       "         ('55111144200', '57205913586'): 1,\n",
       "         ('55111144200', '35227110600'): 1,\n",
       "         ('55441585900', '57653079000'): 1,\n",
       "         ('55441585900', '7005434514'): 1,\n",
       "         ('55441585900', '37110075500'): 1,\n",
       "         ('55441585900', '57195464324'): 1,\n",
       "         ('55441585900', '7005587150'): 1,\n",
       "         ('55441585900', '7003934779'): 1,\n",
       "         ('55441585900', '30567653800'): 1,\n",
       "         ('55441585900', '57205913586'): 1,\n",
       "         ('55441585900', '35227110600'): 1,\n",
       "         ('57653079000', '7005434514'): 1,\n",
       "         ('57653079000', '37110075500'): 1,\n",
       "         ('57653079000', '57195464324'): 1,\n",
       "         ('57653079000', '7005587150'): 1,\n",
       "         ('57653079000', '7003934779'): 1,\n",
       "         ('57653079000', '30567653800'): 1,\n",
       "         ('57653079000', '57205913586'): 1,\n",
       "         ('57653079000', '35227110600'): 1,\n",
       "         ('7005434514', '37110075500'): 1,\n",
       "         ('7005434514', '57195464324'): 1,\n",
       "         ('7005434514', '7005587150'): 1,\n",
       "         ('7005434514', '7003934779'): 1,\n",
       "         ('7005434514', '30567653800'): 1,\n",
       "         ('7005434514', '57205913586'): 1,\n",
       "         ('7005434514', '35227110600'): 1,\n",
       "         ('37110075500', '57195464324'): 1,\n",
       "         ('37110075500', '7005587150'): 1,\n",
       "         ('37110075500', '7003934779'): 1,\n",
       "         ('37110075500', '30567653800'): 1,\n",
       "         ('37110075500', '57205913586'): 1,\n",
       "         ('37110075500', '35227110600'): 1,\n",
       "         ('57195464324', '7005587150'): 1,\n",
       "         ('57195464324', '7003934779'): 1,\n",
       "         ('57195464324', '30567653800'): 1,\n",
       "         ('57195464324', '57205913586'): 1,\n",
       "         ('57195464324', '35227110600'): 1,\n",
       "         ('7005587150', '7003934779'): 1,\n",
       "         ('7005587150', '30567653800'): 1,\n",
       "         ('7005587150', '57205913586'): 1,\n",
       "         ('7005587150', '35227110600'): 1,\n",
       "         ('7003934779', '30567653800'): 1,\n",
       "         ('7003934779', '57205913586'): 1,\n",
       "         ('7003934779', '35227110600'): 1,\n",
       "         ('30567653800', '57205913586'): 1,\n",
       "         ('30567653800', '35227110600'): 1,\n",
       "         ('57205913586', '35227110600'): 1,\n",
       "         ('56465015800', '57200392935'): 1,\n",
       "         ('56465015800', '57896884600'): 1,\n",
       "         ('56465015800', '57220306572'): 1,\n",
       "         ('56465015800', '57360889600'): 1,\n",
       "         ('56465015800', '57188588366'): 1,\n",
       "         ('56465015800', '56861958600'): 1,\n",
       "         ('57200392935', '57896884600'): 1,\n",
       "         ('57200392935', '57220306572'): 1,\n",
       "         ('57200392935', '57360889600'): 1,\n",
       "         ('57200392935', '57188588366'): 1,\n",
       "         ('57200392935', '56861958600'): 1,\n",
       "         ('57896884600', '57220306572'): 1,\n",
       "         ('57896884600', '57360889600'): 1,\n",
       "         ('57896884600', '57188588366'): 1,\n",
       "         ('57896884600', '56861958600'): 1,\n",
       "         ('57220306572', '57360889600'): 1,\n",
       "         ('57220306572', '57188588366'): 1,\n",
       "         ('57220306572', '56861958600'): 1,\n",
       "         ('57360889600', '57188588366'): 1,\n",
       "         ('57360889600', '56861958600'): 1,\n",
       "         ('57188588366', '56861958600'): 1,\n",
       "         ('57209614945', '24338144900'): 2,\n",
       "         ('57209614945', '8296514700'): 2,\n",
       "         ('57209614945', '57201696502'): 1,\n",
       "         ('57209614945', '57786492600'): 1,\n",
       "         ('57209614945', '6602186413'): 1,\n",
       "         ('57209614945', '6701858537'): 2,\n",
       "         ('24338144900', '8296514700'): 1,\n",
       "         ('24338144900', '57201696502'): 1,\n",
       "         ('24338144900', '57786492600'): 1,\n",
       "         ('24338144900', '6602186413'): 1,\n",
       "         ('24338144900', '6701858537'): 1,\n",
       "         ('8296514700', '57201696502'): 1,\n",
       "         ('8296514700', '57786492600'): 1,\n",
       "         ('8296514700', '6602186413'): 1,\n",
       "         ('8296514700', '6701858537'): 2,\n",
       "         ('57201696502', '57786492600'): 1,\n",
       "         ('57201696502', '6602186413'): 1,\n",
       "         ('57201696502', '6701858537'): 1,\n",
       "         ('57786492600', '6602186413'): 1,\n",
       "         ('57786492600', '6701858537'): 1,\n",
       "         ('6602186413', '6701858537'): 1,\n",
       "         ('54937838500', '7202587597'): 1,\n",
       "         ('54937838500', '7102186256'): 1,\n",
       "         ('54937838500', '7005620037'): 1,\n",
       "         ('54937838500', '57204791050'): 1,\n",
       "         ('54937838500', '13807130100'): 1,\n",
       "         ('54937838500', '14832180400'): 1,\n",
       "         ('54937838500', '7202192005'): 1,\n",
       "         ('54937838500', '7003265353'): 1,\n",
       "         ('54937838500', '34570772200'): 1,\n",
       "         ('54937838500', '7102141909'): 1,\n",
       "         ('7202587597', '7102186256'): 1,\n",
       "         ('7202587597', '7005620037'): 1,\n",
       "         ('7202587597', '57204791050'): 1,\n",
       "         ('7202587597', '13807130100'): 1,\n",
       "         ('7202587597', '14832180400'): 1,\n",
       "         ('7202587597', '7202192005'): 1,\n",
       "         ('7202587597', '7003265353'): 1,\n",
       "         ('7202587597', '34570772200'): 1,\n",
       "         ('7202587597', '7102141909'): 1,\n",
       "         ('7102186256', '7005620037'): 1,\n",
       "         ('7102186256', '57204791050'): 1,\n",
       "         ('7102186256', '13807130100'): 1,\n",
       "         ('7102186256', '14832180400'): 1,\n",
       "         ('7102186256', '7202192005'): 1,\n",
       "         ('7102186256', '7003265353'): 1,\n",
       "         ('7102186256', '34570772200'): 1,\n",
       "         ('7102186256', '7102141909'): 1,\n",
       "         ('7005620037', '57204791050'): 1,\n",
       "         ('7005620037', '13807130100'): 1,\n",
       "         ('7005620037', '14832180400'): 1,\n",
       "         ('7005620037', '7202192005'): 1,\n",
       "         ('7005620037', '7003265353'): 1,\n",
       "         ('7005620037', '34570772200'): 1,\n",
       "         ('7005620037', '7102141909'): 1,\n",
       "         ('57204791050', '13807130100'): 1,\n",
       "         ('57204791050', '14832180400'): 1,\n",
       "         ('57204791050', '7202192005'): 1,\n",
       "         ('57204791050', '7003265353'): 1,\n",
       "         ('57204791050', '34570772200'): 1,\n",
       "         ('57204791050', '7102141909'): 1,\n",
       "         ('13807130100', '14832180400'): 1,\n",
       "         ('13807130100', '7202192005'): 1,\n",
       "         ('13807130100', '7003265353'): 1,\n",
       "         ('13807130100', '34570772200'): 1,\n",
       "         ('13807130100', '7102141909'): 1,\n",
       "         ('14832180400', '7202192005'): 1,\n",
       "         ('14832180400', '7003265353'): 1,\n",
       "         ('14832180400', '34570772200'): 1,\n",
       "         ('14832180400', '7102141909'): 1,\n",
       "         ('7202192005', '7003265353'): 1,\n",
       "         ('7202192005', '34570772200'): 1,\n",
       "         ('7202192005', '7102141909'): 1,\n",
       "         ('7003265353', '34570772200'): 1,\n",
       "         ('7003265353', '7102141909'): 1,\n",
       "         ('34570772200', '7102141909'): 1,\n",
       "         ('57221605651', '23980122300'): 1,\n",
       "         ('57221605651', '56147008300'): 1,\n",
       "         ('57221605651', '36455764700'): 1,\n",
       "         ('57221605651', '8964136400'): 1,\n",
       "         ('23980122300', '56147008300'): 1,\n",
       "         ('23980122300', '36455764700'): 1,\n",
       "         ('23980122300', '8964136400'): 1,\n",
       "         ('57189461786', '20333917800'): 1,\n",
       "         ('57189461786', '55225187000'): 1,\n",
       "         ('57189461786', '6603767582'): 1,\n",
       "         ('57189461786', '35243030300'): 1,\n",
       "         ('57189461786', '6603752469'): 1,\n",
       "         ('20333917800', '55225187000'): 1,\n",
       "         ('20333917800', '6603767582'): 1,\n",
       "         ('20333917800', '35243030300'): 1,\n",
       "         ('20333917800', '6603752469'): 1,\n",
       "         ('55225187000', '6603767582'): 1,\n",
       "         ('55225187000', '35243030300'): 1,\n",
       "         ('55225187000', '6603752469'): 1,\n",
       "         ('6603767582', '35243030300'): 1,\n",
       "         ('6603767582', '6603752469'): 1,\n",
       "         ('35243030300', '6603752469'): 1,\n",
       "         ('36611724000', '8835604200'): 1,\n",
       "         ('36611724000', '24576734900'): 1,\n",
       "         ('36611724000', '56580241600'): 1,\n",
       "         ('36611724000', '57209551521'): 1,\n",
       "         ('36611724000', '12761269700'): 1,\n",
       "         ('36611724000', '57190272868'): 1,\n",
       "         ('36611724000', '50760992300'): 1,\n",
       "         ('36611724000', '34569127200'): 1,\n",
       "         ('36611724000', '6602195572'): 1,\n",
       "         ('36611724000', '56463035500'): 1,\n",
       "         ('36611724000', '54789048000'): 1,\n",
       "         ('36611724000', '6507581306'): 1,\n",
       "         ('36611724000', '6603710238'): 1,\n",
       "         ('36611724000', '20735018600'): 1,\n",
       "         ('36611724000', '26434318500'): 1,\n",
       "         ('36611724000', '57219204911'): 1,\n",
       "         ('36611724000', '57209323859'): 1,\n",
       "         ('36611724000', '56480829300'): 1,\n",
       "         ('36611724000', '54986245000'): 1,\n",
       "         ('36611724000', '57213587752'): 1,\n",
       "         ('36611724000', '57204630901'): 1,\n",
       "         ('36611724000', '57208140398'): 1,\n",
       "         ('36611724000', '57202469544'): 1,\n",
       "         ('36611724000', '35380737400'): 1,\n",
       "         ('36611724000', '57220118235'): 1,\n",
       "         ('36611724000', '57205170885'): 1,\n",
       "         ('36611724000', '23134902300'): 1,\n",
       "         ('8835604200', '24576734900'): 1,\n",
       "         ('8835604200', '56580241600'): 1,\n",
       "         ('8835604200', '57209551521'): 1,\n",
       "         ('8835604200', '12761269700'): 1,\n",
       "         ('8835604200', '57190272868'): 1,\n",
       "         ('8835604200', '50760992300'): 1,\n",
       "         ('8835604200', '34569127200'): 1,\n",
       "         ('8835604200', '6602195572'): 1,\n",
       "         ('8835604200', '56463035500'): 1,\n",
       "         ('8835604200', '54789048000'): 1,\n",
       "         ('8835604200', '6507581306'): 1,\n",
       "         ('8835604200', '6603710238'): 1,\n",
       "         ('8835604200', '20735018600'): 1,\n",
       "         ('8835604200', '26434318500'): 1,\n",
       "         ('8835604200', '57219204911'): 1,\n",
       "         ('8835604200', '57209323859'): 1,\n",
       "         ('8835604200', '56480829300'): 1,\n",
       "         ('8835604200', '54986245000'): 1,\n",
       "         ('8835604200', '57213587752'): 1,\n",
       "         ('8835604200', '57204630901'): 1,\n",
       "         ('8835604200', '57208140398'): 1,\n",
       "         ('8835604200', '57202469544'): 1,\n",
       "         ('8835604200', '35380737400'): 1,\n",
       "         ('8835604200', '57220118235'): 1,\n",
       "         ('8835604200', '57205170885'): 1,\n",
       "         ('8835604200', '23134902300'): 1,\n",
       "         ('24576734900', '56580241600'): 1,\n",
       "         ('24576734900', '57209551521'): 1,\n",
       "         ('24576734900', '12761269700'): 1,\n",
       "         ('24576734900', '57190272868'): 1,\n",
       "         ('24576734900', '50760992300'): 1,\n",
       "         ('24576734900', '34569127200'): 1,\n",
       "         ('24576734900', '6602195572'): 1,\n",
       "         ('24576734900', '56463035500'): 1,\n",
       "         ('24576734900', '54789048000'): 1,\n",
       "         ('24576734900', '6507581306'): 1,\n",
       "         ('24576734900', '6603710238'): 1,\n",
       "         ('24576734900', '20735018600'): 1,\n",
       "         ('24576734900', '26434318500'): 1,\n",
       "         ('24576734900', '57219204911'): 1,\n",
       "         ('24576734900', '57209323859'): 1,\n",
       "         ('24576734900', '56480829300'): 1,\n",
       "         ('24576734900', '54986245000'): 1,\n",
       "         ('24576734900', '57213587752'): 1,\n",
       "         ('24576734900', '57204630901'): 1,\n",
       "         ('24576734900', '57208140398'): 1,\n",
       "         ('24576734900', '57202469544'): 1,\n",
       "         ('24576734900', '35380737400'): 1,\n",
       "         ('24576734900', '57220118235'): 1,\n",
       "         ('24576734900', '57205170885'): 1,\n",
       "         ('24576734900', '23134902300'): 1,\n",
       "         ('56580241600', '57209551521'): 1,\n",
       "         ('56580241600', '12761269700'): 1,\n",
       "         ('56580241600', '57190272868'): 1,\n",
       "         ('56580241600', '50760992300'): 1,\n",
       "         ('56580241600', '34569127200'): 1,\n",
       "         ('56580241600', '6602195572'): 1,\n",
       "         ('56580241600', '56463035500'): 1,\n",
       "         ('56580241600', '54789048000'): 1,\n",
       "         ('56580241600', '6507581306'): 1,\n",
       "         ('56580241600', '6603710238'): 1,\n",
       "         ('56580241600', '20735018600'): 1,\n",
       "         ('56580241600', '26434318500'): 1,\n",
       "         ('56580241600', '57219204911'): 1,\n",
       "         ('56580241600', '57209323859'): 1,\n",
       "         ('56580241600', '56480829300'): 1,\n",
       "         ('56580241600', '54986245000'): 1,\n",
       "         ('56580241600', '57213587752'): 1,\n",
       "         ('56580241600', '57204630901'): 1,\n",
       "         ('56580241600', '57208140398'): 1,\n",
       "         ('56580241600', '57202469544'): 1,\n",
       "         ('56580241600', '35380737400'): 1,\n",
       "         ('56580241600', '57220118235'): 1,\n",
       "         ('56580241600', '57205170885'): 1,\n",
       "         ('56580241600', '23134902300'): 1,\n",
       "         ('57209551521', '12761269700'): 1,\n",
       "         ('57209551521', '57190272868'): 1,\n",
       "         ('57209551521', '50760992300'): 1,\n",
       "         ('57209551521', '34569127200'): 1,\n",
       "         ('57209551521', '6602195572'): 1,\n",
       "         ('57209551521', '56463035500'): 1,\n",
       "         ('57209551521', '54789048000'): 1,\n",
       "         ('57209551521', '6507581306'): 1,\n",
       "         ('57209551521', '6603710238'): 1,\n",
       "         ('57209551521', '20735018600'): 1,\n",
       "         ('57209551521', '26434318500'): 1,\n",
       "         ('57209551521', '57219204911'): 1,\n",
       "         ('57209551521', '57209323859'): 1,\n",
       "         ('57209551521', '56480829300'): 1,\n",
       "         ('57209551521', '54986245000'): 1,\n",
       "         ('57209551521', '57213587752'): 1,\n",
       "         ('57209551521', '57204630901'): 1,\n",
       "         ('57209551521', '57208140398'): 1,\n",
       "         ('57209551521', '57202469544'): 1,\n",
       "         ('57209551521', '35380737400'): 1,\n",
       "         ('57209551521', '57220118235'): 1,\n",
       "         ('57209551521', '57205170885'): 1,\n",
       "         ('57209551521', '23134902300'): 1,\n",
       "         ('12761269700', '57190272868'): 1,\n",
       "         ('12761269700', '50760992300'): 1,\n",
       "         ('12761269700', '34569127200'): 1,\n",
       "         ('12761269700', '6602195572'): 1,\n",
       "         ('12761269700', '56463035500'): 1,\n",
       "         ('12761269700', '54789048000'): 1,\n",
       "         ('12761269700', '6507581306'): 1,\n",
       "         ('12761269700', '6603710238'): 1,\n",
       "         ('12761269700', '20735018600'): 1,\n",
       "         ('12761269700', '26434318500'): 1,\n",
       "         ('12761269700', '57219204911'): 1,\n",
       "         ('12761269700', '57209323859'): 1,\n",
       "         ('12761269700', '56480829300'): 1,\n",
       "         ('12761269700', '54986245000'): 1,\n",
       "         ('12761269700', '57213587752'): 1,\n",
       "         ('12761269700', '57204630901'): 1,\n",
       "         ('12761269700', '57208140398'): 1,\n",
       "         ('12761269700', '57202469544'): 1,\n",
       "         ('12761269700', '35380737400'): 1,\n",
       "         ('12761269700', '57220118235'): 1,\n",
       "         ('12761269700', '57205170885'): 1,\n",
       "         ('12761269700', '23134902300'): 1,\n",
       "         ('57190272868', '50760992300'): 1,\n",
       "         ('57190272868', '34569127200'): 1,\n",
       "         ('57190272868', '6602195572'): 1,\n",
       "         ('57190272868', '56463035500'): 1,\n",
       "         ('57190272868', '54789048000'): 1,\n",
       "         ('57190272868', '6507581306'): 1,\n",
       "         ('57190272868', '6603710238'): 1,\n",
       "         ('57190272868', '20735018600'): 1,\n",
       "         ('57190272868', '26434318500'): 1,\n",
       "         ('57190272868', '57219204911'): 1,\n",
       "         ('57190272868', '57209323859'): 1,\n",
       "         ('57190272868', '56480829300'): 1,\n",
       "         ('57190272868', '54986245000'): 1,\n",
       "         ('57190272868', '57213587752'): 1,\n",
       "         ('57190272868', '57204630901'): 1,\n",
       "         ('57190272868', '57208140398'): 1,\n",
       "         ('57190272868', '57202469544'): 1,\n",
       "         ('57190272868', '35380737400'): 1,\n",
       "         ('57190272868', '57220118235'): 1,\n",
       "         ('57190272868', '57205170885'): 1,\n",
       "         ('57190272868', '23134902300'): 1,\n",
       "         ('50760992300', '34569127200'): 1,\n",
       "         ('50760992300', '6602195572'): 1,\n",
       "         ('50760992300', '56463035500'): 1,\n",
       "         ('50760992300', '54789048000'): 1,\n",
       "         ('50760992300', '6507581306'): 1,\n",
       "         ('50760992300', '6603710238'): 1,\n",
       "         ('50760992300', '20735018600'): 1,\n",
       "         ('50760992300', '26434318500'): 1,\n",
       "         ('50760992300', '57219204911'): 1,\n",
       "         ('50760992300', '57209323859'): 1,\n",
       "         ('50760992300', '56480829300'): 1,\n",
       "         ('50760992300', '54986245000'): 1,\n",
       "         ('50760992300', '57213587752'): 1,\n",
       "         ('50760992300', '57204630901'): 1,\n",
       "         ('50760992300', '57208140398'): 1,\n",
       "         ('50760992300', '57202469544'): 1,\n",
       "         ('50760992300', '35380737400'): 1,\n",
       "         ('50760992300', '57220118235'): 1,\n",
       "         ('50760992300', '57205170885'): 1,\n",
       "         ('50760992300', '23134902300'): 1,\n",
       "         ('34569127200', '6602195572'): 1,\n",
       "         ('34569127200', '56463035500'): 1,\n",
       "         ('34569127200', '54789048000'): 1,\n",
       "         ('34569127200', '6507581306'): 1,\n",
       "         ('34569127200', '6603710238'): 1,\n",
       "         ('34569127200', '20735018600'): 1,\n",
       "         ('34569127200', '26434318500'): 1,\n",
       "         ('34569127200', '57219204911'): 1,\n",
       "         ('34569127200', '57209323859'): 1,\n",
       "         ('34569127200', '56480829300'): 1,\n",
       "         ('34569127200', '54986245000'): 1,\n",
       "         ('34569127200', '57213587752'): 1,\n",
       "         ('34569127200', '57204630901'): 1,\n",
       "         ('34569127200', '57208140398'): 1,\n",
       "         ('34569127200', '57202469544'): 1,\n",
       "         ('34569127200', '35380737400'): 1,\n",
       "         ('34569127200', '57220118235'): 1,\n",
       "         ('34569127200', '57205170885'): 1,\n",
       "         ('34569127200', '23134902300'): 1,\n",
       "         ('6602195572', '56463035500'): 1,\n",
       "         ('6602195572', '54789048000'): 1,\n",
       "         ('6602195572', '6507581306'): 1,\n",
       "         ('6602195572', '6603710238'): 1,\n",
       "         ('6602195572', '20735018600'): 1,\n",
       "         ('6602195572', '26434318500'): 1,\n",
       "         ('6602195572', '57219204911'): 1,\n",
       "         ('6602195572', '57209323859'): 1,\n",
       "         ('6602195572', '56480829300'): 1,\n",
       "         ('6602195572', '54986245000'): 1,\n",
       "         ('6602195572', '57213587752'): 1,\n",
       "         ('6602195572', '57204630901'): 1,\n",
       "         ('6602195572', '57208140398'): 1,\n",
       "         ('6602195572', '57202469544'): 1,\n",
       "         ('6602195572', '35380737400'): 1,\n",
       "         ('6602195572', '57220118235'): 1,\n",
       "         ('6602195572', '57205170885'): 1,\n",
       "         ('6602195572', '23134902300'): 1,\n",
       "         ('56463035500', '54789048000'): 1,\n",
       "         ('56463035500', '6507581306'): 1,\n",
       "         ('56463035500', '6603710238'): 1,\n",
       "         ('56463035500', '20735018600'): 1,\n",
       "         ('56463035500', '26434318500'): 1,\n",
       "         ('56463035500', '57219204911'): 1,\n",
       "         ('56463035500', '57209323859'): 1,\n",
       "         ('56463035500', '56480829300'): 1,\n",
       "         ('56463035500', '54986245000'): 1,\n",
       "         ('56463035500', '57213587752'): 1,\n",
       "         ('56463035500', '57204630901'): 1,\n",
       "         ('56463035500', '57208140398'): 1,\n",
       "         ('56463035500', '57202469544'): 1,\n",
       "         ('56463035500', '35380737400'): 1,\n",
       "         ('56463035500', '57220118235'): 1,\n",
       "         ('56463035500', '57205170885'): 1,\n",
       "         ('56463035500', '23134902300'): 1,\n",
       "         ('54789048000', '6507581306'): 1,\n",
       "         ('54789048000', '6603710238'): 1,\n",
       "         ('54789048000', '20735018600'): 1,\n",
       "         ('54789048000', '26434318500'): 1,\n",
       "         ('54789048000', '57219204911'): 1,\n",
       "         ('54789048000', '57209323859'): 1,\n",
       "         ('54789048000', '56480829300'): 1,\n",
       "         ('54789048000', '54986245000'): 1,\n",
       "         ('54789048000', '57213587752'): 1,\n",
       "         ('54789048000', '57204630901'): 1,\n",
       "         ('54789048000', '57208140398'): 1,\n",
       "         ('54789048000', '57202469544'): 1,\n",
       "         ('54789048000', '35380737400'): 1,\n",
       "         ('54789048000', '57220118235'): 1,\n",
       "         ('54789048000', '57205170885'): 1,\n",
       "         ('54789048000', '23134902300'): 1,\n",
       "         ('6507581306', '6603710238'): 1,\n",
       "         ('6507581306', '20735018600'): 1,\n",
       "         ('6507581306', '26434318500'): 1,\n",
       "         ('6507581306', '57219204911'): 1,\n",
       "         ('6507581306', '57209323859'): 1,\n",
       "         ('6507581306', '56480829300'): 1,\n",
       "         ('6507581306', '54986245000'): 1,\n",
       "         ('6507581306', '57213587752'): 1,\n",
       "         ('6507581306', '57204630901'): 1,\n",
       "         ('6507581306', '57208140398'): 1,\n",
       "         ('6507581306', '57202469544'): 1,\n",
       "         ('6507581306', '35380737400'): 1,\n",
       "         ('6507581306', '57220118235'): 1,\n",
       "         ('6507581306', '57205170885'): 1,\n",
       "         ('6507581306', '23134902300'): 1,\n",
       "         ('6603710238', '20735018600'): 1,\n",
       "         ('6603710238', '26434318500'): 1,\n",
       "         ('6603710238', '57219204911'): 1,\n",
       "         ('6603710238', '57209323859'): 1,\n",
       "         ('6603710238', '56480829300'): 1,\n",
       "         ('6603710238', '54986245000'): 1,\n",
       "         ('6603710238', '57213587752'): 1,\n",
       "         ('6603710238', '57204630901'): 1,\n",
       "         ('6603710238', '57208140398'): 1,\n",
       "         ('6603710238', '57202469544'): 1,\n",
       "         ('6603710238', '35380737400'): 1,\n",
       "         ('6603710238', '57220118235'): 1,\n",
       "         ('6603710238', '57205170885'): 1,\n",
       "         ('6603710238', '23134902300'): 1,\n",
       "         ('20735018600', '26434318500'): 1,\n",
       "         ('20735018600', '57219204911'): 1,\n",
       "         ('20735018600', '57209323859'): 1,\n",
       "         ('20735018600', '56480829300'): 1,\n",
       "         ('20735018600', '54986245000'): 1,\n",
       "         ('20735018600', '57213587752'): 1,\n",
       "         ('20735018600', '57204630901'): 1,\n",
       "         ('20735018600', '57208140398'): 1,\n",
       "         ('20735018600', '57202469544'): 1,\n",
       "         ('20735018600', '35380737400'): 1,\n",
       "         ('20735018600', '57220118235'): 1,\n",
       "         ('20735018600', '57205170885'): 1,\n",
       "         ('20735018600', '23134902300'): 1,\n",
       "         ('26434318500', '57219204911'): 1,\n",
       "         ('26434318500', '57209323859'): 1,\n",
       "         ('26434318500', '56480829300'): 1,\n",
       "         ('26434318500', '54986245000'): 1,\n",
       "         ('26434318500', '57213587752'): 1,\n",
       "         ('26434318500', '57204630901'): 1,\n",
       "         ('26434318500', '57208140398'): 1,\n",
       "         ('26434318500', '57202469544'): 1,\n",
       "         ('26434318500', '35380737400'): 1,\n",
       "         ('26434318500', '57220118235'): 1,\n",
       "         ('26434318500', '57205170885'): 1,\n",
       "         ('26434318500', '23134902300'): 1,\n",
       "         ('57219204911', '57209323859'): 1,\n",
       "         ('57219204911', '56480829300'): 1,\n",
       "         ('57219204911', '54986245000'): 1,\n",
       "         ('57219204911', '57213587752'): 1,\n",
       "         ('57219204911', '57204630901'): 1,\n",
       "         ('57219204911', '57208140398'): 1,\n",
       "         ('57219204911', '57202469544'): 1,\n",
       "         ('57219204911', '35380737400'): 1,\n",
       "         ('57219204911', '57220118235'): 1,\n",
       "         ('57219204911', '57205170885'): 1,\n",
       "         ('57219204911', '23134902300'): 1,\n",
       "         ('57209323859', '56480829300'): 1,\n",
       "         ('57209323859', '54986245000'): 1,\n",
       "         ('57209323859', '57213587752'): 1,\n",
       "         ('57209323859', '57204630901'): 1,\n",
       "         ('57209323859', '57208140398'): 1,\n",
       "         ('57209323859', '57202469544'): 1,\n",
       "         ('57209323859', '35380737400'): 1,\n",
       "         ('57209323859', '57220118235'): 1,\n",
       "         ('57209323859', '57205170885'): 1,\n",
       "         ('57209323859', '23134902300'): 1,\n",
       "         ('56480829300', '54986245000'): 1,\n",
       "         ('56480829300', '57213587752'): 1,\n",
       "         ('56480829300', '57204630901'): 1,\n",
       "         ('56480829300', '57208140398'): 1,\n",
       "         ('56480829300', '57202469544'): 1,\n",
       "         ('56480829300', '35380737400'): 1,\n",
       "         ('56480829300', '57220118235'): 1,\n",
       "         ('56480829300', '57205170885'): 1,\n",
       "         ('56480829300', '23134902300'): 1,\n",
       "         ('54986245000', '57213587752'): 1,\n",
       "         ('54986245000', '57204630901'): 1,\n",
       "         ('54986245000', '57208140398'): 1,\n",
       "         ('54986245000', '57202469544'): 1,\n",
       "         ('54986245000', '35380737400'): 1,\n",
       "         ('54986245000', '57220118235'): 1,\n",
       "         ('54986245000', '57205170885'): 1,\n",
       "         ('54986245000', '23134902300'): 1,\n",
       "         ('57213587752', '57204630901'): 1,\n",
       "         ('57213587752', '57208140398'): 1,\n",
       "         ('57213587752', '57202469544'): 1,\n",
       "         ('57213587752', '35380737400'): 1,\n",
       "         ('57213587752', '57220118235'): 1,\n",
       "         ('57213587752', '57205170885'): 1,\n",
       "         ('57213587752', '23134902300'): 1,\n",
       "         ('57204630901', '57208140398'): 1,\n",
       "         ('57204630901', '57202469544'): 1,\n",
       "         ('57204630901', '35380737400'): 1,\n",
       "         ('57204630901', '57220118235'): 1,\n",
       "         ('57204630901', '57205170885'): 1,\n",
       "         ('57204630901', '23134902300'): 1,\n",
       "         ('57208140398', '57202469544'): 1,\n",
       "         ('57208140398', '35380737400'): 1,\n",
       "         ('57208140398', '57220118235'): 1,\n",
       "         ('57208140398', '57205170885'): 1,\n",
       "         ('57208140398', '23134902300'): 1,\n",
       "         ('57202469544', '35380737400'): 1,\n",
       "         ('57202469544', '57220118235'): 1,\n",
       "         ('57202469544', '57205170885'): 1,\n",
       "         ('57202469544', '23134902300'): 1,\n",
       "         ('35380737400', '57220118235'): 1,\n",
       "         ('35380737400', '57205170885'): 1,\n",
       "         ('35380737400', '23134902300'): 1,\n",
       "         ('57220118235', '57205170885'): 1,\n",
       "         ('57220118235', '23134902300'): 1,\n",
       "         ('57205170885', '23134902300'): 1,\n",
       "         ('57195732410', '57189641697'): 1,\n",
       "         ('57195732410', '57489541100'): 2,\n",
       "         ('57195732410', '6603557582'): 1,\n",
       "         ('57195732410', '7103067740'): 2,\n",
       "         ('57195732410', '7003600497'): 2,\n",
       "         ('57195732410', '18040708300'): 1,\n",
       "         ('57189641697', '57489541100'): 2,\n",
       "         ('57189641697', '6603557582'): 1,\n",
       "         ('57189641697', '7103067740'): 2,\n",
       "         ('57189641697', '7003600497'): 2,\n",
       "         ('57189641697', '18040708300'): 1,\n",
       "         ('57489541100', '6603557582'): 1,\n",
       "         ('57489541100', '7103067740'): 2,\n",
       "         ('57489541100', '7003600497'): 2,\n",
       "         ('57489541100', '18040708300'): 1,\n",
       "         ('6603557582', '7103067740'): 1,\n",
       "         ('6603557582', '7003600497'): 1,\n",
       "         ('6603557582', '18040708300'): 1,\n",
       "         ('7103067740', '7003600497'): 2,\n",
       "         ('7103067740', '18040708300'): 1,\n",
       "         ('7003600497', '18040708300'): 1,\n",
       "         ('37053356400', '56411939100'): 1,\n",
       "         ('37053356400', '7102339373'): 1,\n",
       "         ('37053356400', '57190279338'): 1,\n",
       "         ('56411939100', '7102339373'): 1,\n",
       "         ('56411939100', '57190279338'): 1,\n",
       "         ('7102339373', '57190279338'): 1,\n",
       "         ('6507920875', '57115477800'): 1,\n",
       "         ('6507920875', '23135719900'): 1,\n",
       "         ('6507920875', '49863862000'): 1,\n",
       "         ('6507920875', '26534464000'): 1,\n",
       "         ('6507920875', '8530143500'): 1,\n",
       "         ('6507920875', '57830408100'): 1,\n",
       "         ('6507920875', '57830408200'): 1,\n",
       "         ('6507920875', '7202739109'): 1,\n",
       "         ('57115477800', '23135719900'): 1,\n",
       "         ('57115477800', '49863862000'): 1,\n",
       "         ('57115477800', '26534464000'): 1,\n",
       "         ('57115477800', '8530143500'): 1,\n",
       "         ('57115477800', '57830408100'): 1,\n",
       "         ('57115477800', '57830408200'): 1,\n",
       "         ('57115477800', '7202739109'): 1,\n",
       "         ('23135719900', '49863862000'): 1,\n",
       "         ('23135719900', '26534464000'): 1,\n",
       "         ('23135719900', '8530143500'): 1,\n",
       "         ('23135719900', '57830408100'): 1,\n",
       "         ('23135719900', '57830408200'): 1,\n",
       "         ('23135719900', '7202739109'): 1,\n",
       "         ('49863862000', '26534464000'): 1,\n",
       "         ('49863862000', '8530143500'): 1,\n",
       "         ('49863862000', '57830408100'): 1,\n",
       "         ('49863862000', '57830408200'): 1,\n",
       "         ('49863862000', '7202739109'): 1,\n",
       "         ('26534464000', '8530143500'): 1,\n",
       "         ('26534464000', '57830408100'): 1,\n",
       "         ('26534464000', '57830408200'): 1,\n",
       "         ('26534464000', '7202739109'): 1,\n",
       "         ('8530143500', '57830408100'): 1,\n",
       "         ('8530143500', '57830408200'): 1,\n",
       "         ('8530143500', '7202739109'): 1,\n",
       "         ('57830408100', '57830408200'): 1,\n",
       "         ('57830408100', '7202739109'): 1,\n",
       "         ('57830408200', '7202739109'): 1,\n",
       "         ('37052139500', '57210156037'): 1,\n",
       "         ('37052139500', '57222010397'): 1,\n",
       "         ('37052139500', '57777616000'): 1,\n",
       "         ('37052139500', '57776038100'): 1,\n",
       "         ('37052139500', '57195939196'): 1,\n",
       "         ('37052139500', '57777391000'): 1,\n",
       "         ('37052139500', '56430701900'): 1,\n",
       "         ('37052139500', '57777168300'): 1,\n",
       "         ('37052139500', '57189854493'): 1,\n",
       "         ('37052139500', '55111210500'): 1,\n",
       "         ('37052139500', '7005534467'): 1,\n",
       "         ('37052139500', '57777616100'): 1,\n",
       "         ('37052139500', '57199296355'): 1,\n",
       "         ('37052139500', '7005193770'): 1,\n",
       "         ('37052139500', '57205745674'): 1,\n",
       "         ('57210156037', '57222010397'): 1,\n",
       "         ('57210156037', '57777616000'): 1,\n",
       "         ('57210156037', '57776038100'): 1,\n",
       "         ('57210156037', '57195939196'): 1,\n",
       "         ('57210156037', '57777391000'): 1,\n",
       "         ('57210156037', '56430701900'): 1,\n",
       "         ('57210156037', '57777168300'): 1,\n",
       "         ('57210156037', '57189854493'): 1,\n",
       "         ('57210156037', '55111210500'): 1,\n",
       "         ...})"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Num of co-writings with two people -> edge weight?\n",
    "from collections import Counter\n",
    "link_num = Counter(link)\n",
    "link_num"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "2a5c351e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(('6603277503', '57218685959'), 16),\n",
       " (('57454365000', '57218685959'), 13),\n",
       " (('57218685959', '57218685959'), 10),\n",
       " (('35957366900', '57218685959'), 9),\n",
       " (('57203425292', '57218685959'), 9),\n",
       " (('57218685959', '57212263289'), 9),\n",
       " (('57203425292', '6603277503'), 8),\n",
       " (('57454365000', '6603277503'), 8),\n",
       " (('57218685959', '57220962396'), 8),\n",
       " (('57216530713', '6603277503'), 8)]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "link_num.most_common(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "9087f5de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(('6603277503', '57218685959'), 16),\n",
       " (('57454365000', '57218685959'), 13),\n",
       " (('57218685959', '57218685959'), 10),\n",
       " (('35957366900', '57218685959'), 9),\n",
       " (('57203425292', '57218685959'), 9),\n",
       " (('57218685959', '57212263289'), 9),\n",
       " (('57203425292', '6603277503'), 8),\n",
       " (('57454365000', '6603277503'), 8),\n",
       " (('57218685959', '57220962396'), 8),\n",
       " (('57216530713', '6603277503'), 8),\n",
       " (('57216530713', '57218685959'), 8),\n",
       " (('57230977500', '57218685959'), 8),\n",
       " (('57031969300', '57218685959'), 8),\n",
       " (('57192180234', '57218685959'), 8),\n",
       " (('6603277503', '55228809500'), 8),\n",
       " (('6603277503', '23993684800'), 8),\n",
       " (('6603277503', '55823146800'), 8),\n",
       " (('6603277503', '57221929857'), 8),\n",
       " (('6603277503', '57220962396'), 8),\n",
       " (('6603277503', '57212263289'), 8),\n",
       " (('7801624038', '57218685959'), 8),\n",
       " (('56429644500', '57218685959'), 8),\n",
       " (('57367028700', '57218685959'), 8),\n",
       " (('6506014076', '57218685959'), 8),\n",
       " (('6505495562', '57218685959'), 8),\n",
       " (('7003560327', '57218685959'), 8),\n",
       " (('55228809500', '57218685959'), 8),\n",
       " (('57454365000', '23993684800'), 6),\n",
       " (('57454365000', '55823146800'), 6),\n",
       " (('57454365000', '57221929857'), 6),\n",
       " (('57454365000', '57220962396'), 6),\n",
       " (('57454365000', '57212263289'), 6),\n",
       " (('57218685959', '57221929857'), 6),\n",
       " (('6603277503', '6603277503'), 6),\n",
       " (('57204136382', '57218685959'), 5),\n",
       " (('57220337993', '57218685959'), 5),\n",
       " (('55353786600', '57218685959'), 5),\n",
       " (('7003484514', '57218685959'), 5),\n",
       " (('57208092544', '57218685959'), 5),\n",
       " (('57205732943', '57218685959'), 5),\n",
       " (('57207776493', '57218685959'), 5),\n",
       " (('57072675600', '57218685959'), 5),\n",
       " (('56666893900', '57218685959'), 5),\n",
       " (('57201795147', '57218685959'), 5),\n",
       " (('57211973004', '57218685959'), 5),\n",
       " (('57291479000', '57218685959'), 5),\n",
       " (('6701766775', '57218685959'), 5),\n",
       " (('6602856691', '57218685959'), 5),\n",
       " (('36677599700', '57218685959'), 5),\n",
       " (('57204885457', '57218685959'), 5),\n",
       " (('57221264054', '57218685959'), 5),\n",
       " (('57197842652', '57218685959'), 5),\n",
       " (('56108103800', '57218685959'), 5),\n",
       " (('56340274300', '57218685959'), 5),\n",
       " (('55950033500', '57218685959'), 5),\n",
       " (('57190986676', '57218685959'), 5),\n",
       " (('7005443464', '57218685959'), 5),\n",
       " (('34067509400', '57218685959'), 5),\n",
       " (('36105914300', '57218685959'), 5),\n",
       " (('57220193926', '57218685959'), 5),\n",
       " (('57220145221', '57218685959'), 5),\n",
       " (('57223014106', '57218685959'), 5),\n",
       " (('57204776552', '57218685959'), 5),\n",
       " (('57190381886', '57218685959'), 5),\n",
       " (('57211789616', '57218685959'), 5),\n",
       " (('57221350738', '57218685959'), 5),\n",
       " (('56090863600', '57218685959'), 5),\n",
       " (('56784694900', '57218685959'), 5),\n",
       " (('57216598454', '57218685959'), 5),\n",
       " (('55453902300', '57218685959'), 5),\n",
       " (('25824674400', '57218685959'), 5),\n",
       " (('56664953900', '57218685959'), 5),\n",
       " (('36189115500', '57218685959'), 5),\n",
       " (('36061883800', '57218685959'), 5),\n",
       " (('57204537523', '57218685959'), 5),\n",
       " (('25226022600', '57218685959'), 5),\n",
       " (('7005382935', '57218685959'), 5),\n",
       " (('55336592400', '57218685959'), 5),\n",
       " (('55589471700', '57218685959'), 5),\n",
       " (('57221712310', '57218685959'), 5),\n",
       " (('57454325300', '57218685959'), 5),\n",
       " (('57454349100', '57218685959'), 5),\n",
       " (('57221713671', '57218685959'), 5),\n",
       " (('57226273740', '57218685959'), 5),\n",
       " (('35759726100', '57218685959'), 5),\n",
       " (('6603786839', '57218685959'), 5),\n",
       " (('45261113000', '57218685959'), 5),\n",
       " (('57220076751', '57218685959'), 5),\n",
       " (('7005532738', '57218685959'), 5),\n",
       " (('57210732042', '57218685959'), 5),\n",
       " (('57218685959', '23993684800'), 5),\n",
       " (('57218685959', '55823146800'), 5),\n",
       " (('57218685959', '57224306958'), 5),\n",
       " (('57218685959', '57225855031'), 5),\n",
       " (('57218685959', '55473967300'), 5),\n",
       " (('57218685959', '56650735200'), 5),\n",
       " (('57218685959', '57224307376'), 5),\n",
       " (('57218685959', '57224320540'), 5),\n",
       " (('57218685959', '57222642515'), 5),\n",
       " (('57218685959', '57224313768'), 5),\n",
       " (('57218685959', '15765262300'), 5),\n",
       " (('57218685959', '57224305584'), 5),\n",
       " (('57218685959', '56668503400'), 5),\n",
       " (('57218685959', '57224309917'), 5),\n",
       " (('57218685959', '57222642462'), 5),\n",
       " (('57218685959', '7102018381'), 5),\n",
       " (('57218685959', '57224307043'), 5),\n",
       " (('57218685959', '57224319984'), 5),\n",
       " (('57218685959', '57224312528'), 5),\n",
       " (('57218685959', '57208704844'), 5),\n",
       " (('57218685959', '57224307478'), 5),\n",
       " (('57218685959', '57200121679'), 5),\n",
       " (('57218685959', '57224309775'), 5),\n",
       " (('57218685959', '57224312601'), 5),\n",
       " (('57218685959', '7201572443'), 5),\n",
       " (('57218685959', '55904364300'), 5),\n",
       " (('57218685959', '16186296500'), 5),\n",
       " (('57218685959', '57224316445'), 5),\n",
       " (('57218685959', '57220858766'), 5),\n",
       " (('57218685959', '57213590808'), 5),\n",
       " (('57218685959', '57224308997'), 5),\n",
       " (('57218685959', '57225444763'), 5),\n",
       " (('57218685959', '57204240380'), 5),\n",
       " (('57218685959', '57454357600'), 5),\n",
       " (('57218685959', '57224307128'), 5),\n",
       " (('57218685959', '57454341700'), 5),\n",
       " (('57218685959', '57205352866'), 5),\n",
       " (('57218685959', '57454364000'), 5),\n",
       " (('57218685959', '57454348400'), 5),\n",
       " (('57218685959', '56704818900'), 5),\n",
       " (('57218685959', '57221209369'), 5),\n",
       " (('57218685959', '57454364100'), 5),\n",
       " (('57218685959', '57221253997'), 5),\n",
       " (('57218685959', '57216069504'), 5),\n",
       " (('57218685959', '57202361365'), 5),\n",
       " (('57218685959', '57547258300'), 5),\n",
       " (('57218685959', '57351183700'), 5),\n",
       " (('57218685959', '55227463900'), 5),\n",
       " (('57218685959', '57193208558'), 5),\n",
       " (('57218685959', '57208599784'), 5),\n",
       " (('6603277503', '7003560327'), 5),\n",
       " (('7801624038', '6603277503'), 5),\n",
       " (('56429644500', '6603277503'), 5),\n",
       " (('57367028700', '6603277503'), 5),\n",
       " (('6506014076', '6603277503'), 5),\n",
       " (('6505495562', '6603277503'), 5),\n",
       " (('23993684800', '57218685959'), 5),\n",
       " (('55823146800', '57218685959'), 5),\n",
       " (('57204136382', '6603277503'), 4),\n",
       " (('57220337993', '6603277503'), 4),\n",
       " (('55353786600', '6603277503'), 4),\n",
       " (('7003484514', '6603277503'), 4),\n",
       " (('57208092544', '6603277503'), 4),\n",
       " (('57205732943', '6603277503'), 4),\n",
       " (('57207776493', '6603277503'), 4),\n",
       " (('57072675600', '6603277503'), 4),\n",
       " (('56666893900', '6603277503'), 4),\n",
       " (('57201795147', '6603277503'), 4),\n",
       " (('57211973004', '6603277503'), 4),\n",
       " (('57291479000', '6603277503'), 4),\n",
       " (('6701766775', '6603277503'), 4),\n",
       " (('6602856691', '6603277503'), 4),\n",
       " (('36677599700', '6603277503'), 4),\n",
       " (('57204885457', '6603277503'), 4),\n",
       " (('57221264054', '6603277503'), 4),\n",
       " (('57197842652', '6603277503'), 4),\n",
       " (('56108103800', '6603277503'), 4),\n",
       " (('35957366900', '57454365000'), 4),\n",
       " (('35957366900', '6603277503'), 4),\n",
       " (('35957366900', '23993684800'), 4),\n",
       " (('35957366900', '55823146800'), 4),\n",
       " (('35957366900', '57221929857'), 4),\n",
       " (('35957366900', '57220962396'), 4),\n",
       " (('35957366900', '57212263289'), 4),\n",
       " (('56340274300', '6603277503'), 4),\n",
       " (('55950033500', '6603277503'), 4),\n",
       " (('57190986676', '6603277503'), 4),\n",
       " (('7005443464', '6603277503'), 4),\n",
       " (('34067509400', '6603277503'), 4),\n",
       " (('36105914300', '6603277503'), 4),\n",
       " (('57220193926', '6603277503'), 4),\n",
       " (('57220145221', '6603277503'), 4),\n",
       " (('57223014106', '6603277503'), 4),\n",
       " (('57204776552', '6603277503'), 4),\n",
       " (('57190381886', '6603277503'), 4),\n",
       " (('57211789616', '6603277503'), 4),\n",
       " (('57221350738', '6603277503'), 4),\n",
       " (('56090863600', '6603277503'), 4),\n",
       " (('56784694900', '6603277503'), 4),\n",
       " (('57216598454', '6603277503'), 4),\n",
       " (('55453902300', '6603277503'), 4),\n",
       " (('25824674400', '6603277503'), 4),\n",
       " (('56664953900', '6603277503'), 4),\n",
       " (('36189115500', '6603277503'), 4),\n",
       " (('36061883800', '6603277503'), 4),\n",
       " (('57204537523', '6603277503'), 4),\n",
       " (('25226022600', '6603277503'), 4),\n",
       " (('7005382935', '6603277503'), 4),\n",
       " (('55336592400', '6603277503'), 4),\n",
       " (('55589471700', '6603277503'), 4),\n",
       " (('57221712310', '6603277503'), 4),\n",
       " (('57454325300', '6603277503'), 4),\n",
       " (('57203425292', '57454365000'), 4),\n",
       " (('57203425292', '57230977500'), 4),\n",
       " (('57203425292', '57031969300'), 4),\n",
       " (('57203425292', '57192180234'), 4),\n",
       " (('57203425292', '7801624038'), 4),\n",
       " (('57203425292', '56429644500'), 4),\n",
       " (('57203425292', '57367028700'), 4),\n",
       " (('57203425292', '6506014076'), 4),\n",
       " (('57203425292', '6505495562'), 4),\n",
       " (('57203425292', '7003560327'), 4),\n",
       " (('57203425292', '55228809500'), 4),\n",
       " (('57203425292', '23993684800'), 4),\n",
       " (('57203425292', '55823146800'), 4),\n",
       " (('57203425292', '57221929857'), 4),\n",
       " (('57203425292', '57220962396'), 4),\n",
       " (('57203425292', '57212263289'), 4),\n",
       " (('57454349100', '6603277503'), 4),\n",
       " (('57221713671', '6603277503'), 4),\n",
       " (('57226273740', '6603277503'), 4),\n",
       " (('35759726100', '6603277503'), 4),\n",
       " (('6603786839', '6603277503'), 4),\n",
       " (('45261113000', '6603277503'), 4),\n",
       " (('57454365000', '57230977500'), 4),\n",
       " (('57454365000', '57031969300'), 4),\n",
       " (('57454365000', '57192180234'), 4),\n",
       " (('57454365000', '7801624038'), 4),\n",
       " (('57454365000', '56429644500'), 4),\n",
       " (('57454365000', '57367028700'), 4),\n",
       " (('57454365000', '6506014076'), 4),\n",
       " (('57454365000', '6505495562'), 4),\n",
       " (('57454365000', '7003560327'), 4),\n",
       " (('57454365000', '55228809500'), 4),\n",
       " (('57220076751', '6603277503'), 4),\n",
       " (('7005532738', '6603277503'), 4),\n",
       " (('57210732042', '6603277503'), 4),\n",
       " (('57218685959', '6603277503'), 4),\n",
       " (('57218685959', '57224316748'), 4),\n",
       " (('57218685959', '57224311741'), 4),\n",
       " (('57218685959', '57205621869'), 4),\n",
       " (('57218685959', '57204240238'), 4),\n",
       " (('57218685959', '57224307892'), 4),\n",
       " (('57218685959', '44161222100'), 4),\n",
       " (('57218685959', '57454357200'), 4),\n",
       " (('57218685959', '57221416626'), 4),\n",
       " (('57218685959', '57454363600'), 4),\n",
       " (('57218685959', '57221940479'), 4),\n",
       " (('57218685959', '57454372200'), 4),\n",
       " (('57218685959', '57221935454'), 4),\n",
       " (('57218685959', '57224310644'), 4),\n",
       " (('57218685959', '57194603582'), 4),\n",
       " (('57218685959', '57454348200'), 4),\n",
       " (('57218685959', '57221927083'), 4),\n",
       " (('57218685959', '57221939491'), 4),\n",
       " (('57218685959', '57224319598'), 4),\n",
       " (('57218685959', '56763327800'), 4),\n",
       " (('57218685959', '57454363700'), 4),\n",
       " (('57218685959', '57326649000'), 4),\n",
       " (('57218685959', '57224317376'), 4),\n",
       " (('57218685959', '57222596560'), 4),\n",
       " (('57218685959', '57224322464'), 4),\n",
       " (('57218685959', '57224308974'), 4),\n",
       " (('57218685959', '57224307084'), 4),\n",
       " (('57218685959', '57948444700'), 4),\n",
       " (('57218685959', '57454317200'), 4),\n",
       " (('57218685959', '57221935593'), 4),\n",
       " (('57218685959', '56351920200'), 4),\n",
       " (('57218685959', '7201374651'), 4),\n",
       " (('57218685959', '57224309440'), 4),\n",
       " (('57218685959', '57221941212'), 4),\n",
       " (('57218685959', '57221939712'), 4),\n",
       " (('57218685959', '57221943323'), 4),\n",
       " (('57218685959', '57224319248'), 4),\n",
       " (('57218685959', '55184950200'), 4),\n",
       " (('57218685959', '56298923400'), 4),\n",
       " (('57218685959', '57224761351'), 4),\n",
       " (('57218685959', '57224322714'), 4),\n",
       " (('57218685959', '57224321158'), 4),\n",
       " (('57218685959', '57224310546'), 4),\n",
       " (('57218685959', '57224308050'), 4),\n",
       " (('57218685959', '57224313099'), 4),\n",
       " (('57218685959', '57225956523'), 4),\n",
       " (('57218685959', '57454372300'), 4),\n",
       " (('57218685959', '57454357300'), 4),\n",
       " (('57218685959', '57217733811'), 4),\n",
       " (('57218685959', '57195545099'), 4),\n",
       " (('57218685959', '57212452966'), 4),\n",
       " (('57218685959', '57224321693'), 4),\n",
       " (('57218685959', '57214309916'), 4),\n",
       " (('57218685959', '57219092982'), 4),\n",
       " (('57218685959', '57454341600'), 4),\n",
       " (('57218685959', '57454372400'), 4),\n",
       " (('57218685959', '7006633386'), 4),\n",
       " (('57218685959', '57204682023'), 4),\n",
       " (('57218685959', '57224318259'), 4),\n",
       " (('57218685959', '6506673644'), 4),\n",
       " (('57218685959', '26025689900'), 4),\n",
       " (('57218685959', '16424634400'), 4),\n",
       " (('57218685959', '37025910000'), 4),\n",
       " (('57218685959', '57190735159'), 4),\n",
       " (('57218685959', '57224316878'), 4),\n",
       " (('57218685959', '55547122269'), 4),\n",
       " (('57218685959', '36708778900'), 4),\n",
       " (('57218685959', '57454372500'), 4),\n",
       " (('57218685959', '57454325000'), 4),\n",
       " (('57218685959', '57218442259'), 4),\n",
       " (('57218685959', '57204066310'), 4),\n",
       " (('57218685959', '57225962216'), 4),\n",
       " (('57218685959', '57224649260'), 4),\n",
       " (('57218685959', '57224306954'), 4),\n",
       " (('57218685959', '57224312648'), 4),\n",
       " (('57218685959', '57222599210'), 4),\n",
       " (('57218685959', '57221923828'), 4),\n",
       " (('57218685959', '57221931849'), 4),\n",
       " (('57218685959', '57196237024'), 4),\n",
       " (('57218685959', '57222600957'), 4),\n",
       " (('57218685959', '57234253200'), 4),\n",
       " (('57218685959', '57224311472'), 4),\n",
       " (('57218685959', '56607917400'), 4),\n",
       " (('57218685959', '57224315530'), 4),\n",
       " (('57218685959', '57219345101'), 4),\n",
       " (('57218685959', '57221943143'), 4),\n",
       " (('57218685959', '57221929334'), 4),\n",
       " (('57218685959', '57221935942'), 4),\n",
       " (('57218685959', '57221921315'), 4),\n",
       " (('57218685959', '57701086200'), 4),\n",
       " (('57218685959', '57221924311'), 4),\n",
       " (('57218685959', '57224316621'), 4),\n",
       " (('57218685959', '57454357400'), 4),\n",
       " (('57218685959', '57221936217'), 4),\n",
       " (('57218685959', '57531924100'), 4),\n",
       " (('57218685959', '35772489500'), 4),\n",
       " (('57218685959', '57211962880'), 4),\n",
       " (('57218685959', '57454363800'), 4),\n",
       " (('57218685959', '57208378560'), 4),\n",
       " (('57218685959', '57200334643'), 4),\n",
       " (('57218685959', '57204679525'), 4),\n",
       " (('57218685959', '57201332459'), 4),\n",
       " (('57218685959', '7403325890'), 4),\n",
       " (('57218685959', '57221924892'), 4),\n",
       " (('57218685959', '57221925368'), 4),\n",
       " (('57218685959', '57215025692'), 4),\n",
       " (('57218685959', '57224316572'), 4),\n",
       " (('57218685959', '55988399700'), 4),\n",
       " (('57218685959', '57221944793'), 4),\n",
       " (('57218685959', '57224310970'), 4),\n",
       " (('57218685959', '23061421500'), 4),\n",
       " (('57218685959', '57221940477'), 4),\n",
       " (('57218685959', '57193952047'), 4),\n",
       " (('57218685959', '57454325100'), 4),\n",
       " (('57218685959', '57221931421'), 4),\n",
       " (('57218685959', '57201077157'), 4),\n",
       " (('57218685959', '57454325800'), 4),\n",
       " (('57218685959', '57205487643'), 4),\n",
       " (('57218685959', '57221939489'), 4),\n",
       " (('57218685959', '57221934164'), 4),\n",
       " (('57218685959', '23972166600'), 4),\n",
       " (('57218685959', '57221248068'), 4),\n",
       " (('57218685959', '7005440974'), 4),\n",
       " (('57218685959', '55962909800'), 4),\n",
       " (('57218685959', '55046270800'), 4),\n",
       " (('57218685959', '57199371710'), 4),\n",
       " (('57218685959', '56835316200'), 4),\n",
       " (('57218685959', '36165389000'), 4),\n",
       " (('57218685959', '57792861900'), 4),\n",
       " (('57218685959', '57191520077'), 4),\n",
       " (('57218685959', '7402358071'), 4),\n",
       " (('57218685959', '57221771679'), 4),\n",
       " (('57218685959', '7004565877'), 4),\n",
       " (('57218685959', '8836379300'), 4),\n",
       " (('57218685959', '7102023740'), 4),\n",
       " (('57218685959', '57218330005'), 4),\n",
       " (('57218685959', '57193662019'), 4),\n",
       " (('57218685959', '57226178712'), 4),\n",
       " (('57218685959', '57195363908'), 4),\n",
       " (('57218685959', '6603800293'), 4),\n",
       " (('57218685959', '57216823853'), 4),\n",
       " (('57218685959', '6602667380'), 4),\n",
       " (('57218685959', '35601901900'), 4),\n",
       " (('57218685959', '7004230842'), 4),\n",
       " (('57218685959', '7402818741'), 4),\n",
       " (('57218685959', '13611914400'), 4),\n",
       " (('57218685959', '7407335491'), 4),\n",
       " (('57218685959', '24334850600'), 4),\n",
       " (('57218685959', '57191265552'), 4),\n",
       " (('57218685959', '57205617327'), 4),\n",
       " (('57218685959', '6701743013'), 4),\n",
       " (('57218685959', '56377639700'), 4),\n",
       " (('57218685959', '56945667500'), 4),\n",
       " (('57218685959', '57793693600'), 4),\n",
       " (('57218685959', '57213587752'), 4),\n",
       " (('57218685959', '57195496459'), 4),\n",
       " (('57218685959', '56645308000'), 4),\n",
       " (('57218685959', '57211975397'), 4),\n",
       " (('57218685959', '43861903000'), 4),\n",
       " (('57218685959', '7006818684'), 4),\n",
       " (('57218685959', '57203177203'), 4),\n",
       " (('57218685959', '35473647700'), 4),\n",
       " (('57218685959', '6603799300'), 4),\n",
       " (('57218685959', '14033895500'), 4),\n",
       " (('57218685959', '6506082906'), 4),\n",
       " (('57218685959', '57226006976'), 4),\n",
       " (('57218685959', '57203899453'), 4),\n",
       " (('57218685959', '55661699700'), 4),\n",
       " (('57218685959', '57211512404'), 4),\n",
       " (('57218685959', '57193850535'), 4),\n",
       " (('57218685959', '36171971800'), 4),\n",
       " (('57218685959', '55339072600'), 4),\n",
       " (('57218685959', '55355157200'), 4),\n",
       " (('57218685959', '6603621553'), 4),\n",
       " (('57218685959', '57221251125'), 4),\n",
       " (('57218685959', '57221249089'), 4),\n",
       " (('57218685959', '57221250184'), 4),\n",
       " (('57218685959', '57221253988'), 4),\n",
       " (('57218685959', '57214329919'), 4),\n",
       " (('57218685959', '57221252937'), 4),\n",
       " (('57218685959', '57221253782'), 4),\n",
       " (('57218685959', '57221253227'), 4),\n",
       " (('57218685959', '54388530400'), 4),\n",
       " (('57218685959', '57221248970'), 4),\n",
       " (('57218685959', '57221251792'), 4),\n",
       " (('57218685959', '57191854216'), 4),\n",
       " (('57218685959', '57221318803'), 4),\n",
       " (('57218685959', '57221252274'), 4),\n",
       " (('57218685959', '7202634251'), 4),\n",
       " (('57218685959', '57326625400'), 4),\n",
       " (('57218685959', '57221251020'), 4),\n",
       " (('57218685959', '57221248644'), 4),\n",
       " (('57218685959', '57221249084'), 4),\n",
       " (('57218685959', '20734200200'), 4),\n",
       " (('57218685959', '57221250535'), 4),\n",
       " (('57218685959', '57221338357'), 4),\n",
       " (('57218685959', '57221252224'), 4),\n",
       " (('57218685959', '57205602191'), 4),\n",
       " (('57218685959', '56407143600'), 4),\n",
       " (('57218685959', '57221252623'), 4),\n",
       " (('57218685959', '35603377500'), 4),\n",
       " (('57218685959', '57221254353'), 4),\n",
       " (('57218685959', '57948466300'), 4),\n",
       " (('57218685959', '57221251034'), 4),\n",
       " (('57218685959', '57204637245'), 4),\n",
       " (('57218685959', '57198419641'), 4),\n",
       " (('57218685959', '57217336544'), 4),\n",
       " (('57218685959', '57214260594'), 4),\n",
       " (('57218685959', '8618158600'), 4),\n",
       " (('57218685959', '56433701200'), 4),\n",
       " (('57218685959', '8761490000'), 4),\n",
       " (('57218685959', '57221249095'), 4),\n",
       " (('57218685959', '57219724359'), 4),\n",
       " (('57218685959', '57221247140'), 4),\n",
       " (('57218685959', '57221250822'), 4),\n",
       " (('57218685959', '56655966000'), 4),\n",
       " (('57218685959', '57191474359'), 4),\n",
       " (('57218685959', '57200000054'), 4),\n",
       " (('57218685959', '57207760377'), 4),\n",
       " (('57218685959', '57221048034'), 4),\n",
       " (('57218685959', '57221253645'), 4),\n",
       " (('57218685959', '57221248431'), 4),\n",
       " (('57218685959', '57218761773'), 4),\n",
       " (('57218685959', '57221253464'), 4),\n",
       " (('57218685959', '57219516856'), 4),\n",
       " (('57218685959', '55744709300'), 4),\n",
       " (('57218685959', '57215519875'), 4),\n",
       " (('57218685959', '57221250940'), 4),\n",
       " (('57218685959', '57221555947'), 4),\n",
       " (('57218685959', '57221250927'), 4),\n",
       " (('57218685959', '7003372354'), 4),\n",
       " (('57218685959', '57548648800'), 4),\n",
       " (('57218685959', '57454357500'), 4),\n",
       " (('57218685959', '57216321826'), 4),\n",
       " (('57218685959', '57357154500'), 4),\n",
       " (('57218685959', '57221251287'), 4),\n",
       " (('57218685959', '6507295430'), 4),\n",
       " (('57218685959', '57221252312'), 4),\n",
       " (('57218685959', '57194337940'), 4),\n",
       " (('57218685959', '57202008716'), 4),\n",
       " (('57218685959', '57221247892'), 4),\n",
       " (('57218685959', '57221563540'), 4),\n",
       " (('57218685959', '57223605335'), 4),\n",
       " (('57218685959', '57223609783'), 4),\n",
       " (('57218685959', '57223637634'), 4),\n",
       " (('57218685959', '57223628343'), 4),\n",
       " (('57218685959', '57223094973'), 4),\n",
       " (('57218685959', '57223638329'), 4),\n",
       " (('57218685959', '57223620978'), 4),\n",
       " (('57218685959', '23399022300'), 4),\n",
       " (('57218685959', '57222642755'), 4),\n",
       " (('57218685959', '56489487900'), 4),\n",
       " (('7005718665', '6603277503'), 4),\n",
       " (('7005718665', '57218685959'), 4),\n",
       " (('57195587336', '6603277503'), 4),\n",
       " (('57195587336', '57218685959'), 4),\n",
       " (('7006299676', '6603277503'), 4),\n",
       " (('7006299676', '57218685959'), 4),\n",
       " (('55537653100', '6603277503'), 4),\n",
       " (('55537653100', '57218685959'), 4),\n",
       " (('57190031117', '6603277503'), 4),\n",
       " (('57190031117', '57218685959'), 4),\n",
       " (('57216530713', '57230977500'), 4),\n",
       " (('57216530713', '57031969300'), 4),\n",
       " (('57216530713', '57192180234'), 4),\n",
       " (('57216530713', '7801624038'), 4),\n",
       " (('57216530713', '56429644500'), 4),\n",
       " (('57216530713', '57367028700'), 4),\n",
       " (('57216530713', '6506014076'), 4),\n",
       " (('57216530713', '6505495562'), 4),\n",
       " (('57216530713', '7003560327'), 4),\n",
       " (('57216530713', '55228809500'), 4),\n",
       " (('57216530713', '23993684800'), 4),\n",
       " (('57216530713', '55823146800'), 4),\n",
       " (('57216530713', '57221929857'), 4),\n",
       " (('57216530713', '57220962396'), 4),\n",
       " (('57216530713', '57212263289'), 4),\n",
       " (('50461239100', '6603277503'), 4),\n",
       " (('50461239100', '57218685959'), 4),\n",
       " (('53981988900', '6603277503'), 4),\n",
       " (('53981988900', '57218685959'), 4),\n",
       " (('57454349200', '6603277503'), 4),\n",
       " (('57454349200', '57218685959'), 4),\n",
       " (('7004802904', '6603277503'), 4),\n",
       " (('7004802904', '57218685959'), 4),\n",
       " (('7403322256', '6603277503'), 4),\n",
       " (('7403322256', '57218685959'), 4),\n",
       " (('7003464019', '6603277503'), 4),\n",
       " (('7003464019', '57218685959'), 4),\n",
       " (('6603608675', '6603277503'), 4),\n",
       " (('6603608675', '57218685959'), 4),\n",
       " (('35312284500', '6603277503'), 4),\n",
       " (('35312284500', '57218685959'), 4),\n",
       " (('57199371681', '6603277503'), 4),\n",
       " (('57199371681', '57218685959'), 4),\n",
       " (('56644869700', '6603277503'), 4),\n",
       " (('56644869700', '57218685959'), 4),\n",
       " (('23393099600', '6603277503'), 4),\n",
       " (('23393099600', '57218685959'), 4),\n",
       " (('6603228941', '6603277503'), 4),\n",
       " (('6603228941', '57218685959'), 4),\n",
       " (('57220380086', '6603277503'), 4),\n",
       " (('57220380086', '57218685959'), 4),\n",
       " (('57218303684', '6603277503'), 4),\n",
       " (('57218303684', '57218685959'), 4),\n",
       " (('57221489682', '6603277503'), 4),\n",
       " (('57221489682', '57218685959'), 4),\n",
       " (('36801313100', '6603277503'), 4),\n",
       " (('36801313100', '57218685959'), 4),\n",
       " (('7003618645', '6603277503'), 4),\n",
       " (('7003618645', '57218685959'), 4),\n",
       " (('57454343100', '6603277503'), 4),\n",
       " (('57454343100', '57218685959'), 4),\n",
       " (('30467529300', '6603277503'), 4),\n",
       " (('30467529300', '57218685959'), 4),\n",
       " (('24437405900', '6603277503'), 4),\n",
       " (('24437405900', '57218685959'), 4),\n",
       " (('57201953360', '6603277503'), 4),\n",
       " (('57201953360', '57218685959'), 4),\n",
       " (('57221722501', '6603277503'), 4),\n",
       " (('57221722501', '57218685959'), 4),\n",
       " (('37002835000', '6603277503'), 4),\n",
       " (('37002835000', '57218685959'), 4),\n",
       " (('57454326300', '6603277503'), 4),\n",
       " (('57454326300', '57218685959'), 4),\n",
       " (('13003009100', '6603277503'), 4),\n",
       " (('13003009100', '57218685959'), 4),\n",
       " (('57454343200', '6603277503'), 4),\n",
       " (('57454343200', '57218685959'), 4),\n",
       " (('57193001593', '6603277503'), 4),\n",
       " (('57193001593', '57218685959'), 4),\n",
       " (('24779430500', '6603277503'), 4),\n",
       " (('24779430500', '57218685959'), 4),\n",
       " (('54896345200', '6603277503'), 4),\n",
       " (('54896345200', '57218685959'), 4),\n",
       " (('57204348436', '6603277503'), 4),\n",
       " (('57204348436', '57218685959'), 4),\n",
       " (('57204537848', '6603277503'), 4),\n",
       " (('57204537848', '57218685959'), 4),\n",
       " (('57454326400', '6603277503'), 4),\n",
       " (('57454326400', '57218685959'), 4),\n",
       " (('55927746700', '6603277503'), 4),\n",
       " (('55927746700', '57218685959'), 4),\n",
       " (('57193706860', '6603277503'), 4),\n",
       " (('57193706860', '57218685959'), 4),\n",
       " (('57193552968', '6603277503'), 4),\n",
       " (('57193552968', '57218685959'), 4),\n",
       " (('23388903100', '6603277503'), 4),\n",
       " (('23388903100', '57218685959'), 4),\n",
       " (('55617927600', '6603277503'), 4),\n",
       " (('55617927600', '57218685959'), 4),\n",
       " (('57191347043', '6603277503'), 4),\n",
       " (('57191347043', '57218685959'), 4),\n",
       " (('6506017468', '6603277503'), 4),\n",
       " (('6506017468', '57218685959'), 4),\n",
       " (('57700754100', '6603277503'), 4),\n",
       " (('57700754100', '57218685959'), 4),\n",
       " (('25655615300', '6603277503'), 4),\n",
       " (('25655615300', '57218685959'), 4),\n",
       " (('57454365600', '6603277503'), 4),\n",
       " (('57454365600', '57218685959'), 4),\n",
       " (('57191514162', '6603277503'), 4),\n",
       " (('57191514162', '57218685959'), 4),\n",
       " (('24402710900', '6603277503'), 4),\n",
       " (('24402710900', '57218685959'), 4),\n",
       " (('55775358100', '6603277503'), 4),\n",
       " (('55775358100', '57218685959'), 4),\n",
       " (('57538664800', '6603277503'), 4),\n",
       " (('57538664800', '57218685959'), 4),\n",
       " (('57208166236', '6603277503'), 4),\n",
       " (('57208166236', '57218685959'), 4),\n",
       " (('57216952230', '6603277503'), 4),\n",
       " (('57216952230', '57218685959'), 4),\n",
       " (('57201720900', '6603277503'), 4),\n",
       " (('57201720900', '57218685959'), 4),\n",
       " (('57454343500', '6603277503'), 4),\n",
       " (('57454343500', '57218685959'), 4),\n",
       " (('57454343600', '6603277503'), 4),\n",
       " (('57454343600', '57218685959'), 4),\n",
       " (('57454309900', '6603277503'), 4),\n",
       " (('57454309900', '57218685959'), 4),\n",
       " (('57454334800', '6603277503'), 4),\n",
       " (('57454334800', '57218685959'), 4),\n",
       " (('57454358700', '6603277503'), 4),\n",
       " (('57454358700', '57218685959'), 4),\n",
       " (('57454326500', '6603277503'), 4),\n",
       " (('57454326500', '57218685959'), 4),\n",
       " (('57454365700', '6603277503'), 4),\n",
       " (('57454365700', '57218685959'), 4),\n",
       " (('55440387300', '6603277503'), 4),\n",
       " (('55440387300', '57218685959'), 4),\n",
       " (('36100958600', '6603277503'), 4),\n",
       " (('36100958600', '57218685959'), 4),\n",
       " (('57191853432', '6603277503'), 4),\n",
       " (('57191853432', '57218685959'), 4),\n",
       " (('57194519231', '6603277503'), 4),\n",
       " (('57194519231', '57218685959'), 4),\n",
       " (('55553630200', '6603277503'), 4),\n",
       " (('55553630200', '57218685959'), 4),\n",
       " (('55853983400', '6603277503'), 4),\n",
       " (('55853983400', '57218685959'), 4),\n",
       " (('57454365800', '6603277503'), 4),\n",
       " (('57454365800', '57218685959'), 4),\n",
       " (('36917209100', '6603277503'), 4),\n",
       " (('36917209100', '57218685959'), 4),\n",
       " (('57215505815', '6603277503'), 4),\n",
       " (('57215505815', '57218685959'), 4),\n",
       " (('57218259417', '6603277503'), 4),\n",
       " (('57218259417', '57218685959'), 4),\n",
       " (('57454365900', '6603277503'), 4),\n",
       " (('57454365900', '57218685959'), 4),\n",
       " (('57454310000', '6603277503'), 4),\n",
       " (('57454310000', '57218685959'), 4),\n",
       " (('57211300397', '6603277503'), 4),\n",
       " (('57211300397', '57218685959'), 4),\n",
       " (('7004954554', '6603277503'), 4),\n",
       " (('7004954554', '57218685959'), 4),\n",
       " (('18233406900', '6603277503'), 4),\n",
       " (('18233406900', '57218685959'), 4),\n",
       " (('57195237529', '6603277503'), 4),\n",
       " (('57195237529', '57218685959'), 4),\n",
       " (('57215504058', '6603277503'), 4),\n",
       " (('57215504058', '57218685959'), 4),\n",
       " (('57224309545', '6603277503'), 4),\n",
       " (('57224309545', '57218685959'), 4),\n",
       " (('57221411902', '6603277503'), 4),\n",
       " (('57221411902', '57218685959'), 4),\n",
       " (('56912021400', '6603277503'), 4),\n",
       " (('56912021400', '57218685959'), 4),\n",
       " (('6505777250', '6603277503'), 4),\n",
       " (('6505777250', '57218685959'), 4),\n",
       " (('57212854133', '6603277503'), 4),\n",
       " (('57212854133', '57218685959'), 4),\n",
       " (('57204137146', '6603277503'), 4),\n",
       " (('57204137146', '57218685959'), 4),\n",
       " (('57221416670', '6603277503'), 4),\n",
       " (('57221416670', '57218685959'), 4),\n",
       " (('57219828345', '6603277503'), 4),\n",
       " (('57219828345', '57218685959'), 4),\n",
       " (('7005775097', '6603277503'), 4),\n",
       " (('7005775097', '57218685959'), 4),\n",
       " (('7101952271', '6603277503'), 4),\n",
       " (('7101952271', '57218685959'), 4),\n",
       " (('55768247700', '6603277503'), 4),\n",
       " (('55768247700', '57218685959'), 4),\n",
       " (('37034619800', '6603277503'), 4),\n",
       " (('37034619800', '57218685959'), 4),\n",
       " (('7004705151', '6603277503'), 4),\n",
       " (('7004705151', '57218685959'), 4),\n",
       " (('57194256712', '6603277503'), 4),\n",
       " (('57194256712', '57218685959'), 4),\n",
       " (('57454349400', '6603277503'), 4),\n",
       " (('57454349400', '57218685959'), 4),\n",
       " (('57410400100', '6603277503'), 4),\n",
       " (('57410400100', '57218685959'), 4),\n",
       " (('57204398387', '6603277503'), 4),\n",
       " (('57204398387', '57218685959'), 4),\n",
       " (('57454349500', '6603277503'), 4),\n",
       " (('57454349500', '57218685959'), 4),\n",
       " (('57454349600', '6603277503'), 4),\n",
       " (('57454349600', '57218685959'), 4),\n",
       " (('57216343773', '6603277503'), 4),\n",
       " (('57216343773', '57218685959'), 4),\n",
       " (('57454343700', '6603277503'), 4),\n",
       " (('57454343700', '57218685959'), 4),\n",
       " (('57193420762', '6603277503'), 4),\n",
       " (('57193420762', '57218685959'), 4),\n",
       " (('57454334900', '6603277503'), 4),\n",
       " (('57454334900', '57218685959'), 4),\n",
       " (('57454349700', '6603277503'), 4),\n",
       " (('57454349700', '57218685959'), 4),\n",
       " (('7201361573', '6603277503'), 4),\n",
       " (('7201361573', '57218685959'), 4),\n",
       " (('23027393200', '6603277503'), 4),\n",
       " (('23027393200', '57218685959'), 4),\n",
       " (('55872745900', '6603277503'), 4),\n",
       " (('55872745900', '57218685959'), 4),\n",
       " (('15055468800', '6603277503'), 4),\n",
       " (('15055468800', '57218685959'), 4),\n",
       " (('22936120600', '6603277503'), 4),\n",
       " (('22936120600', '57218685959'), 4),\n",
       " (('26432674600', '6603277503'), 4),\n",
       " (('26432674600', '57218685959'), 4),\n",
       " (('57211973500', '6603277503'), 4),\n",
       " (('57211973500', '57218685959'), 4),\n",
       " (('57221805335', '6603277503'), 4),\n",
       " (('57221805335', '57218685959'), 4),\n",
       " (('57224227345', '6603277503'), 4),\n",
       " (('57224227345', '57218685959'), 4),\n",
       " (('57218321919', '6603277503'), 4),\n",
       " (('57218321919', '57218685959'), 4),\n",
       " (('57204743861', '6603277503'), 4),\n",
       " (('57204743861', '57218685959'), 4),\n",
       " (('57224501456', '6603277503'), 4),\n",
       " (('57224501456', '57218685959'), 4),\n",
       " (('57215527494', '6603277503'), 4),\n",
       " (('57215527494', '57218685959'), 4),\n",
       " (('57220195146', '6603277503'), 4),\n",
       " (('57220195146', '57218685959'), 4),\n",
       " (('24832614900', '6603277503'), 4),\n",
       " (('24832614900', '57218685959'), 4),\n",
       " (('6603430249', '6603277503'), 4),\n",
       " (('6603430249', '57218685959'), 4),\n",
       " (('23007527000', '6603277503'), 4),\n",
       " (('23007527000', '57218685959'), 4),\n",
       " (('14070136100', '6603277503'), 4),\n",
       " (('14070136100', '57218685959'), 4),\n",
       " (('51664646500', '6603277503'), 4),\n",
       " (('51664646500', '57218685959'), 4),\n",
       " (('57220504604', '6603277503'), 4),\n",
       " (('57220504604', '57218685959'), 4),\n",
       " (('57201704774', '6603277503'), 4),\n",
       " (('57201704774', '57218685959'), 4),\n",
       " (('6507410474', '6603277503'), 4),\n",
       " (('6507410474', '57218685959'), 4),\n",
       " (('57210155996', '6603277503'), 4),\n",
       " (('57210155996', '57218685959'), 4),\n",
       " (('7004028193', '6603277503'), 4),\n",
       " (('7004028193', '57218685959'), 4),\n",
       " (('57021965300', '6603277503'), 4),\n",
       " (('57021965300', '57218685959'), 4),\n",
       " (('57220196432', '6603277503'), 4),\n",
       " (('57220196432', '57218685959'), 4),\n",
       " (('57202359946', '6603277503'), 4),\n",
       " (('57202359946', '57218685959'), 4),\n",
       " (('6508168536', '6603277503'), 4),\n",
       " (('6508168536', '57218685959'), 4),\n",
       " (('23469369200', '6603277503'), 4),\n",
       " (('23469369200', '57218685959'), 4),\n",
       " (('57208618126', '6603277503'), 4),\n",
       " (('57208618126', '57218685959'), 4),\n",
       " (('7004576211', '6603277503'), 4),\n",
       " (('7004576211', '57218685959'), 4),\n",
       " (('35279736700', '6603277503'), 4),\n",
       " (('35279736700', '57218685959'), 4),\n",
       " (('57220201855', '6603277503'), 4),\n",
       " (('57220201855', '57218685959'), 4),\n",
       " (('57204671434', '6603277503'), 4),\n",
       " (('57204671434', '57218685959'), 4),\n",
       " (('24757974900', '6603277503'), 4),\n",
       " (('24757974900', '57218685959'), 4),\n",
       " (('57536089400', '6603277503'), 4),\n",
       " (('57536089400', '57218685959'), 4),\n",
       " (('57203465782', '6603277503'), 4),\n",
       " (('57203465782', '57218685959'), 4),\n",
       " (('57220804615', '6603277503'), 4),\n",
       " (('57220804615', '57218685959'), 4),\n",
       " (('57222990193', '6603277503'), 4),\n",
       " (('57222990193', '57218685959'), 4),\n",
       " (('57190388566', '6603277503'), 4),\n",
       " (('57190388566', '57218685959'), 4),\n",
       " (('54890773600', '6603277503'), 4),\n",
       " (('54890773600', '57218685959'), 4),\n",
       " (('33967777600', '6603277503'), 4),\n",
       " (('33967777600', '57218685959'), 4),\n",
       " (('57216850822', '6603277503'), 4),\n",
       " (('57216850822', '57218685959'), 4),\n",
       " (('57454326600', '6603277503'), 4),\n",
       " (('57454326600', '57218685959'), 4),\n",
       " (('7203079884', '6603277503'), 4),\n",
       " (('7203079884', '57218685959'), 4),\n",
       " (('6506458290', '6603277503'), 4),\n",
       " (('6506458290', '57218685959'), 4),\n",
       " (('7201366838', '6603277503'), 4),\n",
       " (('7201366838', '57218685959'), 4),\n",
       " (('57219410395', '6603277503'), 4),\n",
       " (('57219410395', '57218685959'), 4),\n",
       " (('57211535641', '6603277503'), 4),\n",
       " (('57211535641', '57218685959'), 4),\n",
       " (('6602562011', '6603277503'), 4),\n",
       " (('6602562011', '57218685959'), 4),\n",
       " (('55348239500', '6603277503'), 4),\n",
       " (('55348239500', '57218685959'), 4),\n",
       " (('24281788300', '6603277503'), 4),\n",
       " (('24281788300', '57218685959'), 4),\n",
       " (('57217629913', '6603277503'), 4),\n",
       " (('57217629913', '57218685959'), 4),\n",
       " (('7101676266', '6603277503'), 4),\n",
       " (('7101676266', '57218685959'), 4),\n",
       " (('56246061200', '6603277503'), 4),\n",
       " (('56246061200', '57218685959'), 4),\n",
       " (('7102303470', '6603277503'), 4),\n",
       " (('7102303470', '57218685959'), 4),\n",
       " (('56805279000', '6603277503'), 4),\n",
       " (('56805279000', '57218685959'), 4),\n",
       " (('57209137418', '6603277503'), 4),\n",
       " (('57209137418', '57218685959'), 4),\n",
       " (('57211339218', '6603277503'), 4),\n",
       " (('57211339218', '57218685959'), 4),\n",
       " (('7202779648', '6603277503'), 4),\n",
       " (('7202779648', '57218685959'), 4),\n",
       " (('57200545121', '6603277503'), 4),\n",
       " (('57200545121', '57218685959'), 4),\n",
       " (('15849336900', '6603277503'), 4),\n",
       " (('15849336900', '57218685959'), 4),\n",
       " (('57221822883', '6603277503'), 4),\n",
       " (('57221822883', '57218685959'), 4),\n",
       " (('7801371515', '6603277503'), 4),\n",
       " (('7801371515', '57218685959'), 4),\n",
       " (('57222249457', '6603277503'), 4),\n",
       " (('57222249457', '57218685959'), 4),\n",
       " (('6701704380', '6603277503'), 4),\n",
       " (('6701704380', '57218685959'), 4),\n",
       " (('56290046700', '6603277503'), 4),\n",
       " (('56290046700', '57218685959'), 4),\n",
       " (('57219977284', '6603277503'), 4),\n",
       " (('57219977284', '57218685959'), 4),\n",
       " (('7007185669', '6603277503'), 4),\n",
       " (('7007185669', '57218685959'), 4),\n",
       " (('8664453800', '6603277503'), 4),\n",
       " (('8664453800', '57218685959'), 4),\n",
       " (('57191927861', '6603277503'), 4),\n",
       " (('57191927861', '57218685959'), 4),\n",
       " (('57469021400', '6603277503'), 4),\n",
       " (('57469021400', '57218685959'), 4),\n",
       " (('57220200167', '6603277503'), 4),\n",
       " (('57220200167', '57218685959'), 4),\n",
       " (('55403880400', '6603277503'), 4),\n",
       " (('55403880400', '57218685959'), 4),\n",
       " (('57454358800', '6603277503'), 4),\n",
       " (('57454358800', '57218685959'), 4),\n",
       " (('16445330000', '6603277503'), 4),\n",
       " (('16445330000', '57218685959'), 4),\n",
       " (('6602433268', '6603277503'), 4),\n",
       " (('6602433268', '57218685959'), 4),\n",
       " (('57205532859', '6603277503'), 4),\n",
       " (('57205532859', '57218685959'), 4),\n",
       " (('57221414360', '6603277503'), 4),\n",
       " (('57221414360', '57218685959'), 4),\n",
       " (('57194699001', '6603277503'), 4),\n",
       " (('57194699001', '57218685959'), 4),\n",
       " (('57196190877', '6603277503'), 4),\n",
       " (('57196190877', '57218685959'), 4),\n",
       " (('57527166100', '6603277503'), 4),\n",
       " (('57527166100', '57218685959'), 4),\n",
       " (('57221420775', '6603277503'), 4),\n",
       " (('57221420775', '57218685959'), 4),\n",
       " (('56423602300', '6603277503'), 4),\n",
       " (('56423602300', '57218685959'), 4),\n",
       " (('6504710212', '6603277503'), 4),\n",
       " (('6504710212', '57218685959'), 4),\n",
       " (('57221420543', '6603277503'), 4),\n",
       " (('57221420543', '57218685959'), 4),\n",
       " (('57520311500', '6603277503'), 4),\n",
       " (('57520311500', '57218685959'), 4),\n",
       " (('57221419164', '6603277503'), 4),\n",
       " (('57221419164', '57218685959'), 4),\n",
       " (('57454335200', '6603277503'), 4),\n",
       " (('57454335200', '57218685959'), 4),\n",
       " (('57222063482', '6603277503'), 4),\n",
       " (('57222063482', '57218685959'), 4),\n",
       " (('57546570200', '6603277503'), 4),\n",
       " (('57546570200', '57218685959'), 4),\n",
       " (('7102538707', '6603277503'), 4),\n",
       " (('7102538707', '57218685959'), 4),\n",
       " (('55807317400', '6603277503'), 4),\n",
       " (('55807317400', '57218685959'), 4),\n",
       " (('6602502186', '6603277503'), 4),\n",
       " (('6602502186', '57218685959'), 4),\n",
       " (('35792218800', '6603277503'), 4),\n",
       " (('35792218800', '57218685959'), 4),\n",
       " (('57221702598', '6603277503'), 4),\n",
       " (('57221702598', '57218685959'), 4),\n",
       " (('57210570167', '6603277503'), 4),\n",
       " (('57210570167', '57218685959'), 4),\n",
       " (('7101775825', '6603277503'), 4),\n",
       " (('7101775825', '57218685959'), 4),\n",
       " (('57223012847', '6603277503'), 4),\n",
       " (('57223012847', '57218685959'), 4),\n",
       " (('57221348779', '6603277503'), 4),\n",
       " (('57221348779', '57218685959'), 4),\n",
       " (('57221711727', '6603277503'), 4),\n",
       " (('57221711727', '57218685959'), 4),\n",
       " (('57208570427', '6603277503'), 4),\n",
       " (('57208570427', '57218685959'), 4),\n",
       " (('57188730260', '6603277503'), 4),\n",
       " (('57188730260', '57218685959'), 4),\n",
       " (('6603940183', '6603277503'), 4),\n",
       " (('6603940183', '57218685959'), 4),\n",
       " (('16939770600', '6603277503'), 4),\n",
       " (('16939770600', '57218685959'), 4),\n",
       " (('7005367074', '6603277503'), 4),\n",
       " (('7005367074', '57218685959'), 4),\n",
       " (('55078253600', '6603277503'), 4),\n",
       " (('55078253600', '57218685959'), 4),\n",
       " (('6506658107', '6603277503'), 4),\n",
       " (('6506658107', '57218685959'), 4),\n",
       " (('26643211400', '6603277503'), 4),\n",
       " (('26643211400', '57218685959'), 4),\n",
       " (('57202560371', '6603277503'), 4),\n",
       " (('57202560371', '57218685959'), 4),\n",
       " (('57516527200', '6603277503'), 4),\n",
       " (('57516527200', '57218685959'), 4),\n",
       " (('56768198900', '6603277503'), 4),\n",
       " (('56768198900', '57218685959'), 4),\n",
       " (('7005428980', '6603277503'), 4),\n",
       " (('7005428980', '57218685959'), 4),\n",
       " (('57202198322', '6603277503'), 4),\n",
       " (('57202198322', '57218685959'), 4),\n",
       " (('57292281600', '6603277503'), 4),\n",
       " (('57292281600', '57218685959'), 4),\n",
       " (('23984794100', '6603277503'), 4),\n",
       " (('23984794100', '57218685959'), 4),\n",
       " (('7801629502', '6603277503'), 4),\n",
       " (('7801629502', '57218685959'), 4),\n",
       " (('57454326700', '6603277503'), 4),\n",
       " (('57454326700', '57218685959'), 4),\n",
       " (('57202582820', '6603277503'), 4),\n",
       " (('57202582820', '57218685959'), 4),\n",
       " (('57208030690', '6603277503'), 4),\n",
       " (('57208030690', '57218685959'), 4),\n",
       " (('57454335400', '6603277503'), 4),\n",
       " (('57454335400', '57218685959'), 4),\n",
       " (('48461423400', '6603277503'), 4),\n",
       " (('48461423400', '57218685959'), 4),\n",
       " (('22135611200', '6603277503'), 4),\n",
       " (('22135611200', '57218685959'), 4),\n",
       " (('57454358900', '6603277503'), 4),\n",
       " (('57454358900', '57218685959'), 4),\n",
       " (('57454310200', '6603277503'), 4),\n",
       " (('57454310200', '57218685959'), 4),\n",
       " (('23987477400', '6603277503'), 4),\n",
       " (('23987477400', '57218685959'), 4),\n",
       " (('57316212000', '6603277503'), 4),\n",
       " (('57316212000', '57218685959'), 4),\n",
       " (('57213265213', '6603277503'), 4),\n",
       " (('57213265213', '57218685959'), 4),\n",
       " (('57204244525', '6603277503'), 4),\n",
       " (('57204244525', '57218685959'), 4),\n",
       " (('23134251100', '6603277503'), 4),\n",
       " (('23134251100', '57218685959'), 4),\n",
       " (('57454326800', '6603277503'), 4),\n",
       " (('57454326800', '57218685959'), 4),\n",
       " (('57223894721', '6603277503'), 4),\n",
       " (('57223894721', '57218685959'), 4),\n",
       " (('57193679093', '6603277503'), 4),\n",
       " (('57193679093', '57218685959'), 4),\n",
       " (('57454310300', '6603277503'), 4),\n",
       " (('57454310300', '57218685959'), 4),\n",
       " (('57200282943', '6603277503'), 4),\n",
       " (('57200282943', '57218685959'), 4),\n",
       " (('57226187564', '6603277503'), 4),\n",
       " (('57226187564', '57218685959'), 4),\n",
       " (('57501952300', '6603277503'), 4),\n",
       " (('57501952300', '57218685959'), 4),\n",
       " (('57454310400', '6603277503'), 4),\n",
       " (('57454310400', '57218685959'), 4),\n",
       " (('6602779888', '6603277503'), 4),\n",
       " (('6602779888', '57218685959'), 4),\n",
       " (('56922111700', '6603277503'), 4),\n",
       " (('56922111700', '57218685959'), 4),\n",
       " (('57211513293', '6603277503'), 4),\n",
       " (('57211513293', '57218685959'), 4),\n",
       " (('56960478500', '6603277503'), 4),\n",
       " (('56960478500', '57218685959'), 4),\n",
       " (('57210378406', '6603277503'), 4),\n",
       " (('57210378406', '57218685959'), 4),\n",
       " (('55357645900', '6603277503'), 4),\n",
       " (('55357645900', '57218685959'), 4),\n",
       " (('57454335500', '6603277503'), 4),\n",
       " (('57454335500', '57218685959'), 4),\n",
       " (('57191908241', '6603277503'), 4),\n",
       " (('57191908241', '57218685959'), 4),\n",
       " ...]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(link_num.items(), key = lambda x: x[1], reverse = True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3cbe215e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "66645be7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5717\n",
      "5488\n",
      "5728\n",
      "5432\n"
     ]
    }
   ],
   "source": [
    "name = []\n",
    "for i in metadata[\"Authors\"]:\n",
    "    name = name + i.split(\", \")\n",
    "aid = []\n",
    "for j in metadata[\"Author(s) ID\"]:\n",
    "    id_tem = j.split(\";\")\n",
    "    id_tem.pop()\n",
    "    aid = aid + id_tem\n",
    "    \n",
    "print(len(aid))\n",
    "print(len(set(aid)))\n",
    "print(len(name))\n",
    "print(len(set(name)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "eb208b16",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('57221419164', '57454369800'),\n",
       " ('57226680789', '57221943520'),\n",
       " ('7005862625', '7402098322'),\n",
       " ('56448994500', '57454339100'),\n",
       " ('24723671900', '57191352082'),\n",
       " ('55856048200', '56868468600'),\n",
       " ('48561311800', '57221922242'),\n",
       " ('35957366900', '57454334500'),\n",
       " ('7003647481', '7006453644'),\n",
       " ('57211142178', '56222324500'),\n",
       " ('7102554029', '57219594588'),\n",
       " ('23985807900', '6603685624'),\n",
       " ('57224323209', '57224317368'),\n",
       " ('57193800277', '57221921899'),\n",
       " ('34067509400', '57454339400'),\n",
       " ('57208618126', '57221942200'),\n",
       " ('15924910600', '57454337500'),\n",
       " ('56833814500', '57221924943'),\n",
       " ('57216035433', '57221930247'),\n",
       " ('7101775825', '57501952300'),\n",
       " ('57193315387', '57224312561'),\n",
       " ('57215613785', '57195543181'),\n",
       " ('57200339338', '57221253227'),\n",
       " ('57643094300', '57534874900'),\n",
       " ('57195543181', '57219855665'),\n",
       " ('57221252734', '56489487900'),\n",
       " ('57211137010', '57224318045'),\n",
       " ('6506130990', '16556241700'),\n",
       " ('7004003394', '7201357349'),\n",
       " ('7004705151', '57221943384'),\n",
       " ('57218142419', '57220658751'),\n",
       " ('57096280900', '7006453644'),\n",
       " ('55777166300', '57224316171'),\n",
       " ('57454326300', '57223269312'),\n",
       " ('57221247940', '57215597090'),\n",
       " ('57224501456', '57224311658'),\n",
       " ('57205701841', '57211975397'),\n",
       " ('57194060946', '57204679899'),\n",
       " ('57517422800', '57221597178'),\n",
       " ('7402546786', '57223603985'),\n",
       " ('57221943000', '57223120527'),\n",
       " ('55336592400', '57225305987'),\n",
       " ('57221396658', '57454348600'),\n",
       " ('57454353900', '57188711087'),\n",
       " ('36503607000', '57224317514'),\n",
       " ('57211535641', '57217993277'),\n",
       " ('35352322800', '57208885647'),\n",
       " ('57948519500', '57201076490'),\n",
       " ('36637697300', '57205495634'),\n",
       " ('7006336628', '57224306256'),\n",
       " ('7006927974', '57425280300'),\n",
       " ('57221937826', '57221944793'),\n",
       " ('57193756577', '56689410400'),\n",
       " ('57212508727', '57454322800'),\n",
       " ('56310667700', '57221937573'),\n",
       " ('55531171800', '57221924892'),\n",
       " ('57194256712', '57221934938'),\n",
       " ('57454369600', '57224318601'),\n",
       " ('6602092672', '22939498800'),\n",
       " ('57221419164', '57221943323'),\n",
       " ('57192315805', '57216632343'),\n",
       " ('57224961245', '57190346906'),\n",
       " ('57454343500', '14819472700'),\n",
       " ('57217992024', '57221249089'),\n",
       " ('8842055200', '57210510046'),\n",
       " ('12774155500', '57217937045'),\n",
       " ('57224308326', '57221929374'),\n",
       " ('57209047246', '57221925153'),\n",
       " ('37088090500', '57208855566'),\n",
       " ('7005636955', '57454309300'),\n",
       " ('6603264816', '57224323426'),\n",
       " ('35286113200', '57454329300'),\n",
       " ('57204136382', '54788484800'),\n",
       " ('6602603515', '57323185600'),\n",
       " ('57201979248', '28767846100'),\n",
       " ('57221252368', '57224322006'),\n",
       " ('57221938245', '57224309194'),\n",
       " ('57225770109', '7102018381'),\n",
       " ('7005952236', '57224323594'),\n",
       " ('57226611989', '57221939239'),\n",
       " ('57221251783', '57200588468'),\n",
       " ('57221940700', '57203902849'),\n",
       " ('57218283241', '57221248135'),\n",
       " ('57454311600', '57214973925'),\n",
       " ('57220196432', '55207776500'),\n",
       " ('57193951700', '57454338700'),\n",
       " ('57224323667', '57191242291'),\n",
       " ('57454365000', '23061421500'),\n",
       " ('7003647481', '57224322714'),\n",
       " ('12785974500', '57409737200'),\n",
       " ('26666073000', '57221251068'),\n",
       " ('51863718400', '57203817506'),\n",
       " ('56090686800', '57221932672'),\n",
       " ('56654989300', '57211137010'),\n",
       " ('53981988900', '57221921628'),\n",
       " ('57219977284', '57225674853'),\n",
       " ('57211508955', '16556241700'),\n",
       " ('57194030325', '57221929490'),\n",
       " ('57200422694', '57224318418'),\n",
       " ('54880102900', '57221250822'),\n",
       " ('57201075413', '57201095238'),\n",
       " ('55856048200', '57216157215'),\n",
       " ('57230977500', '56767830300'),\n",
       " ('6506552133', '57190067074'),\n",
       " ('7003622403', '57225444763'),\n",
       " ('57216850822', '57454331400'),\n",
       " ('57203019863', '57948444300'),\n",
       " ('57216275547', '55801109300'),\n",
       " ('25723868100', '57224311467'),\n",
       " ('57326066100', '36840499700'),\n",
       " ('57221926322', '57221934608'),\n",
       " ('57221414443', '57191362654'),\n",
       " ('57222120758', '57221935546'),\n",
       " ('57191347043', '57221922356'),\n",
       " ('57096280900', '57224322714'),\n",
       " ('25653169900', '35772489500'),\n",
       " ('57212431315', '57201067714'),\n",
       " ('26323230800', '13204582500'),\n",
       " ('55880004900', '57454316600'),\n",
       " ('9037555100', '57213587752'),\n",
       " ('57509558500', '57222600957'),\n",
       " ('7003659893', '55118463400'),\n",
       " ('57454326700', '57221250026'),\n",
       " ('7005504048', '57219724359'),\n",
       " ('7005938482', '6508303735'),\n",
       " ('35279736700', '57224310788'),\n",
       " ('56090863600', '57204398387'),\n",
       " ('35104408100', '57326460600'),\n",
       " ('16557276500', '57221339169'),\n",
       " ('36061883800', '57201076197'),\n",
       " ('57211789616', '57221419164'),\n",
       " ('56674917400', '57507621800'),\n",
       " ('57192639481', '57454328300'),\n",
       " ('57454354200', '57209048487'),\n",
       " ('7007185669', '57201300587'),\n",
       " ('57470324400', '57224305446'),\n",
       " ('6603126066', '57454345500'),\n",
       " ('7003540682', '6603452371'),\n",
       " ('57221249691', '57224313612'),\n",
       " ('7202322545', '57219933469'),\n",
       " ('57204682023', '6506673644'),\n",
       " ('6701330428', '57221934608'),\n",
       " ('6602726311', '57454371200'),\n",
       " ('13204582500', '36160653100'),\n",
       " ('36011342200', '57224318540'),\n",
       " ('57454326500', '57221929384'),\n",
       " ('24465051900', '6701802648'),\n",
       " ('57523058200', '57224312243'),\n",
       " ('7402230187', '57195545412'),\n",
       " ('57454355300', '57211137010'),\n",
       " ('7101775825', '57225770109'),\n",
       " ('37036899900', '57221922964'),\n",
       " ('57209256843', '7102812136'),\n",
       " ('57221254539', '57221938763'),\n",
       " ('7101962523', '57202099099'),\n",
       " ('57226418447', '57213587752'),\n",
       " ('57454323200', '57215570088'),\n",
       " ('57223122254', '54684618000'),\n",
       " ('56922111700', '56090876000'),\n",
       " ('57216157215', '57221250613'),\n",
       " ('57224317186', '55367759400'),\n",
       " ('35312284500', '57225804127'),\n",
       " ('7004003394', '57203821280'),\n",
       " ('57219425183', '56395058900'),\n",
       " ('57274033200', '57222312302'),\n",
       " ('7005755132', '57454345900'),\n",
       " ('57224309332', '57454363400'),\n",
       " ('8289320500', '57224877583'),\n",
       " ('56108103800', '57208481456'),\n",
       " ('57469021400', '8243215200'),\n",
       " ('57221925342', '57219594588'),\n",
       " ('56044592900', '57225799765'),\n",
       " ('57188876186', '57224318755'),\n",
       " ('57454354600', '56970924200'),\n",
       " ('57193279010', '22952609700'),\n",
       " ('57218143206', '57454371800'),\n",
       " ('57202569480', '35741920500'),\n",
       " ('57221929534', '56433701200'),\n",
       " ('57230977600', '57274056300'),\n",
       " ('57190986676', '57221934419'),\n",
       " ('7003592406', '57221250184'),\n",
       " ('57210171789', '55138098700'),\n",
       " ('57211021896', '16549482100'),\n",
       " ('56939048400', '57370612200'),\n",
       " ('7004276725', '57221935942'),\n",
       " ('57454355400', '57218329157'),\n",
       " ('6701640137', '57224307400'),\n",
       " ('57217144278', '57224312745'),\n",
       " ('57212431315', '56589950200'),\n",
       " ('36100958600', '57224312031'),\n",
       " ('57217993316', '57208855566'),\n",
       " ('57224311746', '57224313437'),\n",
       " ('57323017400', '57221253645'),\n",
       " ('57454331700', '57211686976'),\n",
       " ('24479446100', '57212789200'),\n",
       " ('57195266886', '57221938859'),\n",
       " ('57215647150', '57221943208'),\n",
       " ('57204537523', '57224322523'),\n",
       " ('25621160900', '57222599210'),\n",
       " ('23975787300', '56668503400'),\n",
       " ('6603839820', '57226622986'),\n",
       " ('57220196432', '57209047246'),\n",
       " ('57221323231', '57202354327'),\n",
       " ('57233760500', '57454357300'),\n",
       " ('57224322349', '57221248644'),\n",
       " ('55886290100', '7005840067'),\n",
       " ('57192315805', '57221923828'),\n",
       " ('57224312031', '57454361500'),\n",
       " ('57323185600', '57454316400'),\n",
       " ('57612555000', '57219516856'),\n",
       " ('57225157091', '8886276100'),\n",
       " ('16027795500', '57218803045'),\n",
       " ('25936599900', '7801355510'),\n",
       " ('57454365500', '55555716600'),\n",
       " ('57216632442', '57454330700'),\n",
       " ('57221420543', '57194137707'),\n",
       " ('56375339800', '57217018569'),\n",
       " ('57498031400', '57192298470'),\n",
       " ('20834455400', '57224310546'),\n",
       " ('36801313100', '57454369500'),\n",
       " ('57454342800', '57205487643'),\n",
       " ('57274033200', '57221944737'),\n",
       " ('57211329094', '57454356600'),\n",
       " ('7103359132', '57191242291'),\n",
       " ('57220076751', '57224307043'),\n",
       " ('7201366838', '8761490000'),\n",
       " ('14066864600', '57454369000'),\n",
       " ('57191275219', '6506435773'),\n",
       " ('57211606200', '57325885700'),\n",
       " ('36194876600', '57418909600'),\n",
       " ('51664646500', '57207903298'),\n",
       " ('6701640137', '57204241288'),\n",
       " ('6602721168', '57224308502'),\n",
       " ('56108103800', '57221931158'),\n",
       " ('57218296259', '57224312859'),\n",
       " ('57220846574', '7003363686'),\n",
       " ('7003656388', '26666073000'),\n",
       " ('6602779888', '36059752800'),\n",
       " ('55919193200', '57214444115'),\n",
       " ('55964096100', '57224322857'),\n",
       " ('8420912800', '57323301300'),\n",
       " ('57217222253', '57325707800'),\n",
       " ('57221714474', '57454325800'),\n",
       " ('36193512000', '6603800293'),\n",
       " ('57224321099', '57223119818'),\n",
       " ('57220145221', '57190063894'),\n",
       " ('55950033500', '7201572443'),\n",
       " ('57218296259', '57454315100'),\n",
       " ('57200330388', '57224305633'),\n",
       " ('56246061200', '57454337100'),\n",
       " ('57454319400', '57454371400'),\n",
       " ('57221924714', '57221925342'),\n",
       " ('57454366300', '7201374651'),\n",
       " ('56888654700', '57454337700'),\n",
       " ('57454322700', '57323185600'),\n",
       " ('56815974900', '57454371200'),\n",
       " ('50461239100', '57224322045'),\n",
       " ('57217995127', '57454346000'),\n",
       " ('26642875600', '57225762000'),\n",
       " ('7006336628', '57218219136'),\n",
       " ('55956891500', '57224320299'),\n",
       " ('57454369200', '57200330952'),\n",
       " ('6603481179', '57221926418'),\n",
       " ('56431173200', '7402415882'),\n",
       " ('57454343100', '57223119591'),\n",
       " ('57203136893', '57224310788'),\n",
       " ('57211946662', '57221250582'),\n",
       " ('57218321919', '57454338900'),\n",
       " ('55242134400', '57219933469'),\n",
       " ('57454361700', '57224319019'),\n",
       " ('51863718400', '37027145000'),\n",
       " ('57242162800', '57190736954'),\n",
       " ('7404667771', '57201077157'),\n",
       " ('6603410789', '57193951700'),\n",
       " ('57221370745', '57224313405'),\n",
       " ('57454314600', '57323185600'),\n",
       " ('19640122000', '57226785391'),\n",
       " ('57366446400', '57221929374'),\n",
       " ('57454350100', '57221937407'),\n",
       " ('56290071000', '57454366600'),\n",
       " ('27168215300', '57454360200'),\n",
       " ('57539350500', '14319127900'),\n",
       " ('57201493483', '7403216862'),\n",
       " ('8157312000', '36469979800'),\n",
       " ('20834455400', '57221249809'),\n",
       " ('57218135333', '57204671688'),\n",
       " ('57224313135', '57454349300'),\n",
       " ('57207891696', '57223121588'),\n",
       " ('55210719100', '57224316028'),\n",
       " ('57220504604', '57222990193'),\n",
       " ('36189115500', '57221940268'),\n",
       " ('55802050800', '57203349428'),\n",
       " ('7102639708', '6603780906'),\n",
       " ('23469369200', '57224312194'),\n",
       " ('57454310300', '57203745921'),\n",
       " ('22938086900', '57224315837'),\n",
       " ('53164662800', '57191362654'),\n",
       " ('7101952271', '8267285300'),\n",
       " ('6505495562', '57221251068'),\n",
       " ('57454353600', '57221929064'),\n",
       " ('57221252077', '57221936263'),\n",
       " ('57204136382', '57202093773'),\n",
       " ('57221249978', '57454313200'),\n",
       " ('8923810600', '7003885172'),\n",
       " ('7003528261', '57224312332'),\n",
       " ('57206415009', '57221933491'),\n",
       " ('56330316500', '57204673027'),\n",
       " ('36194876600', '57224307159'),\n",
       " ('57219934571', '57221943129'),\n",
       " ('57202859348', '54955251800'),\n",
       " ('6701480432', '57948466600'),\n",
       " ('57194030325', '57454311600'),\n",
       " ('57201979248', '57221931523'),\n",
       " ('57454358900', '57224313612'),\n",
       " ('57221249809', '56607917400'),\n",
       " ('57454370100', '6603264816'),\n",
       " ('26031725600', '57221796970'),\n",
       " ('7004603641', '18634285700'),\n",
       " ('57221921617', '57221944852'),\n",
       " ('6507092700', '57454341200'),\n",
       " ('56206861200', '57221250026'),\n",
       " ('57454343700', '57193315387'),\n",
       " ('57454311200', '57196809161'),\n",
       " ('55801109300', '57221928239'),\n",
       " ('6602721168', '58021775300'),\n",
       " ('57454365000', '57454324100'),\n",
       " ('6602186342', '55702521824'),\n",
       " ('23975787300', '57214766775'),\n",
       " ('57218645174', '57205449459'),\n",
       " ('35610072300', '57204673095'),\n",
       " ('7003636460', '57222605141'),\n",
       " ('57204674247', '57454345500'),\n",
       " ('57223003246', '57214021799'),\n",
       " ('55612516600', '57221934423'),\n",
       " ('55394254500', '57221921330'),\n",
       " ('56912021400', '6508168536'),\n",
       " ('7005428980', '57454321400'),\n",
       " ('35407637300', '57221179706'),\n",
       " ('57454368800', '57357154500'),\n",
       " ('57212147323', '57221944737'),\n",
       " ('23985807900', '57350211200'),\n",
       " ('57612555000', '37025910000'),\n",
       " ('57212489443', '16556241700'),\n",
       " ('57221416670', '7102916409'),\n",
       " ('6505966338', '57454314400'),\n",
       " ('57454351100', '57221931192'),\n",
       " ('7006674071', '9436804100'),\n",
       " ('55336856700', '57454327300'),\n",
       " ('57201720900', '56963946900'),\n",
       " ('57201076490', '57221940633'),\n",
       " ('57218144485', '57223123263'),\n",
       " ('16679534600', '57195545412'),\n",
       " ('57220201855', '57211786788'),\n",
       " ('56768198900', '57204674247'),\n",
       " ('7402586556', '16480785500'),\n",
       " ('6604021222', '57224320578'),\n",
       " ('57220075251', '57221251020'),\n",
       " ('7402598650', '57224307952'),\n",
       " ('56794804900', '57454322800'),\n",
       " ('23398486700', '57454369600'),\n",
       " ('55261869400', '57454348600'),\n",
       " ('57948488000', '57223119649'),\n",
       " ('57213143732', '57221247892'),\n",
       " ('57221250613', '57454368600'),\n",
       " ('6701641117', '57224308997'),\n",
       " ('57454350100', '57454356400'),\n",
       " ('57203678052', '57221934954'),\n",
       " ('55801109300', '57454341200'),\n",
       " ('35270159800', '57694837900'),\n",
       " ('55799331300', '57205487643'),\n",
       " ('57454331800', '57454341600'),\n",
       " ('57193756577', '57224503907'),\n",
       " ('57221472317', '57224318601'),\n",
       " ('57454326600', '57219560720'),\n",
       " ('57326825200', '7006089242'),\n",
       " ('57326269400', '57195548486'),\n",
       " ('57192167092', '57222596098'),\n",
       " ('57454347100', '56763327800'),\n",
       " ('57022137500', '57225203260'),\n",
       " ('9279703700', '6603621553'),\n",
       " ('57193881918', '57221933491'),\n",
       " ('57454320500', '57325885800'),\n",
       " ('55453902300', '57190972727'),\n",
       " ('56108103800', '15756295600'),\n",
       " ('35270159800', '57208704844'),\n",
       " ('26666513900', '57454361500'),\n",
       " ('23474184700', '55823146800'),\n",
       " ('56805279000', '57221922022'),\n",
       " ('57219315083', '57454317100'),\n",
       " ('57454326500', '57221941210'),\n",
       " ('16177136600', '36059752800'),\n",
       " ('57221250554', '57221936217'),\n",
       " ('55808491400', '57221924062'),\n",
       " ('57211284402', '57215570088'),\n",
       " ('44161222100', '57205487643'),\n",
       " ('57072675600', '57221924714'),\n",
       " ('35242679500', '57224817010'),\n",
       " ('55879315800', '57218135333'),\n",
       " ('57225937647', '57224312319'),\n",
       " ('57211721091', '57207891696'),\n",
       " ('57454326700', '57454320300'),\n",
       " ('57454319000', '57224314792'),\n",
       " ('57210425984', '57224307400'),\n",
       " ('57517602600', '57221555947'),\n",
       " ('57613633600', '57205487643'),\n",
       " ('57221250582', '57223609783'),\n",
       " ('55185187200', '57203902849'),\n",
       " ('7003647481', '57221927336'),\n",
       " ('57205441168', '57454320600'),\n",
       " ('57219332480', '57454322600'),\n",
       " ('57203425292', '56429644500'),\n",
       " ('7004403489', '34975069600'),\n",
       " ('57201795147', '57221938859'),\n",
       " ('57191514162', '57454361000'),\n",
       " ('57210171789', '55843546500'),\n",
       " ('57219594588', '7102812136'),\n",
       " ('25823206800', '56195553500'),\n",
       " ('37026813300', '57213455630'),\n",
       " ('55775358100', '57454339200'),\n",
       " ('57210846479', '57454309300'),\n",
       " ('28767846100', '57221249461'),\n",
       " ('35468845900', '57221254554'),\n",
       " ('57218144763', '6603264816'),\n",
       " ('6602562011', '57221934419'),\n",
       " ('6602358994', '6602881911'),\n",
       " ('57189961179', '57204664849'),\n",
       " ('54896345200', '57224305382'),\n",
       " ('7403322256', '57221930844'),\n",
       " ('57216576947', '57224316886'),\n",
       " ('57223012847', '57195062785'),\n",
       " ('57189961179', '57221922970'),\n",
       " ('55808491400', '57454353300'),\n",
       " ('57454326400', '57202098114'),\n",
       " ('7102132624', '57221932772'),\n",
       " ('6601980300', '57222283945'),\n",
       " ('57201399171', '7006012650'),\n",
       " ('57214346757', '35736759700'),\n",
       " ('55490604400', '57454330800'),\n",
       " ('57221923499', '57326272900'),\n",
       " ('57224314044', '57220092478'),\n",
       " ('57210425984', '57204241288'),\n",
       " ('7003592406', '57189495794'),\n",
       " ('57221933525', '57224315746'),\n",
       " ('35362301200', '55531171800'),\n",
       " ('57223122611', '57223987065'),\n",
       " ('57221250026', '57211402875'),\n",
       " ('56655338200', '57194588339'),\n",
       " ('57204673634', '57454320900'),\n",
       " ('57454350500', '57190677419'),\n",
       " ('23469369200', '55981282000'),\n",
       " ('56429644500', '35957366900'),\n",
       " ('57204885457', '57454323800'),\n",
       " ('57217164954', '57221940028'),\n",
       " ('57224309612', '57195550820'),\n",
       " ('57195296740', '57219933303'),\n",
       " ('57211903056', '35830546400'),\n",
       " ('57512109500', '57195545118'),\n",
       " ('57519177300', '57219345101'),\n",
       " ('57218143311', '56844297100'),\n",
       " ('7003528261', '7003496993'),\n",
       " ('57222284309', '57224311460'),\n",
       " ('57217985609', '57454320000'),\n",
       " ('57454359500', '57502312500'),\n",
       " ('56340274300', '57454343700'),\n",
       " ('57191339595', '57454357400'),\n",
       " ('7005664186', '57210510046'),\n",
       " ('7004206966', '22833341000'),\n",
       " ('57203425292', '25655615300'),\n",
       " ('57221249395', '57221940700'),\n",
       " ('56006385100', '7005814772'),\n",
       " ('57454349700', '7201923392'),\n",
       " ('57198890719', '7402521306'),\n",
       " ('57199419746', '57223123263'),\n",
       " ('55078253600', '57221250554'),\n",
       " ('57211535641', '57192408353'),\n",
       " ('20834455400', '57224312243'),\n",
       " ('54581261100', '57201091422'),\n",
       " ('57948455900', '23052986500'),\n",
       " ('57454354000', '57224314527'),\n",
       " ('16549482100', '57219013125'),\n",
       " ('57224322006', '57222596560'),\n",
       " ('57201069148', '57218330005'),\n",
       " ('57221722501', '57219097433'),\n",
       " ('33967777600', '57221938630'),\n",
       " ('14421078400', '57224314373'),\n",
       " ('57221929857', '56868294700'),\n",
       " ('57221928846', '57211329485'),\n",
       " ('35352322800', '57454366300'),\n",
       " ('57201070687', '56607917400'),\n",
       " ('57210570167', '57205408446'),\n",
       " ('57220206242', '57221253997'),\n",
       " ('57507917200', '57221416626'),\n",
       " ('36469979800', '12782644400'),\n",
       " ('57217992807', '57201919660'),\n",
       " ('26424923900', '17345446300'),\n",
       " ('16239073600', '57221416626'),\n",
       " ('57454350500', '12766560800'),\n",
       " ('57290491600', '57218371358'),\n",
       " ('57224116217', '6506435773'),\n",
       " ('57454360800', '57221544505'),\n",
       " ('57216772749', '57454313200'),\n",
       " ('9737156900', '57211142178'),\n",
       " ('56335390000', '57454352600'),\n",
       " ('23059810000', '57224309194'),\n",
       " ('55838509629', '57454312300'),\n",
       " ('7005632732', '57224318601'),\n",
       " ('57230394400', '57454330900'),\n",
       " ('57191347043', '55768247700'),\n",
       " ('7006525793', '57221254086'),\n",
       " ('37102535800', '57223119825'),\n",
       " ('24462882700', '57223991186'),\n",
       " ('57224319060', '24474332300'),\n",
       " ('8837787500', '55917186500'),\n",
       " ('48661310300', '56335390000'),\n",
       " ('57454328000', '57217831203'),\n",
       " ('7801392109', '7003709891'),\n",
       " ('24376377600', '6603279526'),\n",
       " ('57216275684', '57454366900'),\n",
       " ('57454331400', '35371312200'),\n",
       " ('35407637300', '57225799765'),\n",
       " ('57222605141', '57454340600'),\n",
       " ('9133994600', '57454316500'),\n",
       " ('7004572247', '55094468200'),\n",
       " ('7004102999', '57221254414'),\n",
       " ('57221944301', '57216471496'),\n",
       " ('55348239500', '57222118338'),\n",
       " ('57221939102', '57454371900'),\n",
       " ('57948476800', '57221252274'),\n",
       " ('57216952786', '57204676862'),\n",
       " ('7402546786', '57454335800'),\n",
       " ('7004206966', '57948444200'),\n",
       " ('57214515639', '57224316445'),\n",
       " ('57192320359', '57221249395'),\n",
       " ('7005938482', '55884760500'),\n",
       " ('57202570404', '6506177247'),\n",
       " ('57454327000', '57454344900'),\n",
       " ('7004304063', '57454338800'),\n",
       " ('55214122000', '57454315300'),\n",
       " ('57454345600', '57221416626'),\n",
       " ('57292281600', '57205701841'),\n",
       " ('56006385100', '57454363600'),\n",
       " ('57202569480', '57205494846'),\n",
       " ('36100958600', '57454369200'),\n",
       " ('57192979926', '57224316878'),\n",
       " ('57217986263', '7006334144'),\n",
       " ('35779215700', '57221248664'),\n",
       " ('57454336900', '57224322714'),\n",
       " ('7003963460', '7005887627'),\n",
       " ('36848322200', '57195363908'),\n",
       " ('57211237140', '57572468800'),\n",
       " ('7003933583', '57223122495'),\n",
       " ('23008040100', '57224877583'),\n",
       " ('57208570427', '57498031400'),\n",
       " ('48461423400', '57221934941'),\n",
       " ('57217987537', '57221922913'),\n",
       " ('57454369600', '57224321158'),\n",
       " ('57230977500', '57393561300'),\n",
       " ('7402586556', '57454372200'),\n",
       " ('7006137696', '56685681000'),\n",
       " ('57454329000', '57221939834'),\n",
       " ('6603481179', '57204669446'),\n",
       " ('57190982325', '6602667380'),\n",
       " ('57221923202', '57221928846'),\n",
       " ('16480785500', '57454368700'),\n",
       " ('7004003394', '57225243574'),\n",
       " ('57221923202', '57200311337'),\n",
       " ('56646265300', '57225814723'),\n",
       " ('15129377900', '57221932867'),\n",
       " ('7005254436', '57221929111'),\n",
       " ('56037650100', '6508157038'),\n",
       " ('56292326300', '7202605211'),\n",
       " ('6602978425', '56222324500'),\n",
       " ('57224321108', '57224307903'),\n",
       " ('57190982325', '57193800277'),\n",
       " ('57264029900', '57221252623'),\n",
       " ('57221931782', '57454329600'),\n",
       " ('57205441168', '57224320389'),\n",
       " ('57454338100', '57217501509'),\n",
       " ('12779561100', '57454319500'),\n",
       " ('57223121416', '57454325800'),\n",
       " ('57218143507', '57221247550'),\n",
       " ('56218097300', '57214260594'),\n",
       " ('57208733620', '57224309515'),\n",
       " ('57233522700', '57221923777'),\n",
       " ('57200294509', '57454353600'),\n",
       " ('57550959900', '57221941974'),\n",
       " ('57221369795', '14010798100'),\n",
       " ('7102856656', '57217163922'),\n",
       " ('57211334314', '56645308000'),\n",
       " ('56041664100', '57072957100'),\n",
       " ('57211789616', '57454313400'),\n",
       " ('35778123700', '57224306649'),\n",
       " ('9038778200', '56386609500'),\n",
       " ('57225666041', '57220714919'),\n",
       " ('57218135333', '57221328210'),\n",
       " ('57326252900', '57195543633'),\n",
       " ('6508236898', '57454368300'),\n",
       " ('22034709700', '56827294000'),\n",
       " ('12786571000', '6603963634'),\n",
       " ('57203048581', '55270572600'),\n",
       " ('57225102301', '57221931192'),\n",
       " ('57221945148', '57221925521'),\n",
       " ('57948476100', '57217646050'),\n",
       " ('57221929415', '57224316748'),\n",
       " ('7801355510', '57454340600'),\n",
       " ('57206260356', '57224316786'),\n",
       " ('22956670300', '14319127900'),\n",
       " ('57226487430', '57224315311'),\n",
       " ('35189150500', '57224317006'),\n",
       " ('6508261692', '8645153700'),\n",
       " ('57230977600', '57202906261'),\n",
       " ('8247573400', '57521326100'),\n",
       " ('57224319765', '57221251125'),\n",
       " ('57230977700', '57224317371'),\n",
       " ('57221416806', '22939498800'),\n",
       " ('7004802904', '7403267603'),\n",
       " ('57221967187', '57226325732'),\n",
       " ('57221942966', '57224307043'),\n",
       " ('57221369735', '57470324400'),\n",
       " ('19640122000', '57188711087'),\n",
       " ('57219332480', '57221250184'),\n",
       " ('57226755956', '57221249095'),\n",
       " ('57196350860', '57948487600'),\n",
       " ('6602694967', '57224315475'),\n",
       " ('57194256712', '57454329300'),\n",
       " ('57210155996', '23984794100'),\n",
       " ('57202569480', '57221925368'),\n",
       " ('6507671879', '35830546400'),\n",
       " ('57211968524', '57222598868'),\n",
       " ('26640267200', '57221473652'),\n",
       " ('56044592900', '55973477500'),\n",
       " ('56716455300', '57195154406'),\n",
       " ('57212111835', '57224322714'),\n",
       " ('57193744135', '57224305416'),\n",
       " ('57218144763', '25623599800'),\n",
       " ('57216875002', '7006343245'),\n",
       " ('37117653500', '57224306134'),\n",
       " ('57221944737', '7102023740'),\n",
       " ('57217055938', '57454316900'),\n",
       " ('6506017468', '6603435894'),\n",
       " ('57221938882', '55962909800'),\n",
       " ('7201374651', '7402818741'),\n",
       " ('57190031117', '57221252734'),\n",
       " ('7102538707', '7004417687'),\n",
       " ('57323775100', '57454336900'),\n",
       " ('6603363716', '57203080999'),\n",
       " ('6603320527', '57948466300'),\n",
       " ('7005367074', '57224312747'),\n",
       " ('54581261100', '57204680443'),\n",
       " ('7004309751', '57224323531'),\n",
       " ('7005073319', '7006940003'),\n",
       " ('7006488444', '57195547909'),\n",
       " ('6603685624', '57224322349'),\n",
       " ('57454359600', '16239073600'),\n",
       " ('57535655000', '57221921315'),\n",
       " ('57454343700', '25626248700'),\n",
       " ('57188730260', '57948487600'),\n",
       " ('57216156739', '56828936500'),\n",
       " ('57191854216', '57948466300'),\n",
       " ('57546570200', '57454321300'),\n",
       " ('55336856700', '57454361200'),\n",
       " ('57367028700', '57221930102'),\n",
       " ('6603199799', '23474184700'),\n",
       " ('57948466200', '57193788769'),\n",
       " ('57224313058', '57221938632'),\n",
       " ('57454358100', '57222327723'),\n",
       " ('7004254649', '57221925160'),\n",
       " ('6504375533', '57221938797'),\n",
       " ('57217992024', '57454330800'),\n",
       " ('57221200001', '35482685300'),\n",
       " ('57221940700', '57454341600'),\n",
       " ('57948476100', '57204681714'),\n",
       " ('9744154400', '57221933217'),\n",
       " ('57208090961', '25621160900'),\n",
       " ('16686840700', '57221935049'),\n",
       " ('57222063482', '35468845900'),\n",
       " ('57221934419', '57221940479'),\n",
       " ('57211946662', '57454366700'),\n",
       " ('57216875131', '6603264816'),\n",
       " ('12766560800', '57224649260'),\n",
       " ('57454325300', '26025689900'),\n",
       " ('6504207080', '57221254322'),\n",
       " ('6508303735', '57193730694'),\n",
       " ('57221254362', '57224306649'),\n",
       " ('57224307159', '57221935329'),\n",
       " ('57224322349', '57454363200'),\n",
       " ('57211903056', '57694837900'),\n",
       " ('6602517591', '57454338000'),\n",
       " ('56429644500', '57224312319'),\n",
       " ('57454311600', '57201091422'),\n",
       " ('57190381886', '6507087260'),\n",
       " ('57454365500', '57204682537'),\n",
       " ('56872898400', '57572468800'),\n",
       " ('57202569480', '57221935435'),\n",
       " ('57218851970', '56868294700'),\n",
       " ('57225732228', '57195543181'),\n",
       " ('57454331600', '57224311602'),\n",
       " ('56395137300', '57078712600'),\n",
       " ('57221927875', '57224305382'),\n",
       " ('57454315800', '57224315746'),\n",
       " ('57224317925', '35772489500'),\n",
       " ('35311398500', '57193208558'),\n",
       " ('57203048581', '56433701200'),\n",
       " ('57217995127', '57224307043'),\n",
       " ('57211903056', '57208704844'),\n",
       " ('57454345000', '7403325890'),\n",
       " ('57221253192', '57221937088'),\n",
       " ('57221250192', '57213143732'),\n",
       " ('57194588339', '57221247140'),\n",
       " ('7406547916', '57221939161'),\n",
       " ('6603547405', '57224312748'),\n",
       " ('36917209100', '22936120600'),\n",
       " ('57192934758', '43861903000'),\n",
       " ('57216035433', '14066864600'),\n",
       " ('35402411600', '57454340900'),\n",
       " ('57219560720', '55904364300'),\n",
       " ('57221253040', '57210092283'),\n",
       " ('56960478500', '57454315300'),\n",
       " ('35951889200', '57224314988'),\n",
       " ('7201667856', '57221922105'),\n",
       " ('57210171789', '57454337600'),\n",
       " ('57229995400', '57454313400'),\n",
       " ('57220193926', '57224314792'),\n",
       " ('57221923202', '57222602814'),\n",
       " ('56833814500', '57454372300'),\n",
       " ('16939770600', '7006089242'),\n",
       " ('7201508348', '57221248690'),\n",
       " ('57221943919', '57224322714'),\n",
       " ('7003528261', '57454360400'),\n",
       " ('57454339600', '23466821300'),\n",
       " ('57454353200', '56404237600'),\n",
       " ('57409737200', '57210179740'),\n",
       " ('57351175400', '57221931250'),\n",
       " ('57222312302', '57221252224'),\n",
       " ('57222118338', '57454352300'),\n",
       " ('11738887200', '57201070687'),\n",
       " ('57222043615', '56970924200'),\n",
       " ('57218283241', '57214309916'),\n",
       " ('57223119818', '57454321200'),\n",
       " ('13611914400', '20734200200'),\n",
       " ('7202322545', '6603525360'),\n",
       " ('57194699001', '57215071976'),\n",
       " ('55242134400', '57219844804'),\n",
       " ('57211339218', '57221924062'),\n",
       " ('57221369735', '7003647481'),\n",
       " ('57216275309', '57221248753'),\n",
       " ('8289320500', '57212431315'),\n",
       " ('57224322349', '57200918382'),\n",
       " ('7102576008', '15765137100'),\n",
       " ('7201774556', '57224309970'),\n",
       " ('57218143206', '57701853500'),\n",
       " ('57221945148', '36708778900'),\n",
       " ('57226234589', '57221937379'),\n",
       " ('57221419164', '57454368600'),\n",
       " ('57535655000', '57204682023'),\n",
       " ('57223993469', '57207605921'),\n",
       " ('55510867300', '24492323500'),\n",
       " ('57700754100', '56446293300'),\n",
       " ('57584164500', '7005755132'),\n",
       " ('57214596611', '55367759400'),\n",
       " ('57224307675', '57454348400'),\n",
       " ('57221250026', '57454324700'),\n",
       " ('57454349100', '57219712874'),\n",
       " ('12779561100', '57221251859'),\n",
       " ('7004802904', '57454335900'),\n",
       " ('55534294500', '57223599932'),\n",
       " ('45261113000', '57550935200'),\n",
       " ('57221938245', '57454338900'),\n",
       " ('57214121395', '57190110013'),\n",
       " ('57454360200', '57224311741'),\n",
       " ('57207528404', '57454357100'),\n",
       " ('6506017468', '57221939836'),\n",
       " ('56862324300', '7005755132'),\n",
       " ('57221930844', '57326809000'),\n",
       " ('57216875131', '57221252521'),\n",
       " ('57454352400', '57224322464'),\n",
       " ('15756295600', '56827294000'),\n",
       " ('23008235300', '55333044600'),\n",
       " ('57221328210', '57225801885'),\n",
       " ('57221420543', '57221938859'),\n",
       " ('57207891696', '57454321100'),\n",
       " ('20734200200', '57221253464'),\n",
       " ('57454320600', '57221252390'),\n",
       " ('57224600040', '57454321300'),\n",
       " ('57224315475', '57221940442'),\n",
       " ('56835316200', '6506082906'),\n",
       " ('57454310700', '57220658751'),\n",
       " ('15849336900', '7202342767'),\n",
       " ('24402710900', '57957538900'),\n",
       " ('57190986676', '57224312194'),\n",
       " ('56998409700', '57221943156'),\n",
       " ('57224129974', '7103129209'),\n",
       " ('6507424395', '57222605141'),\n",
       " ('57454322100', '57221937641'),\n",
       " ('7003921883', '57221249809'),\n",
       " ('57221921330', '57224309775'),\n",
       " ('57217031688', '57498031400'),\n",
       " ('35482685300', '26635388800'),\n",
       " ('57221930102', '43861903000'),\n",
       " ('57224322318', '57221932220'),\n",
       " ('57210570167', '57454356300'),\n",
       " ('57228128600', '57221942644'),\n",
       " ('57191246660', '55962909800'),\n",
       " ('55736000300', '57221926831'),\n",
       " ('57219712874', '57221929535'),\n",
       " ('57224318540', '57201078907'),\n",
       " ('57201078907', '57219934879'),\n",
       " ('56424163100', '57221932220'),\n",
       " ('24437405900', '57538664800'),\n",
       " ('56340274300', '6506981950'),\n",
       " ('55843546500', '57454345300'),\n",
       " ('57203074384', '57201075399'),\n",
       " ('55603389800', '57223123371'),\n",
       " ('57202582820', '57325885700'),\n",
       " ('57203745921', '57203177203'),\n",
       " ('57454335800', '7101962523'),\n",
       " ('57225414218', '57224313099'),\n",
       " ('57224317368', '57224307128'),\n",
       " ('57195550820', '55046270800'),\n",
       " ('57202555419', '57191530686'),\n",
       " ('57454309300', '57454324400'),\n",
       " ('57454337100', '57224316740'),\n",
       " ('57218128061', '57197309621'),\n",
       " ('57216351128', '57224313811'),\n",
       " ('7004259279', '57224305382'),\n",
       " ('57454346900', '57221923883'),\n",
       " ('55214122000', '57217993316'),\n",
       " ('57194694837', '57224312859'),\n",
       " ('57221254554', '57224322857'),\n",
       " ('57221254539', '57224319019'),\n",
       " ('57454360800', '57221922543'),\n",
       " ('57220145221', '57224320299'),\n",
       " ('55775358100', '21234190100'),\n",
       " ('57221944772', '57326081200'),\n",
       " ('57410400100', '57454313600'),\n",
       " ('7005775097', '54795430500'),\n",
       " ('15924910600', '57225811342'),\n",
       " ('57217993277', '57225674853'),\n",
       " ('57220092478', '57224313811'),\n",
       " ('36469979800', '56389193500'),\n",
       " ('56896844100', '6602270825'),\n",
       " ('57454371900', '56407143600'),\n",
       " ('57517422800', '24492323500'),\n",
       " ('57194694837', '57454315100'),\n",
       " ('57224308326', '57221939491'),\n",
       " ('57195588807', '57454372500'),\n",
       " ('57193420762', '36503607000'),\n",
       " ('24757974900', '57224316018'),\n",
       " ('26643211400', '57221932672'),\n",
       " ('24334850600', '55227463900'),\n",
       " ('57217144278', '36848322200'),\n",
       " ('55702835700', '57216597476'),\n",
       " ('57194873664', '57454338800'),\n",
       " ('9039666200', '57221249084'),\n",
       " ('57218143396', '57224308326'),\n",
       " ('57221597178', '57224320299'),\n",
       " ('15849336900', '6603277503'),\n",
       " ('6603214152', '57224305734'),\n",
       " ('57215071976', '7403228654'),\n",
       " ('57224314044', '57204668171'),\n",
       " ('57193500089', '57191362654'),\n",
       " ('15047606300', '6602438755'),\n",
       " ('57454311600', '57204680443'),\n",
       " ('57211789616', '57224319366'),\n",
       " ('57211606200', '57224318203'),\n",
       " ('57199371681', '57204673095'),\n",
       " ('57219810179', '57205487643'),\n",
       " ('6603017789', '57221251449'),\n",
       " ('35268685000', '13611914400'),\n",
       " ('57205306317', '57224316886'),\n",
       " ('57454366700', '57218329157'),\n",
       " ('6507671879', '57694837900'),\n",
       " ('56888654700', '57218296259'),\n",
       " ('57223120241', '57645409700'),\n",
       " ('15052454400', '25654127400'),\n",
       " ('57225745596', '57225855293'),\n",
       " ('23984232500', '57224322781'),\n",
       " ('57217646050', '57205490529'),\n",
       " ('9532485500', '55426366100'),\n",
       " ('57222541501', '57224307244'),\n",
       " ('57454335500', '57200330952'),\n",
       " ('57221928249', '57195550820'),\n",
       " ('57454324700', '57221932220'),\n",
       " ('57189468284', '57203520135'),\n",
       " ('26432674600', '57224322857'),\n",
       " ('57225425174', '57218599577'),\n",
       " ('57224322349', '57211140731'),\n",
       " ('16836282400', '57454330900'),\n",
       " ('57192408353', '57454311700'),\n",
       " ('55265957900', '8055289000'),\n",
       " ('57454328400', '57226633914'),\n",
       " ('57454319500', '58060964700'),\n",
       " ('57207940521', '18633282000'),\n",
       " ('6507671879', '57208704844'),\n",
       " ('57221472317', '57224321158'),\n",
       " ('7102538707', '23399022300'),\n",
       " ('57675486400', '57224309194'),\n",
       " ('57220804615', '57454353600'),\n",
       " ('57221921628', '57454371900'),\n",
       " ('57208618126', '6603940183'),\n",
       " ('25224723100', '23993684800'),\n",
       " ('56814993200', '24075896600'),\n",
       " ('57454332600', '57216823853'),\n",
       " ('57454323200', '56184516400'),\n",
       " ('57454337800', '57214973925'),\n",
       " ('57214577447', '35369845200'),\n",
       " ('57326269400', '56463857900'),\n",
       " ('55834323900', '55385651700'),\n",
       " ('57221943356', '57204671688'),\n",
       " ('57221710487', '57197779349'),\n",
       " ('57196190877', '57224309403'),\n",
       " ('6603818486', '57224129118'),\n",
       " ('57221947689', '7402306357'),\n",
       " ('57224323999', '57221940700'),\n",
       " ('57191907674', '57195543181'),\n",
       " ('57190986676', '57205246545'),\n",
       " ('54896345200', '57221250582'),\n",
       " ('57221944394', '57218761773'),\n",
       " ('35243381600', '57221924496'),\n",
       " ('57350211200', '56801266700'),\n",
       " ('57219653299', '57220206242'),\n",
       " ('7005718665', '36463920000'),\n",
       " ('57519177300', '57454325000'),\n",
       " ('25224723100', '57224322211'),\n",
       " ('35352322800', '57198904041'),\n",
       " ('36468055100', '57215613785'),\n",
       " ('55897430300', '57222598095'),\n",
       " ('57205368772', '57194192257'),\n",
       " ('57454328100', '57225937647'),\n",
       " ('57221931250', '57223120521'),\n",
       " ('57221929415', '55465964200'),\n",
       " ('57204112346', '55041735300'),\n",
       " ('57224315286', '35217737200'),\n",
       " ('57218139373', '57627979500'),\n",
       " ('57201076197', '57201090047'),\n",
       " ('9038778200', '6602978425'),\n",
       " ('57211329094', '6508277910'),\n",
       " ('57224319488', '57454347800'),\n",
       " ('7402230187', '55357418100'),\n",
       " ('16431901300', '57454332900'),\n",
       " ('7005120333', '56440139100'),\n",
       " ('51161538600', '57210774697'),\n",
       " ('57213845477', '7404045199'),\n",
       " ('57454355400', '57204669372'),\n",
       " ('56425273600', '56533577200'),\n",
       " ('57190619921', '57200334643'),\n",
       " ('57212854133', '7003340975'),\n",
       " ('24832614900', '57206260356'),\n",
       " ('57454321400', '55059740300'),\n",
       " ('56200473600', '57217532334'),\n",
       " ('24832614900', '56145533200'),\n",
       " ('57520311500', '57224761351'),\n",
       " ('57220804615', '57219620314'),\n",
       " ('57218599577', '57224313480'),\n",
       " ('57454352500', '57206433438'),\n",
       " ('57202906261', '57193951700'),\n",
       " ('7403534478', '57454324300'),\n",
       " ('57454330800', '36022352400'),\n",
       " ('55617927600', '57454319000'),\n",
       " ('35242500200', '57201341873'),\n",
       " ('57222990193', '57221934941'),\n",
       " ('57218443557', '57221252698'),\n",
       " ('57212111835', '57221927336'),\n",
       " ('57216952230', '57642286800'),\n",
       " ('57454355000', '26666073000'),\n",
       " ('6602186342', '57204559275'),\n",
       " ('57201704774', '57454313600'),\n",
       " ('57454350700', '57221930844'),\n",
       " ('57192373009', '57219933993'),\n",
       " ('57454321200', '57454317100'),\n",
       " ('56784694900', '15764579300'),\n",
       " ('7005981560', '57221943208'),\n",
       " ('57501952300', '57454311200'),\n",
       " ('57221478095', '57224319797'),\n",
       " ('55336856700', '55523277900'),\n",
       " ('57221328210', '57221253782'),\n",
       " ('6602779888', '57221928492'),\n",
       " ('37102535800', '57454353500'),\n",
       " ('57454350600', '57203026252'),\n",
       " ('57202435533', '24449530100'),\n",
       " ('37034619800', '57224313811'),\n",
       " ('6508101557', '35583019600'),\n",
       " ('57454326700', '57454354200'),\n",
       " ('47561603300', '55553693100'),\n",
       " ('6602721168', '56868540200'),\n",
       " ('37088153100', '57221248753'),\n",
       " ('8442609600', '55534294500'),\n",
       " ('57205495634', '57225334790'),\n",
       " ('57454325900', '57222581715'),\n",
       " ('57226207751', '57454360300'),\n",
       " ('57217018569', '55046270800'),\n",
       " ('55469722600', '7005988634'),\n",
       " ('57230195300', '57226680789'),\n",
       " ('16230073100', '57221771679'),\n",
       " ('57325885800', '57454372200'),\n",
       " ('57190031117', '57193744135'),\n",
       " ('57221350738', '57501952300'),\n",
       " ('57221396658', '57221250139'),\n",
       " ('57218138375', '57454336000'),\n",
       " ('57229995400', '57224319366'),\n",
       " ...]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set_link = list(set(link))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "123a5ee9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6215971"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set_link)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "b14f4cb4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('57221419164', '57454369800')"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set_link[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "ac4ced02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "link.count(('6603277503', '57218685959'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "71f0931d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import networkx as nx\n",
    "G = nx.Graph() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "f9dea60d",
   "metadata": {},
   "outputs": [],
   "source": [
    "G = nx.empty_graph()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "62ebf3b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "G.add_edge(id_name[set_link[0][0]], id_name[set_link[0][1]], weight=link.count(set_link[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "2cefe9f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "G.add_edge(id_name[set_link[0][0]], id_name[set_link[0][1]], weight=link.count(set_link[0]))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "49b496fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAgMAAAGFCAYAAABg2vAPAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAA9hAAAPYQGoP6dpAAA2mUlEQVR4nO3dd3RUBfrG8WeS0IXQBEVKFBBFsAABBJGqVMEAmUkhpADSpIiIoCIWLKuisuqqK13AjoiggCahE0gbRCyASFE6gSSEhCQz9/eHP7NEWgJJ7kzm+zmHc3Zm7tw8ye7ZeeZ9p1gMwzAEAAA8lpfZAQAAgLkoAwAAeDjKAAAAHo4yAACAh6MMAADg4SgDAAB4OMoAAAAezqcgBzmdTh08eFCVK1eWxWIp7kwAAKAIGIah9PR01alTR15eF3/+X6AycPDgQdWrV6/IwgEAgJJz4MAB1a1b96K3F6gMVK5cOe9kVapUKZpkAACgWKWlpalevXp5j+MXU6Ay8PdqoEqVKpQBAADczOVW/LyAEAAAD0cZAADAw1EGAADwcJQBAAA8HGUAAAAPRxkAAMDDUQYAAPBwlAEAADwcZQAAAA9HGQAAwMNRBgAA8HCUAQAAPBxlAAAAD0cZAADAw1EGAADwcJQBAAA8nI/ZAQAAcBcZZ3O190SGsnOdKuvjJb8alVSpnPs/lLr/bwAAQDHadSRdi7bsV+yvR7U/5YyMc26zSKpfvaI6N6ml0Db11bh2ZbNiXhWLYRjG5Q5KS0uTr6+vUlNTVaVKlZLIBQCAqQ6knNETX27X+t3H5e1lkcN58YfLv2/v0KimXgxornrVK5Zg0osr6OM3rxkAAOAfPo7fr25vrNWmPSck6ZJF4NzbN+05oW5vrNXH8fuLPWNRYk0AAMA53o7dpddW77yi+zqchhxOQ5OXbNfx02f1cOfGRZyueDAZAADg/30cv/+Ki8A/vbZ6pz5xkwkBZQAAAP31GoFpy3YU6TmfXrZDB1LOFOk5iwNlAADgsn744QcNGTJEDRs2VIUKFVShQgU1btxYw4cPV0JCQpH+LNvEl7Rzei/lnjpy2WP/+E+Uji9/I+9y1r4ftO/lPsra90O+43Kdhp74cnuhcjzzzDOyWCzy8vLSnj17zrs9IyNDVapUkcViUURERKHOfTGUAQCAS3r//ffVsmVLbdmyRePGjdPy5cu1YsUKjR8/Xjt27JC/v79+++23IvlZu46ka+eR01d8/7LXNdJ1Ya+p7HWN8l3vcBpav/u4dh9NL/Q5r7nmGs2dO/e86z/77DPl5OSoTJkyV5z3nygDAACXs3HjRo0aNUo9e/ZUUlKSxo4dq65du6pLly4aPXq0NmzYoE8//VQVKlQokp+3aMt+eV3FI6JXuYoqd8Mt8ip3/lsKvb0sWhhX+NcO2Gw2zZ8/X06nM9/1s2fPVkBAgMqWLXvFef+JMgAAcDkvvviivL299f7771/0QS8wMFB16tTJu5yQkKCgoCD5+fmpQoUK8vPzU3BwsPbt23fefePi4tS+fXuVL19ederU0Zw3X5AjN/e84wxHrk7GztGBtwZp/2sDdHjhJJ09+Ot5x11oTZBz6rCOffUv7ft3mJ7rf5dq166trl27ym63F+hvEBUVpQMHDui7777Lu27nzp3asGGDoqKiCnSOguKthQAAl+JwOBQbG6tWrVrp+uuvL/D99u7dqyZNmigoKEjVq1fXoUOH9O6778rf318//fSTatasKUn66aef1LVrV/n5+WnevHmylCmniMemK+fEgfPOeeLbt5TxY4yqtAlQeb+7lHNsn459+aKc2ZmXzXP002ckw6GqnSJVpsq1mtH3JiXFb9GpU6cK9Ps0btxYHTp00Jw5c9S9e3dJ0pw5c+Tn56euXbsW+O9SEJQBAIBLOX78uDIzM9WgQYPzbnM4HDr3g3O9vb1lsVgkSQMHDtTAgQPzHdunTx/Vrl1bixcv1tixYyVJzz33nAzDUExMjGrXrq0dB1NVK9BbB2ePzvezck4cUMaP0ars30/VOv/1TLzCjXfJu1JVHf/6tUv+Do7MNOWm/KFqXYfpmmadJUl3dLhHIbbAQv0toqKiNGLECKWkpMjX11cLFizQ8OHD837nosKaAADgNlq2bKkyZcrk/ZsxY0bebadPn9bjjz+uRo0aycfHRz4+PrrmmmuUkZGhn3/+Oe+42NhYde3aVbVr15YkZec6ZfHyVqVbOuT7WVn7/noXQKXbOue7vuKtHSQv70vm9CpfWT5Vr1fa1iVK2/qlsg//pqzs89cQlxMYGKiyZctq0aJF+uabb3T48OEiewfBuZgMAABcSs2aNVWhQoUL7voXL16sM2fO6NChQ+rbt2++20JCQhQdHa2pU6fK398/7+13vXr1Umbm/8b6J06c0HXXXZd3uazPX8+Lva+plu98jsy0v66vVDXf9RYvb3lVuPQXElksFtUOfkGnNn6k1C1f6GTMbHVd+owGhw3SCy+8oMqVC/aFRpUqVZLNZtOcOXPUoEEDdevW7YITk6tFGQAAuBRvb2916dJFq1ev1qFDh/K9bqBp06aS/np9wLlSU1O1fPlyTZs2TZMnT867/uzZs0pJScl3bI0aNXT48OG8y341KskiyXH6ZP4cFf76Yh9Hxin5VK6Zd73hdMiZefm3Cvr41lLNXuMkSbkpfyrq+sN6cfpzys7O1nvvvXfZ+/8tKipKs2bN0g8//KBFixYV+H6FwZoAAOBypkyZIofDoREjRignJ+eyx1ssFhmGoXLlyuW7ftasWXI4HPmu69y5s6Kjo3XkyF8fLlSpnI/qVS2njF/W5zuufP3mkqSMHbH5rj/z83rJmf+cl9OwcWM998zTat68uZKSkgp137vvvltRUVEKCAhQQEBAoe5bUEwGAAAup3379nrnnXc0ZswYtWjRQg899JBuu+02eXl56dChQ/riiy8kKe9reatUqaJ7771Xr776qmrWrCk/Pz+tXbtWs2fPVtWqVfOd+6mnntKyZcvUpUsXPf3006pYsaIOfvIvGTln8x1XpmY9Vbqts9Ljl8ni5aPyfncq59g+pW1dIssFPk/gXNlHf1fK6vdU8ZZ7VK5GHflVqqunnorWDz/8kG9yUVCzZ88u0HE+Pj7q2LGjoqOjC3V+ygAAwCWNGDFCd999t2bOnKk33nhDBw8elMViUd26ddWuXTtFR0erS5cueccvXrxY48aN06RJk5Sbm6v27dvru+++U+/evfOdt1mzZvr+++/16KOPKjw8XNWqVVOf/jYtqX2XUla+ne/YGr3GyrtSVZ3eHq30xK9VptaNujbgCR376pVLZveuVE0+1a5XevIKnUo7ri+W+6hhw5s0Y8YMjRkzpuj+SP/gcDjOm4QUhMU49z0aF5GWliZfX1+lpqbmtTAAAEqTsNlbtHH3MTlVdG/b8/ayqN1NNfThkDZFds7CKOjjN68ZAAB4vDNnzsjYuli5OdnS5Z8jF5iPl0UvBjQvsvMVF8oAAMCj/fzzz2rTpo2+WPBf9a2TJRXhB/o81/c21at+6dcXuALKAADAY82fP1+tWrWSw+FQfHy83p4Qoon331wk537s/iay+dcvknMVN8oAAMDjZGRkKCIiQhEREQoMDFR8fLyaNWsmSXq4c2O93L+5yvl4ydurcFMCby+Lyvl46V/9m2t050aXv4OL4N0EAACPsn37dlmtVu3fv1/z5s1TeHj4eccE+ddX+4Y19cSX27V+93F5e1nkcF78tQR/397uphp6MaC5W6wGzkUZAAB4BMMwNHv2bI0ZM0aNGjVSQkKCbr311oseX696RX04pI12HUnXoi37FbvzqPafOKNzK4FFUv0aFdX55loa1La+GtUq2McMuxreWggAKPXS09M1YsQILV68WEOHDtXMmTNVsWLhn71nnM3V3hMZys51qqyPl/xqVFKlcq77vLqgj9+u+xsAAFAE7Ha7rFarDh06pEWLFikkJOSKz1WpnI9uq+NbhOlcAy8gBACUSoZh6N1331Xbtm1VsWJFJSYmXlURKM0oAwCAUic1NVU2m02jRo3SkCFDFBcXp5tvLpq3DJZGrAkAAKVKQkKCbDabjh8/rk8//VSBgYFmR3J5TAYAAKWCYRj697//rXbt2qlatWpKSkqiCBQQZQAA4PZOnjypAQMGaNy4cRo5cqQ2btyohg0bmh3LbbAmAAC4tS1btshmsyk1NVVffvmlHnzwQbMjuR0mAwAAt2QYhmbMmKF77rlH1113nZKTkykCV4gyAABwOydOnFDfvn01ceJEjRs3TuvWrZOfn5/ZsdwWawIAgFvZuHGjgoKCdObMGX399dfq06eP2ZHcHpMBAIBbcDqdevnll9WxY0c1aNBAdrudIlBEKAMAAJd37Ngx9e7dW1OmTNGkSZMUGxurevXqmR2r1GBNAABwaWvXrlVISIiys7O1cuVKde/e3exIpQ6TAQCAS3I4HHr++efVpUsXNW7cWNu2baMIFBMmAwAAl3P48GENGjRIMTExeuqpp/T000/Lx4eHrOLCXxYA4FKio6MVGhoqSVq9erW6detmcqLSjzUBAMAlOBwOTZs2Tffdd5+aNWsmu91OESghTAYAAKY7ePCgQkJCtH79ej377LN64okn5O3tbXYsj0EZAACYatWqVRo0aJDKlCmjmJgYdezY0exIHoc1AQDAFLm5uZoyZYp69Oihli1bym63UwRMwmQAAFDiDhw4oODgYMXFxemll17SpEmT5OXF81OzUAYAACVqxYoVGjx4sCpWrKi1a9eqffv2ZkfyeNQwAECJyMnJ0cSJE9WnTx+1a9dOdrudIuAimAwAAIrd3r17FRQUpMTERL322muaMGGCLBaL2bHw/ygDAIBitXTpUkVGRsrX11fr169X27ZtzY6Ef2BNAAAoFmfPntX48eMVEBCgTp06KTk5mSLgopgMAACK3J49e2S1WvXDDz9o5syZGjNmDGsBF0YZAAAUqc8//1xDhgxRzZo1tWnTJrVq1crsSLgM1gQAgCKRlZWlUaNGKTAwUPfff7+SkpIoAm6CyQAA4Krt2rVLVqtVP//8s/7zn/9oxIgRrAXcCJMBAMBV+eijj9SiRQtlZGQoLi5OI0eOpAi4GcoAAOCKZGZmatiwYQoJCVHfvn2VmJioO++80+xYuAKsCQAAhfbzzz/LarVq9+7d+uCDDzRkyBCmAW6MyQAAoFAWLFigVq1aKTc3V1u3btXQoUMpAm6OMgAAKJCMjAxFRkYqPDxcgYGBSkhIUPPmzc2OhSLAmgAAcFk//vijrFar9u3bp3nz5ik8PNzsSChCTAYAABdlGIZmz56t1q1by8vLS/Hx8RSBUogyAAC4oPT0dIWFhWno0KEKDQ3V1q1b1bRpU7NjoRiwJgAAnGfbtm2yWq06ePCgFi1apJCQELMjoRgxGQAA5DEMQ++9957atGmjChUqKDExkSLgASgDAABJUmpqqoKCgjRy5EhFRUUpLi5ON998s9mxUAJYEwAAlJiYKJvNpqNHj+qTTz6R1Wo1OxJKEJMBAPBghmHorbfeUrt27VS1alUlJydTBDwQZQAAPNTJkyc1YMAAjR07VsOHD9fGjRvVsGFDs2PBBKwJAMADbd26VTabTadOndKSJUsUEBBgdiSYiMkAAHgQwzD0+uuvq3379qpVq5aSk5MpAqAMAICnSElJUb9+/fToo49q3LhxWr9+vfz8/MyOBRfAmgAAPMCmTZsUFBSkjIwMLVu2TA888IDZkeBCmAwAQCnmdDr1r3/9S/fee6/q1aun5ORkigDOQxkAgFLq2LFj6t27tyZPnqzHHntMa9asUf369c2OBRfEmgAASqF169YpODhY2dnZ+vbbb9WjRw+zI8GFMRkAgFLE4XBo+vTp6ty5sxo1aiS73U4RwGUxGQCAUuLIkSMaNGiQoqOj9eSTT2ratGny8eH/5nF5/K8EAEqBmJiYvG8XXL16tbp162ZyIrgT1gQA4MYcDoemTZumbt266bbbbpPdbqcIoNCYDACAmzp48KBCQ0O1bt06PfPMM3ryySfl7e1tdiy4IcoAALih1atXa9CgQfLx8VF0dLQ6depkdiS4MdYEAOBGcnNz9cQTT6h79+666667ZLfbKQK4akwGAMBN/PHHHwoODtbmzZv10ksvadKkSfLy4jkdrh5lAADcwIoVKxQeHq4KFSpozZo1uueee8yOhFKESgkALiwnJ0ePPfaY+vTpo7Zt2yo5OZkigCLHZAAAXNS+ffsUFBSkhIQEvfrqq5owYQJrARQLygAAuKCvvvpKERER8vX11fr169W2bVuzI6EUo2ICgAvJzs7W+PHj9eCDD6pjx45KTk6mCKDYMRkAABexZ88e2Ww2bdu2TW+++abGjh0ri8Vidix4AMoAALiAzz//XEOGDFGNGjW0ceNG+fv7mx0JHoQ1AQCYKCsrS6NHj1ZgYKDuv/9+JScnUwRQ4pgMAIBJdu3aJZvNph07duidd97RyJEjWQvAFEwGAMAEH3/8sVq0aKH09HTFxcVp1KhRFAGYhjIAACUoMzNTw4cPV3BwsB544AElJSXprrvuMjsWPBxrAgAoIb/88ousVqt27dql//73vxo6dCjTALgEJgMAUAIWLFigli1bKicnR1u3btWwYcMoAnAZlAEAKEYZGRmKjIxUeHi4Bg4cqPj4eDVv3tzsWEA+rAkAoJjs2LFDVqtVe/fu1dy5cxUREWF2JOCCmAwAQBEzDENz5syRv7+/LBaL4uPjKQJwaZQBAChCp0+fVlhYmIYMGaKQkBBt3bpVTZs2NTsWcEmsCQCgiGzbtk1Wq1V//vmnFi5cqNDQULMjAQXCZAAArpJhGHr//ffVpk0blS9fXomJiRQBuBXKAABchbS0NAUHB2vEiBGKjIxUXFycmjRpYnYsoFBYEwDAFUpKSpLVatXRo0f1ySefyGq1mh0JuCJMBgCgkAzD0Ntvv627775bvr6+eaUAcFeUAQAohFOnTmngwIEaM2aMhg8frk2bNqlRo0ZmxwKuCmsCACigrVu3ymaz6eTJk/riiy/Uv39/syMBRYLJAABchmEYeuONN3TPPffo2muvVXJyMkUApQplAAAuISUlRf369dOECRM0ZswYbdiwQTfeeKPZsYAixZoAAC5i06ZNCgoK0unTp7Vs2TI98MADZkcCigWTAQD4B6fTqVdeeUX33nuv6tatK7vdThFAqUYZAIBzHDt2TH369NHjjz+uiRMnau3atapfv77ZsYBixZoAAP7f+vXrFRQUpOzsbH3zzTfq2bOn2ZGAEsFkAIDHczqdeuGFF9SpUyc1bNhQdrudIgCPQhkA4NGOHDmiHj16aOrUqXriiScUExOjG264wexYQIliTQDAY8XExCg0NFROp1OrVq3SfffdZ3YkwBRMBgB4HIfDoWeeeUbdunXTrbfeKrvdThGAR2MyAMCjHDp0SCEhIVq3bp2eeeYZPfnkk/L29jY7FmAqygAAj7F69WoNGjRI3t7eio6OVqdOncyOBLgE1gQASr3c3Fw9+eST6tGjh+666y5t27aNIgCcg8kAgFLtjz/+UEhIiDZu3Kjp06dr8uTJ8vLieRBwLsoAgFLrm2++0eDBg1W+fHmtWbNGHTp0MDsS4JKoxwBKnZycHE2aNEm9e/dWmzZtZLfbKQLAJTAZAFCq7N+/X0FBQYqPj9err76qCRMmsBYALoMyAKDUWLZsmSIiIlS5cmWtW7dOd999t9mRALdAXQbg9rKzs/XII4+oX79+6tChg5KTkykCQCEwGQDg1n7//XfZbDbZ7Xa9+eabGjt2rCwWi9mxALdCGQDgtr744gsNGTJE1atX18aNG+Xv7292JMAtsSYA4HaysrL08MMPa+DAgerWrZuSkpIoAsBVYDIAwK3s3r1bVqtVO3bs0Ntvv61Ro0axFgCuEpMBAG7j448/VosWLZSenq64uDiNHj2aIgAUAcoAAJeXmZmp4cOHKzg4WL1791ZiYqLuuusus2MBpQZrAgAu7ddff5XVatXOnTv13//+V0OHDmUaABQxJgMAXNbChQvVsmVLnT17Vlu2bNGwYcMoAkAxoAwAcDlnzpxRVFSUwsLC1L9/fyUkJOj22283OxZQarEmAOBSduzYIavVqt9//11z5sxRREQE0wCgmDEZAOASDMPQ3Llz8z4vICEhQZGRkRQBoARQBgCY7vTp0xo8eLCioqIUHBys+Ph4NW3a1OxYgMdgTQDAVD/88IOsVqv++OMPffjhhxo0aJDZkQCPw2QAgCkMw9B///tftW7dWuXKlVNCQgJFADAJZQBAiUtLS1NISIiGDx+uiIgIxcXF6ZZbbjE7FuCxWBMAKFHJycmyWq06cuSIPv74Y9lsNrMjAR6PyQCAEmEYht555x21bdtWVapUUVJSEkUAcBGUAQDF7tSpUwoMDNTDDz+shx56SJs2bVKjRo3MjgXg/7EmAFCs4uPjZbPZlJKSos8//1wDBgwwOxKAf2AyAKBYGIahN998U+3bt1fNmjWVnJxMEQBcFGUAQJFLSUnRgw8+qEceeUQPP/ywNmzYoBtvvNHsWAAugjUBgCK1efNmBQUFKT09XV999ZX69u1rdiQAl8FkAECRcDqdevXVV3XvvffqhhtuUHJyMkUAcBOUAQBX7fjx43rggQc0adIkTZgwQWvXrlWDBg3MjgWggFgTALgq69evV3BwsLKysrRixQr16tXL7EgAConJAIAr4nQ69eKLL6pz58666aabZLfbKQKAm6IMACi0o0ePqkePHnrqqac0efJkxcTEqG7dumbHAnCFWBMAKJTY2FiFhITI6XRq1apVuu+++8yOBOAqMRkAUCAOh0PPPvusunXrpltvvVV2u50iAJQSTAYAXNahQ4c0aNAgxcbGatq0aXrqqafk7e1tdiwARYQyAOCSvvvuOw0aNEheXl6Kjo5W586dzY4EoIixJgBwQbm5uXrqqafUvXt33XHHHbLb7RQBoJRiMgDgPH/88YdCQkK0ceNGTZ8+XZMnT5aXF88dgNKKMgAgn2+//VZhYWEqX7681qxZow4dOpgdCUAxo+oDkCTl5OTo8ccfV69evdS6dWvZ7XaKAOAhmAwA0P79+xUUFKStW7fqlVde0aOPPspaAPAglAHAwy1btkwRERGqXLmy1q9fr7vvvtvsSABKGNUf8FDZ2dmaMGGC+vXrpw4dOig5OZkiAHgoJgOAB/r9998VFBSk5ORkvfHGGxo3bpwsFovZsQCYhDIAeJglS5YoKipK1apV08aNG+Xv7292JAAmY00AeIizZ89qzJgxGjBggLp27ark5GSKAABJTAYAj7B7927ZbDb9+OOPevvttzVq1CjWAgDyMBkASrlPPvlELVq0UGpqqjZv3qzRo0dTBADkQxkASqnMzEyNGDFCQUFB6tWrl5KSktSiRQuzYwFwQawJgFLo119/ldVq1a+//qr3339fw4YNYxoA4KKYDAClzMKFC9WyZUudPXtWW7du1UMPPUQRAHBJlAGglDhz5oyGDBmisLAwBQQEKCEhQbfffrvZsQC4AdYEQCnw008/yWq1as+ePZozZ44iIiKYBgAoMCYDgJubN2+eWrVqJcMwFB8fr8jISIoAgEKhDABu6vTp0woPD1dkZGTeNw7edtttZscC4IZYEwBuaPv27bJarTpw4IAWLFigsLAwsyMBcGNMBgA3YhiGPvjgA7Vu3VplypRRQkICRQDAVaMMAG4iLS1NISEheuihhzR48GBt2bJFt9xyi9mxAJQCrAkAN5CcnCyr1arDhw/ro48+UlBQkNmRAJQiTAYAF2YYhv7zn/+obdu2qly5spKSkigCAIocZQBwUampqbJarRo9erSGDRumTZs2qXHjxmbHAlAKsSYAXFB8fLxsNptSUlL0+eefa8CAAWZHAlCKMRkAXIhhGJo5c6bat2+vmjVrKikpiSIAoNhRBgAXkZKSooCAAI0fP16jR4/Whg0bdNNNN5kdC4AHYE0AuIC4uDjZbDalpaVp6dKl6tevn9mRAHgQJgOAiZxOp1577TV16NBBderUkd1upwgAKHGUAcAkx48fV9++ffXYY4/pkUce0bp169SgQQOzYwHwQKwJABNs2LBBwcHByszM1PLly9W7d2+zIwHwYEwGgBLkdDr10ksvqVOnTvLz85PdbqcIADAdZQAoIUePHlXPnj315JNP6vHHH1dsbKzq1q1rdiwAYE0AlIQ1a9YoJCREubm5Wrlype6//36zIwFAHiYDQDFyOBx67rnn1LVrVzVp0kTbtm2jCABwOUwGgGJy+PBhhYaGKjY2Vk8//bSmTp0qb29vs2MBwHkoA0Ax+P777xUaGiovLy99//336tKli9mRAOCiWBMARSg3N1dTp07V/fffr9tvv112u50iAMDlMRkAisiff/6pkJAQbdiwQc8//7ymTJkiLy/6NgDXRxkAisDKlSsVFhamsmXLKjY2Vvfee6/ZkQCgwHjaAlyFnJwcTZ48WT179pS/v7/sdjtFAIDbYTIAXKH9+/crODhYW7Zs0b/+9S9NnDiRtQAAt0QZAK7A119/rYiICFWqVEnr1q1Tu3btzI4EAFeMpzFAIWRnZ+vRRx9V37591b59e9ntdooAALfHZAAooN9//11BQUFKTk7W66+/rvHjx8tisZgdCwCuGmUAKIAvv/xSkZGRqlatmjZs2KDWrVubHQkAigxrAuASzp49q7Fjx6p///7q2rWrkpOTKQIASh0mA8BF/Pbbb7LZbNq+fbveeustjR49mrUAgFKJMgBcwKeffqqhQ4eqVq1a2rRpk1q2bGl2JAAoNqwJgHNkZWVp5MiRstls6tmzp5KSkigCAEo9JgPA/9u5c6esVqt++eUXvffee3rooYdYCwDwCEwGAEmLFi1SixYtlJmZqS1btmj48OEUAQAegzIAj3bmzBkNHTpUgwYNUkBAgBITE3XHHXeYHQsAShRrAnisn376SVarVXv27NHs2bMVGRnJNACAR2IyAI80b948+fv7y+l0Kj4+XlFRURQBAB6LMgCPcvr0aYWHhysyMlI2m03x8fG67bbbzI4FAKZiTQCPsX37dlmtVu3fv18LFixQWFiY2ZEAwCUwGUCpZxiGZs2apdatW6tMmTJKTEykCADAOSgDKNXS09MVGhqqYcOGKSwsTFu2bNEtt9xidiwAcCmsCVBq2e12Wa1WHTp0SIsXL1ZwcLDZkQDAJTEZQKljGIbeffddtW3bVpUqVVJSUhJFAAAugTKAUiU1NVU2m02jRo3S0KFDtXnzZjVu3NjsWADg0lgToNRISEiQzWbT8ePH9dlnn2ngwIFmRwIAt8BkAG7PMAzNnDlT7dq1U/Xq1ZWcnEwRAIBCoAzArZ08eVL9+/fX+PHjNXr0aG3YsEE33XST2bEAwK2wJoDb2rJli2w2m1JTU7V06VL169fP7EgA4JaYDMDtOJ1OzZgxQ/fcc4+uv/562e12igAAXAXKANzKiRMn1LdvX02cOFHjx4/XunXr1KBBA7NjAYBbY00At7Fx40YFBQUpMzNTy5cvV+/evc2OBAClApMBuDyn06mXX35ZHTt2lJ+fn+x2O0UAAIoQZQAu7ejRo+rVq5emTJmiSZMmKTY2VnXr1jU7FgCUKqwJ4LLWrl2r4OBg5ebmauXKlerevbvZkQCgVGIyAJfjcDj0/PPPq0uXLmrSpInsdjtFAACKEZMBuJTDhw9r0KBBiomJ0dSpU/X000/L29vb7FgAUKpRBuAyoqOjFRoaKkn6/vvv1aVLF5MTAYBnYE0A0+Xm5urpp5/Wfffdp+bNm2vbtm0UAQAoQUwGYKqDBw8qODhYGzZs0HPPPacpU6awFgCAEkYZgGlWrlypsLAwlS1bVjExMerYsaPZkQDAI7EmQInLzc3VlClT1LNnT7Vq1Up2u50iAAAmYjKAEnXgwAEFBwcrLi5OL7/8sh577DF5edFJAcBMlAGUmOXLlys8PFyVKlXSunXr1K5dO7MjAQDEmgAlIDs7WxMnTtQDDzyg9u3bKzk5mSIAAC6EyQCK1d69exUUFKTExETNmDFDjzzyiCwWi9mxAADnoAyg2CxdulSRkZHy9fXVhg0b1KZNG7MjAQAugDUBitzZs2c1btw4BQQEqHPnzkpOTqYIAIALYzKAIvXbb7/JZrNp+/bt+ve//62HH36YtQAAuDjKAIrMZ599pqFDh6pmzZratGmTWrZsaXYkAEABsCbAVcvKytKoUaNktVrVo0cPJSUlUQQAwI0wGcBV2blzp6xWq3755Re9++67Gj58OGsBAHAzTAZwxRYvXqyWLVsqMzNTW7Zs0YgRIygCAOCGKAMotDNnzmjYsGEKDQ1Vv379lJCQoDvuuMPsWACAK8SaAIXy888/y2q16rffftOsWbMUFRXFNAAA3ByTARTY/Pnz1apVKzkcDm3dulVDhgyhCABAKUAZwGVlZGQoIiJCERERslqtio+PV7NmzcyOBQAoIqwJcEk//vijAgMDtX//fs2fP1+DBw82OxIAoIgxGcAFGYahWbNmyd/fXz4+PkpISKAIAEApRRnAedLT0zVo0CANGzZMYWFh2rp1q2699VazYwEAiglrAuRjt9tls9l08OBBLVq0SCEhIWZHAgAUMyYDkPTXWuDdd99V27ZtVbFiRSUmJlIEAMBDUAag1NRUBQUFadSoURoyZIg2b96sm2++2exYAIASwprAwyUmJspqter48eP69NNPFRgYaHYkAEAJYzLgoQzD0FtvvaV27dqpWrVqSkpKoggAgIeiDHigkydPasCAARo7dqxGjhypjRs3qmHDhmbHAgCYhDWBh9myZYuCgoJ06tQpffnll3rwwQfNjgQAMBmTAQ9hGIZmzJihe+65R7Vr11ZycjJFAAAgiTLgEU6cOKG+fftq4sSJGjdunNatWyc/Pz+zYwEAXARrglJu48aNCg4OVkZGhr7++mv16dPH7EgAABfDZKCUcjqdevnll9WxY0fVr19fdrudIgAAuCDKQCl07Ngx9e7dW1OmTNGkSZMUGxurevXqmR0LAOCiWBOUMuvWrVNwcLBycnK0cuVKde/e3exIAAAXx2SglHA4HJo+fbo6d+6sxo0by263UwQAAAXCZKAUOHLkiEJDQxUTE6OpU6dq6tSp8vHhv1oAQMHwiOHmoqOjFRoaKkn67rvv1LVrV5MTAQDcDWsCN+VwODRt2jTdd999atasmex2O0UAAHBFmAy4oYMHDyo0NFTr1q3Ts88+qyeeeELe3t5mxwIAuCnKgJtZtWqVwsLCVKZMGcXExKhjx45mRwIAuDnWBG4iNzdXU6ZMUY8ePdSiRQvZ7XaKAACgSDAZcAMHDhxQcHCw4uLi9NJLL2nSpEny8qLHAQCKBmXAxa1YsUKDBw9WxYoVtXbtWrVv397sSACAUoanly4qJydHjz32mPr06aN27drJbrdTBAAAxYLJgAvat2+fbDabEhMTNWPGDD3yyCOyWCxmxwIAlFKUARezdOlSRUZGytfXVxs2bFCbNm3MjgQAKOVYE7iI7OxsjR8/XgEBAerUqZOSk5MpAgCAEsFkwAXs2bNHNptN27Zt08yZMzVmzBjWAgCAEkMZMNnnn3+uIUOGqGbNmtq0aZNatWpldiQAgIdhTWCSrKwsjR49WoGBgerevbuSkpIoAgAAUzAZMMGuXbtktVr1888/6z//+Y9GjBjBWgAAYBomAyXso48+UosWLZSRkaG4uDiNHDmSIgAAMBVloIRkZmbqoYceUkhIiPr27avExETdeeedZscCAIA1QUn45ZdfFBgYqN27d2vWrFmKiopiGgAAcBlMBorZggUL1LJlS+Xm5io+Pl5DhgyhCAAAXAploJhkZGQoMjJS4eHhCgwMVEJCgpo1a2Z2LAAAzsOaoBjs2LFDVqtVe/fu1bx58xQeHm52JAAALorJQBEyDEOzZ8+Wv7+/vLy8FB8fTxEAALg8ykARSU9PV1hYmIYOHarQ0FBt2bJFTZs2NTsWAACXxZqgCGzbtk1Wq1UHDx7UokWLFBISYnYkAAAKjMnAVTAMQ++9957atGmjChUqKDExkSIAAHA7lIErlJaWpqCgII0cOVJRUVGKi4vTzTffbHYsAAAKjTXBFUhMTJTNZtOxY8f0ySefyGq1mh0JAIArxmSgEAzD0FtvvaV27dqpatWqSkpKoggAANweZaCATp06pYEDB2rs2LEaMWKENm7cqIYNG5odCwCAq8aaoAC2bt0qm82mU6dOacmSJQoICDA7EgAARYbJwCUYhqHXX39d7du3V61atZScnEwRAACUOpSBi0hJSVG/fv306KOPaty4cVq/fr38/PzMjgUAQJFjTXABmzZtUlBQkDIyMrRs2TI98MADZkcCAKDYlKrJQMbZXO04mKrk/Se142CqMs7mFur+TqdTr7zyiu69917Vq1dPdrudIgAAKPXcfjKw60i6Fm3Zr9hfj2p/yhkZ59xmkVS/ekV1blJLoW3qq3Htyhc9z7FjxxQeHq5vv/1WkydP1nPPPacyZcoUe34AAMxmMQzDuNxBaWlp8vX1VWpqqqpUqVISuS7rQMoZPfHldq3ffVzeXhY5nBf/Nf6+vUOjmnoxoLnqVa+Y7/Z169YpODhY2dnZ+vDDD9WjR4/ijg8AQLEr6OO3W64JPo7fr25vrNWmPSck6ZJF4NzbN+05oW5vrNXH8fv/ut7h0PTp09W5c2c1atRIdrudIgAA8DhutyZ4O3aXXlu984ru63AacjgNTV6yXXsPn1D0vycpOjpaTz31lJ5++mn5+LjdnwMAgKvmVo9+H8fvv+Ii8E/vbTqonCxfrV69Wt26dSuScwIA4I5KbE3w4osvaunSpYW6z5o1a2SxWPT555/rQMoZTVu2o+gCGYau6RilJi3aFd05S5DT6dSHH36obt26qWbNmipTpoxq1aqlPn366Ouvv5bT6TQ7IgDATbh0GTjXE19uV+5lXhtQKBaLco2/zutusrKy1KtXL4WHh6tWrVp69913FRMTo/fee0916tRRYGCgvv76a7NjAgDchFusCQ6dytT6344X+XkdTkPrdx/X7qPpalTr4m87dDUTJkzQqlWrNH/+fA0ePDjfbf3799djjz2mzMxMk9IBANxNoSYDp0+flq+vr0aPHi3pr2eojz76qO688075+vqqevXquvvuu/XVV1/lu5/FYlFGRobmz58vi8Uii8WiTp06Ffjnrt/119sHJSnn5EEdX/Gm/nx/mPa/NkB/vD1YRz97VtlH9553P2fWaaVEz9Kf7w7Rvlcf1IF/h+rIp9OUc+JA3jFGRor69Q9U5cqV5evrK5vNpri4OFksFs2bNy/vuE6dOl0wc0RExHkfU5ydna3p06frlltuUbly5XTttdcqMjJSx44dy3dcTEyMOnXqpBo1aqhChQqqX7++BgwYoDNnzlz0b3H48GHNmjVL3bt3P68I/K1x48a6/fbbL3oOAADOVajJwEcffaS0tLS8MnD27FmlpKRo4sSJuuGGG5Sdna3vv/9e/fv319y5c/MerDZv3qwuXbqoc+fOmjp1qiQV6vMKfjyYKkfdv1YEjvQUeVeorKodI+Rd0VfOrHSd3h6twwse1fWRM1WmRl1JkvPsGR1eOEm5qUdVpe0AlavTREZ2lrIO/CjH6ZMqU6OenDlndXjxk3JmpOjNV1/WzTffrBUrVshmsxXmz5KP0+lUv379tH79ek2aNEnt2rXTvn37NG3aNHXq1EkJCQmqUKGC9u7dq969e6tDhw6aM2eOqlatqj///FMrV65Udna2KlaseMHzx8bGKicnRw8++OAVZwQA4FyFKgOzZs1S586d1bRpU0mSr6+v5s6dm3e7w+FQ165ddfLkSb355pt5ZaBt27by8vLStddeq7Zt2xY65PHTZ/X3Q2P5+s1Uvn6zvNsMp0MVGvrr4KxRSrd/q+pdh0mS0hK+Us7x/aoVNF0V/O7MO75ik/+9YDDjx2jlnDigWgOmKnLYCFUq56P7779fmZmZ+uCDDwqdU5I+/fRTrVy5Ul988YX69++fd/0dd9whf39/zZs3TyNHjlRiYqKysrL06quv6o477sg7LiQk5JLn37//r89IuPHGG68oHwAA/1SoMvDLL7/ohRdeyHfdZ599pjfffFPbtm1TRkZG3vXly5cv0Dlzc/N/f4C3t7csFku+68592aDhdCgt7gtl7IhVzslDkvN/98858Ufef87akyif6jfkKwL/lLVvuyxlK6hC4zbaeyJDt9XxlfTXA/KVloHly5eratWqeuCBB/L9bnfeeaeuu+46rVmzRiNHjtSdd96psmXL6qGHHtKoUaPUoUMH3XTTTVf0MwEAuBqFes3ADTfcoH79+uVdXrJkiaxWq2644QYtXLhQmzdvVnx8vKKiopSVlVWgc5YpUybfv/nz51/y+JPRs3Rq/UJVuLmtag2cqusGz9B14W+oTK0bZeSczTvOcSZVPpVrXPJczsw0eVeqKknKzv3fW/Guu+66AmW/kCNHjujUqVMqW7bseb/b4cOHdfz4Xy+EbNiwob7//nvVqlVLo0ePVsOGDdWwYUPNnDnzkuevX7++JOn333+/4owAAJyrUJOByMhIeXt7511euHChbrzxRn3yySf5ns2fPXv2Qne/oPj4+HyXLzf+ztgRq0rNuqhax/B81zsz0+RVrlLeZe+KvspNP3HJc3lVqCLHob8+xKisz/960eHDh887tnz58kpNTT3v+r8f3P9Ws2ZN1ahRQytXrrzgz6xc+X/vWujQoYM6dOggh8OhhIQEvfXWWxo/frxq166toKCgC96/c+fOKlOmjJYuXaoRI0Zc8vcDAKAgCjUZiIiIyHfZYrGobNmy+YrA4cOHz3s3gSSVK1fugm93a9WqVb5/NWqc/2ze8o9LFu/8HebM7ng5/vHAX/6mlspN+VOZe7dd9Pcp36C5jOxMZe7aIr8a/ysSixcvPu9YPz8/7dy5M1/ROXHihDZt2pTvuD59+ujEiRNyOBzn/W6tWrVSkyZNzju3t7e32rRpo3feeUeSlJSUdNHM1113nYYOHapVq1ZpwYIFFzzmt99+0w8//HDRcwAAcK5CTQauvfbafJf79OmjJUuWaNSoURo4cKAOHDig559/Xtdff7127dqV79jmzZtrzZo1+vrrr3X99dercuXKF3xgvJCalcvr7zfbVWjkr9Pbo1WmRj2VudZP2Ud2K23LEnlXrpnvPlVa9dOZn9fr2JLp8m07UGWvv1lGbray9m9XxUatVb7B7arUrIvS4r/SiRWva96s2mrcuLG++eYbrVq16rwMYWFhev/99zVo0CANGzZMJ06c0CuvvHLeuyKCgoK0aNEi9erVS+PGjVPr1q1VpkwZ/fHHH4qNjVW/fv0UEBCg9957TzExMerdu7fq16+vrKwszZkzR5Iu+/HIr7/+uvbs2aOIiAitWrVKAQEBql27to4fP67vvvtOc+fO1ccff5z39kIfHx917NhR0dHRBfp7AwA8jFEAqamphiQjNTX1vNtefvllw8/PzyhXrpxx6623Gh988IExbdo045+nttvtRvv27Y2KFSsakoyOHTte9ueuWLHCkGQEP/W2cdMTK4wGk5cb9cZ/bFxz+/2GV8WqhqVMOaNc3aZG7dB/GeXqNTPK1WtmNJi8PO9fvfEfG5Vb9TW8q1xryMvH8KpY1ajQ0N+oM+y9vGPqj5lv3Hr3fcY111xjVK5c2RgwYICxadMmQ5Ixd+7cfHnmz59v3HrrrUb58uWNpk2bGp988okRHh5uNGjQIN9xOTk5xmuvvWbccccdRvny5Y1rrrnGuOWWW4zhw4cbu3btMgzDMDZv3mwEBAQYDRo0MMqVK2fUqFHD6Nixo7Fs2bKC/Fdi5ObmGvPnzze6dOliVK9e3fDx8TGuvfZao2fPnsbixYsNh8ORd2xB/94AgNLlUo/f57IYhnHZz/gt6PchF7U33nhDEyZM0Ddrt2rkt0eL7ed8/8i9+T6BcO/evbrxxhs1d+7c81YjAAC4i4I+fpfYdxMUxo4dO7Rw4UK98soruvPOO9XzXn91aFQz71MIi4q3l0UdGtV0q48iBgCgqLlkGXj44Yc1atQotWrVSl9++aUk6cWA5vIp4jLg42XRiwHNi/ScAAC4G5f8oqLY2NjzrqtXvaKe7XubJi8pum8ZfK7vbapX/fyP/fXz81MBticAAJQKLjkZuJgg//qaeP/NRXKux+5vIpt//SI5FwAA7swlJwOX8nDnxqp5TTlNW7ZDuU5DDmfBn8F7e1nk42XRc31vowgAAPD/3Goy8Lcg//r6/pGOanfTXx9QdLkXFv59e7ubauj7RzpSBAAAOIfbTQb+Vq96RX04pI12HUnXoi37FbvzqPafOJPvS40skurXqKjON9fSoLb1edcAAAAX4NKfM1BYGWdztfdEhrJznSrr4yW/GpVUqZzb9h0AAK5KQR+/S9UjZaVyPnlfQwwAAArGLV8zAAAAig5lAAAAD0cZAADAw1EGAADwcJQBAAA8HGUAAAAPRxkAAMDDUQYAAPBwlAEAADwcZQAAAA9HGQAAwMNRBgAA8HCUAQAAPBxlAAAAD0cZAADAw1EGAADwcD4FOcgwDElSWlpasYYBAABF5+/H7b8fxy+mQGUgPT1dklSvXr2rjAUAAEpaenq6fH19L3q7xbhcXZDkdDp18OBBVa5cWRaLpUgDAgCA4mEYhtLT01WnTh15eV38lQEFKgMAAKD04gWEAAB4OMoAAAAejjIAAICHowwAAODhKAMAAHg4ygAAAB6OMgAAgIf7Px00KTtM7maFAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nx.draw_networkx(G)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "ca94df13",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_18454/346134509.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m     \u001b[0msource\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mtarget\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0mweight\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlink\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "sources = []\n",
    "targets = []\n",
    "weights = []\n",
    "\n",
    "for i in link_num:\n",
    "    sources.append(i[0])\n",
    "    targets.append(i[1])\n",
    "    weights.append(link.count(i))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67e00f76",
   "metadata": {},
   "outputs": [],
   "source": [
    "link_df = pd.DataFrame()\n",
    "link_df[\"source\"] = source\n",
    "link_df[\"target\"] = target\n",
    "link_df[\"weight\"] = weight"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74eb5d91",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "e6d23e78",
   "metadata": {},
   "outputs": [],
   "source": [
    "source_target = {}\n",
    "\n",
    "for i in set_link:\n",
    "#     G.add_edge(id_name[set_link[0][0]], id_name[set_link[0][1]], weight=link.count(set_link[0]))\n",
    "    source_target = {\"source\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "149c0369",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9a6e419",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "4b7a2ecc",
   "metadata": {},
   "source": [
    "## Much smaller dataset\n",
    "\n",
    "|Search within|Search document|\n",
    "|-----|-----|\n",
    "|Language|english|\n",
    "|Affiliation country|netherlands|\n",
    "|Keywords|social AND networks AND analysis AND collaboration|\n",
    "|Published from|2018|"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4952c2c2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "metadata2 = pd.read_csv(\"./smaller.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4223f853",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['57219404780', '7006530227', '18934205300', '57191728473']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test1 = metadata2['Author(s) ID'][0]\n",
    "currentline2 = test1.split(\";\")\n",
    "currentline2.pop()\n",
    "currentline2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4fed599c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "89\n",
      "84\n",
      "89\n",
      "84\n"
     ]
    }
   ],
   "source": [
    "name = []\n",
    "for i in metadata2[\"Authors\"]:\n",
    "    name = name + i.split(\", \")\n",
    "aid = []\n",
    "for j in metadata2[\"Author(s) ID\"]:\n",
    "    id_tem = j.split(\";\")\n",
    "    id_tem.pop()\n",
    "    aid = aid + id_tem\n",
    "    \n",
    "print(len(aid))\n",
    "print(len(set(aid)))\n",
    "print(len(name))\n",
    "print(len(set(name)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8affe153",
   "metadata": {},
   "outputs": [],
   "source": [
    "name = []\n",
    "aid = []\n",
    "id_name = {}\n",
    "link = []\n",
    "\n",
    "def getCombinations(seq):\n",
    "    combinations = list()\n",
    "    for i in range(0,len(seq)):\n",
    "        for j in range(i+1,len(seq)):\n",
    "            combinations.append([seq[i],seq[j]])\n",
    "    return combinations\n",
    "\n",
    "for i in metadata2.index:\n",
    "    name_temp = metadata2[\"Authors\"][i].split(\", \")\n",
    "    aid_temp = metadata2[\"Author(s) ID\"][i].split(\";\")\n",
    "    aid_temp.pop()\n",
    "    if len(name_temp) >= len(aid_temp):\n",
    "        for i in range(len(aid_temp)):\n",
    "            id_name[aid_temp[i]] = name_temp[i]\n",
    "\n",
    "    link = link+list(itertools.combinations(aid_temp, 2))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "fafb2d2b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "795"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(link)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2a2a3364",
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter\n",
    "link_num = Counter(link)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "69157e15",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(('57477841700', '55348668300'), 2),\n",
       " (('57477841700', '6602098289'), 2),\n",
       " (('57477841700', '7004691405'), 2),\n",
       " (('57477841700', '6701326090'), 2),\n",
       " (('55348668300', '6602098289'), 2),\n",
       " (('55348668300', '7004691405'), 2),\n",
       " (('55348668300', '6701326090'), 2),\n",
       " (('6602098289', '7004691405'), 2),\n",
       " (('6602098289', '6701326090'), 2),\n",
       " (('7004691405', '6701326090'), 2)]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "link_num.most_common(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f285d177",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "785"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(link_num)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cc0ef8f",
   "metadata": {},
   "source": [
    "### pyvis\n",
    "\n",
    "https://pyvis.readthedocs.io/en/latest/index.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "14d16fa1",
   "metadata": {},
   "outputs": [],
   "source": [
    "sources = []\n",
    "targets = []\n",
    "weights = []\n",
    "\n",
    "for i in link_num:\n",
    "    sources.append(i[0])\n",
    "    targets.append(i[1])\n",
    "    weights.append(link.count(i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "2a42b26c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "a = 0\n",
    "for i in range(len(sources)):\n",
    "    for j in range(len(sources)):\n",
    "        if i != j and sources[i] ==targets[j] and sources[j] ==targets[i]:\n",
    "            i = i+1\n",
    "print(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "cb15f43f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "N = Network(height=800, width=\"100%\", bgcolor=\"#222222\", font_color=\"white\", select_menu=True)\n",
    "N.toggle_hide_edges_on_drag(False)\n",
    "N.barnes_hut()\n",
    "\n",
    "edge_data = zip(sources, targets, weights)\n",
    "\n",
    "for e in edge_data:\n",
    "    src = e[0]\n",
    "    dst = e[1]\n",
    "    w = e[2]\n",
    "\n",
    "    N.add_node(src, src, title=id_name[src])\n",
    "    N.add_node(dst, dst, title=id_name[dst])\n",
    "    N.add_edge(src, dst, value=5*w)\n",
    "\n",
    "neighbor_map = N.get_adj_list()\n",
    "\n",
    "# add neighbor data to node hover data\n",
    "for node in N.nodes:\n",
    "    neighbors = []\n",
    "    for neighbor_id in neighbor_map[node[\"id\"]]:\n",
    "        neighbors.append(id_name[neighbor_id])\n",
    "    node[\"title\"] = \" Neighbors: \\n\" + \" \\n\".join(neighbors)\n",
    "    node[\"value\"] = 5*len(neighbor_map[node[\"id\"]])\n",
    "    node[\"label\"] = id_name[node[\"id\"]]\n",
    "    node[\"font_size\"] = 5*len(neighbor_map[node[\"id\"]])\n",
    "\n",
    "N.show(\"coauthorship.html\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc3dfab6",
   "metadata": {},
   "source": [
    "### d3graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "6098b09d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[d3graph] INFO> Set directed=True to see the markers!\n",
      "[d3graph] INFO> Keep only edges with weight>0\n",
      "[d3graph] INFO> Number of unique nodes: 84\n",
      "[d3graph] INFO> Slider range is set to [0, 4]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[colourmap]> Warning: Colormap [Set1] can not create [12] unique colors! Available unique colors: [9].\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[d3graph] INFO> Write to path: [/tmp/tmp363t42b3/d3graph.html]\n",
      "[d3graph] INFO> File already exists and will be overwritten: [/tmp/tmp363t42b3/d3graph.html]\n"
     ]
    }
   ],
   "source": [
    "from d3graph import d3graph, vec2adjmat\n",
    "\n",
    "adjmat = vec2adjmat(sources, targets, weight=weights)\n",
    "\n",
    "# Initialize\n",
    "d3 = d3graph()\n",
    "d3.graph(adjmat)\n",
    "d3.show(\"d3graph.html\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e7ba015",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "84249b96",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Authors                      0\n",
       "Author(s) ID                 0\n",
       "Title                        0\n",
       "Year                         0\n",
       "Cited by                     2\n",
       "Link                         0\n",
       "Affiliations                 0\n",
       "Authors with affiliations    0\n",
       "Abstract                     0\n",
       "Author Keywords              1\n",
       "Index Keywords               1\n",
       "References                   0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata2.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9c6c0ccb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     /Users/jennifer/nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "import nltk\n",
    "from nltk.corpus import stopwords\n",
    "nltk.download(\"stopwords\")\n",
    "stop_words = set(stopwords.words(\"english\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5e7a15b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_text(text):\n",
    "    text = text.replace('\\n',\" \") \n",
    "    text = re.sub(r\"-\", \" \", text) \n",
    "    text = re.sub(r\"\\d+/\\d+/\\d+\", \"\", text) \n",
    "    text = re.sub(r\"[0-2]?[0-9]:[0-6][0-9]\", \"\", text) \n",
    "    text = re.sub(r\"[\\w]+@[\\.\\w]+\", \"\", text) \n",
    "    text = re.sub(r\"/[a-zA-Z]*[:\\//\\]*[A-Za-z0-9\\-_]+\\.+[A-Za-z0-9\\.\\/%&=\\?\\-_]+/i\", \"\", text)\n",
    "    pure_text = ''\n",
    "    \n",
    "    for letter in text:\n",
    "        # Leave only letters and spaces\n",
    "        if letter.isalpha() or letter==' ':\n",
    "            letter = letter.lower()\n",
    "            pure_text += letter\n",
    "    text = ' '.join(word for word in pure_text.split() if word not in stop_words)\n",
    "    return text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7bd42fd8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[['introduction', 'students', 'formal', 'networks', 'formed', 'formal', 'curriculum', 'design', 'formally', 'organized', 'study', 'groups', 'within', 'learning', 'communities', 'lcs', 'may', 'benefit', 'students', 'interactions', 'learning', 'unclear', 'large scale', 'lcs', 'contribute', 'formation', 'different', 'informal', 'peer', 'relationships', 'refers', 'student', 'self organized', 'class', 'relationships', 'two', 'mechanisms', 'explain', 'relationship', 'formation', 'lcs', 'propinquity', 'within', 'formal', 'networks', 'homophily', 'students', 'characteristics', 'nationality', 'sex', 'academic', 'performance', 'may', 'promote', 'students', 'peer', 'relationships', 'study', 'explores', 'extent', 'formation', 'students', 'informal', 'networks', 'determined', 'formal', 'networks', 'lcs', 'controlling', 'students', 'characteristics', 'mechanisms', 'play', 'important', 'role', 'methods', 'online', 'surveys', 'data', 'collected', 'five', 'informal', 'networks', 'help seeking', 'collaboration', 'information', 'sharing', 'friendship', 'learn', 'first', 'second', 'bachelor', 'year', 'medical', 'students', 'relationships', 'students', 'divided', 'four', 'lcs', 'formal', 'curriculum', 'compared', 'students', 'five', 'informal', 'network', 'structures', 'first', 'second year', 'students', 'domestic', 'international', 'students', 'within', 'formal', 'networks', 'besides', 'used', 'quadratic', 'assignment', 'procedure', 'qap', 'regression', 'analysis', 'ucinet', 'investigate', 'associations', 'students', 'informal', 'formal', 'networks', 'lcs', 'students', 'characteristics', 'results', 'propinquity', 'lc', 'plays', 'role', 'since', 'students', 'informal', 'connections', 'within', 'lcs', 'lcs', 'furthermore', 'seems', 'play', 'greater', 'role', 'second year', 'students', 'first year', 'students', 'homophily', 'nationality', 'important', 'informal', 'networking', 'since', 'students', 'likely', 'connect', 'others', 'similar', 'nationalities', 'conclusion', 'students', 'become', 'connected', 'within', 'lc', 'remain', 'lc', 'longer', 'period', 'formal', 'networks', 'enhance', 'students', 'informal', 'interactions', 'within', 'lcs', 'seem', 'restrict', 'interactions', 'among', 'students', 'lcs', 'international', 'students', 'need', 'support', 'order', 'integrate', 'domestic', 'students', 'lcs', 'authors', 'published', 'informa', 'uk', 'limited', 'trading', 'taylor', 'francis', 'group'], ['background', 'needs', 'families', 'social', 'behavioral', 'health', 'problems', 'often', 'exceed', 'expertise', 'possibilities', 'single', 'professional', 'service', 'organization', 'cross service', 'collaboration', 'indispensable', 'adequately', 'meeting', 'needs', 'despite', 'progressive', 'focus', 'organizing', 'integrated', 'care', 'service', 'fragmentation', 'service', 'duplication', 'remain', 'persistent', 'problems', 'child', 'welfare', 'healthcare', 'service', 'delivery', 'systems', 'crucial', 'factor', 'overcome', 'problems', 'information', 'exchange', 'organizations', 'study', 'explores', 'compares', 'development', 'time', 'structures', 'information', 'exchange', 'networks', 'concerning', 'material', 'knowledge based', 'information', 'methods', 'comparative', 'case', 'study', 'social', 'network', 'analysis', 'three', 'inter organizational', 'networks', 'child', 'welfare', 'healthcare', 'services', 'different sized', 'municipalities', 'netherlands', 'research', 'population', 'consisted', 'organizations', 'various', 'sectors', 'participating', 'networks', 'data', 'collected', 'two', 'moments', 'time', 'mixed', 'method', 'semi structured', 'interviews', 'network', 'managers', 'online', 'questionnaire', 'network', 'members', 'density', 'degree', 'centralization', 'used', 'examine', 'information', 'exchange', 'structures', 'ucinet', 'used', 'analyze', 'data', 'use', 'statistical', 'tests', 'compare', 'density', 'procedure', 'quadratic', 'assignment', 'procedure', 'results', 'study', 'shows', 'different', 'structures', 'information', 'exchange', 'distinguished', 'concerning', 'material', 'knowledge based', 'information', 'overall', 'connectedness', 'studied', 'structures', 'networks', 'quite', 'similar', 'way', 'involvement', 'structured', 'turns', 'different', 'networks', 'time', 'overall', 'connectedness', 'structures', 'appears', 'stable', 'internal', 'dynamics', 'reveals', 'major', 'change', 'relationships', 'organizations', 'networks', 'conclusions', 'study', 'yields', 'empirical', 'evidence', 'existence', 'differences', 'structures', 'dynamics', 'material', 'knowledge based', 'information', 'exchange', 'relationships', 'loss', 'half', 'relations', 'year', 'relationships', 'organizations', 'network', 'stable', 'time', 'contrast', 'major', 'internal', 'dynamics', 'stable', 'overall', 'connectedness', 'important', 'point', 'concern', 'network', 'managers', 'public', 'officials', 'since', 'impermanence', 'relations', 'means', 'long term', 'integrated', 'care', 'cannot', 'guaranteed', 'authors'], ['purpose', 'help', 'ensure', 'children', 'social', 'behavioral', 'health', 'problems', 'get', 'support', 'services', 'need', 'organizations', 'collaborate', 'cross sectoral', 'networks', 'article', 'authors', 'explore', 'compare', 'structure', 'complex', 'child', 'service', 'delivery', 'networks', 'terms', 'differentiation', 'composition', 'integration', 'interconnection', 'particular', 'authors', 'investigate', 'structure', 'client', 'referral', 'identify', 'organizations', 'prominent', 'within', 'network', 'structure', 'could', 'therefore', 'fulfill', 'coordinating', 'role', 'designmethodologyapproach', 'authors', 'used', 'comparative', 'case', 'study', 'approach', 'social', 'network', 'analysis', 'three', 'interorganizational', 'networks', 'consisting', 'organizations', 'within', 'dutch', 'child', 'service', 'delivery', 'system', 'semi structured', 'interviews', 'network', 'managers', 'conducted', 'online', 'questionnaire', 'sent', 'representatives', 'network', 'members', 'findings', 'networks', 'similarly', 'differentiated', 'sectors', 'various', 'tasks', 'remarkably', 'network', 'members', 'contact', 'average', 'organizations', 'fairly', 'high', 'number', 'handled', 'successfully', 'terms', 'integration', 'authors', 'found', 'striking', 'diversity', 'structures', 'client', 'referral', 'organizations', 'gatekeeper', 'task', 'hold', 'central', 'positions', 'originalityvalue', 'due', 'scarcity', 'comparative', 'whole', 'network', 'research', 'field', 'strength', 'study', 'deeper', 'understanding', 'differentiation', 'integration', 'complex', 'child', 'service', 'delivery', 'systems', 'insights', 'crucial', 'order', 'deliver', 'needed', 'services', 'minimize', 'service', 'silos', 'fragmentation', 'mariëlle', 'blanken', 'jolanda', 'mathijssen', 'chijs', 'van', 'nieuwenhuizen', 'jörg', 'raab', 'hans', 'van', 'oers'], ['sustainability', 'requires', 'combination', 'meaningful', 'co production', 'locally', 'relevant', 'solutions', 'synthesis', 'insights', 'gained', 'across', 'regions', 'increased', 'cooperation', 'science', 'policy', 'practice', 'programme', 'ecosystem', 'change', 'society', 'pecs', 'coordinating', 'place based', 'social ecological', 'sustainability', 'research', 'pbsesr', 'across', 'globe', 'emphasizes', 'need', 'regional', 'scientific', 'nodes', 'diverse', 'biocultural', 'regions', 'inform', 'sustainability', 'science', 'action', 'paper', 'assess', 'strengths', 'pbsesr', 'communities', 'latin', 'america', 'caribbean', 'lac', 'provide', 'overview', 'pbsesr', 'literature', 'associated', 'region', 'highlight', 'achievements', 'two', 'prominent', 'regional', 'networks', 'social ecological', 'systems', 'sustainability', 'research', 'network', 'mexico', 'socioecos', 'south', 'american', 'institute', 'resilience', 'sustainability', 'studies', 'uruguay', 'saras', 'institute', 'finally', 'identify', 'potential', 'nodes', 'constitute', 'regional', 'pecs', 'node', 'latin', 'america', 'discuss', 'capacity', 'needed', 'ensure', 'function', 'results', 'literature', 'review', 'show', 'still', 'loosely', 'interconnected', 'across', 'region', 'networks', 'play', 'key', 'roles', 'connecting', 'otherwise', 'cloistered', 'teams', 'illustrate', 'socioecos', 'network', 'focusing', 'transdisciplinary', 'co production', 'knowledge', 'towards', 'sustainability', 'saras', 'institute', 'focusing', 'innovative', 'approaches', 'looking', 'complex', 'social ecological', 'problems', 'rooted', 'slow', 'science', 'arts', 'operate', 'key', 'connectors', 'region', 'conclude', 'organizations', 'combined', 'embody', 'latin', 'american', 'node', 'pecs', 'would', 'thereby', 'contribute', 'regional', 'also', 'global', 'capacities', 'advance', 'sustainability', 'agenda', 'authors', 'published', 'informa', 'uk', 'limited', 'trading', 'taylor', 'francis', 'group'], ['background', 'due', 'multimorbidity', 'geriatric', 'problems', 'older', 'people', 'often', 'require', 'psychosocial', 'medical', 'care', 'collaboration', 'medical', 'social', 'professionals', 'prerequisite', 'deliver', 'high quality', 'care', 'community living', 'older', 'people', 'effective', 'safe', 'person centered', 'care', 'relies', 'skilled', 'interprofessional', 'collaboration', 'practice', 'little', 'known', 'interprofessional', 'education', 'increase', 'interprofessional', 'collaboration', 'practice', 'ipcp', 'context', 'community', 'care', 'older', 'people', 'study', 'examines', 'feasibility', 'implementation', 'ipcp', 'program', 'three', 'community', 'districts', 'determines', 'potential', 'increase', 'interprofessional', 'collaboration', 'primary', 'healthcare', 'professionals', 'caring', 'older', 'people', 'method', 'feasibility', 'study', 'conducted', 'determine', 'acceptability', 'feasibility', 'data', 'collection', 'analysis', 'regarding', 'interprofessional', 'collaboration', 'network', 'development', 'questionnaire', 'used', 'measure', 'learning', 'experience', 'acquisition', 'knowledge', 'skills', 'regarding', 'program', 'network', 'development', 'assessed', 'distributing', 'social', 'network', 'survey', 'among', 'professionals', 'attending', 'program', 'well', 'professionals', 'attending', 'program', 'baseline', 'months', 'network', 'development', 'determined', 'calculating', 'number', 'reciprocity', 'value', 'diversity', 'contacts', 'professionals', 'using', 'social', 'network', 'analysis', 'results', 'ipcp', 'program', 'found', 'instructive', 'knowledge', 'skills', 'gained', 'applicable', 'practice', 'social', 'network', 'analysis', 'feasible', 'conduct', 'revealed', 'spill', 'effect', 'regarding', 'network', 'development', 'program', 'participants', 'well', 'non program', 'participants', 'larger', 'reciprocal', 'diverse', 'interprofessional', 'networks', 'program', 'conclusions', 'study', 'showed', 'feasibility', 'implementing', 'ipcp', 'program', 'terms', 'acceptability', 'feasibility', 'data', 'collection', 'social', 'network', 'analysis', 'measure', 'network', 'development', 'indicated', 'potential', 'increase', 'interprofessional', 'collaboration', 'primary', 'healthcare', 'professionals', 'program', 'participants', 'non program', 'participants', 'developed', 'larger', 'collaborative', 'diverse', 'interprofessional', 'network', 'authors'], ['climate', 'change', 'various', 'economic', 'environmental', 'social', 'consequences', 'one', 'greatest', 'challenges', 'faced', 'society', 'climate', 'change', 'governance', 'forestry', 'nature', 'conservation', 'includes', 'developing', 'joint', 'activities', 'collaboration', 'among', 'stakeholders', 'combine', 'different', 'interests', 'influences', 'competences', 'national', 'regional', 'local', 'levels', 'research', 'aims', 'classify', 'climate', 'change', 'stakeholders', 'within', 'forestry', 'nature', 'conservation', 'sectors', 'serbia', 'classified', 'according', 'interests', 'perceived', 'influences', 'analyze', 'factors', 'impacting', 'development', 'different', 'areas', 'collaboration', 'combining', 'stakeholder', 'analysis', 'social', 'network', 'analysis', 'total', 'representatives', 'civil', 'society', 'public', 'sector', 'organizations', 'forestry', 'nature', 'conservation', 'different', 'governance', 'levels', 'expertise', 'climate', 'change', 'participated', 'survey', 'results', 'show', 'civil', 'sector', 'organizations', 'distributed', 'subject', 'quadrant', 'lower', 'perceived', 'influence', 'well', 'interconnected', 'seven', 'different', 'areas', 'collaboration', 'identified', 'disconnected', 'stakeholders', 'limited', 'representation', 'mostly', 'peripheral', 'position', 'civil', 'society', 'organizations', 'except', 'case', 'area', 'collaboration', 'workshop', 'seminars', 'knowledge', 'exchange', 'analyzed', 'factors', 'different', 'positive', 'negative', 'effects', 'development', 'different', 'areas', 'collaboration', 'frequency', 'contacts', 'standing', 'significant', 'factor', 'collaboration', 'level', 'whole', 'collaboration', 'network', 'strong', 'indication', 'centralized', 'top', 'approach', 'climate', 'change', 'governance', 'forestry', 'nature', 'conservation', 'serbia', 'multilevel', 'horizontal', 'stakeholder', 'governance', 'needed', 'achieve', 'effective', 'implementation', 'strategic', 'climate change', 'policy', 'commitments', 'important', 'step', 'achieve', 'structure', 'empowerment', 'local level', 'organizations', 'climate', 'change', 'collaboration', 'authors', 'licensee', 'mdpi', 'basel', 'switzerland'], ['interdisciplinary', 'research', 'essential', 'study', 'complex', 'systems', 'growing', 'need', 'understand', 'factors', 'facilitate', 'collaboration', 'across', 'diverse', 'fields', 'inquiry', 'exploratory', 'study', 'examine', 'composition', 'self organized', 'project', 'groups', 'structure', 'collaboration', 'networks', 'santa', 'fe', 'institutes', 'complex', 'systems', 'summer', 'school', 'using', 'data', 'iterations', 'summer', 'school', 'comprising', 'participants', 'projects', 'investigate', 'factors', 'contribute', 'group', 'composition', 'first', 'test', 'homophily', 'respect', 'individual level', 'attributes', 'finding', 'group', 'composition', 'largely', 'consistent', 'random', 'mixing', 'based', 'gender', 'career', 'position', 'institutional', 'prestige', 'country', 'study', 'however', 'find', 'evidence', 'homophilic', 'preference', 'group', 'composition', 'based', 'disciplinary', 'background', 'conduct', 'analyses', 'level', 'group', 'projects', 'finding', 'project', 'topics', 'social', 'behavioral', 'sciences', 'represented', 'could', 'due', 'higher', 'level', 'baseline', 'interest', 'knowledge', 'social', 'behavioral', 'sciences', 'common', 'application', 'methods', 'natural', 'sciences', 'problems', 'social', 'sciences', 'consequently', 'future', 'research', 'explore', 'discrepancy', 'examine', 'whether', 'mitigated', 'policies', 'aimed', 'making', 'topics', 'disciplines', 'accessible', 'appealing', 'collaboration', 'brown', 'et', 'al', 'open', 'access', 'article', 'distributed', 'terms', 'creative', 'commons', 'attribution', 'license', 'permits', 'unrestricted', 'use', 'distribution', 'reproduction', 'medium', 'provided', 'original', 'author', 'source', 'credited'], ['work', 'aims', 'understand', 'intra', 'interprofessional', 'networks', 'general', 'practitioners', 'gps', 'ear', 'nose', 'throat', 'specialists', 'ent', 'specialists', 'manner', 'supervisors', 'specialties', 'involve', 'interns', 'professional', 'network', 'help', 'learn', 'intra', 'interprofessional', 'collaboration', 'egocentric', 'social', 'network', 'approach', 'used', 'collect', 'analyze', 'quantitative', 'well', 'qualitative', 'data', 'semi structured', 'interviews', 'held', 'ten', 'gp', 'ten', 'ent', 'specialists', 'gps', 'significantly', 'interprofessional', 'contacts', 'ent', 'specialists', 'p', 'significant', 'difference', 'network', 'sizes', 'professions', 'p', 'supervisors', 'involved', 'interns', 'egonetwork', 'actively', 'well', 'passively', 'actively', 'discussed', 'collaboration', 'professionals', 'evolved', 'passively', 'assumed', 'intern', 'would', 'learn', 'observing', 'supervisors', 'network', 'interactions', 'many', 'supervisors', 'considered', 'interns', 'initiative', 'essential', 'deciding', 'involve', 'intern', 'network', 'although', 'workplace', 'gps', 'differed', 'notably', 'hospital', 'settings', 'ent', 'specialists', 'work', 'network', 'sizes', 'comparable', 'clerkships', 'general', 'practice', 'seemed', 'provide', 'opportunities', 'learn', 'interprofessional', 'collaboration', 'example', 'medical', 'nurse', 'supervisors', 'specialties', 'could', 'involve', 'interns', 'actively', 'intra', 'interprofessional', 'network', 'interns', 'could', 'take', 'initiative', 'learn', 'collaboration', 'supervisors', 'network', 'authors', 'published', 'license', 'taylor', 'francis', 'group', 'llc'], ['paper', 'conducts', 'explorative', 'analysis', 'uks', 'nanotechnology', 'research', 'collaboration', 'network', 'understand', 'contributions', 'different', 'institutions', 'development', 'generation', 'knowledge', 'framed', 'national', 'innovation', 'system', 'nis', 'triple', 'helix', 'th', 'model', 'literature', 'paper', 'makes', 'use', 'social', 'network', 'analysis', 'sna', 'tools', 'identify', 'role', 'involvement', 'different', 'institutional', 'actors', 'interactions', 'collaborations', 'within', 'nanotechnology', 'network', 'building', 'traditional', 'universityindustrygovernment', 'three helix', 'interaction', 'model', 'paper', 'includes', 'two', 'extra', 'dimensions', 'model', 'account', 'increase', 'international', 'collaboration', 'increasingly', 'important', 'role', 'non profit', 'organizations', 'npos', 'knowledge', 'generation', 'way', 'paper', 'responds', 'recent', 'calls', 'adapt', 'traditional', 'nis', 'models', 'reflect', 'new', 'realities', 'scientific', 'collaboration', 'elsevier', 'inc'], ['inclusive', 'early', 'childhood', 'education', 'care', 'ecec', 'requires', 'interprofessional', 'collaboration', 'professionals', 'diverse', 'professional', 'backgrounds', 'following', 'developments', 'human', 'services', 'growing', 'interest', 'role', 'interprofessional', 'teams', 'community based', 'settings', 'young', 'children', 'three year', 'longitudinal', 'study', 'investigated', 'interprofessional', 'collaboration', 'ipc', 'staff', 'childcare', 'education', 'youth', 'care', 'individual', 'level', 'survey', 'used', 'analysis', 'ipc', 'competences', 'network', 'level', 'investigated', 'professional', 'relationships', 'individuals', 'using', 'social', 'network', 'analysis', 'results', 'multilevel', 'mixed', 'linear', 'model', 'showed', 'increase', 'interdependence', 'reflection', 'process', 'individual', 'staff', 'followed', 'progressive', 'development', 'perceived', 'team', 'performance', 'smaller', 'networks', 'higher', 'density', 'professionals', 'centrality', 'predicted', 'positive', 'perceptions', 'inclusive', 'ecec', 'discuss', 'findings', 'context', 'growing', 'interest', 'interprofessional', 'teams', 'ecec', 'taylor', 'francis', 'group', 'llc'], ['social', 'intellectual', 'relations', 'structured', 'given', 'shape', 'within', 'research', 'universities', 'answer', 'questions', 'test', 'extent', 'various', 'theoretically', 'predicted', 'processes', 'explain', 'dynamics', 'academics', 'networks', 'collaboration', 'shared', 'language', 'use', 'unique', 'longitudinal', 'data', 'set', 'faculty', 'large', 'private', 'american', 'university', 'using', 'latest', 'advances', 'stochastic', 'actor oriented', 'models', 'rsiena', 'text', 'analysis', 'found', 'social', 'intellectual', 'relations', 'clustered', 'centralized', 'bridging', 'faculty', 'form', 'broader', 'interdisciplinary', 'hub', 'research', 'university', 'time', 'hub', 'disseminated', 'style', 'interdisciplinary', 'research', 'faculty', 'networks', 'shaped', 'selection', 'based', 'age', 'gender', 'race', 'academic', 'rank', 'well', 'coevolution', 'social', 'intellectual', 'relations', 'time', 'clear', 'differences', 'emerge', 'science', 'technology', 'engineering', 'mathematics', 'stem', 'fields', 'strongly', 'driven', 'structural', 'mechanisms', 'clustering', 'centralization', 'whereas', 'non stem', 'fields', 'social', 'sciences', 'humanities', 'strongly', 'driven', 'personal', 'preferences', 'faculty', 'members', 'authors', 'open access', 'article', 'published', 'creative', 'commons', 'attribution', 'license', 'allows', 'unrestricted', 'use', 'distribution', 'reproduction', 'form', 'long', 'original', 'author', 'source', 'credited'], ['background', 'south south', 'collaboration', 'health', 'development', 'research', 'critical', 'mechanism', 'social', 'economic', 'progress', 'allows', 'sharing', 'replicating', 'experiences', 'find', 'southern', 'solution', 'meet', 'shared', 'health', 'challenges', 'access', 'adequate', 'hivaids', 'prevention', 'treatment', 'study', 'aimed', 'generate', 'evidence', 'dynamics', 'south south', 'collaboration', 'hivaidsresearch', 'could', 'ultimately', 'inform', 'stakeholders', 'progress', 'nature', 'collaboration', 'towards', 'increased', 'research', 'capacities', 'low', 'middle income', 'countries', 'lmic', 'methods', 'bibliometric', 'social', 'network', 'analysis', 'methods', 'used', 'assess', 'year', 'scientific', 'contribution', 'lmic', 'analysis', 'scientific', 'publications', 'hivaids', 'prevention', 'andor', 'treatment', 'five', 'dimensions', 'oriented', 'study', 'knowledge', 'production', 'co authorship', 'analysis', 'research', 'themes', 'mapping', 'research', 'types', 'classification', 'funding', 'sources', 'results', 'publications', 'involving', 'lmic', 'substantially', 'increased', 'overtime', 'despite', 'small', 'expression', 'south south', 'collaboration', 'research', 'themes', 'mapping', 'revealed', 'publication', 'focus', 'varied', 'according', 'collaborating', 'countries', 'income', 'categories', 'diagnosis', 'opportunistic', 'infections', 'laboratory based', 'research', 'lmic', 'single', 'lmic lmic', 'human', 'behavior', 'healthcare', 'drug', 'therapy', 'mother', 'child', 'transmission', 'lmic hic', 'analysis', 'research', 'types', 'showed', 'south south', 'collaborations', 'frequently', 'targeted', 'social', 'sciences', 'issues', 'funding', 'agencies', 'acknowledged', 'south south', 'collaboration', 'also', 'showed', 'diverse', 'focus', 'lmic based', 'funders', 'tended', 'support', 'basic', 'biomedical', 'research', 'whereas', 'internationalhic based', 'funders', 'seem', 'cover', 'predominantly', 'social', 'sciences oriented', 'research', 'conclusions', 'although', 'global', 'environment', 'fostered', 'increasing', 'participation', 'lmic', 'collaborative', 'learning', 'models', 'south south', 'collaboration', 'hivaids', 'prevention', 'andor', 'treatment', 'research', 'seemed', 'lower', 'expected', 'stressing', 'need', 'strategies', 'foster', 'partnerships', 'evidence', 'presented', 'study', 'used', 'strengthen', 'knowledge', 'platform', 'inform', 'future', 'policy', 'planning', 'funding', 'decisions', 'contributing', 'development', 'enhanced', 'collaboration', 'priority', 'research', 'agenda', 'lmics', 'authors'], ['leadership', 'competencies', 'regularly', 'identified', 'demand', 'workplace', 'competencies', 'however', 'development', 'competencies', 'requires', 'appropriate', 'assessments', 'often', 'either', 'highly', 'subjective', 'eg', 'manager', 'appraisals', 'prohibitively', 'expensive', 'eg', 'roleplays', 'trained', 'actors', 'increasing', 'usage', 'workplace', 'social', 'networks', 'increasing', 'prevalence', 'digital', 'collaboration', 'tools', 'presents', 'continuous', 'stream', 'social', 'interactions', 'contain', 'evidence', 'leadership', 'occurring', 'situ', 'paper', 'present', 'initial', 'research', 'feasibility', 'social', 'network', 'analysis', 'workplace', 'assess', 'leadership', 'competencies', 'examine', 'assessment', 'terms', 'content', 'construct', 'criterion', 'validity', 'present', 'hypotheses', 'assessment', 'conducted', 'including', 'algorithms', 'necessary', 'extract', 'relevant', 'features', 'social', 'network', 'graph', 'model', 'initial', 'research', 'surprise', 'shows', 'weak', 'correlation', 'individuals', 'degree', 'centrality', 'betweenness', 'centrality', 'leadership', 'competency', 'self reported', 'however', 'experiments', 'indicated', 'strong', 'positive', 'correlation', 'network', 'structure', 'based', 'social', 'collaboration', 'activities', 'based', 'features', 'characteristics', 'leadership', 'competencies', 'initial', 'machine', 'learning', 'experiments', 'achieved', 'area', 'curve', 'auc', 'score', 'social', 'network', 'collaboration', 'activity', 'based', 'features', 'leveraged', 'distinguish', 'individuals', 'self reported', 'leadership', 'competencies', 'others', 'finally', 'discuss', 'findings', 'practicality', 'approach', 'future', 'work', 'validating', 'improving', 'results', 'obtained', 'using', 'parallel', 'conventional', 'assessments', 'leadership', 'competencies', 'springer', 'nature', 'switzerland', 'ag']]\n"
     ]
    }
   ],
   "source": [
    "data_set=[]  \n",
    "for i in range(len(metadata2[\"Abstract\"])):\n",
    "    result=[]\n",
    "    seg_list = metadata2[\"Abstract\"][i].split()\n",
    "    \n",
    "    for w in seg_list :  \n",
    "        clean_w = clean_text(w)\n",
    "        if clean_w != '':\n",
    "            result.append(clean_w)\n",
    "    data_set.append(result)\n",
    "print(data_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "9fa6a054",
   "metadata": {},
   "outputs": [],
   "source": [
    "dictionary = corpora.Dictionary(data_set) \n",
    "corpus = [dictionary.doc2bow(text) for text in data_set]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "7d55bfee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(0, 1), (1, 1), (2, 1), (3, 1), (4, 1), (5, 1), (6, 1), (7, 1), (8, 1), (9, 1), (10, 3), (11, 1), (12, 1), (13, 1), (14, 1), (15, 1), (16, 1), (17, 1), (18, 1), (19, 1), (20, 1), (21, 1), (22, 2), (23, 1), (24, 1), (25, 1), (26, 1), (27, 1), (28, 2), (29, 1)]\n"
     ]
    }
   ],
   "source": [
    "print(corpus[:1][0][:30])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "9dc30e88",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[d3graph] WARNING> too few updates, training might not converge; consider increasing the number of passes or iterations to improve accuracy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(0,\n",
      "  '0.013*\"students\" + 0.010*\"social\" + 0.009*\"networks\" + 0.007*\"faculty\" + '\n",
      "  '0.007*\"research\" + 0.007*\"authors\" + 0.006*\"lcs\" + 0.006*\"informal\" + '\n",
      "  '0.006*\"intellectual\" + 0.006*\"relations\"'),\n",
      " (1,\n",
      "  '0.018*\"network\" + 0.016*\"collaboration\" + 0.012*\"networks\" + 0.012*\"social\" '\n",
      "  '+ 0.011*\"study\" + 0.010*\"research\" + 0.009*\"analysis\" + '\n",
      "  '0.008*\"interprofessional\" + 0.007*\"organizations\" + 0.007*\"development\"'),\n",
      " (2,\n",
      "  '0.014*\"social\" + 0.012*\"collaboration\" + 0.011*\"network\" + 0.009*\"research\" '\n",
      "  '+ 0.006*\"networks\" + 0.006*\"interprofessional\" + 0.006*\"analysis\" + '\n",
      "  '0.005*\"leadership\" + 0.005*\"development\" + 0.005*\"study\"'),\n",
      " (3,\n",
      "  '0.021*\"network\" + 0.017*\"collaboration\" + 0.013*\"research\" + 0.012*\"social\" '\n",
      "  '+ 0.010*\"supervisors\" + 0.008*\"interns\" + 0.007*\"specialists\" + '\n",
      "  '0.007*\"interprofessional\" + 0.007*\"ent\" + 0.006*\"learn\"'),\n",
      " (4,\n",
      "  '0.019*\"students\" + 0.018*\"network\" + 0.014*\"collaboration\" + '\n",
      "  '0.013*\"networks\" + 0.011*\"lcs\" + 0.009*\"within\" + 0.009*\"paper\" + '\n",
      "  '0.009*\"social\" + 0.009*\"formal\" + 0.008*\"analysis\"'),\n",
      " (5,\n",
      "  '0.014*\"networks\" + 0.012*\"network\" + 0.011*\"students\" + '\n",
      "  '0.007*\"organizations\" + 0.007*\"structures\" + 0.006*\"study\" + '\n",
      "  '0.006*\"service\" + 0.006*\"social\" + 0.006*\"information\" + 0.005*\"research\"'),\n",
      " (6,\n",
      "  '0.022*\"students\" + 0.019*\"network\" + 0.015*\"collaboration\" + '\n",
      "  '0.015*\"interprofessional\" + 0.011*\"analysis\" + 0.011*\"study\" + '\n",
      "  '0.010*\"program\" + 0.010*\"lcs\" + 0.009*\"networks\" + 0.008*\"social\"'),\n",
      " (7,\n",
      "  '0.019*\"collaboration\" + 0.013*\"social\" + 0.012*\"research\" + 0.011*\"network\" '\n",
      "  '+ 0.009*\"organizations\" + 0.008*\"change\" + 0.008*\"different\" + '\n",
      "  '0.008*\"analysis\" + 0.007*\"climate\" + 0.006*\"nature\"'),\n",
      " (8,\n",
      "  '0.013*\"network\" + 0.011*\"networks\" + 0.009*\"social\" + 0.007*\"organizations\" '\n",
      "  '+ 0.007*\"collaboration\" + 0.007*\"study\" + 0.006*\"research\" + '\n",
      "  '0.006*\"information\" + 0.005*\"service\" + 0.005*\"sustainability\"'),\n",
      " (9,\n",
      "  '0.015*\"network\" + 0.014*\"collaboration\" + 0.013*\"social\" + '\n",
      "  '0.007*\"sustainability\" + 0.007*\"climate\" + 0.006*\"change\" + '\n",
      "  '0.006*\"interprofessional\" + 0.006*\"development\" + 0.006*\"different\" + '\n",
      "  '0.006*\"organizations\"')]\n"
     ]
    }
   ],
   "source": [
    "from pprint import pprint\n",
    "# number of topics\n",
    "num_topics = 10\n",
    "# Build LDA model\n",
    "lda_model = gensim.models.LdaMulticore(corpus=corpus,\n",
    "                                       id2word=dictionary,\n",
    "                                       num_topics=num_topics)\n",
    "# Print the Keyword in the 10 topics\n",
    "pprint(lda_model.print_topics())\n",
    "doc_lda = lda_model[corpus]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f45d43e4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c47de0ac",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'metadata2' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_31887/1517449037.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mdata_set\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmetadata2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Abstract\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m     \u001b[0mresult\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mseg_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmetadata2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Abstract\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'metadata2' is not defined"
     ]
    }
   ],
   "source": [
    "data_set=[]  \n",
    "for i in range(len(metadata2[\"Abstract\"])):\n",
    "    result=[]\n",
    "    seg_list = metadata2[\"Abstract\"][i].split()\n",
    "    \n",
    "    for w in seg_list :  \n",
    "        clean_w = clean_text(w)\n",
    "        if clean_w != '':\n",
    "            result.append(clean_w)\n",
    "    data_set.append(result)\n",
    "print(data_set)\n",
    "\n",
    "dictionary = corpora.Dictionary(data_set) \n",
    "corpus = [dictionary.doc2bow(text) for text in data_set]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce56de94",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(corpus[:1][0][:30])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "284813e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pprint import pprint\n",
    "# number of topics\n",
    "num_topics = 10\n",
    "# Build LDA model\n",
    "lda_model = gensim.models.LdaMulticore(corpus=corpus,\n",
    "                                       id2word=dictionary,\n",
    "                                       num_topics=num_topics)\n",
    "# Print the Keyword in the 10 topics\n",
    "pprint(lda_model.print_topics())\n",
    "doc_lda = lda_model[corpus]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e786f91",
   "metadata": {},
   "source": [
    "### pybliometrics: Python-based API-Wrapper to access Scopus\n",
    "pybliometrics is an easy to use Python library to pull, cache and extract data from the Scopus database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a6921bda",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pybliometrics.scopus import AuthorSearch\n",
    "from pybliometrics.scopus import AuthorRetrieval\n",
    "from pybliometrics.scopus import AbstractRetrieval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "12a9fdc6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pybliometrics: Scriptable bibliometrics using a Python interface to Scopus\n",
      "[Author(auid=57209617104, indexed_name='Rose M.E.', surname='Rose', given_name='Michael E.', affiliation='60105007'), Author(auid=7004212771, indexed_name='Kitchin J.R.', surname='Kitchin', given_name='John R.', affiliation='60027950')]\n"
     ]
    }
   ],
   "source": [
    "ab = AbstractRetrieval(\"10.1016/j.softx.2019.100263\")\n",
    "print(ab.title)\n",
    "print(ab.authors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cda5b516",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Search 'AUTHLAST(Selten) and AUTHFIRST(Reinhard)' yielded 2 authors as of 2023-02-14:\n",
      "    Selten, Reinhard; AUTHOR_ID:6602907525 (74 document(s))\n",
      "    Selten, Reinhard; AUTHOR_ID:57213632570 (1 document(s))\n"
     ]
    }
   ],
   "source": [
    "s = AuthorSearch('AUTHLAST(Selten) and AUTHFIRST(Reinhard)')\n",
    "print(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "5a632f68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Affiliation(id=110785688, parent=60027950, type='dept', relationship='author', afdispname=None, preferred_name='Department of Chemical Engineering', parent_preferred_name='Carnegie Mellon University', country_code='usa', country='United States', address_part='5000 Forbes Avenue', city='Pittsburgh', state='PA', postal_code='15213-3890', org_domain='cmu.edu', org_URL='https://www.cmu.edu/')]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "au2 = AuthorRetrieval(ab.authors[1].auid)\n",
    "aff2 = au2.affiliation_current\n",
    "aff2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1b49a831",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'usa'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aff2[0][7]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "b9fe5b58",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bochud P. from Department of Medicine in Switzerland,\n",
      "published 172 document(s) since 1994 \n",
      "which were cited by 6,422 author(s) in 8,098 document(s) as of 2023-02-14\n"
     ]
    }
   ],
   "source": [
    "author_id = 6603277503\n",
    "au = AuthorRetrieval(author_id)\n",
    "print(au)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "816d6a0c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pierre Yves\n",
      "Bochud\n"
     ]
    }
   ],
   "source": [
    "print(au.given_name)\n",
    "print(au.surname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "492cff86",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Affiliation(id=109652521, parent=60006027, type='dept', relationship='author', afdispname=None, preferred_name='Department of Health Sciences', parent_preferred_name='Universitair Medisch Centrum Groningen', country_code='nld', country='Netherlands', address_part='Hanzeplein 1', city='Groningen', state='Groningen', postal_code='9713 GZ', org_domain='umcg.nl', org_URL='https://www.umcg.nl/EN/corporate/paginas/default.aspx')]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "au = AuthorRetrieval(6701589266)\n",
    "aff = au.affiliation_current\n",
    "aff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "be4983d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'nor'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aff[1][7]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "22d5a9b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Brouwer S. from Department of Health Sciences in Netherlands,\n",
      "published 174 document(s) since 2003 \n",
      "which were cited by 2,703 author(s) in 3,408 document(s) as of 2023-02-14\n"
     ]
    }
   ],
   "source": [
    "print(au)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "d36449f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2003, 2023)"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "au.publication_range\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "948638ba",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(174, 36)"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "docs = pd.DataFrame(au.get_documents(refresh=10))\n",
    "docs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "0485cef2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['eid', 'doi', 'pii', 'pubmed_id', 'title', 'subtype',\n",
       "       'subtypeDescription', 'creator', 'afid', 'affilname',\n",
       "       'affiliation_city', 'affiliation_country', 'author_count',\n",
       "       'author_names', 'author_ids', 'author_afids', 'coverDate',\n",
       "       'coverDisplayDate', 'publicationName', 'issn', 'source_id', 'eIssn',\n",
       "       'aggregationType', 'volume', 'issueIdentifier', 'article_number',\n",
       "       'pageRange', 'description', 'authkeywords', 'citedby_count',\n",
       "       'openaccess', 'freetoread', 'freetoreadLabel', 'fund_acr', 'fund_no',\n",
       "       'fund_sponsor'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "docs.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "15f449fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "au_id = docs['author_ids']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "cec3a90e",
   "metadata": {},
   "outputs": [],
   "source": [
    "coau_id = []\n",
    "link = []\n",
    "all_node = []\n",
    "for i in au_id:\n",
    "    coau_id = i.split(\";\")\n",
    "    new_coau_id = []\n",
    "    for i in coau_id:\n",
    "        aff = AuthorRetrieval(i).affiliation_current\n",
    "        if AuthorRetrieval(i).affiliation_current:\n",
    "            if AuthorRetrieval(i).affiliation_current[0][7] == 'nld':\n",
    "                all_node.append(i)\n",
    "                new_coau_id.append(i)\n",
    "    link = link+list(itertools.combinations(new_coau_id, 2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "f0edf524",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(('7004580260', '6701589266'), 31),\n",
       " (('6603920310', '6701589266'), 30),\n",
       " (('6701589266', '7004580260'), 29),\n",
       " (('6701589266', '6603414230'), 18),\n",
       " (('8518459100', '6701589266'), 16),\n",
       " (('6701589266', '6603920310'), 16),\n",
       " (('7004580260', '7005631576'), 16),\n",
       " (('41861260100', '6701589266'), 15),\n",
       " (('6701589266', '6603547181'), 15),\n",
       " (('55566583100', '6701589266'), 15)]"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "links = Counter(link)\n",
    "links.most_common(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "1174434b",
   "metadata": {},
   "outputs": [],
   "source": [
    "sources = []\n",
    "targets = []\n",
    "weights = []\n",
    "\n",
    "for i in links:\n",
    "    sources.append(i[0])\n",
    "    targets.append(i[1])\n",
    "    weights.append(link.count(i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "0b6f0ecb",
   "metadata": {},
   "outputs": [],
   "source": [
    "N = Network(height=800, width=\"100%\", bgcolor=\"#222222\", font_color=\"white\", select_menu=True)\n",
    "N.toggle_hide_edges_on_drag(False)\n",
    "N.barnes_hut()\n",
    "\n",
    "edge_data = zip(sources, targets, weights)\n",
    "\n",
    "for e in edge_data:\n",
    "    src = e[0]\n",
    "    dst = e[1]\n",
    "    w = e[2]\n",
    "\n",
    "    N.add_node(src, src, title=src)\n",
    "    N.add_node(dst, dst, title=dst)\n",
    "    N.add_edge(src, dst, value=w)\n",
    "\n",
    "neighbor_map = N.get_adj_list()\n",
    "\n",
    "# add neighbor data to node hover data\n",
    "for node in N.nodes:\n",
    "    neighbors = []\n",
    "    for neighbor_id in neighbor_map[node[\"id\"]]:\n",
    "        neigh = AuthorRetrieval(neighbor_id)\n",
    "        neighbors.append(neigh.given_name+neigh.surname)\n",
    "    node[\"title\"] = \" Neighbors: \\n\" + \" \\n\".join(neighbors)\n",
    "    node[\"value\"] = all_node.count(node[\"id\"])\n",
    "    node[\"label\"] = AuthorRetrieval(node[\"id\"]).given_name+AuthorRetrieval(node[\"id\"]).surname\n",
    "\n",
    "N.show(\"pybliometrics.html\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7bfa21a3",
   "metadata": {},
   "source": [
    "### scholarly \n",
    "Scholarly is a module that allows you to retrieve author and publication information from Google Scholar in a friendly, Pythonic way without having to solve CAPTCHAs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8849cf5e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def build_graph(G, sch_id, depth: int, bredth: int, last_author=[]):\n",
    "    author_search = scholarly.search_author_id(sch_id)\n",
    "    author = scholarly.fill(author_search,sections=['coauthors','indices','counts'])\n",
    "    author_name = author['name']\n",
    "    author_hin = author['hindex']\n",
    "    author_ncite = author['citedby']\n",
    "    G.add_node(author_name,hindex=author_hin,name=author_name,ncite=author_ncite)\n",
    "    print(\"now at %s, depth=%d\"%(author_name,depth))\n",
    "    if depth>0:\n",
    "        for coauthor in author['coauthors'][:int(bredth)]:\n",
    "            G.add_edge(author_name,coauthor['name'])\n",
    "            if coauthor['name'] not in last_author:\n",
    "                last_author.append(author_name)\n",
    "                build_graph(G,coauthor['scholar_id'],depth-1,bredth,last_author)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2367438a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LmPLt9AAAAAJ\n"
     ]
    }
   ],
   "source": [
    "#input and parameters\n",
    "name = 'Francine Chen'\n",
    "sch_id = next(scholarly.search_author(name))['scholar_id']\n",
    "print(sch_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "14f0fa4b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "now at francine chen, depth=3\n"
     ]
    }
   ],
   "source": [
    "depth = 3\n",
    "bredth = 10\n",
    "\n",
    "G = nx.DiGraph()\n",
    "output_path = '%s_%d_%d.gexf'%(name.replace(' ','_'),depth,bredth)\n",
    "\n",
    "build_graph(G,sch_id,depth,bredth)\n",
    "nx.write_gexf(G,output_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "141607e1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
